\hypertarget{classorg_1_1jgap_1_1_genotype}{\section{org.\-jgap.\-Genotype Class Reference}
\label{classorg_1_1jgap_1_1_genotype}\index{org.\-jgap.\-Genotype@{org.\-jgap.\-Genotype}}
}


Inheritance diagram for org.\-jgap.\-Genotype\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=222pt]{classorg_1_1jgap_1_1_genotype__inherit__graph}
\end{center}
\end{figure}


Collaboration diagram for org.\-jgap.\-Genotype\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype__coll__graph}
\end{center}
\end{figure}
\subsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
\hyperlink{classorg_1_1jgap_1_1_genotype_a00b11af9a68e9d84bbc52afbf661607f}{Genotype} (\hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} a\-\_\-configuration, \hyperlink{interfaceorg_1_1jgap_1_1_i_chromosome}{I\-Chromosome}\mbox{[}$\,$\mbox{]} a\-\_\-initial\-Chromosomes)  throws Invalid\-Configuration\-Exception 
\item 
\hyperlink{classorg_1_1jgap_1_1_genotype_a7b08ffd2154cd3ccc8636b795666c996}{Genotype} (\hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} a\-\_\-configuration, \hyperlink{classorg_1_1jgap_1_1_population}{Population} a\-\_\-population)  throws Invalid\-Configuration\-Exception 
\item 
\hyperlink{classorg_1_1jgap_1_1_genotype_afc52d6b649eea1eefd92e376ea3762ff}{Genotype} (\hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} a\-\_\-configuration)  throws Invalid\-Configuration\-Exception 
\item 
synchronized \hyperlink{interfaceorg_1_1jgap_1_1_i_chromosome}{I\-Chromosome}\mbox{[}$\,$\mbox{]} \hyperlink{classorg_1_1jgap_1_1_genotype_ad6795433da5538c6f520253ff9d2382e}{get\-Chromosomes} ()
\item 
\hyperlink{classorg_1_1jgap_1_1_population}{Population} \hyperlink{classorg_1_1jgap_1_1_genotype_a8241b2d89d32b283abe5c9b42974d64d}{get\-Population} ()
\item 
synchronized \hyperlink{interfaceorg_1_1jgap_1_1_i_chromosome}{I\-Chromosome} \hyperlink{classorg_1_1jgap_1_1_genotype_a05380af4c0737841c09d3a5533f9f942}{get\-Fittest\-Chromosome} ()
\item 
synchronized \hyperlink{interfaceorg_1_1jgap_1_1_i_chromosome}{I\-Chromosome} \hyperlink{classorg_1_1jgap_1_1_genotype_a6c40835533352287db8d34e5fae6875f}{get\-Fittest\-Chromosome} (int a\-\_\-start\-Index, int a\-\_\-end\-Index)
\item 
synchronized List \hyperlink{classorg_1_1jgap_1_1_genotype_a602d6b1413007ef28880cf88ccd2f912}{get\-Fittest\-Chromosomes} (int a\-\_\-number\-Of\-Chromosomes)
\item 
synchronized void \hyperlink{classorg_1_1jgap_1_1_genotype_a54a6791fbb6e13920e79f4e2dd719eb4}{evolve} ()
\item 
void \hyperlink{classorg_1_1jgap_1_1_genotype_ab15d88cf1a6084f9efea2dc935968471}{evolve} (int a\-\_\-number\-Of\-Evolutions)
\item 
List$<$ String $>$ \hyperlink{classorg_1_1jgap_1_1_genotype_aa740dd753d31620166979199a38d5a50}{evolve} (I\-Evolution\-Monitor a\-\_\-monitor)
\item 
String \hyperlink{classorg_1_1jgap_1_1_genotype_a658a541940e9be8ef474375091ff31ae}{to\-String} ()
\item 
void \hyperlink{classorg_1_1jgap_1_1_genotype_ac622bbf487c0ab7344d46ca669607cef}{fill\-Population} (final int a\-\_\-num)  throws Invalid\-Configuration\-Exception 
\item 
boolean \hyperlink{classorg_1_1jgap_1_1_genotype_ab9bba698920b358bb53131338c9d130a}{equals} (Object a\-\_\-other)
\item 
void \hyperlink{classorg_1_1jgap_1_1_genotype_adf2dbe6d60d6c74592cb1e6398cece2c}{apply\-Genetic\-Operators} ()
\item 
\hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} \hyperlink{classorg_1_1jgap_1_1_genotype_a786848d2da4fd3d87f9f4caa8f9dd58b}{get\-Configuration} ()
\item 
int \hyperlink{classorg_1_1jgap_1_1_genotype_abe044a5473e935c2093ee3c6c18775f7}{hash\-Code} ()
\item 
void \hyperlink{classorg_1_1jgap_1_1_genotype_a0d14c5400197e3c43f9d1f0666309a85}{run} ()
\item 
List$<$ I\-Evolve\-Job $>$ \hyperlink{classorg_1_1jgap_1_1_genotype_a84fd1b399188ea10c0af4b7c1152e0c2}{get\-Evolves} (I\-Population\-Splitter a\-\_\-splitter)  throws Exception 
\item 
void \hyperlink{classorg_1_1jgap_1_1_genotype_aa2b0341edc420ad61e797508b8bf4dd5}{merge\-Results} (I\-Population\-Merger a\-\_\-merger, Evolve\-Result\mbox{[}$\,$\mbox{]} a\-\_\-results)  throws Exception 
\item 
void \hyperlink{classorg_1_1jgap_1_1_genotype_abba6c68b3e8388be6884a35f5600e8cb}{set\-Use\-Monitor} (boolean a\-\_\-use\-Monitor)
\item 
void \hyperlink{classorg_1_1jgap_1_1_genotype_af3f49ba90cc8515817366792b7bce320}{set\-Monitor} (I\-Evolution\-Monitor a\-\_\-monitor)
\end{DoxyCompactItemize}
\subsection*{Static Public Member Functions}
\begin{DoxyCompactItemize}
\item 
static \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} \hyperlink{classorg_1_1jgap_1_1_genotype_a930184e2920b4642f20d0b8bc8412005}{random\-Initial\-Genotype} (\hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} a\-\_\-configuration)  throws Invalid\-Configuration\-Exception 
\item 
static \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} \hyperlink{classorg_1_1jgap_1_1_genotype_aaf3d4bf0b470aa4655e85262289443eb}{get\-Static\-Configuration} ()
\item 
static void \hyperlink{classorg_1_1jgap_1_1_genotype_a2f44a7709eca83cddd6d794ee59e663c}{set\-Static\-Configuration} (\hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} a\-\_\-configuration)
\end{DoxyCompactItemize}
\subsection*{Protected Member Functions}
\begin{DoxyCompactItemize}
\item 
void \hyperlink{classorg_1_1jgap_1_1_genotype_a6c14c3a36aa8e4291840f0b9c0facb35}{apply\-Natural\-Selectors} (boolean a\-\_\-process\-Before\-Genetic\-Operators)
\item 
void \hyperlink{classorg_1_1jgap_1_1_genotype_ac56a4d112334db07ba333602c4a24706}{set\-Population} (\hyperlink{classorg_1_1jgap_1_1_population}{Population} a\-\_\-pop)
\item 
void \hyperlink{classorg_1_1jgap_1_1_genotype_a554d1eaf21a0a9437cdbc5b0fbeedef4}{apply\-Genetic\-Operator} (\hyperlink{interfaceorg_1_1jgap_1_1_genetic_operator}{Genetic\-Operator} a\-\_\-operator, \hyperlink{classorg_1_1jgap_1_1_population}{Population} a\-\_\-population, List a\-\_\-chromosomes)
\item 
void \hyperlink{classorg_1_1jgap_1_1_genotype_af5e224c5b7c05dbe02b3324d5ec16f2a}{keep\-Pop\-Size\-Constant} (\hyperlink{classorg_1_1jgap_1_1_population}{Population} a\-\_\-pop, int a\-\_\-max\-Size)
\end{DoxyCompactItemize}
\subsection*{Private Attributes}
\begin{DoxyCompactItemize}
\item 
\hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} \hyperlink{classorg_1_1jgap_1_1_genotype_aeb3fac2610b8ef8fa8599fcac1b6a199}{m\-\_\-active\-Configuration}
\item 
\hyperlink{classorg_1_1jgap_1_1_population}{Population} \hyperlink{classorg_1_1jgap_1_1_genotype_aa2ba4539110aaa8d853c808679f4f0e2}{m\-\_\-population}
\item 
boolean \hyperlink{classorg_1_1jgap_1_1_genotype_ab1b1b976deab7f2faea463d1b8d52abe}{m\-\_\-use\-Monitor}
\item 
I\-Evolution\-Monitor \hyperlink{classorg_1_1jgap_1_1_genotype_acac1218542ab91833483ebd42ac8636b}{m\-\_\-monitor}
\end{DoxyCompactItemize}
\subsection*{Static Private Attributes}
\begin{DoxyCompactItemize}
\item 
static final String \hyperlink{classorg_1_1jgap_1_1_genotype_ac9f8e9d203b7a4efdf8c2e32a08f84bb}{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N} = \char`\"{}\$Revision\-: 1.\-109 \$\char`\"{}
\item 
static transient \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} \hyperlink{classorg_1_1jgap_1_1_genotype_a8b93a0b2044e40520d4a966ef3ae0bbe}{m\-\_\-static\-Configuration}
\end{DoxyCompactItemize}


\subsection{Detailed Description}
Genotypes are fixed-\/length populations of chromosomes. As an instance of a \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} is evolved, all of its Chromosomes are also evolved. A \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} may be constructed normally via constructor, or the static \hyperlink{classorg_1_1jgap_1_1_genotype_a930184e2920b4642f20d0b8bc8412005}{random\-Initial\-Genotype()} method can be used to generate a \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} with a randomized \hyperlink{classorg_1_1jgap_1_1_chromosome}{Chromosome} population. 

Please note that among all created \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} instances there may only be one configuration, used by all \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} instances.

\begin{DoxyAuthor}{Author}
Neil Rotstan 

Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
1.\-0 
\end{DoxySince}


Definition at line 33 of file Genotype.\-java.



\subsection{Constructor \& Destructor Documentation}
\hypertarget{classorg_1_1jgap_1_1_genotype_a00b11af9a68e9d84bbc52afbf661607f}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!Genotype@{Genotype}}
\index{Genotype@{Genotype}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{Genotype}]{\setlength{\rightskip}{0pt plus 5cm}org.\-jgap.\-Genotype.\-Genotype (
\begin{DoxyParamCaption}
\item[{{\bf Configuration}}]{a\-\_\-configuration, }
\item[{{\bf I\-Chromosome}\mbox{[}$\,$\mbox{]}}]{a\-\_\-initial\-Chromosomes}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_a00b11af9a68e9d84bbc52afbf661607f}
Constructs a new \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} instance with the given array of Chromosomes and the given active \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} instance. Note that the \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} object must be in a valid state when this method is invoked, or a \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception} will be thrown.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-configuration} & the \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} object to use \\
\hline
{\em a\-\_\-initial\-Chromosomes} & the \hyperlink{classorg_1_1jgap_1_1_chromosome}{Chromosome} population to be managed by this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} instance \\
\hline
\end{DoxyParams}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & if the given \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} object is in an invalid state\\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Neil Rotstan 

Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
1.\-0 
\end{DoxySince}
\begin{DoxyRefDesc}{Deprecated}
\item[\hyperlink{deprecated__deprecated000007}{Deprecated}]use \hyperlink{classorg_1_1jgap_1_1_genotype_a7b08ffd2154cd3ccc8636b795666c996}{Genotype(\-Configuration, Population)} instead \end{DoxyRefDesc}


Definition at line 76 of file Genotype.\-java.



Referenced by org.\-jgap.\-Genotype.\-equals(), and org.\-jgap.\-Genotype.\-random\-Initial\-Genotype().

\hypertarget{classorg_1_1jgap_1_1_genotype_a7b08ffd2154cd3ccc8636b795666c996}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!Genotype@{Genotype}}
\index{Genotype@{Genotype}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{Genotype}]{\setlength{\rightskip}{0pt plus 5cm}org.\-jgap.\-Genotype.\-Genotype (
\begin{DoxyParamCaption}
\item[{{\bf Configuration}}]{a\-\_\-configuration, }
\item[{{\bf Population}}]{a\-\_\-population}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_a7b08ffd2154cd3ccc8636b795666c996}
Constructs a new \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} instance with the given array of Chromosomes and the given active \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} instance. Note that the \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} object must be in a valid state when this method is invoked, or a Invalidconfiguration\-Exception will be thrown.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-configuration} & the \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} object to use \\
\hline
{\em a\-\_\-population} & the \hyperlink{classorg_1_1jgap_1_1_chromosome}{Chromosome} population to be managed by this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} instance \\
\hline
\end{DoxyParams}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Neil Rotstan 

Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-0 
\end{DoxySince}


Definition at line 98 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-m\-\_\-active\-Configuration, and org.\-jgap.\-Genotype.\-m\-\_\-population.

\hypertarget{classorg_1_1jgap_1_1_genotype_afc52d6b649eea1eefd92e376ea3762ff}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!Genotype@{Genotype}}
\index{Genotype@{Genotype}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{Genotype}]{\setlength{\rightskip}{0pt plus 5cm}org.\-jgap.\-Genotype.\-Genotype (
\begin{DoxyParamCaption}
\item[{{\bf Configuration}}]{a\-\_\-configuration}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_afc52d6b649eea1eefd92e376ea3762ff}
Don't use this constructor, it's only for internal use.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-configuration} & not used here! \\
\hline
\end{DoxyParams}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 136 of file Genotype.\-java.



\subsection{Member Function Documentation}
\hypertarget{classorg_1_1jgap_1_1_genotype_a554d1eaf21a0a9437cdbc5b0fbeedef4}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!apply\-Genetic\-Operator@{apply\-Genetic\-Operator}}
\index{apply\-Genetic\-Operator@{apply\-Genetic\-Operator}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{apply\-Genetic\-Operator}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Genotype.\-apply\-Genetic\-Operator (
\begin{DoxyParamCaption}
\item[{{\bf Genetic\-Operator}}]{a\-\_\-operator, }
\item[{{\bf Population}}]{a\-\_\-population, }
\item[{List}]{a\-\_\-chromosomes}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1_genotype_a554d1eaf21a0a9437cdbc5b0fbeedef4}
Overwritable method that calls a \hyperlink{interfaceorg_1_1jgap_1_1_genetic_operator}{Genetic\-Operator} to operate on a given population and asks him to store the result in the list of chromosomes. Override this method if you want to ensure that a\-\_\-chromosomes is not part of a\-\_\-population resp. if you want to use a different list.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-operator} & the \hyperlink{interfaceorg_1_1jgap_1_1_genetic_operator}{Genetic\-Operator} to call \\
\hline
{\em a\-\_\-population} & the population to use \\
\hline
{\em a\-\_\-chromosomes} & the list of \hyperlink{classorg_1_1jgap_1_1_chromosome}{Chromosome} objects to return\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-4 
\end{DoxySince}


Definition at line 587 of file Genotype.\-java.



Referenced by org.\-jgap.\-Genotype.\-apply\-Genetic\-Operators().

\hypertarget{classorg_1_1jgap_1_1_genotype_adf2dbe6d60d6c74592cb1e6398cece2c}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!apply\-Genetic\-Operators@{apply\-Genetic\-Operators}}
\index{apply\-Genetic\-Operators@{apply\-Genetic\-Operators}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{apply\-Genetic\-Operators}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Genotype.\-apply\-Genetic\-Operators (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_adf2dbe6d60d6c74592cb1e6398cece2c}
Applies all Genetic\-Operators registered with the \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration}.

\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 501 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-apply\-Genetic\-Operator(), org.\-jgap.\-Genotype.\-get\-Chromosomes(), and org.\-jgap.\-Genotype.\-get\-Population().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_adf2dbe6d60d6c74592cb1e6398cece2c_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_a6c14c3a36aa8e4291840f0b9c0facb35}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!apply\-Natural\-Selectors@{apply\-Natural\-Selectors}}
\index{apply\-Natural\-Selectors@{apply\-Natural\-Selectors}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{apply\-Natural\-Selectors}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Genotype.\-apply\-Natural\-Selectors (
\begin{DoxyParamCaption}
\item[{boolean}]{a\-\_\-process\-Before\-Genetic\-Operators}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1_genotype_a6c14c3a36aa8e4291840f0b9c0facb35}
Applies all Natural\-Selectors registered with the \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration}.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-process\-Before\-Genetic\-Operators} & true apply Natural\-Selectors applicable before Genetic\-Operators, false\-: apply the ones applicable after Genetic\-Operators\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-0 
\end{DoxySince}
\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000118}{Todo}]optionally use working pool \end{DoxyRefDesc}


Definition at line 442 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-get\-Configuration(), org.\-jgap.\-Genotype.\-get\-Population(), org.\-jgap.\-Genotype.\-m\-\_\-active\-Configuration, and org.\-jgap.\-Genotype.\-set\-Population().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_a6c14c3a36aa8e4291840f0b9c0facb35_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_ab9bba698920b358bb53131338c9d130a}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!equals@{equals}}
\index{equals@{equals}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{equals}]{\setlength{\rightskip}{0pt plus 5cm}boolean org.\-jgap.\-Genotype.\-equals (
\begin{DoxyParamCaption}
\item[{Object}]{a\-\_\-other}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_ab9bba698920b358bb53131338c9d130a}
Compares this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} against the specified object. The result is true if the argument is an instance of the \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} class, has exactly the same number of chromosomes as the given \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype}, and, for each \hyperlink{classorg_1_1jgap_1_1_chromosome}{Chromosome} in this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype}, there is an equal chromosome in the given \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype}. The chromosomes do not need to appear in the same order within the population.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-other} & the object to compare against \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
true if the objects are the same, false otherwise
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Neil Rotstan 

Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
1.\-0 
\end{DoxySince}


Definition at line 397 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-Genotype(), org.\-jgap.\-Genotype.\-get\-Chromosomes(), and org.\-jgap.\-Genotype.\-get\-Population().



Referenced by org.\-jgap.\-xml.\-X\-M\-L\-Manager\-Test.\-test\-Get\-Genotype\-From\-Document\-\_\-1().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_ab9bba698920b358bb53131338c9d130a_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_a54a6791fbb6e13920e79f4e2dd719eb4}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!evolve@{evolve}}
\index{evolve@{evolve}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{evolve}]{\setlength{\rightskip}{0pt plus 5cm}synchronized void org.\-jgap.\-Genotype.\-evolve (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_a54a6791fbb6e13920e79f4e2dd719eb4}
Evolves the population of Chromosomes within this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype}. This will execute all of the genetic operators added to the present active configuration and then invoke the natural selector to choose which chromosomes will be included in the next generation population. Note that the population size not always remains constant (dependent on the Natural\-Selectors used!). To consecutively call this method, use \hyperlink{classorg_1_1jgap_1_1_genotype_ab15d88cf1a6084f9efea2dc935968471}{evolve(int)}!!!

\begin{DoxyAuthor}{Author}
Neil Rotstan 

Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
1.\-0 
\end{DoxySince}


Definition at line 231 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-get\-Configuration(), org.\-jgap.\-Genotype.\-get\-Population(), and org.\-jgap.\-Genotype.\-set\-Population().



Referenced by org.\-jgap.\-Genotype.\-evolve(), and org.\-jgap.\-Genotype.\-run().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_a54a6791fbb6e13920e79f4e2dd719eb4_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_ab15d88cf1a6084f9efea2dc935968471}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!evolve@{evolve}}
\index{evolve@{evolve}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{evolve}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Genotype.\-evolve (
\begin{DoxyParamCaption}
\item[{int}]{a\-\_\-number\-Of\-Evolutions}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_ab15d88cf1a6084f9efea2dc935968471}
Evolves this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} the specified number of times. This is equivalent to invoking the standard \hyperlink{classorg_1_1jgap_1_1_genotype_a54a6791fbb6e13920e79f4e2dd719eb4}{evolve()} method the given number of times in a row.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-number\-Of\-Evolutions} & the number of times to evolve this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} before returning\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
1.\-1 
\end{DoxySince}


Definition at line 248 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-evolve(), org.\-jgap.\-Genotype.\-get\-Population(), org.\-jgap.\-Configuration.\-get\-Population\-Size(), org.\-jgap.\-Configuration.\-is\-Keep\-Population\-Size\-Constant(), org.\-jgap.\-Genotype.\-keep\-Pop\-Size\-Constant(), and org.\-jgap.\-Genotype.\-m\-\_\-active\-Configuration.



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_ab15d88cf1a6084f9efea2dc935968471_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_aa740dd753d31620166979199a38d5a50}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!evolve@{evolve}}
\index{evolve@{evolve}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{evolve}]{\setlength{\rightskip}{0pt plus 5cm}List$<$String$>$ org.\-jgap.\-Genotype.\-evolve (
\begin{DoxyParamCaption}
\item[{I\-Evolution\-Monitor}]{a\-\_\-monitor}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_aa740dd753d31620166979199a38d5a50}
Evolves this genotype until the given monitor asks to quit the evolution cycle.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-monitor} & the monitor used to decide when to stop evolution\\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
messages of the registered evolution monitor. May indicate why the evolution was asked to be stopped. May be empty, depending on the implementation of the used monitor
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-4.\-4 
\end{DoxySince}


Definition at line 271 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-evolve(), org.\-jgap.\-Genotype.\-get\-Configuration(), and org.\-jgap.\-Genotype.\-get\-Population().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_aa740dd753d31620166979199a38d5a50_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_ac622bbf487c0ab7344d46ca669607cef}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!fill\-Population@{fill\-Population}}
\index{fill\-Population@{fill\-Population}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{fill\-Population}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Genotype.\-fill\-Population (
\begin{DoxyParamCaption}
\item[{final int}]{a\-\_\-num}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_ac622bbf487c0ab7344d46ca669607cef}
Fills up the population with random chromosomes if necessary.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-num} & the number of chromosomes to add \\
\hline
\end{DoxyParams}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2 
\end{DoxySince}


Definition at line 354 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-get\-Configuration(), and org.\-jgap.\-Genotype.\-get\-Population().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_ac622bbf487c0ab7344d46ca669607cef_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_ad6795433da5538c6f520253ff9d2382e}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!get\-Chromosomes@{get\-Chromosomes}}
\index{get\-Chromosomes@{get\-Chromosomes}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{get\-Chromosomes}]{\setlength{\rightskip}{0pt plus 5cm}synchronized {\bf I\-Chromosome} \mbox{[}$\,$\mbox{]} org.\-jgap.\-Genotype.\-get\-Chromosomes (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_ad6795433da5538c6f520253ff9d2382e}
Retrieves the array of Chromosomes that make up the population of this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} instance.

\begin{DoxyReturn}{Returns}
the \hyperlink{classorg_1_1jgap_1_1_population}{Population} of Chromosomes
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Neil Rotstan 

Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
1.\-0 
\end{DoxySince}
\begin{DoxyRefDesc}{Deprecated}
\item[\hyperlink{deprecated__deprecated000008}{Deprecated}]uses \hyperlink{classorg_1_1jgap_1_1_genotype_a8241b2d89d32b283abe5c9b42974d64d}{get\-Population()} instead \end{DoxyRefDesc}


Definition at line 151 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-get\-Population().



Referenced by org.\-jgap.\-Genotype.\-apply\-Genetic\-Operators(), and org.\-jgap.\-Genotype.\-equals().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_ad6795433da5538c6f520253ff9d2382e_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_a786848d2da4fd3d87f9f4caa8f9dd58b}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!get\-Configuration@{get\-Configuration}}
\index{get\-Configuration@{get\-Configuration}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{get\-Configuration}]{\setlength{\rightskip}{0pt plus 5cm}{\bf Configuration} org.\-jgap.\-Genotype.\-get\-Configuration (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_a786848d2da4fd3d87f9f4caa8f9dd58b}


Definition at line 535 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-m\-\_\-active\-Configuration.



Referenced by org.\-jgap.\-Genotype.\-apply\-Natural\-Selectors(), org.\-jgap.\-Genotype.\-evolve(), org.\-jgap.\-Genotype.\-fill\-Population(), org.\-jgap.\-Genotype.\-get\-Evolves(), and org.\-jgap.\-Genotype.\-merge\-Results().

\hypertarget{classorg_1_1jgap_1_1_genotype_a84fd1b399188ea10c0af4b7c1152e0c2}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!get\-Evolves@{get\-Evolves}}
\index{get\-Evolves@{get\-Evolves}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{get\-Evolves}]{\setlength{\rightskip}{0pt plus 5cm}List$<$I\-Evolve\-Job$>$ org.\-jgap.\-Genotype.\-get\-Evolves (
\begin{DoxyParamCaption}
\item[{I\-Population\-Splitter}]{a\-\_\-splitter}
\end{DoxyParamCaption}
) throws Exception\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_a84fd1b399188ea10c0af4b7c1152e0c2}
Splits a population into pieces that can be evolved independently.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-splitter} & splits the population \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
list of I\-Evolve\-Job objects 
\end{DoxyReturn}

\begin{DoxyExceptions}{Exceptions}
{\em Exception} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}


Definition at line 644 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-get\-Configuration(), and org.\-jgap.\-Genotype.\-get\-Population().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_a84fd1b399188ea10c0af4b7c1152e0c2_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_a05380af4c0737841c09d3a5533f9f942}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!get\-Fittest\-Chromosome@{get\-Fittest\-Chromosome}}
\index{get\-Fittest\-Chromosome@{get\-Fittest\-Chromosome}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{get\-Fittest\-Chromosome}]{\setlength{\rightskip}{0pt plus 5cm}synchronized {\bf I\-Chromosome} org.\-jgap.\-Genotype.\-get\-Fittest\-Chromosome (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_a05380af4c0737841c09d3a5533f9f942}
Retrieves the \hyperlink{classorg_1_1jgap_1_1_chromosome}{Chromosome} in the \hyperlink{classorg_1_1jgap_1_1_population}{Population} with the highest fitness value.

\begin{DoxyReturn}{Returns}
the \hyperlink{classorg_1_1jgap_1_1_chromosome}{Chromosome} with the highest fitness value, or null if there are no chromosomes in this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype}
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Neil Rotstan 

Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
1.\-0 
\end{DoxySince}


Definition at line 182 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-get\-Population().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_a05380af4c0737841c09d3a5533f9f942_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_a6c40835533352287db8d34e5fae6875f}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!get\-Fittest\-Chromosome@{get\-Fittest\-Chromosome}}
\index{get\-Fittest\-Chromosome@{get\-Fittest\-Chromosome}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{get\-Fittest\-Chromosome}]{\setlength{\rightskip}{0pt plus 5cm}synchronized {\bf I\-Chromosome} org.\-jgap.\-Genotype.\-get\-Fittest\-Chromosome (
\begin{DoxyParamCaption}
\item[{int}]{a\-\_\-start\-Index, }
\item[{int}]{a\-\_\-end\-Index}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_a6c40835533352287db8d34e5fae6875f}
Retrieves the \hyperlink{classorg_1_1jgap_1_1_chromosome}{Chromosome} in the \hyperlink{classorg_1_1jgap_1_1_population}{Population} with the highest fitness value within the given indices.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-start\-Index} & the index to start the determination with \\
\hline
{\em a\-\_\-end\-Index} & the index to end the determination with \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
the \hyperlink{classorg_1_1jgap_1_1_chromosome}{Chromosome} with the highest fitness value within the given indices, or null if there are no chromosomes in this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype}
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 198 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-get\-Population().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_a6c40835533352287db8d34e5fae6875f_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_a602d6b1413007ef28880cf88ccd2f912}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!get\-Fittest\-Chromosomes@{get\-Fittest\-Chromosomes}}
\index{get\-Fittest\-Chromosomes@{get\-Fittest\-Chromosomes}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{get\-Fittest\-Chromosomes}]{\setlength{\rightskip}{0pt plus 5cm}synchronized List org.\-jgap.\-Genotype.\-get\-Fittest\-Chromosomes (
\begin{DoxyParamCaption}
\item[{int}]{a\-\_\-number\-Of\-Chromosomes}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_a602d6b1413007ef28880cf88ccd2f912}
Retrieves the top n Chromsomes in the population (the ones with the best fitness values).


\begin{DoxyParams}{Parameters}
{\em a\-\_\-number\-Of\-Chromosomes} & the number of chromosomes desired \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
the list of Chromosomes with the highest fitness values, or null if there are no chromosomes in this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype}
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Charles Kevin Hill 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-4 
\end{DoxySince}


Definition at line 214 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-get\-Population().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_a602d6b1413007ef28880cf88ccd2f912_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_a8241b2d89d32b283abe5c9b42974d64d}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!get\-Population@{get\-Population}}
\index{get\-Population@{get\-Population}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{get\-Population}]{\setlength{\rightskip}{0pt plus 5cm}{\bf Population} org.\-jgap.\-Genotype.\-get\-Population (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_a8241b2d89d32b283abe5c9b42974d64d}
\begin{DoxyReturn}{Returns}
the current population of chromosomes
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-1 
\end{DoxySince}


Definition at line 167 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-m\-\_\-population.



Referenced by org.\-jgap.\-Genotype.\-apply\-Genetic\-Operators(), org.\-jgap.\-Genotype.\-apply\-Natural\-Selectors(), org.\-jgap.\-Genotype\-Test.\-do\-Test\-Evolve\-\_\-5(), org.\-jgap.\-Genotype.\-equals(), org.\-jgap.\-Genotype.\-evolve(), org.\-jgap.\-Genotype.\-fill\-Population(), org.\-jgap.\-Genotype.\-get\-Chromosomes(), org.\-jgap.\-Genotype.\-get\-Evolves(), org.\-jgap.\-Genotype.\-get\-Fittest\-Chromosome(), org.\-jgap.\-Genotype.\-get\-Fittest\-Chromosomes(), org.\-jgap.\-Genotype.\-hash\-Code(), org.\-jgap.\-audit.\-Evaluator.\-store\-Genotype(), org.\-jgap.\-Genotype\-Test.\-test\-Evolve\-\_\-1(), and org.\-jgap.\-Genotype.\-to\-String().

\hypertarget{classorg_1_1jgap_1_1_genotype_aaf3d4bf0b470aa4655e85262289443eb}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!get\-Static\-Configuration@{get\-Static\-Configuration}}
\index{get\-Static\-Configuration@{get\-Static\-Configuration}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{get\-Static\-Configuration}]{\setlength{\rightskip}{0pt plus 5cm}static {\bf Configuration} org.\-jgap.\-Genotype.\-get\-Static\-Configuration (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1_genotype_aaf3d4bf0b470aa4655e85262289443eb}
\begin{DoxyReturn}{Returns}
the configuration to use with the Genetic Algorithm
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-0 
\end{DoxySince}


Definition at line 520 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-m\-\_\-static\-Configuration.



Referenced by org.\-jgap.\-Chromosome\-For\-Testing.\-Chromosome\-For\-Testing(), org.\-jgap.\-Chromosome\-For\-Testing2.\-Chromosome\-For\-Testing2(), org.\-jgap.\-impl.\-Crossover\-Operator.\-Crossover\-Operator(), org.\-jgap.\-impl.\-Greedy\-Crossover.\-Greedy\-Crossover(), org.\-jgap.\-Natural\-Selector\-Ext.\-Natural\-Selector\-Ext(), org.\-jgap.\-impl.\-Tournament\-Selector.\-Tournament\-Selector(), and org.\-jgap.\-impl.\-Two\-Way\-Mutation\-Operator.\-Two\-Way\-Mutation\-Operator().

\hypertarget{classorg_1_1jgap_1_1_genotype_abe044a5473e935c2093ee3c6c18775f7}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!hash\-Code@{hash\-Code}}
\index{hash\-Code@{hash\-Code}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{hash\-Code}]{\setlength{\rightskip}{0pt plus 5cm}int org.\-jgap.\-Genotype.\-hash\-Code (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_abe044a5473e935c2093ee3c6c18775f7}


Definition at line 555 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-get\-Population().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_abe044a5473e935c2093ee3c6c18775f7_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_af5e224c5b7c05dbe02b3324d5ec16f2a}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!keep\-Pop\-Size\-Constant@{keep\-Pop\-Size\-Constant}}
\index{keep\-Pop\-Size\-Constant@{keep\-Pop\-Size\-Constant}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{keep\-Pop\-Size\-Constant}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Genotype.\-keep\-Pop\-Size\-Constant (
\begin{DoxyParamCaption}
\item[{{\bf Population}}]{a\-\_\-pop, }
\item[{int}]{a\-\_\-max\-Size}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1_genotype_af5e224c5b7c05dbe02b3324d5ec16f2a}
Cares that the population size does not exceed the given maximum size.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-pop} & the population to keep constant in size \\
\hline
{\em a\-\_\-max\-Size} & the maximum size allowed for the population\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-5 
\end{DoxySince}
\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000119}{Todo}]use Standard\-Post\-Selector instead? \end{DoxyRefDesc}


Definition at line 602 of file Genotype.\-java.



Referenced by org.\-jgap.\-Genotype.\-evolve().

\hypertarget{classorg_1_1jgap_1_1_genotype_aa2b0341edc420ad61e797508b8bf4dd5}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!merge\-Results@{merge\-Results}}
\index{merge\-Results@{merge\-Results}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{merge\-Results}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Genotype.\-merge\-Results (
\begin{DoxyParamCaption}
\item[{I\-Population\-Merger}]{a\-\_\-merger, }
\item[{Evolve\-Result\mbox{[}$\,$\mbox{]}}]{a\-\_\-results}
\end{DoxyParamCaption}
) throws Exception\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_aa2b0341edc420ad61e797508b8bf4dd5}
\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000120}{Todo}]use/enhance I\-Population\-Merger \end{DoxyRefDesc}


Definition at line 673 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-get\-Configuration(), and org.\-jgap.\-Genotype.\-set\-Population().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_aa2b0341edc420ad61e797508b8bf4dd5_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_a930184e2920b4642f20d0b8bc8412005}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!random\-Initial\-Genotype@{random\-Initial\-Genotype}}
\index{random\-Initial\-Genotype@{random\-Initial\-Genotype}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{random\-Initial\-Genotype}]{\setlength{\rightskip}{0pt plus 5cm}static {\bf Genotype} org.\-jgap.\-Genotype.\-random\-Initial\-Genotype (
\begin{DoxyParamCaption}
\item[{{\bf Configuration}}]{a\-\_\-configuration}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1_genotype_a930184e2920b4642f20d0b8bc8412005}
Convenience method that returns a newly constructed \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} instance configured according to the given \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} instance. The population of Chromosomes will be created according to the setup of the sample \hyperlink{classorg_1_1jgap_1_1_chromosome}{Chromosome} in the \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} object, but the gene values (alleles) will be set to random legal values.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-configuration} & the current active \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} object \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
a newly constructed \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} instance
\end{DoxyReturn}

\begin{DoxyExceptions}{Exceptions}
{\em Illegal\-Argument\-Exception} & if the given \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} object is null \\
\hline
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & if the given \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} instance is not in a valid state\\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Neil Rotstan 

Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-3 
\end{DoxySince}


Definition at line 321 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-Genotype().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_a930184e2920b4642f20d0b8bc8412005_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_a0d14c5400197e3c43f9d1f0666309a85}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!run@{run}}
\index{run@{run}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{run}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Genotype.\-run (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_a0d14c5400197e3c43f9d1f0666309a85}
If used in a Thread\-: runs the evolution forever. You have to implement a listener to stop computation sometime. See \hyperlink{namespaceexamples_1_1simple_boolean_threaded}{examples.\-simple\-Boolean\-Threaded} for a possible implementation of such a listener.

\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-01 
\end{DoxySince}


Definition at line 624 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-evolve(), org.\-jgap.\-Genotype.\-m\-\_\-monitor, and org.\-jgap.\-Genotype.\-m\-\_\-use\-Monitor.



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_a0d14c5400197e3c43f9d1f0666309a85_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_genotype_af3f49ba90cc8515817366792b7bce320}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!set\-Monitor@{set\-Monitor}}
\index{set\-Monitor@{set\-Monitor}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{set\-Monitor}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Genotype.\-set\-Monitor (
\begin{DoxyParamCaption}
\item[{I\-Evolution\-Monitor}]{a\-\_\-monitor}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_af3f49ba90cc8515817366792b7bce320}
Sets the evolution monitor to use, activate it via set\-Use\-Monitor(true). Only to be used when running \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} as a thread.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-monitor} & the I\-Evolution\-Monitor to use\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-6 
\end{DoxySince}


Definition at line 716 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-m\-\_\-monitor.

\hypertarget{classorg_1_1jgap_1_1_genotype_ac56a4d112334db07ba333602c4a24706}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!set\-Population@{set\-Population}}
\index{set\-Population@{set\-Population}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{set\-Population}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Genotype.\-set\-Population (
\begin{DoxyParamCaption}
\item[{{\bf Population}}]{a\-\_\-pop}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1_genotype_ac56a4d112334db07ba333602c4a24706}


Definition at line 570 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-m\-\_\-population.



Referenced by org.\-jgap.\-Genotype.\-apply\-Natural\-Selectors(), org.\-jgap.\-Genotype.\-evolve(), and org.\-jgap.\-Genotype.\-merge\-Results().

\hypertarget{classorg_1_1jgap_1_1_genotype_a2f44a7709eca83cddd6d794ee59e663c}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!set\-Static\-Configuration@{set\-Static\-Configuration}}
\index{set\-Static\-Configuration@{set\-Static\-Configuration}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{set\-Static\-Configuration}]{\setlength{\rightskip}{0pt plus 5cm}static void org.\-jgap.\-Genotype.\-set\-Static\-Configuration (
\begin{DoxyParamCaption}
\item[{{\bf Configuration}}]{a\-\_\-configuration}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1_genotype_a2f44a7709eca83cddd6d794ee59e663c}
Sets the configuration to use with the Genetic Algorithm. 
\begin{DoxyParams}{Parameters}
{\em a\-\_\-configuration} & the configuration to use\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-0 
\end{DoxySince}


Definition at line 531 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-m\-\_\-static\-Configuration.

\hypertarget{classorg_1_1jgap_1_1_genotype_abba6c68b3e8388be6884a35f5600e8cb}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!set\-Use\-Monitor@{set\-Use\-Monitor}}
\index{set\-Use\-Monitor@{set\-Use\-Monitor}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{set\-Use\-Monitor}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Genotype.\-set\-Use\-Monitor (
\begin{DoxyParamCaption}
\item[{boolean}]{a\-\_\-use\-Monitor}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_abba6c68b3e8388be6884a35f5600e8cb}
Use an evolution monitor, only to be used when running \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} as a thread. Otherwise use method \hyperlink{classorg_1_1jgap_1_1_genotype_aa740dd753d31620166979199a38d5a50}{evolve(\-I\-Evolution\-Monitor)}


\begin{DoxyParams}{Parameters}
{\em a\-\_\-use\-Monitor} & true\-: use evolution monitor, set it via set\-Monitor\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-6 
\end{DoxySince}


Definition at line 703 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-m\-\_\-use\-Monitor.

\hypertarget{classorg_1_1jgap_1_1_genotype_a658a541940e9be8ef474375091ff31ae}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!to\-String@{to\-String}}
\index{to\-String@{to\-String}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{to\-String}]{\setlength{\rightskip}{0pt plus 5cm}String org.\-jgap.\-Genotype.\-to\-String (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_genotype_a658a541940e9be8ef474375091ff31ae}
\begin{DoxyReturn}{Returns}
string representation of this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} instance, useful for display or debug purposes
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Neil Rotstan 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
1.\-0 
\end{DoxySince}


Definition at line 292 of file Genotype.\-java.



References org.\-jgap.\-Genotype.\-get\-Population().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_genotype_a658a541940e9be8ef474375091ff31ae_cgraph}
\end{center}
\end{figure}




\subsection{Member Data Documentation}
\hypertarget{classorg_1_1jgap_1_1_genotype_ac9f8e9d203b7a4efdf8c2e32a08f84bb}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N@{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N}}
\index{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N@{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N}]{\setlength{\rightskip}{0pt plus 5cm}final String org.\-jgap.\-Genotype.\-C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N = \char`\"{}\$Revision\-: 1.\-109 \$\char`\"{}\hspace{0.3cm}{\ttfamily [static]}, {\ttfamily [private]}}}\label{classorg_1_1jgap_1_1_genotype_ac9f8e9d203b7a4efdf8c2e32a08f84bb}
String containing the C\-V\-S revision. Read out via reflection! 

Definition at line 36 of file Genotype.\-java.

\hypertarget{classorg_1_1jgap_1_1_genotype_aeb3fac2610b8ef8fa8599fcac1b6a199}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!m\-\_\-active\-Configuration@{m\-\_\-active\-Configuration}}
\index{m\-\_\-active\-Configuration@{m\-\_\-active\-Configuration}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{m\-\_\-active\-Configuration}]{\setlength{\rightskip}{0pt plus 5cm}{\bf Configuration} org.\-jgap.\-Genotype.\-m\-\_\-active\-Configuration\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1_genotype_aeb3fac2610b8ef8fa8599fcac1b6a199}
The current \hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} instance. \begin{DoxySince}{Since}
1.\-0 
\end{DoxySince}


Definition at line 42 of file Genotype.\-java.



Referenced by org.\-jgap.\-Genotype.\-apply\-Natural\-Selectors(), org.\-jgap.\-Genotype.\-evolve(), org.\-jgap.\-Genotype.\-Genotype(), and org.\-jgap.\-Genotype.\-get\-Configuration().

\hypertarget{classorg_1_1jgap_1_1_genotype_acac1218542ab91833483ebd42ac8636b}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!m\-\_\-monitor@{m\-\_\-monitor}}
\index{m\-\_\-monitor@{m\-\_\-monitor}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{m\-\_\-monitor}]{\setlength{\rightskip}{0pt plus 5cm}I\-Evolution\-Monitor org.\-jgap.\-Genotype.\-m\-\_\-monitor\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1_genotype_acac1218542ab91833483ebd42ac8636b}


Definition at line 57 of file Genotype.\-java.



Referenced by org.\-jgap.\-Genotype.\-run(), and org.\-jgap.\-Genotype.\-set\-Monitor().

\hypertarget{classorg_1_1jgap_1_1_genotype_aa2ba4539110aaa8d853c808679f4f0e2}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!m\-\_\-population@{m\-\_\-population}}
\index{m\-\_\-population@{m\-\_\-population}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{m\-\_\-population}]{\setlength{\rightskip}{0pt plus 5cm}{\bf Population} org.\-jgap.\-Genotype.\-m\-\_\-population\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1_genotype_aa2ba4539110aaa8d853c808679f4f0e2}
The array of Chromosomes that make-\/up the \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype}'s population. \begin{DoxySince}{Since}
2.\-0 
\end{DoxySince}


Definition at line 50 of file Genotype.\-java.



Referenced by org.\-jgap.\-Genotype.\-Genotype(), org.\-jgap.\-Genotype.\-get\-Population(), and org.\-jgap.\-Genotype.\-set\-Population().

\hypertarget{classorg_1_1jgap_1_1_genotype_a8b93a0b2044e40520d4a966ef3ae0bbe}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!m\-\_\-static\-Configuration@{m\-\_\-static\-Configuration}}
\index{m\-\_\-static\-Configuration@{m\-\_\-static\-Configuration}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{m\-\_\-static\-Configuration}]{\setlength{\rightskip}{0pt plus 5cm}transient {\bf Configuration} org.\-jgap.\-Genotype.\-m\-\_\-static\-Configuration\hspace{0.3cm}{\ttfamily [static]}, {\ttfamily [private]}}}\label{classorg_1_1jgap_1_1_genotype_a8b93a0b2044e40520d4a966ef3ae0bbe}


Definition at line 44 of file Genotype.\-java.



Referenced by org.\-jgap.\-Genotype.\-get\-Static\-Configuration(), and org.\-jgap.\-Genotype.\-set\-Static\-Configuration().

\hypertarget{classorg_1_1jgap_1_1_genotype_ab1b1b976deab7f2faea463d1b8d52abe}{\index{org\-::jgap\-::\-Genotype@{org\-::jgap\-::\-Genotype}!m\-\_\-use\-Monitor@{m\-\_\-use\-Monitor}}
\index{m\-\_\-use\-Monitor@{m\-\_\-use\-Monitor}!org::jgap::Genotype@{org\-::jgap\-::\-Genotype}}
\subsubsection[{m\-\_\-use\-Monitor}]{\setlength{\rightskip}{0pt plus 5cm}boolean org.\-jgap.\-Genotype.\-m\-\_\-use\-Monitor\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1_genotype_ab1b1b976deab7f2faea463d1b8d52abe}
Use an enolution monitor \begin{DoxySince}{Since}
3.\-6 
\end{DoxySince}


Definition at line 55 of file Genotype.\-java.



Referenced by org.\-jgap.\-Genotype.\-run(), and org.\-jgap.\-Genotype.\-set\-Use\-Monitor().



The documentation for this class was generated from the following file\-:\begin{DoxyCompactItemize}
\item 
L\-D\-H\-\_\-\-Git/src/org/jgap/\hyperlink{_genotype_8java}{Genotype.\-java}\end{DoxyCompactItemize}
