\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{\section{org.\-jgap.\-gp.\-impl.\-G\-P\-Population Class Reference}
\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}\index{org.\-jgap.\-gp.\-impl.\-G\-P\-Population@{org.\-jgap.\-gp.\-impl.\-G\-P\-Population}}
}


Inheritance diagram for org.\-jgap.\-gp.\-impl.\-G\-P\-Population\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=233pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population__inherit__graph}
\end{center}
\end{figure}


Collaboration diagram for org.\-jgap.\-gp.\-impl.\-G\-P\-Population\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population__coll__graph}
\end{center}
\end{figure}
\subsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a2495394d8ede9583b5ac922b8149e37b}{G\-P\-Population} ()  throws Exception 
\item 
\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ad19b2b5571851230b86e6f1f0f74fe0b}{G\-P\-Population} (\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} a\-\_\-config, int a\-\_\-size)  throws Invalid\-Configuration\-Exception 
\item 
\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a69cc3bdcc9e18685c208485f0c3d0a52}{G\-P\-Population} (\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{G\-P\-Population} a\-\_\-pop)  throws Invalid\-Configuration\-Exception 
\item 
\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a4dd4478edeca976efc6f31bcdafa19f0}{G\-P\-Population} (\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{G\-P\-Population} a\-\_\-pop, boolean a\-\_\-keep\-Programs)  throws Invalid\-Configuration\-Exception 
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a9d49b2b18535658aae05ecdc2263918b}{sort} (Comparator c)
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_aeffef58a29ebb252060a7e2ed91227e2}{create} (Class\mbox{[}$\,$\mbox{]} a\-\_\-types, Class\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-arg\-Types, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-node\-Sets, int\mbox{[}$\,$\mbox{]} a\-\_\-min\-Depths, int\mbox{[}$\,$\mbox{]} a\-\_\-max\-Depths, int a\-\_\-max\-Nodes, boolean\mbox{[}$\,$\mbox{]} a\-\_\-full\-Mode\-Allowed)  throws Invalid\-Configuration\-Exception 
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a0f3f874207800be662835fcbc58092cf}{create} (Class\mbox{[}$\,$\mbox{]} a\-\_\-types, Class\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-arg\-Types, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-node\-Sets, int\mbox{[}$\,$\mbox{]} a\-\_\-min\-Depths, int\mbox{[}$\,$\mbox{]} a\-\_\-max\-Depths, int a\-\_\-max\-Nodes, boolean\mbox{[}$\,$\mbox{]} a\-\_\-full\-Mode\-Allowed, \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_program_creator}{I\-Program\-Creator} a\-\_\-program\-Creator)  throws Invalid\-Configuration\-Exception 
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a21670302d6d9c5ad4bdc2903797eabff}{create} (Class\mbox{[}$\,$\mbox{]} a\-\_\-types, Class\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-arg\-Types, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-node\-Sets, int\mbox{[}$\,$\mbox{]} a\-\_\-min\-Depths, int\mbox{[}$\,$\mbox{]} a\-\_\-max\-Depths, int a\-\_\-max\-Nodes, boolean\mbox{[}$\,$\mbox{]} a\-\_\-full\-Mode\-Allowed, \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_program_creator}{I\-Program\-Creator} a\-\_\-program\-Creator, int a\-\_\-offset)  throws Invalid\-Configuration\-Exception 
\item 
\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a3ba57ecc975cad55e2868567baf248e9}{create} (Class\mbox{[}$\,$\mbox{]} a\-\_\-types, Class\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-arg\-Types, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-node\-Sets, int\mbox{[}$\,$\mbox{]} a\-\_\-min\-Depths, int\mbox{[}$\,$\mbox{]} a\-\_\-max\-Depths, int a\-\_\-depth, boolean a\-\_\-grow, int a\-\_\-max\-Nodes, boolean\mbox{[}$\,$\mbox{]} a\-\_\-full\-Mode\-Allowed, int a\-\_\-tries)  throws Invalid\-Configuration\-Exception 
\item 
\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a4d68eefb94b971473cb1487ecf9f92a0}{create} (int a\-\_\-program\-Index, Class\mbox{[}$\,$\mbox{]} a\-\_\-types, Class\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-arg\-Types, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-node\-Sets, int\mbox{[}$\,$\mbox{]} a\-\_\-min\-Depths, int\mbox{[}$\,$\mbox{]} a\-\_\-max\-Depths, int a\-\_\-depth, boolean a\-\_\-grow, int a\-\_\-max\-Nodes, boolean\mbox{[}$\,$\mbox{]} a\-\_\-full\-Mode\-Allowed, int a\-\_\-tries)  throws Invalid\-Configuration\-Exception 
\item 
\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ada6ec95d4f4e1b59e0270d6c00b92664}{create} (int a\-\_\-program\-Index, Class\mbox{[}$\,$\mbox{]} a\-\_\-types, Class\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-arg\-Types, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-node\-Sets, int\mbox{[}$\,$\mbox{]} a\-\_\-min\-Depths, int\mbox{[}$\,$\mbox{]} a\-\_\-max\-Depths, int a\-\_\-depth, boolean a\-\_\-grow, int a\-\_\-max\-Nodes, boolean\mbox{[}$\,$\mbox{]} a\-\_\-full\-Mode\-Allowed, int a\-\_\-tries, \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_program_creator}{I\-Program\-Creator} a\-\_\-program\-Creator)  throws Invalid\-Configuration\-Exception 
\item 
int \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a22d30c1d8d7bf5e1340fb6bf6bf36af5}{get\-Pop\-Size} ()
\item 
\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a6c4ab9ac1358c54cce879696e56376df}{get\-G\-P\-Configuration} ()
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a11dce280ef2ec6ff8a7e1078500feda0}{set\-G\-P\-Program} (final int a\-\_\-index, final \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} a\-\_\-program)
\item 
\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a85692cb78096a06ba3446f4847d0fd1e}{get\-G\-P\-Program} (int a\-\_\-index)
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a2e36a0da99ef591a5980b0f1eef83b31}{set\-G\-P\-Programs} (final \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{G\-P\-Population} a\-\_\-pop)
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a8d9d5bb131077ec72be3a0137716775e}{set\-G\-P\-Programs} (final \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program}\mbox{[}$\,$\mbox{]} a\-\_\-progs)
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a81449bc4b63442f9296c2f65508a155a}{copy\-G\-P\-Programs} (final \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{G\-P\-Population} a\-\_\-pop)
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a45a7caf3ae473c7f75979a76526cb67a}{set\-G\-P\-Programs} (final List a\-\_\-pop)
\item 
\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program}\mbox{[}$\,$\mbox{]} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a6e608302ac1da22597e9b5dbc2c0faae}{get\-G\-P\-Programs} ()
\item 
int \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a73feb5c387bfb23ae12620126e392ebc}{size} ()
\item 
\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a6daa11823812dd94682ae3a590dd38f1}{determine\-Fittest\-Program} ()
\item 
\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a4086d0d89287220377311d23251e10ba}{determine\-Fittest\-Program\-Computed} ()
\item 
List \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_afb008e4b0e98c45736c53eb82021f94c}{determine\-Fittest\-Chromosomes} (final int a\-\_\-number\-Of\-Programs)
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a0d7b58819ffefbefbc0baaa1d12a9cb6}{sort\-By\-Fitness} ()
\item 
float\mbox{[}$\,$\mbox{]} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_aa780fcbd721a87a70b46abbfab096e3c}{get\-Fitness\-Ranks} ()
\item 
float \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_add444d666bfb69e116878039b634a4c1}{get\-Fitness\-Rank} (int a\-\_\-index)
\item 
boolean \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a1d9efd7c3d81d1e70168ef872e587142}{is\-Changed} ()
\item 
int \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a0e749981f33246815f8124532e3136c6}{compare\-To} (Object a\-\_\-pop)
\item 
boolean \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a45ced80a38dc1904286b4eb3ff52df96}{equals} (Object a\-\_\-pop)
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a28f3cef918cd1f82d3613d93f6e0e4e9}{add\-Fittest\-Program} (final \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} a\-\_\-to\-Add)
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a815c413fb2bdd4b2b9c404868ee0426e}{clear} ()
\item 
boolean \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_aae705ed824ebf7c7bf152f7998158239}{is\-First\-Empty} ()
\item 
String \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a9cdf68c27b023404e7c51b02ef2ec756}{get\-Persistent\-Representation} ()
\end{DoxyCompactItemize}
\subsection*{Static Public Attributes}
\begin{DoxyCompactItemize}
\item 
static final double \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ab8c06531dff207ecdff0c86b3d36651e}{D\-E\-L\-T\-A} = 0.\-0000001d
\end{DoxyCompactItemize}
\subsection*{Protected Member Functions}
\begin{DoxyCompactItemize}
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a77008e555c1d551716a52eb849694a7b}{set\-Changed} (final boolean a\-\_\-changed)
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a863f5c2070d60fbf599fcf49230baec1}{set\-Sorted} (final boolean a\-\_\-sorted)
\item 
boolean \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ac6941b5054925d718727279f7e67416a}{contained\-In\-Array} (\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program}\mbox{[}$\,$\mbox{]} a\-\_\-progs, \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} a\-\_\-prog)
\item 
String \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ac97f9a8c11750bed24a8cfc486b7f44a}{encode} (String a\-\_\-string)
\item 
String \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a0a0818d5aae810ab518784a75743933a}{decode} (String a\-\_\-string)
\end{DoxyCompactItemize}
\subsection*{Private Attributes}
\begin{DoxyCompactItemize}
\item 
transient Logger \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a2e511782d083ba5853395ac9cb59b95a}{L\-O\-G\-G\-E\-R} = Logger.\-get\-Logger(G\-P\-Population.\-class)
\item 
transient int \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_aeb4067371d53afb6f4e0a4939725dd36}{warning\-Prototype\-Reused} = 0
\item 
\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program}\mbox{[}$\,$\mbox{]} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a3bf2729a02a6d7daa8d9badb52cd6c3a}{m\-\_\-programs}
\item 
float\mbox{[}$\,$\mbox{]} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a7f43165486ff8fd8d69f6c9ad7484823}{m\-\_\-fitness\-Rank}
\item 
int \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a82de82c54ab2617c745e5493ed025e00}{m\-\_\-pop\-Size}
\item 
transient \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a13678da0768ccaf2dc3b5a892c434cb7}{m\-\_\-fittest\-Program}
\item 
\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a54cd610adaf8ce845d6e9ac796f682dc}{m\-\_\-config}
\item 
boolean \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a4454c7e30fa80ed50d68c9e06b7f6897}{m\-\_\-changed}
\item 
boolean \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a522f220885c90a850191b89116bee75f}{m\-\_\-sorted}
\item 
\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_afeeb7ca5a1210d473c18fdfe84844811}{m\-\_\-fittest\-To\-Add}
\end{DoxyCompactItemize}
\subsection*{Static Private Attributes}
\begin{DoxyCompactItemize}
\item 
static final String \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ab209b6a7ffefa14c8d2f133e34f4d7e5}{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N} = \char`\"{}\$Revision\-: 1.\-43 \$\char`\"{}
\end{DoxyCompactItemize}


\subsection{Detailed Description}
\hyperlink{classorg_1_1jgap_1_1_population}{Population} for G\-P programs.

\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 26 of file G\-P\-Population.\-java.



\subsection{Constructor \& Destructor Documentation}
\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a2495394d8ede9583b5ac922b8149e37b}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!G\-P\-Population@{G\-P\-Population}}
\index{G\-P\-Population@{G\-P\-Population}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{G\-P\-Population}]{\setlength{\rightskip}{0pt plus 5cm}org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-G\-P\-Population (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
) throws Exception\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a2495394d8ede9583b5ac922b8149e37b}
Default constructor, only for dynamic instantiation.


\begin{DoxyExceptions}{Exceptions}
{\em Exception} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-3.\-4 
\end{DoxySince}


Definition at line 78 of file G\-P\-Population.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-compare\-To().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ad19b2b5571851230b86e6f1f0f74fe0b}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!G\-P\-Population@{G\-P\-Population}}
\index{G\-P\-Population@{G\-P\-Population}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{G\-P\-Population}]{\setlength{\rightskip}{0pt plus 5cm}org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-G\-P\-Population (
\begin{DoxyParamCaption}
\item[{{\bf G\-P\-Configuration}}]{a\-\_\-config, }
\item[{int}]{a\-\_\-size}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ad19b2b5571851230b86e6f1f0f74fe0b}


Definition at line 89 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-config, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-fitness\-Rank, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-pop\-Size, and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a69cc3bdcc9e18685c208485f0c3d0a52}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!G\-P\-Population@{G\-P\-Population}}
\index{G\-P\-Population@{G\-P\-Population}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{G\-P\-Population}]{\setlength{\rightskip}{0pt plus 5cm}org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-G\-P\-Population (
\begin{DoxyParamCaption}
\item[{{\bf G\-P\-Population}}]{a\-\_\-pop}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a69cc3bdcc9e18685c208485f0c3d0a52}


Definition at line 116 of file G\-P\-Population.\-java.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a4dd4478edeca976efc6f31bcdafa19f0}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!G\-P\-Population@{G\-P\-Population}}
\index{G\-P\-Population@{G\-P\-Population}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{G\-P\-Population}]{\setlength{\rightskip}{0pt plus 5cm}org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-G\-P\-Population (
\begin{DoxyParamCaption}
\item[{{\bf G\-P\-Population}}]{a\-\_\-pop, }
\item[{boolean}]{a\-\_\-keep\-Programs}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a4dd4478edeca976efc6f31bcdafa19f0}

\begin{DoxyParams}{Parameters}
{\em a\-\_\-pop} & the population to retrieve the parameters from \\
\hline
{\em a\-\_\-keep\-Programs} & true copy programs of given population to this one \\
\hline
\end{DoxyParams}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}


Definition at line 129 of file G\-P\-Population.\-java.



References org.\-jgap.\-util.\-I\-Cloneable.\-clone(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-changed, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-config, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-fitness\-Rank, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-fittest\-Program, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-pop\-Size, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-sorted, and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Changed().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a4dd4478edeca976efc6f31bcdafa19f0_cgraph}
\end{center}
\end{figure}




\subsection{Member Function Documentation}
\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a28f3cef918cd1f82d3613d93f6e0e4e9}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!add\-Fittest\-Program@{add\-Fittest\-Program}}
\index{add\-Fittest\-Program@{add\-Fittest\-Program}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{add\-Fittest\-Program}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-add\-Fittest\-Program (
\begin{DoxyParamCaption}
\item[{final {\bf I\-G\-P\-Program}}]{a\-\_\-to\-Add}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a28f3cef918cd1f82d3613d93f6e0e4e9}
Adds a G\-P program to this \hyperlink{classorg_1_1jgap_1_1_population}{Population}. Does nothing when given null. The injection is actually executed in method create(..)


\begin{DoxyParams}{Parameters}
{\em a\-\_\-to\-Add} & the program to add\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2 
\end{DoxySince}


Definition at line 907 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-fittest\-To\-Add.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a815c413fb2bdd4b2b9c404868ee0426e}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!clear@{clear}}
\index{clear@{clear}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{clear}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-clear (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a815c413fb2bdd4b2b9c404868ee0426e}
Clears the list of programs. Normally, this should not be necessary. But especially in distributed computing, a fresh population has to be provided sometimes.

\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2 
\end{DoxySince}


Definition at line 921 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-changed, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-fittest\-Program, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs, and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-sorted.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a0e749981f33246815f8124532e3136c6}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!compare\-To@{compare\-To}}
\index{compare\-To@{compare\-To}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{compare\-To}]{\setlength{\rightskip}{0pt plus 5cm}int org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-compare\-To (
\begin{DoxyParamCaption}
\item[{Object}]{a\-\_\-pop}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a0e749981f33246815f8124532e3136c6}
This method is not producing symmetric results as -\/1 is more often returned than 1 (see description of return value).


\begin{DoxyParams}{Parameters}
{\em a\-\_\-pop} & the other population to compare \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
1\-: a\-\_\-pop is null or having fewer programs or equal number of programs but at least one not contained. 0\-: both populations containing exactly the same programs. -\/1\-: this population contains fewer programs than a\-\_\-pop
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-6 
\end{DoxySince}


Definition at line 831 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-contained\-In\-Array(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-G\-P\-Population(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs, and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-size().



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-equals().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a0e749981f33246815f8124532e3136c6_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ac6941b5054925d718727279f7e67416a}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!contained\-In\-Array@{contained\-In\-Array}}
\index{contained\-In\-Array@{contained\-In\-Array}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{contained\-In\-Array}]{\setlength{\rightskip}{0pt plus 5cm}boolean org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-contained\-In\-Array (
\begin{DoxyParamCaption}
\item[{{\bf I\-G\-P\-Program}\mbox{[}$\,$\mbox{]}}]{a\-\_\-progs, }
\item[{{\bf I\-G\-P\-Program}}]{a\-\_\-prog}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ac6941b5054925d718727279f7e67416a}
Checks if a program is contained within an array of programs. Assumes that in the array no element will follow after the first null element.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-progs} & the array to search thru \\
\hline
{\em a\-\_\-prog} & the program to find \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
true\-: program found in array via equals-\/method
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 866 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-equals().



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-compare\-To().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ac6941b5054925d718727279f7e67416a_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a81449bc4b63442f9296c2f65508a155a}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!copy\-G\-P\-Programs@{copy\-G\-P\-Programs}}
\index{copy\-G\-P\-Programs@{copy\-G\-P\-Programs}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{copy\-G\-P\-Programs}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-copy\-G\-P\-Programs (
\begin{DoxyParamCaption}
\item[{final {\bf G\-P\-Population}}]{a\-\_\-pop}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a81449bc4b63442f9296c2f65508a155a}
Sets the G\-P\-Programs of the given population to this population.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-pop} & the population to use as template\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-3.\-3 
\end{DoxySince}


Definition at line 610 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Changed(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-size().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a81449bc4b63442f9296c2f65508a155a_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_aeffef58a29ebb252060a7e2ed91227e2}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!create@{create}}
\index{create@{create}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{create}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create (
\begin{DoxyParamCaption}
\item[{Class\mbox{[}$\,$\mbox{]}}]{a\-\_\-types, }
\item[{Class}]{a\-\_\-arg\-Types\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{{\bf Command\-Gene}}]{a\-\_\-node\-Sets\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-min\-Depths, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-max\-Depths, }
\item[{int}]{a\-\_\-max\-Nodes, }
\item[{boolean\mbox{[}$\,$\mbox{]}}]{a\-\_\-full\-Mode\-Allowed}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_aeffef58a29ebb252060a7e2ed91227e2}
Creates a population.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-types} & the type for each chromosome, the length of the array represents the number of chromosomes \\
\hline
{\em a\-\_\-arg\-Types} & the types of the arguments to each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of arguments to the chromosome \\
\hline
{\em a\-\_\-node\-Sets} & the nodes which are allowed to be used by each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of nodes \\
\hline
{\em a\-\_\-min\-Depths} & contains the minimum depth allowed for each chromosome \\
\hline
{\em a\-\_\-max\-Depths} & contains the maximum depth allowed for each chromosome \\
\hline
{\em a\-\_\-max\-Nodes} & reserve space for a\-\_\-max\-Nodes number of nodes \\
\hline
{\em a\-\_\-full\-Mode\-Allowed} & array of boolean values. For each chromosome there is one value indicating whether the full mode for creating chromosome generations during evolution is allowed (true) or not (false) \\
\hline
\end{DoxyParams}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 205 of file G\-P\-Population.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a0f3f874207800be662835fcbc58092cf}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!create@{create}}
\index{create@{create}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{create}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create (
\begin{DoxyParamCaption}
\item[{Class\mbox{[}$\,$\mbox{]}}]{a\-\_\-types, }
\item[{Class}]{a\-\_\-arg\-Types\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{{\bf Command\-Gene}}]{a\-\_\-node\-Sets\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-min\-Depths, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-max\-Depths, }
\item[{int}]{a\-\_\-max\-Nodes, }
\item[{boolean\mbox{[}$\,$\mbox{]}}]{a\-\_\-full\-Mode\-Allowed, }
\item[{{\bf I\-Program\-Creator}}]{a\-\_\-program\-Creator}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a0f3f874207800be662835fcbc58092cf}
Creates a population.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-types} & the type for each chromosome, the length of the array represents the number of chromosomes \\
\hline
{\em a\-\_\-arg\-Types} & the types of the arguments to each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of arguments to the chromosome \\
\hline
{\em a\-\_\-node\-Sets} & the nodes which are allowed to be used by each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of nodes \\
\hline
{\em a\-\_\-min\-Depths} & contains the minimum depth allowed for each chromosome \\
\hline
{\em a\-\_\-max\-Depths} & contains the maximum depth allowed for each chromosome \\
\hline
{\em a\-\_\-max\-Nodes} & reserve space for a\-\_\-max\-Nodes number of nodes \\
\hline
{\em a\-\_\-full\-Mode\-Allowed} & array of boolean values. For each chromosome there is one value indicating whether the full mode for creating chromosome generations during evolution is allowed (true) or not (false) \\
\hline
{\em a\-\_\-program\-Creator} & service to create new programs with \\
\hline
\end{DoxyParams}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2.\-2 
\end{DoxySince}


Definition at line 238 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a0f3f874207800be662835fcbc58092cf_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a21670302d6d9c5ad4bdc2903797eabff}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!create@{create}}
\index{create@{create}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{create}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create (
\begin{DoxyParamCaption}
\item[{Class\mbox{[}$\,$\mbox{]}}]{a\-\_\-types, }
\item[{Class}]{a\-\_\-arg\-Types\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{{\bf Command\-Gene}}]{a\-\_\-node\-Sets\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-min\-Depths, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-max\-Depths, }
\item[{int}]{a\-\_\-max\-Nodes, }
\item[{boolean\mbox{[}$\,$\mbox{]}}]{a\-\_\-full\-Mode\-Allowed, }
\item[{{\bf I\-Program\-Creator}}]{a\-\_\-program\-Creator, }
\item[{int}]{a\-\_\-offset}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a21670302d6d9c5ad4bdc2903797eabff}
Creates a population.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-types} & the type for each chromosome, the length of the array represents the number of chromosomes \\
\hline
{\em a\-\_\-arg\-Types} & the types of the arguments to each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of arguments to the chromosome \\
\hline
{\em a\-\_\-node\-Sets} & the nodes which are allowed to be used by each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of nodes \\
\hline
{\em a\-\_\-min\-Depths} & contains the minimum depth allowed for each chromosome \\
\hline
{\em a\-\_\-max\-Depths} & contains the maximum depth allowed for each chromosome \\
\hline
{\em a\-\_\-max\-Nodes} & reserve space for a\-\_\-max\-Nodes number of nodes \\
\hline
{\em a\-\_\-full\-Mode\-Allowed} & array of boolean values. For each chromosome there is one value indicating whether the full mode for creating chromosome generations during evolution is allowed (true) or not (false) \\
\hline
{\em a\-\_\-program\-Creator} & service to create new programs with \\
\hline
{\em a\-\_\-offset} & start index for new programs to put into the configuration\\
\hline
\end{DoxyParams}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-3.\-3 
\end{DoxySince}
\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000155}{Todo}]add element of randomness \end{DoxyRefDesc}


\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000156}{Todo}]5\-: make configurable \end{DoxyRefDesc}


Definition at line 274 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-G\-P\-Configuration(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-pop\-Size, org.\-jgap.\-I\-Handler.\-perform(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Changed(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-G\-P\-Program(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-warning\-Prototype\-Reused.



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a21670302d6d9c5ad4bdc2903797eabff_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a3ba57ecc975cad55e2868567baf248e9}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!create@{create}}
\index{create@{create}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{create}]{\setlength{\rightskip}{0pt plus 5cm}{\bf I\-G\-P\-Program} org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create (
\begin{DoxyParamCaption}
\item[{Class\mbox{[}$\,$\mbox{]}}]{a\-\_\-types, }
\item[{Class}]{a\-\_\-arg\-Types\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{{\bf Command\-Gene}}]{a\-\_\-node\-Sets\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-min\-Depths, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-max\-Depths, }
\item[{int}]{a\-\_\-depth, }
\item[{boolean}]{a\-\_\-grow, }
\item[{int}]{a\-\_\-max\-Nodes, }
\item[{boolean\mbox{[}$\,$\mbox{]}}]{a\-\_\-full\-Mode\-Allowed, }
\item[{int}]{a\-\_\-tries}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a3ba57ecc975cad55e2868567baf248e9}
Creates a population.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-types} & Class\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-arg\-Types} & Class\mbox{[}\mbox{]}\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-node\-Sets} & \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}\mbox{]}\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-min\-Depths} & int\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-max\-Depths} & int\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-depth} & int \\
\hline
{\em a\-\_\-grow} & boolean \\
\hline
{\em a\-\_\-max\-Nodes} & int \\
\hline
{\em a\-\_\-full\-Mode\-Allowed} & boolean\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-tries} & int \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} 
\end{DoxyReturn}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2.\-2 
\end{DoxySince}


Definition at line 413 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a3ba57ecc975cad55e2868567baf248e9_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a4d68eefb94b971473cb1487ecf9f92a0}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!create@{create}}
\index{create@{create}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{create}]{\setlength{\rightskip}{0pt plus 5cm}{\bf I\-G\-P\-Program} org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create (
\begin{DoxyParamCaption}
\item[{int}]{a\-\_\-program\-Index, }
\item[{Class\mbox{[}$\,$\mbox{]}}]{a\-\_\-types, }
\item[{Class}]{a\-\_\-arg\-Types\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{{\bf Command\-Gene}}]{a\-\_\-node\-Sets\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-min\-Depths, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-max\-Depths, }
\item[{int}]{a\-\_\-depth, }
\item[{boolean}]{a\-\_\-grow, }
\item[{int}]{a\-\_\-max\-Nodes, }
\item[{boolean\mbox{[}$\,$\mbox{]}}]{a\-\_\-full\-Mode\-Allowed, }
\item[{int}]{a\-\_\-tries}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a4d68eefb94b971473cb1487ecf9f92a0}

\begin{DoxyParams}{Parameters}
{\em a\-\_\-program\-Index} & int \\
\hline
{\em a\-\_\-types} & Class\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-arg\-Types} & Class\mbox{[}\mbox{]}\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-node\-Sets} & \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}\mbox{]}\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-min\-Depths} & int\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-max\-Depths} & int\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-depth} & int \\
\hline
{\em a\-\_\-grow} & boolean \\
\hline
{\em a\-\_\-max\-Nodes} & int \\
\hline
{\em a\-\_\-full\-Mode\-Allowed} & boolean\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-tries} & int \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} 
\end{DoxyReturn}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-3 
\end{DoxySince}


Definition at line 442 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a4d68eefb94b971473cb1487ecf9f92a0_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ada6ec95d4f4e1b59e0270d6c00b92664}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!create@{create}}
\index{create@{create}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{create}]{\setlength{\rightskip}{0pt plus 5cm}{\bf I\-G\-P\-Program} org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create (
\begin{DoxyParamCaption}
\item[{int}]{a\-\_\-program\-Index, }
\item[{Class\mbox{[}$\,$\mbox{]}}]{a\-\_\-types, }
\item[{Class}]{a\-\_\-arg\-Types\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{{\bf Command\-Gene}}]{a\-\_\-node\-Sets\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-min\-Depths, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-max\-Depths, }
\item[{int}]{a\-\_\-depth, }
\item[{boolean}]{a\-\_\-grow, }
\item[{int}]{a\-\_\-max\-Nodes, }
\item[{boolean\mbox{[}$\,$\mbox{]}}]{a\-\_\-full\-Mode\-Allowed, }
\item[{int}]{a\-\_\-tries, }
\item[{{\bf I\-Program\-Creator}}]{a\-\_\-program\-Creator}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ada6ec95d4f4e1b59e0270d6c00b92664}
Creates a valid \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program}. No fitness computation is initiated here!


\begin{DoxyParams}{Parameters}
{\em a\-\_\-program\-Index} & index of the program in the population \\
\hline
{\em a\-\_\-types} & the type of each chromosome, the length is the number of chromosomes \\
\hline
{\em a\-\_\-arg\-Types} & the types of the arguments to each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of arguments to the chromosome \\
\hline
{\em a\-\_\-node\-Sets} & the nodes which are allowed to be used by each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of nodes \\
\hline
{\em a\-\_\-min\-Depths} & contains the minimum depth allowed for each chromosome \\
\hline
{\em a\-\_\-max\-Depths} & contains the maximum depth allowed for each chromosome \\
\hline
{\em a\-\_\-depth} & the maximum depth of the program to create \\
\hline
{\em a\-\_\-grow} & true\-: grow mode, false\-: full mode \\
\hline
{\em a\-\_\-max\-Nodes} & reserve space for a\-\_\-max\-Nodes number of nodes \\
\hline
{\em a\-\_\-full\-Mode\-Allowed} & array of boolean values. For each chromosome there is one value indicating whether the full mode for creating chromosomes during evolution is allowed (true) or not (false) \\
\hline
{\em a\-\_\-tries} & maximum number of tries to get a valid program \\
\hline
{\em a\-\_\-program\-Creator} & strategy class to create programs for the population\\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
valid program
\end{DoxyReturn}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 486 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-G\-P\-Configuration(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-fittest\-To\-Add, and org.\-jgap.\-I\-Handler.\-perform().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ada6ec95d4f4e1b59e0270d6c00b92664_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a0a0818d5aae810ab518784a75743933a}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!decode@{decode}}
\index{decode@{decode}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{decode}]{\setlength{\rightskip}{0pt plus 5cm}String org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-decode (
\begin{DoxyParamCaption}
\item[{String}]{a\-\_\-string}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a0a0818d5aae810ab518784a75743933a}


Definition at line 964 of file G\-P\-Population.\-java.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_afb008e4b0e98c45736c53eb82021f94c}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!determine\-Fittest\-Chromosomes@{determine\-Fittest\-Chromosomes}}
\index{determine\-Fittest\-Chromosomes@{determine\-Fittest\-Chromosomes}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{determine\-Fittest\-Chromosomes}]{\setlength{\rightskip}{0pt plus 5cm}List org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-determine\-Fittest\-Chromosomes (
\begin{DoxyParamCaption}
\item[{final int}]{a\-\_\-number\-Of\-Programs}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_afb008e4b0e98c45736c53eb82021f94c}
Sorts the G\-P\-Programs list and returns the fittest n G\-P\-Programs in the population.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-number\-Of\-Programs} & number of top performer G\-P\-Programs to be returned \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
list of the fittest n G\-P\-Programs of the population, or the fittest x G\-P\-Programs with x = number of G\-P\-Programs in case n $>$ x.
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 743 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-changed, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-sorted, and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-sort\-By\-Fitness().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_afb008e4b0e98c45736c53eb82021f94c_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a6daa11823812dd94682ae3a590dd38f1}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!determine\-Fittest\-Program@{determine\-Fittest\-Program}}
\index{determine\-Fittest\-Program@{determine\-Fittest\-Program}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{determine\-Fittest\-Program}]{\setlength{\rightskip}{0pt plus 5cm}{\bf I\-G\-P\-Program} org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-determine\-Fittest\-Program (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a6daa11823812dd94682ae3a590dd38f1}
Determines the fittest \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program}{G\-P\-Program} in the population (the one with the highest fitness value) and memorizes it. This is an optimized version compared to calling determine\-Fittes\-Programs(1).

\begin{DoxyReturn}{Returns}
the fittest \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program}{G\-P\-Program} of the population
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 657 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-D\-E\-L\-T\-A, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-G\-P\-Configuration(), org.\-jgap.\-gp.\-I\-G\-P\-Fitness\-Evaluator.\-is\-Fitter(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-changed, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-fittest\-Program, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs, org.\-jgap.\-I\-Handler.\-perform(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Changed().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a6daa11823812dd94682ae3a590dd38f1_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a4086d0d89287220377311d23251e10ba}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!determine\-Fittest\-Program\-Computed@{determine\-Fittest\-Program\-Computed}}
\index{determine\-Fittest\-Program\-Computed@{determine\-Fittest\-Program\-Computed}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{determine\-Fittest\-Program\-Computed}]{\setlength{\rightskip}{0pt plus 5cm}{\bf I\-G\-P\-Program} org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-determine\-Fittest\-Program\-Computed (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a4086d0d89287220377311d23251e10ba}
Determines the fittest \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program}{G\-P\-Program} in the population, but only considers programs with already computed fitness value.

\begin{DoxyReturn}{Returns}
the fittest \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program}{G\-P\-Program} of the population
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2 
\end{DoxySince}


Definition at line 706 of file G\-P\-Population.\-java.



References org.\-jgap.\-Fitness\-Function.\-D\-E\-L\-T\-A, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-G\-P\-Configuration(), org.\-jgap.\-gp.\-I\-G\-P\-Fitness\-Evaluator.\-is\-Fitter(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs, and org.\-jgap.\-Fitness\-Function.\-N\-O\-\_\-\-F\-I\-T\-N\-E\-S\-S\-\_\-\-V\-A\-L\-U\-E.



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a4086d0d89287220377311d23251e10ba_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ac97f9a8c11750bed24a8cfc486b7f44a}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!encode@{encode}}
\index{encode@{encode}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{encode}]{\setlength{\rightskip}{0pt plus 5cm}String org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-encode (
\begin{DoxyParamCaption}
\item[{String}]{a\-\_\-string}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ac97f9a8c11750bed24a8cfc486b7f44a}


Definition at line 960 of file G\-P\-Population.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-Persistent\-Representation().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a45ced80a38dc1904286b4eb3ff52df96}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!equals@{equals}}
\index{equals@{equals}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{equals}]{\setlength{\rightskip}{0pt plus 5cm}boolean org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-equals (
\begin{DoxyParamCaption}
\item[{Object}]{a\-\_\-pop}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a45ced80a38dc1904286b4eb3ff52df96}
The equals-\/method.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-pop} & the population instance to compare with \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
true\-: given object equal to comparing one
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 887 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-compare\-To().



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-contained\-In\-Array().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a45ced80a38dc1904286b4eb3ff52df96_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_add444d666bfb69e116878039b634a4c1}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!get\-Fitness\-Rank@{get\-Fitness\-Rank}}
\index{get\-Fitness\-Rank@{get\-Fitness\-Rank}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{get\-Fitness\-Rank}]{\setlength{\rightskip}{0pt plus 5cm}float org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-Fitness\-Rank (
\begin{DoxyParamCaption}
\item[{int}]{a\-\_\-index}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_add444d666bfb69e116878039b634a4c1}


Definition at line 779 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-fitness\-Rank.



Referenced by org.\-jgap.\-gp.\-impl.\-Fitness\-Proportionate\-Selection.\-select().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_aa780fcbd721a87a70b46abbfab096e3c}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!get\-Fitness\-Ranks@{get\-Fitness\-Ranks}}
\index{get\-Fitness\-Ranks@{get\-Fitness\-Ranks}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{get\-Fitness\-Ranks}]{\setlength{\rightskip}{0pt plus 5cm}float \mbox{[}$\,$\mbox{]} org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-Fitness\-Ranks (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_aa780fcbd721a87a70b46abbfab096e3c}


Definition at line 775 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-fitness\-Rank.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a6c4ab9ac1358c54cce879696e56376df}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!get\-G\-P\-Configuration@{get\-G\-P\-Configuration}}
\index{get\-G\-P\-Configuration@{get\-G\-P\-Configuration}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{get\-G\-P\-Configuration}]{\setlength{\rightskip}{0pt plus 5cm}{\bf G\-P\-Configuration} org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-G\-P\-Configuration (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a6c4ab9ac1358c54cce879696e56376df}
\begin{DoxyReturn}{Returns}
the \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} set
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 544 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-config.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Pop(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-determine\-Fittest\-Program(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-determine\-Fittest\-Program\-Computed(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-sort\-By\-Fitness().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a85692cb78096a06ba3446f4847d0fd1e}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!get\-G\-P\-Program@{get\-G\-P\-Program}}
\index{get\-G\-P\-Program@{get\-G\-P\-Program}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{get\-G\-P\-Program}]{\setlength{\rightskip}{0pt plus 5cm}{\bf I\-G\-P\-Program} org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-G\-P\-Program (
\begin{DoxyParamCaption}
\item[{int}]{a\-\_\-index}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a85692cb78096a06ba3446f4847d0fd1e}


Definition at line 566 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-calc\-Fitness().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a6e608302ac1da22597e9b5dbc2c0faae}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!get\-G\-P\-Programs@{get\-G\-P\-Programs}}
\index{get\-G\-P\-Programs@{get\-G\-P\-Programs}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{get\-G\-P\-Programs}]{\setlength{\rightskip}{0pt plus 5cm}{\bf I\-G\-P\-Program} \mbox{[}$\,$\mbox{]} org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-G\-P\-Programs (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a6e608302ac1da22597e9b5dbc2c0faae}


Definition at line 639 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a9cdf68c27b023404e7c51b02ef2ec756}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!get\-Persistent\-Representation@{get\-Persistent\-Representation}}
\index{get\-Persistent\-Representation@{get\-Persistent\-Representation}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{get\-Persistent\-Representation}]{\setlength{\rightskip}{0pt plus 5cm}String org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-Persistent\-Representation (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a9cdf68c27b023404e7c51b02ef2ec756}
\begin{DoxyReturn}{Returns}
the persistent representation of the population, including all G\-P programs
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2.\-3 
\end{DoxySince}


Definition at line 947 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-encode(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs.



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a9cdf68c27b023404e7c51b02ef2ec756_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a22d30c1d8d7bf5e1340fb6bf6bf36af5}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!get\-Pop\-Size@{get\-Pop\-Size}}
\index{get\-Pop\-Size@{get\-Pop\-Size}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{get\-Pop\-Size}]{\setlength{\rightskip}{0pt plus 5cm}int org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-Pop\-Size (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a22d30c1d8d7bf5e1340fb6bf6bf36af5}
\begin{DoxyReturn}{Returns}
fixed size of the population
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 534 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-pop\-Size.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population\-Test.\-test\-Construct\-\_\-0().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a1d9efd7c3d81d1e70168ef872e587142}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!is\-Changed@{is\-Changed}}
\index{is\-Changed@{is\-Changed}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{is\-Changed}]{\setlength{\rightskip}{0pt plus 5cm}boolean org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-is\-Changed (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a1d9efd7c3d81d1e70168ef872e587142}
\begin{DoxyReturn}{Returns}
true\-: population's programs (maybe) were changed, false\-: not changed for sure
\end{DoxyReturn}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 803 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-changed.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_aae705ed824ebf7c7bf152f7998158239}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!is\-First\-Empty@{is\-First\-Empty}}
\index{is\-First\-Empty@{is\-First\-Empty}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{is\-First\-Empty}]{\setlength{\rightskip}{0pt plus 5cm}boolean org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-is\-First\-Empty (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_aae705ed824ebf7c7bf152f7998158239}


Definition at line 930 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs, and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-size().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_aae705ed824ebf7c7bf152f7998158239_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a77008e555c1d551716a52eb849694a7b}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!set\-Changed@{set\-Changed}}
\index{set\-Changed@{set\-Changed}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{set\-Changed}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Changed (
\begin{DoxyParamCaption}
\item[{final boolean}]{a\-\_\-changed}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a77008e555c1d551716a52eb849694a7b}
Mark that for the population the fittest program may have changed.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-changed} & true\-: population's fittest program may have changed, false\-: fittest program evaluated earlier is still valid\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 792 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-changed, and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Sorted().



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-copy\-G\-P\-Programs(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-determine\-Fittest\-Program(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-G\-P\-Population(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-G\-P\-Program(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-G\-P\-Programs(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-sort\-By\-Fitness().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a77008e555c1d551716a52eb849694a7b_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a11dce280ef2ec6ff8a7e1078500feda0}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!set\-G\-P\-Program@{set\-G\-P\-Program}}
\index{set\-G\-P\-Program@{set\-G\-P\-Program}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{set\-G\-P\-Program}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-G\-P\-Program (
\begin{DoxyParamCaption}
\item[{final int}]{a\-\_\-index, }
\item[{final {\bf I\-G\-P\-Program}}]{a\-\_\-program}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a11dce280ef2ec6ff8a7e1078500feda0}
Sets the given \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program}{G\-P\-Program} at the given index in the list of \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program}{G\-P\-Program}'s. If the given index is exceeding the list by one, the chromosome is appended.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-index} & the index to set the \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program}{G\-P\-Program} in \\
\hline
{\em a\-\_\-program} & the \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program}{G\-P\-Program} to be set\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 559 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs, and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Changed().



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a11dce280ef2ec6ff8a7e1078500feda0_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a2e36a0da99ef591a5980b0f1eef83b31}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!set\-G\-P\-Programs@{set\-G\-P\-Programs}}
\index{set\-G\-P\-Programs@{set\-G\-P\-Programs}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{set\-G\-P\-Programs}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-G\-P\-Programs (
\begin{DoxyParamCaption}
\item[{final {\bf G\-P\-Population}}]{a\-\_\-pop}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a2e36a0da99ef591a5980b0f1eef83b31}
Sets the G\-P\-Programs of the given population to this population.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-pop} & the population to use as template\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-3.\-3 
\end{DoxySince}


Definition at line 578 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-pop\-Size, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs, and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Changed().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a2e36a0da99ef591a5980b0f1eef83b31_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a8d9d5bb131077ec72be3a0137716775e}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!set\-G\-P\-Programs@{set\-G\-P\-Programs}}
\index{set\-G\-P\-Programs@{set\-G\-P\-Programs}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{set\-G\-P\-Programs}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-G\-P\-Programs (
\begin{DoxyParamCaption}
\item[{final {\bf I\-G\-P\-Program}\mbox{[}$\,$\mbox{]}}]{a\-\_\-progs}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a8d9d5bb131077ec72be3a0137716775e}
Sets the G\-P\-Programs of the given array to this population.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-progs} & the programs to set\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-3.\-3 
\end{DoxySince}


Definition at line 594 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-pop\-Size, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs, and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Changed().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a8d9d5bb131077ec72be3a0137716775e_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a45a7caf3ae473c7f75979a76526cb67a}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!set\-G\-P\-Programs@{set\-G\-P\-Programs}}
\index{set\-G\-P\-Programs@{set\-G\-P\-Programs}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{set\-G\-P\-Programs}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-G\-P\-Programs (
\begin{DoxyParamCaption}
\item[{final List}]{a\-\_\-pop}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a45a7caf3ae473c7f75979a76526cb67a}
Sets the G\-P\-Programs of the given population to this population.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-pop} & the population to use as template\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-3.\-3 
\end{DoxySince}


Definition at line 628 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Changed(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-size().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a45a7caf3ae473c7f75979a76526cb67a_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a863f5c2070d60fbf599fcf49230baec1}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!set\-Sorted@{set\-Sorted}}
\index{set\-Sorted@{set\-Sorted}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{set\-Sorted}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Sorted (
\begin{DoxyParamCaption}
\item[{final boolean}]{a\-\_\-sorted}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a863f5c2070d60fbf599fcf49230baec1}
Mark the population as sorted. 
\begin{DoxyParams}{Parameters}
{\em a\-\_\-sorted} & true\-: mark population as sorted\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 814 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-sorted.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Changed(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-sort\-By\-Fitness().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a73feb5c387bfb23ae12620126e392ebc}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!size@{size}}
\index{size@{size}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{size}]{\setlength{\rightskip}{0pt plus 5cm}int org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-size (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a73feb5c387bfb23ae12620126e392ebc}


Definition at line 643 of file G\-P\-Population.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-compare\-To(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-copy\-G\-P\-Programs(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-is\-First\-Empty(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-G\-P\-Programs(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population\-Test.\-test\-Construct\-\_\-0().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a9d49b2b18535658aae05ecdc2263918b}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!sort@{sort}}
\index{sort@{sort}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{sort}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-sort (
\begin{DoxyParamCaption}
\item[{Comparator}]{c}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a9d49b2b18535658aae05ecdc2263918b}
Sorts the population into \char`\"{}ascending\char`\"{} order using some criterion for \char`\"{}ascending\char`\"{}. A Comparator is given which will compare two individuals, and if one individual compares lower than another individual, the first individual will appear in the population before the second individual.


\begin{DoxyParams}{Parameters}
{\em c} & the Comparator to use\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 169 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-fitness\-Rank, and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-sort\-By\-Fitness().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a0d7b58819ffefbefbc0baaa1d12a9cb6}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!sort\-By\-Fitness@{sort\-By\-Fitness}}
\index{sort\-By\-Fitness@{sort\-By\-Fitness}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{sort\-By\-Fitness}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-sort\-By\-Fitness (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a0d7b58819ffefbefbc0baaa1d12a9cb6}
Sorts the programs within the population according to their fitness value using \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program_fitness_comparator}{G\-P\-Program\-Fitness\-Comparator}.

\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 764 of file G\-P\-Population.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-G\-P\-Configuration(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-fittest\-Program, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs, org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Changed(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Sorted(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-sort().



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-determine\-Fittest\-Chromosomes().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a0d7b58819ffefbefbc0baaa1d12a9cb6_cgraph}
\end{center}
\end{figure}




\subsection{Member Data Documentation}
\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ab209b6a7ffefa14c8d2f133e34f4d7e5}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N@{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N}}
\index{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N@{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N}]{\setlength{\rightskip}{0pt plus 5cm}final String org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N = \char`\"{}\$Revision\-: 1.\-43 \$\char`\"{}\hspace{0.3cm}{\ttfamily [static]}, {\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ab209b6a7ffefa14c8d2f133e34f4d7e5}
String containing the C\-V\-S revision. Read out via reflection! 

Definition at line 29 of file G\-P\-Population.\-java.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ab8c06531dff207ecdff0c86b3d36651e}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!D\-E\-L\-T\-A@{D\-E\-L\-T\-A}}
\index{D\-E\-L\-T\-A@{D\-E\-L\-T\-A}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{D\-E\-L\-T\-A}]{\setlength{\rightskip}{0pt plus 5cm}final double org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-D\-E\-L\-T\-A = 0.\-0000001d\hspace{0.3cm}{\ttfamily [static]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_ab8c06531dff207ecdff0c86b3d36651e}


Definition at line 37 of file G\-P\-Population.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-determine\-Fittest\-Program().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a2e511782d083ba5853395ac9cb59b95a}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!L\-O\-G\-G\-E\-R@{L\-O\-G\-G\-E\-R}}
\index{L\-O\-G\-G\-E\-R@{L\-O\-G\-G\-E\-R}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{L\-O\-G\-G\-E\-R}]{\setlength{\rightskip}{0pt plus 5cm}transient Logger org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-L\-O\-G\-G\-E\-R = Logger.\-get\-Logger(G\-P\-Population.\-class)\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a2e511782d083ba5853395ac9cb59b95a}


Definition at line 39 of file G\-P\-Population.\-java.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a4454c7e30fa80ed50d68c9e06b7f6897}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!m\-\_\-changed@{m\-\_\-changed}}
\index{m\-\_\-changed@{m\-\_\-changed}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{m\-\_\-changed}]{\setlength{\rightskip}{0pt plus 5cm}boolean org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-changed\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a4454c7e30fa80ed50d68c9e06b7f6897}
Indicates whether at least one of the programs has been changed (deleted, added, modified). 

Definition at line 61 of file G\-P\-Population.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-clear(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-determine\-Fittest\-Chromosomes(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-determine\-Fittest\-Program(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-G\-P\-Population(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-is\-Changed(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Changed().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a54cd610adaf8ce845d6e9ac796f682dc}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!m\-\_\-config@{m\-\_\-config}}
\index{m\-\_\-config@{m\-\_\-config}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{m\-\_\-config}]{\setlength{\rightskip}{0pt plus 5cm}{\bf G\-P\-Configuration} org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-config\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a54cd610adaf8ce845d6e9ac796f682dc}


Definition at line 55 of file G\-P\-Population.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-G\-P\-Configuration(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-G\-P\-Population().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a7f43165486ff8fd8d69f6c9ad7484823}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!m\-\_\-fitness\-Rank@{m\-\_\-fitness\-Rank}}
\index{m\-\_\-fitness\-Rank@{m\-\_\-fitness\-Rank}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{m\-\_\-fitness\-Rank}]{\setlength{\rightskip}{0pt plus 5cm}float \mbox{[}$\,$\mbox{]} org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-fitness\-Rank\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a7f43165486ff8fd8d69f6c9ad7484823}


Definition at line 49 of file G\-P\-Population.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-Fitness\-Rank(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-Fitness\-Ranks(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-G\-P\-Population(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-sort().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a13678da0768ccaf2dc3b5a892c434cb7}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!m\-\_\-fittest\-Program@{m\-\_\-fittest\-Program}}
\index{m\-\_\-fittest\-Program@{m\-\_\-fittest\-Program}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{m\-\_\-fittest\-Program}]{\setlength{\rightskip}{0pt plus 5cm}transient {\bf I\-G\-P\-Program} org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-fittest\-Program\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a13678da0768ccaf2dc3b5a892c434cb7}


Definition at line 53 of file G\-P\-Population.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-clear(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-determine\-Fittest\-Program(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-G\-P\-Population(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-sort\-By\-Fitness().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_afeeb7ca5a1210d473c18fdfe84844811}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!m\-\_\-fittest\-To\-Add@{m\-\_\-fittest\-To\-Add}}
\index{m\-\_\-fittest\-To\-Add@{m\-\_\-fittest\-To\-Add}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{m\-\_\-fittest\-To\-Add}]{\setlength{\rightskip}{0pt plus 5cm}{\bf I\-G\-P\-Program} org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-fittest\-To\-Add\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_afeeb7ca5a1210d473c18fdfe84844811}


Definition at line 68 of file G\-P\-Population.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-add\-Fittest\-Program(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a82de82c54ab2617c745e5493ed025e00}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!m\-\_\-pop\-Size@{m\-\_\-pop\-Size}}
\index{m\-\_\-pop\-Size@{m\-\_\-pop\-Size}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{m\-\_\-pop\-Size}]{\setlength{\rightskip}{0pt plus 5cm}int org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-pop\-Size\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a82de82c54ab2617c745e5493ed025e00}


Definition at line 51 of file G\-P\-Population.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-Pop\-Size(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-G\-P\-Population(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-G\-P\-Programs().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a3bf2729a02a6d7daa8d9badb52cd6c3a}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!m\-\_\-programs@{m\-\_\-programs}}
\index{m\-\_\-programs@{m\-\_\-programs}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{m\-\_\-programs}]{\setlength{\rightskip}{0pt plus 5cm}{\bf I\-G\-P\-Program} \mbox{[}$\,$\mbox{]} org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-programs\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a3bf2729a02a6d7daa8d9badb52cd6c3a}
The array of \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program}{G\-P\-Program}'s that make-\/up the \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype}'s population. 

Definition at line 46 of file G\-P\-Population.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-clear(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-compare\-To(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-copy\-G\-P\-Programs(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-determine\-Fittest\-Chromosomes(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-determine\-Fittest\-Program(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-determine\-Fittest\-Program\-Computed(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-G\-P\-Program(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-G\-P\-Programs(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-Persistent\-Representation(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-G\-P\-Population(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-is\-First\-Empty(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-G\-P\-Program(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-G\-P\-Programs(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-sort(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-sort\-By\-Fitness().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a522f220885c90a850191b89116bee75f}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!m\-\_\-sorted@{m\-\_\-sorted}}
\index{m\-\_\-sorted@{m\-\_\-sorted}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{m\-\_\-sorted}]{\setlength{\rightskip}{0pt plus 5cm}boolean org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-m\-\_\-sorted\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_a522f220885c90a850191b89116bee75f}
Indicates that the list of G\-P\-Programs has been sorted. 

Definition at line 66 of file G\-P\-Population.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-clear(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-determine\-Fittest\-Chromosomes(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-G\-P\-Population(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-set\-Sorted().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_aeb4067371d53afb6f4e0a4939725dd36}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}!warning\-Prototype\-Reused@{warning\-Prototype\-Reused}}
\index{warning\-Prototype\-Reused@{warning\-Prototype\-Reused}!org::jgap::gp::impl::GPPopulation@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Population}}
\subsubsection[{warning\-Prototype\-Reused}]{\setlength{\rightskip}{0pt plus 5cm}transient int org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-warning\-Prototype\-Reused = 0\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_aeb4067371d53afb6f4e0a4939725dd36}


Definition at line 41 of file G\-P\-Population.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-create().



The documentation for this class was generated from the following file\-:\begin{DoxyCompactItemize}
\item 
L\-D\-H\-\_\-\-Git/src/org/jgap/gp/impl/\hyperlink{_g_p_population_8java}{G\-P\-Population.\-java}\end{DoxyCompactItemize}
