\hypertarget{classorg_1_1jgap_1_1_base_gene}{\section{org.\-jgap.\-Base\-Gene Class Reference}
\label{classorg_1_1jgap_1_1_base_gene}\index{org.\-jgap.\-Base\-Gene@{org.\-jgap.\-Base\-Gene}}
}


Inheritance diagram for org.\-jgap.\-Base\-Gene\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_base_gene__inherit__graph}
\end{center}
\end{figure}


Collaboration diagram for org.\-jgap.\-Base\-Gene\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_base_gene__coll__graph}
\end{center}
\end{figure}
\subsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
\hyperlink{classorg_1_1jgap_1_1_base_gene_acea53de5bb68978322967e8019607f24}{Base\-Gene} (\hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} a\-\_\-configuration)  throws Invalid\-Configuration\-Exception 
\item 
Object \hyperlink{classorg_1_1jgap_1_1_base_gene_a35e754e8e217bdd26962e1c3c31d0169}{get\-Allele} ()
\item 
int \hyperlink{classorg_1_1jgap_1_1_base_gene_af842a106d90ca973956f560ae448464e}{hash\-Code} ()
\item 
void \hyperlink{classorg_1_1jgap_1_1_base_gene_a569b99cf460d27d2c1acbea2b7775975}{cleanup} ()
\item 
String \hyperlink{classorg_1_1jgap_1_1_base_gene_a935dfec06cb77bff0d2a45859d6a35b7}{to\-String} ()
\item 
int \hyperlink{classorg_1_1jgap_1_1_base_gene_a26c00775111b327b178288164ed9c667}{size} ()
\item 
boolean \hyperlink{classorg_1_1jgap_1_1_base_gene_a9fb0e302ae57ee3f2f5d1ee08a75bf43}{equals} (final Object a\-\_\-other)
\item 
double \hyperlink{classorg_1_1jgap_1_1_base_gene_a48f71c4c1282de2ca2cf8f4009c4615a}{get\-Energy} ()
\item 
void \hyperlink{classorg_1_1jgap_1_1_base_gene_afb39fbd8997971a140a9a4380ceecc4a}{set\-Energy} (final double a\-\_\-energy)
\item 
void \hyperlink{classorg_1_1jgap_1_1_base_gene_afeb9a5f42018c057c8197fb0e7d57354}{set\-Application\-Data} (final Object a\-\_\-new\-Data)
\item 
Object \hyperlink{classorg_1_1jgap_1_1_base_gene_a0520139dc694dc6034aeb631ae7d70bd}{get\-Application\-Data} ()
\item 
void \hyperlink{classorg_1_1jgap_1_1_base_gene_adad4ccd46efd7b2a3dc81bd0ba0c410c}{set\-Compare\-Application\-Data} (final boolean a\-\_\-do\-Compare)
\item 
boolean \hyperlink{classorg_1_1jgap_1_1_base_gene_acd6431c06c81bc03b4abbe32e4d9d8ef}{is\-Compare\-Application\-Data} ()
\item 
void \hyperlink{classorg_1_1jgap_1_1_base_gene_a334db8a543f3008719d78ee219826b65}{set\-Constraint\-Checker} (final \hyperlink{interfaceorg_1_1jgap_1_1_i_gene_constraint_checker}{I\-Gene\-Constraint\-Checker} a\-\_\-constraint\-Checker)
\item 
\hyperlink{interfaceorg_1_1jgap_1_1_i_gene_constraint_checker}{I\-Gene\-Constraint\-Checker} \hyperlink{classorg_1_1jgap_1_1_base_gene_ae9823bef7c550fc1533afb8ab2b34d93}{get\-Constraint\-Checker} ()
\item 
\hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} \hyperlink{classorg_1_1jgap_1_1_base_gene_a40f8f3d249145ad028892cd706ba9dae}{new\-Gene} ()
\item 
\hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} \hyperlink{classorg_1_1jgap_1_1_base_gene_a32a316c9cc9e273bd75182cbdf188fb6}{get\-Configuration} ()
\item 
String \hyperlink{classorg_1_1jgap_1_1_base_gene_ac4e7c0fad73465365953b375cfa802a8}{get\-Business\-Key} ()
\item 
String \hyperlink{classorg_1_1jgap_1_1_base_gene_ab53d562a62f5c72fc88c7e5315a23ed7}{get\-Unique\-I\-D} ()
\item 
void \hyperlink{classorg_1_1jgap_1_1_base_gene_aa7d9f1c3c2e727d496d66a85818c17d0}{set\-Unique\-I\-D\-Template} (String a\-\_\-template\-I\-D, int a\-\_\-index)
\item 
String \hyperlink{classorg_1_1jgap_1_1_base_gene_ad3c210e440f2b542391db5a909fda3b7}{get\-Unique\-I\-D\-Template} (int a\-\_\-index)
\end{DoxyCompactItemize}
\subsection*{Static Public Attributes}
\begin{DoxyCompactItemize}
\item 
static final String \hyperlink{classorg_1_1jgap_1_1_base_gene_acc4b1c46f3508b835bf95b9058eae3a0}{S\-\_\-\-A\-P\-P\-L\-I\-C\-A\-T\-I\-O\-N\-\_\-\-D\-A\-T\-A} = \char`\"{}Application data\char`\"{}
\item 
static final double \hyperlink{classorg_1_1jgap_1_1_base_gene_a86edb3d2c29e5064305fabc2dd59a3b9}{D\-E\-L\-T\-A} = 0.\-0000001
\end{DoxyCompactItemize}
\subsection*{Protected Member Functions}
\begin{DoxyCompactItemize}
\item 
abstract Object \hyperlink{classorg_1_1jgap_1_1_base_gene_ad20da6b23ba524a95caa28f727c26dcd}{get\-Internal\-Value} ()
\item 
int \hyperlink{classorg_1_1jgap_1_1_base_gene_a72fa365cdcc9f2eb399fb8346a0a511b}{compare\-Application\-Data} (final Object a\-\_\-appdata1, final Object a\-\_\-appdata2)
\item 
abstract \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} \hyperlink{classorg_1_1jgap_1_1_base_gene_aa423e96ffac5a9589fb3a4fbda791b3c}{new\-Gene\-Internal} ()
\item 
String \hyperlink{classorg_1_1jgap_1_1_base_gene_a5e8fc72c1fccdb665113c0c34e3fa4b5}{encode} (String a\-\_\-string)
\item 
String \hyperlink{classorg_1_1jgap_1_1_base_gene_ae9277d0b77c1e8ca244687a8ab8cf42f}{decode} (String a\-\_\-string)
\end{DoxyCompactItemize}
\subsection*{Private Attributes}
\begin{DoxyCompactItemize}
\item 
double \hyperlink{classorg_1_1jgap_1_1_base_gene_a8553da900cf30fc4b0f5463404b86a8e}{m\-\_\-energy}
\item 
Object \hyperlink{classorg_1_1jgap_1_1_base_gene_ab2e121e812be7a4dead5ca7e300ac3d7}{m\-\_\-application\-Data}
\item 
boolean \hyperlink{classorg_1_1jgap_1_1_base_gene_a27bb3cd52e8cd52fffc4953f172709a2}{m\-\_\-compare\-App\-Data}
\item 
\hyperlink{classorg_1_1jgap_1_1_configuration}{Configuration} \hyperlink{classorg_1_1jgap_1_1_base_gene_a4d41d885b45ce031c33c88214f98019e}{m\-\_\-configuration}
\item 
String \hyperlink{classorg_1_1jgap_1_1_base_gene_a8cd57f3a57f5a892f590df9e06b82e47}{m\-\_\-unique\-I\-D}
\item 
Map$<$ Integer, String $>$ \hyperlink{classorg_1_1jgap_1_1_base_gene_a3f14fef3c7107830f24306cf7bb0abf0}{m\-\_\-unique\-I\-D\-Templates}
\item 
\hyperlink{interfaceorg_1_1jgap_1_1_i_gene_constraint_checker}{I\-Gene\-Constraint\-Checker} \hyperlink{classorg_1_1jgap_1_1_base_gene_a846344a347601679d0935dcdba7f7e6f}{m\-\_\-gene\-Allele\-Checker}
\end{DoxyCompactItemize}
\subsection*{Static Private Attributes}
\begin{DoxyCompactItemize}
\item 
static final String \hyperlink{classorg_1_1jgap_1_1_base_gene_a7e83775c4b2b9391ec833869808d0154}{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N} = \char`\"{}\$Revision\-: 1.\-28 \$\char`\"{}
\end{DoxyCompactItemize}


\subsection{Detailed Description}
Abstract base class for all genes. Provides default implementations.

\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-2 
\end{DoxySince}


Definition at line 23 of file Base\-Gene.\-java.



\subsection{Constructor \& Destructor Documentation}
\hypertarget{classorg_1_1jgap_1_1_base_gene_acea53de5bb68978322967e8019607f24}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!Base\-Gene@{Base\-Gene}}
\index{Base\-Gene@{Base\-Gene}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{Base\-Gene}]{\setlength{\rightskip}{0pt plus 5cm}org.\-jgap.\-Base\-Gene.\-Base\-Gene (
\begin{DoxyParamCaption}
\item[{{\bf Configuration}}]{a\-\_\-configuration}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_acea53de5bb68978322967e8019607f24}

\begin{DoxyParams}{Parameters}
{\em a\-\_\-configuration} & the configuration to use \\
\hline
\end{DoxyParams}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 82 of file Base\-Gene.\-java.



References org.\-jgap.\-Configuration.\-get\-J\-G\-A\-P\-Factory(), org.\-jgap.\-Configuration.\-is\-Unique\-Keys\-Active(), org.\-jgap.\-Base\-Gene.\-m\-\_\-configuration, org.\-jgap.\-Base\-Gene.\-m\-\_\-unique\-I\-D, and org.\-jgap.\-Base\-Gene.\-m\-\_\-unique\-I\-D\-Templates.



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_base_gene_acea53de5bb68978322967e8019607f24_cgraph}
\end{center}
\end{figure}




\subsection{Member Function Documentation}
\hypertarget{classorg_1_1jgap_1_1_base_gene_a569b99cf460d27d2c1acbea2b7775975}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!cleanup@{cleanup}}
\index{cleanup@{cleanup}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{cleanup}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Base\-Gene.\-cleanup (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_a569b99cf460d27d2c1acbea2b7775975}
Executed by the genetic engine when this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} instance is no longer needed and should perform any necessary resource cleanup. If you need a special cleanup, override this method.

\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
1.\-0 
\end{DoxySince}


Implements \hyperlink{interfaceorg_1_1jgap_1_1_gene_a5e353bbf6a9576b702bf466a5f952b82}{org.\-jgap.\-Gene}.



Definition at line 138 of file Base\-Gene.\-java.

\hypertarget{classorg_1_1jgap_1_1_base_gene_a72fa365cdcc9f2eb399fb8346a0a511b}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!compare\-Application\-Data@{compare\-Application\-Data}}
\index{compare\-Application\-Data@{compare\-Application\-Data}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{compare\-Application\-Data}]{\setlength{\rightskip}{0pt plus 5cm}int org.\-jgap.\-Base\-Gene.\-compare\-Application\-Data (
\begin{DoxyParamCaption}
\item[{final Object}]{a\-\_\-appdata1, }
\item[{final Object}]{a\-\_\-appdata2}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1_base_gene_a72fa365cdcc9f2eb399fb8346a0a511b}


Definition at line 306 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-get\-Configuration().



Referenced by org.\-jgap.\-impl.\-Number\-Gene.\-compare\-To(), org.\-jgap.\-impl.\-Boolean\-Gene.\-compare\-To(), org.\-jgap.\-impl.\-String\-Gene.\-compare\-To(), org.\-jgap.\-impl.\-Fixed\-Binary\-Gene.\-compare\-To(), org.\-jgap.\-impl.\-Composite\-Gene.\-compare\-To(), and org.\-jgap.\-Base\-Gene.\-equals().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_base_gene_a72fa365cdcc9f2eb399fb8346a0a511b_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_base_gene_ae9277d0b77c1e8ca244687a8ab8cf42f}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!decode@{decode}}
\index{decode@{decode}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{decode}]{\setlength{\rightskip}{0pt plus 5cm}String org.\-jgap.\-Base\-Gene.\-decode (
\begin{DoxyParamCaption}
\item[{String}]{a\-\_\-string}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1_base_gene_ae9277d0b77c1e8ca244687a8ab8cf42f}


Definition at line 426 of file Base\-Gene.\-java.



Referenced by org.\-jgap.\-supergenes.\-Abstract\-Supergene.\-create\-Validator(), org.\-jgap.\-impl.\-String\-Gene.\-set\-Value\-From\-Persistent\-Representation(), org.\-jgap.\-impl.\-Composite\-Gene.\-set\-Value\-From\-Persistent\-Representation(), and org.\-jgap.\-supergenes.\-Abstract\-Supergene.\-set\-Value\-From\-Persistent\-Representation().

\hypertarget{classorg_1_1jgap_1_1_base_gene_a5e8fc72c1fccdb665113c0c34e3fa4b5}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!encode@{encode}}
\index{encode@{encode}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{encode}]{\setlength{\rightskip}{0pt plus 5cm}String org.\-jgap.\-Base\-Gene.\-encode (
\begin{DoxyParamCaption}
\item[{String}]{a\-\_\-string}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1_base_gene_a5e8fc72c1fccdb665113c0c34e3fa4b5}


Definition at line 422 of file Base\-Gene.\-java.



Referenced by org.\-jgap.\-impl.\-String\-Gene.\-get\-Persistent\-Representation(), org.\-jgap.\-impl.\-Composite\-Gene.\-get\-Persistent\-Representation(), and org.\-jgap.\-supergenes.\-Abstract\-Supergene.\-get\-Persistent\-Representation().

\hypertarget{classorg_1_1jgap_1_1_base_gene_a9fb0e302ae57ee3f2f5d1ee08a75bf43}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!equals@{equals}}
\index{equals@{equals}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{equals}]{\setlength{\rightskip}{0pt plus 5cm}boolean org.\-jgap.\-Base\-Gene.\-equals (
\begin{DoxyParamCaption}
\item[{final Object}]{a\-\_\-other}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_a9fb0e302ae57ee3f2f5d1ee08a75bf43}
Compares this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} with the given object and returns true if the other object is a \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} of the same type and has the same value (allele) as this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene}. Otherwise it returns false.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-other} & the object to compare to this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} for equality \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
true if this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} is equal to the given object, false otherwise
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
1.\-1 
\end{DoxySince}


Definition at line 193 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-compare\-Application\-Data(), org.\-jgap.\-Gene.\-get\-Application\-Data(), org.\-jgap.\-Base\-Gene.\-get\-Application\-Data(), and org.\-jgap.\-Base\-Gene.\-is\-Compare\-Application\-Data().



Referenced by org.\-jgap.\-impl.\-Composite\-Gene.\-add\-Gene(), org.\-jgap.\-impl.\-Set\-Gene\-Test.\-test\-Equals\-\_\-2(), org.\-jgap.\-impl.\-Composite\-Gene\-Test.\-test\-Equals\-\_\-5(), org.\-jgap.\-impl.\-Composite\-Gene\-Test.\-test\-Equals\-\_\-5\-\_\-2(), org.\-jgap.\-impl.\-Integer\-Gene\-Test.\-test\-Equals\-\_\-8(), org.\-jgap.\-impl.\-Integer\-Gene\-Test.\-test\-Equals\-\_\-9(), org.\-jgap.\-impl.\-Integer\-Gene\-Test.\-test\-Equals\-\_\-9\-\_\-2(), org.\-jgap.\-impl.\-Integer\-Gene\-Test.\-test\-Equals\-\_\-9\-\_\-3(), org.\-jgap.\-impl.\-Composite\-Gene\-Test.\-test\-New\-Gene\-\_\-0(), org.\-jgap.\-impl.\-Fixed\-Binary\-Gene\-Test.\-test\-New\-Gene\-\_\-0(), org.\-jgap.\-impl.\-Map\-Gene\-Test.\-test\-New\-Gene\-\_\-2(), and org.\-jgap.\-impl.\-Map\-Gene\-Test.\-test\-New\-Gene\-\_\-3().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_base_gene_a9fb0e302ae57ee3f2f5d1ee08a75bf43_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_base_gene_a35e754e8e217bdd26962e1c3c31d0169}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!get\-Allele@{get\-Allele}}
\index{get\-Allele@{get\-Allele}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{get\-Allele}]{\setlength{\rightskip}{0pt plus 5cm}Object org.\-jgap.\-Base\-Gene.\-get\-Allele (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_a35e754e8e217bdd26962e1c3c31d0169}
Retrieves the allele value represented by this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene}.

\begin{DoxyReturn}{Returns}
the allele value of this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} 
\end{DoxyReturn}
\begin{DoxySince}{Since}
1.\-0 
\end{DoxySince}


Implements \hyperlink{interfaceorg_1_1jgap_1_1_gene_aa06c80639659ddbcfa1cfe7b7bb109f9}{org.\-jgap.\-Gene}.



Definition at line 103 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-get\-Internal\-Value().



Referenced by org.\-jgap.\-impl.\-Integer\-Gene.\-apply\-Mutation(), org.\-jgap.\-impl.\-Mutiple\-Integer\-Gene.\-apply\-Mutation(), org.\-jgap.\-impl.\-Double\-Gene.\-double\-Value(), org.\-jgap.\-Base\-Gene.\-get\-Business\-Key(), org.\-jgap.\-impl.\-Integer\-Gene.\-int\-Value(), org.\-jgap.\-impl.\-Double\-Gene.\-map\-Value\-To\-Within\-Bounds(), org.\-jgap.\-impl.\-Integer\-Gene.\-map\-Value\-To\-Within\-Bounds(), org.\-jgap.\-impl.\-Mutiple\-Integer\-Gene.\-map\-Value\-To\-Within\-Bounds(), org.\-jgap.\-impl.\-Number\-Gene\-Test.\-Number\-Gene\-Impl.\-map\-Value\-To\-Within\-Bounds(), org.\-jgap.\-impl.\-Map\-Gene.\-new\-Gene\-Internal(), org.\-jgap.\-impl.\-Set\-Gene\-Test.\-test\-Get\-Allele\-\_\-0(), org.\-jgap.\-impl.\-Map\-Gene\-Test.\-test\-Get\-Allele\-\_\-0(), org.\-jgap.\-impl.\-Double\-Gene\-Test.\-test\-Map\-Value\-To\-Within\-Bounds\-Supports\-Full\-Double\-Range(), org.\-jgap.\-impl.\-Double\-Gene\-Test.\-test\-Map\-Value\-To\-Within\-Bounds\-Supports\-Full\-Double\-Range2(), org.\-jgap.\-impl.\-Double\-Gene\-Test.\-test\-Map\-Value\-To\-Within\-Bounds\-Supports\-Full\-Double\-Range3(), org.\-jgap.\-impl.\-Double\-Gene\-Test.\-test\-Map\-Value\-To\-Within\-Bounds\-Supports\-Full\-Double\-Range4(), org.\-jgap.\-impl.\-Double\-Gene\-Test.\-test\-Map\-Value\-To\-Within\-Bounds\-Supports\-Full\-Double\-Range5(), org.\-jgap.\-impl.\-Integer\-Gene\-Test.\-test\-Map\-Value\-To\-Within\-Bounds\-Supports\-Full\-Integer\-Range(), org.\-jgap.\-impl.\-Inversion\-Operator\-Test.\-test\-Operate\-\_\-2(), org.\-jgap.\-impl.\-Composite\-Gene\-Test.\-test\-Persistent\-Presentation\-\_\-1(), org.\-jgap.\-impl.\-Map\-Gene\-Test.\-test\-Persistent\-Representation\-\_\-2(), org.\-jgap.\-impl.\-Integer\-Gene\-Test.\-test\-Persistent\-Representation\-\_\-2(), org.\-jgap.\-impl.\-Integer\-Gene\-Test.\-test\-Persistent\-Representation\-\_\-3(), org.\-jgap.\-impl.\-Composite\-Gene\-Test.\-test\-Set\-Allele\-\_\-4(), org.\-jgap.\-impl.\-Composite\-Gene\-Test.\-test\-Set\-Allele\-\_\-5(), org.\-jgap.\-impl.\-String\-Gene\-Test.\-test\-Set\-To\-Random\-Value\-\_\-0(), org.\-jgap.\-impl.\-String\-Gene\-Test.\-test\-Set\-To\-Random\-Value\-\_\-3(), org.\-jgap.\-impl.\-String\-Gene\-Test.\-test\-Set\-To\-Random\-Value\-\_\-4(), org.\-jgap.\-impl.\-Set\-Gene\-Test.\-test\-Set\-Value\-From\-Persistent\-Representation\-\_\-1(), org.\-jgap.\-impl.\-Boolean\-Gene\-Test.\-test\-Set\-Value\-From\-Persistent\-Representation\-\_\-1(), org.\-jgap.\-impl.\-Set\-Gene\-Test.\-test\-Set\-Value\-From\-Persistent\-Representation\-\_\-2(), org.\-jgap.\-impl.\-Boolean\-Gene\-Test.\-test\-Set\-Value\-From\-Persistent\-Representation\-\_\-2(), org.\-jgap.\-impl.\-Set\-Gene\-Test.\-test\-Set\-Value\-From\-Persistent\-Representation\-\_\-3(), and org.\-jgap.\-impl.\-Boolean\-Gene\-Test.\-test\-Set\-Value\-From\-Persistent\-Representation\-\_\-3().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_base_gene_a35e754e8e217bdd26962e1c3c31d0169_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_base_gene_a0520139dc694dc6034aeb631ae7d70bd}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!get\-Application\-Data@{get\-Application\-Data}}
\index{get\-Application\-Data@{get\-Application\-Data}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{get\-Application\-Data}]{\setlength{\rightskip}{0pt plus 5cm}Object org.\-jgap.\-Base\-Gene.\-get\-Application\-Data (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_a0520139dc694dc6034aeb631ae7d70bd}
Retrieves the application-\/specific data that is attached to this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene}. Attaching application-\/specific data may be useful for some applications when it comes time to distinguish a \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} from another. J\-G\-A\-P ignores this data functionally.

\begin{DoxyReturn}{Returns}
the application-\/specific data previously attached to this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene}, or null if there is no data attached
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-4 
\end{DoxySince}


Implements \hyperlink{interfaceorg_1_1jgap_1_1_gene_a4eb1d079f14465a2324ad41a6ee3b77a}{org.\-jgap.\-Gene}.



Definition at line 278 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-m\-\_\-application\-Data.



Referenced by org.\-jgap.\-impl.\-Number\-Gene.\-compare\-To(), org.\-jgap.\-impl.\-Boolean\-Gene.\-compare\-To(), org.\-jgap.\-impl.\-String\-Gene.\-compare\-To(), org.\-jgap.\-impl.\-Fixed\-Binary\-Gene.\-compare\-To(), org.\-jgap.\-impl.\-Composite\-Gene.\-compare\-To(), org.\-jgap.\-Base\-Gene.\-equals(), org.\-jgap.\-Base\-Gene.\-new\-Gene(), and org.\-jgap.\-Base\-Gene.\-to\-String().

\hypertarget{classorg_1_1jgap_1_1_base_gene_ac4e7c0fad73465365953b375cfa802a8}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!get\-Business\-Key@{get\-Business\-Key}}
\index{get\-Business\-Key@{get\-Business\-Key}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{get\-Business\-Key}]{\setlength{\rightskip}{0pt plus 5cm}String org.\-jgap.\-Base\-Gene.\-get\-Business\-Key (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_ac4e7c0fad73465365953b375cfa802a8}
\begin{DoxyReturn}{Returns}
business key
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2 
\end{DoxySince}


Implements \hyperlink{interfaceorg_1_1jgap_1_1_i_business_key_ab1b12087d8726e9020441a1fed6be0f8}{org.\-jgap.\-I\-Business\-Key}.



Definition at line 413 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-get\-Allele(), and org.\-jgap.\-Gene.\-P\-E\-R\-S\-I\-S\-T\-E\-N\-T\-\_\-\-F\-I\-E\-L\-D\-\_\-\-D\-E\-L\-I\-M\-I\-T\-E\-R.



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_base_gene_ac4e7c0fad73465365953b375cfa802a8_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_base_gene_a32a316c9cc9e273bd75182cbdf188fb6}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!get\-Configuration@{get\-Configuration}}
\index{get\-Configuration@{get\-Configuration}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{get\-Configuration}]{\setlength{\rightskip}{0pt plus 5cm}{\bf Configuration} org.\-jgap.\-Base\-Gene.\-get\-Configuration (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_a32a316c9cc9e273bd75182cbdf188fb6}
\begin{DoxyReturn}{Returns}
the configuration set
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Implements \hyperlink{interfaceorg_1_1jgap_1_1_gene_aa178414303649867b6176de80c6ba892}{org.\-jgap.\-Gene}.



Definition at line 409 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-m\-\_\-configuration.



Referenced by org.\-jgap.\-impl.\-Set\-Gene.\-apply\-Mutation(), org.\-jgap.\-impl.\-Map\-Gene.\-apply\-Mutation(), org.\-jgap.\-supergenes.\-Abstract\-Supergene.\-apply\-Mutation(), org.\-jgap.\-impl.\-String\-Gene.\-apply\-Mutation(), org.\-jgap.\-impl.\-Fixed\-Binary\-Gene.\-clone(), org.\-jgap.\-Base\-Gene.\-compare\-Application\-Data(), org.\-jgap.\-impl.\-Set\-Gene.\-compare\-To(), org.\-jgap.\-impl.\-Composite\-Gene.\-create\-Gene(), org.\-jgap.\-supergenes.\-Abstract\-Supergene.\-create\-Gene(), org.\-jgap.\-supergenes.\-Abstract\-Supergene.\-create\-Validator(), org.\-jgap.\-impl.\-Double\-Gene.\-map\-Value\-To\-Within\-Bounds(), org.\-jgap.\-impl.\-Integer\-Gene.\-map\-Value\-To\-Within\-Bounds(), org.\-jgap.\-impl.\-Mutiple\-Integer\-Gene.\-map\-Value\-To\-Within\-Bounds(), org.\-jgap.\-impl.\-Set\-Gene.\-new\-Gene\-Internal(), org.\-jgap.\-impl.\-Fixed\-Binary\-Gene.\-new\-Gene\-Internal(), org.\-jgap.\-impl.\-Map\-Gene.\-new\-Gene\-Internal(), org.\-jgap.\-impl.\-Double\-Gene.\-new\-Gene\-Internal(), org.\-jgap.\-impl.\-Integer\-Gene.\-new\-Gene\-Internal(), org.\-jgap.\-impl.\-Mutiple\-Integer\-Gene.\-new\-Gene\-Internal(), org.\-jgap.\-impl.\-Boolean\-Gene.\-new\-Gene\-Internal(), org.\-jgap.\-supergenes.\-Abstract\-Supergene.\-new\-Gene\-Internal(), org.\-jgap.\-impl.\-String\-Gene.\-new\-Gene\-Internal(), org.\-jgap.\-impl.\-Composite\-Gene.\-new\-Gene\-Internal(), org.\-jgap.\-impl.\-Number\-Gene\-Test.\-Number\-Gene\-Impl.\-new\-Gene\-Internal(), org.\-jgap.\-impl.\-Fixed\-Binary\-Gene.\-substring(), and org.\-jgap.\-impl.\-Set\-Gene\-Test.\-test\-Construct\-\_\-0().

\hypertarget{classorg_1_1jgap_1_1_base_gene_ae9823bef7c550fc1533afb8ab2b34d93}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!get\-Constraint\-Checker@{get\-Constraint\-Checker}}
\index{get\-Constraint\-Checker@{get\-Constraint\-Checker}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{get\-Constraint\-Checker}]{\setlength{\rightskip}{0pt plus 5cm}{\bf I\-Gene\-Constraint\-Checker} org.\-jgap.\-Base\-Gene.\-get\-Constraint\-Checker (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_ae9823bef7c550fc1533afb8ab2b34d93}
\begin{DoxyReturn}{Returns}
\hyperlink{interfaceorg_1_1jgap_1_1_i_gene_constraint_checker}{I\-Gene\-Constraint\-Checker} the constraint checker to be used whenever method \hyperlink{interfaceorg_1_1jgap_1_1_gene_af46a0c31e0fecea483497e87ca278127}{set\-Allele(\-Object)} is called.
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-5 (moved from Composite\-Gene, where it was since 2.\-0) 
\end{DoxySince}


Definition at line 370 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-m\-\_\-gene\-Allele\-Checker.



Referenced by org.\-jgap.\-Base\-Gene.\-new\-Gene(), org.\-jgap.\-impl.\-String\-Gene.\-new\-Gene\-Internal(), org.\-jgap.\-impl.\-Composite\-Gene.\-new\-Gene\-Internal(), org.\-jgap.\-impl.\-Fixed\-Binary\-Gene.\-set\-Allele(), org.\-jgap.\-impl.\-Number\-Gene.\-set\-Allele(), org.\-jgap.\-impl.\-String\-Gene.\-set\-Allele(), org.\-jgap.\-impl.\-Composite\-Gene.\-set\-Allele(), org.\-jgap.\-impl.\-Double\-Gene\-Test.\-test\-New\-Gene\-\_\-0(), org.\-jgap.\-impl.\-Number\-Gene\-Test.\-test\-Set\-Constraint\-Checker\-\_\-0(), org.\-jgap.\-impl.\-Fixed\-Binary\-Gene\-Test.\-test\-Set\-Constraint\-Checker\-\_\-0(), org.\-jgap.\-impl.\-Double\-Gene\-Test.\-test\-Set\-Constraint\-Checker\-\_\-0(), org.\-jgap.\-impl.\-Composite\-Gene\-Test.\-test\-Set\-Constraint\-Checker\-\_\-0(), and org.\-jgap.\-impl.\-String\-Gene\-Test.\-test\-Set\-Constraint\-Checker\-\_\-0().

\hypertarget{classorg_1_1jgap_1_1_base_gene_a48f71c4c1282de2ca2cf8f4009c4615a}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!get\-Energy@{get\-Energy}}
\index{get\-Energy@{get\-Energy}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{get\-Energy}]{\setlength{\rightskip}{0pt plus 5cm}double org.\-jgap.\-Base\-Gene.\-get\-Energy (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_a48f71c4c1282de2ca2cf8f4009c4615a}
\begin{DoxyReturn}{Returns}
energy of the gene
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-3 
\end{DoxySince}


Implements \hyperlink{interfaceorg_1_1jgap_1_1_gene_aa1794038b5018cdb49998639c79cb57a}{org.\-jgap.\-Gene}.



Definition at line 235 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-m\-\_\-energy.



Referenced by org.\-jgap.\-Base\-Gene.\-new\-Gene(), org.\-jgap.\-impl.\-Number\-Gene\-Test.\-test\-Set\-Energy\-\_\-0(), org.\-jgap.\-impl.\-Set\-Gene\-Test.\-test\-Set\-Energy\-\_\-0(), org.\-jgap.\-impl.\-Boolean\-Gene\-Test.\-test\-Set\-Energy\-\_\-0(), org.\-jgap.\-impl.\-Double\-Gene\-Test.\-test\-Set\-Energy\-\_\-0(), org.\-jgap.\-impl.\-Integer\-Gene\-Test.\-test\-Set\-Energy\-\_\-0(), org.\-jgap.\-impl.\-Composite\-Gene\-Test.\-test\-Set\-Energy\-\_\-0(), org.\-jgap.\-impl.\-String\-Gene\-Test.\-test\-Set\-Energy\-\_\-0(), org.\-jgap.\-impl.\-Fixed\-Binary\-Gene\-Test.\-test\-Set\-Energy\-\_\-0(), org.\-jgap.\-impl.\-Number\-Gene\-Test.\-test\-Set\-Energy\-\_\-1(), org.\-jgap.\-impl.\-Set\-Gene\-Test.\-test\-Set\-Energy\-\_\-1(), org.\-jgap.\-impl.\-Boolean\-Gene\-Test.\-test\-Set\-Energy\-\_\-1(), org.\-jgap.\-impl.\-Double\-Gene\-Test.\-test\-Set\-Energy\-\_\-1(), org.\-jgap.\-impl.\-Integer\-Gene\-Test.\-test\-Set\-Energy\-\_\-1(), org.\-jgap.\-impl.\-Composite\-Gene\-Test.\-test\-Set\-Energy\-\_\-1(), org.\-jgap.\-impl.\-String\-Gene\-Test.\-test\-Set\-Energy\-\_\-1(), and org.\-jgap.\-impl.\-Fixed\-Binary\-Gene\-Test.\-test\-Set\-Energy\-\_\-1().

\hypertarget{classorg_1_1jgap_1_1_base_gene_ad20da6b23ba524a95caa28f727c26dcd}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!get\-Internal\-Value@{get\-Internal\-Value}}
\index{get\-Internal\-Value@{get\-Internal\-Value}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{get\-Internal\-Value}]{\setlength{\rightskip}{0pt plus 5cm}abstract Object org.\-jgap.\-Base\-Gene.\-get\-Internal\-Value (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [protected]}, {\ttfamily [pure virtual]}}}\label{classorg_1_1jgap_1_1_base_gene_ad20da6b23ba524a95caa28f727c26dcd}
Each \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} implementation holds its own m\-\_\-value object keeping the allele value. In your \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} implementation, just return it with this method (see \hyperlink{classorg_1_1jgap_1_1impl_1_1_boolean_gene}{org.\-jgap.\-impl.\-Boolean\-Gene} for example) \begin{DoxyReturn}{Returns}
the m\-\_\-value object
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-2 
\end{DoxySince}


Implemented in \hyperlink{classorg_1_1jgap_1_1impl_1_1_composite_gene_a9ee20f6169c0d441cdacb3685d5336f6}{org.\-jgap.\-impl.\-Composite\-Gene}, \hyperlink{classorg_1_1jgap_1_1supergenes_1_1_abstract_supergene_af4b0e1f740b6aa78679ab71c51568cea}{org.\-jgap.\-supergenes.\-Abstract\-Supergene}, \hyperlink{classorg_1_1jgap_1_1impl_1_1_string_gene_a2510aeb97b5362ab5ce9e00d412873dd}{org.\-jgap.\-impl.\-String\-Gene}, \hyperlink{classorg_1_1jgap_1_1impl_1_1_map_gene_a20a2cdecf740152ac3454df33e0d8449}{org.\-jgap.\-impl.\-Map\-Gene}, \hyperlink{classorg_1_1jgap_1_1impl_1_1_fixed_binary_gene_a9c3d7ad5bd78dbdf8b6c17278fe14f11}{org.\-jgap.\-impl.\-Fixed\-Binary\-Gene}, \hyperlink{classorg_1_1jgap_1_1impl_1_1_boolean_gene_ad1d9fb9f74b0905a92472f22050ab5b8}{org.\-jgap.\-impl.\-Boolean\-Gene}, \hyperlink{classorg_1_1jgap_1_1impl_1_1_set_gene_a5877331914ff77741c6bcde9531008a2}{org.\-jgap.\-impl.\-Set\-Gene}, and \hyperlink{classorg_1_1jgap_1_1impl_1_1_number_gene_a123a17deb1c4d63e97ade4ec25e2917f}{org.\-jgap.\-impl.\-Number\-Gene}.



Referenced by org.\-jgap.\-Base\-Gene.\-get\-Allele(), org.\-jgap.\-Base\-Gene.\-hash\-Code(), and org.\-jgap.\-Base\-Gene.\-to\-String().

\hypertarget{classorg_1_1jgap_1_1_base_gene_ab53d562a62f5c72fc88c7e5315a23ed7}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!get\-Unique\-I\-D@{get\-Unique\-I\-D}}
\index{get\-Unique\-I\-D@{get\-Unique\-I\-D}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{get\-Unique\-I\-D}]{\setlength{\rightskip}{0pt plus 5cm}String org.\-jgap.\-Base\-Gene.\-get\-Unique\-I\-D (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_ab53d562a62f5c72fc88c7e5315a23ed7}
\begin{DoxyReturn}{Returns}
unique I\-D of the gene, which allows to distinct this instance from others, in the best case worldwide
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-5 
\end{DoxySince}


Implements \hyperlink{interfaceorg_1_1jgap_1_1_i_unique_key_a1276412b558e169199771997a190f876}{org.\-jgap.\-I\-Unique\-Key}.



Definition at line 437 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-m\-\_\-unique\-I\-D.

\hypertarget{classorg_1_1jgap_1_1_base_gene_ad3c210e440f2b542391db5a909fda3b7}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!get\-Unique\-I\-D\-Template@{get\-Unique\-I\-D\-Template}}
\index{get\-Unique\-I\-D\-Template@{get\-Unique\-I\-D\-Template}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{get\-Unique\-I\-D\-Template}]{\setlength{\rightskip}{0pt plus 5cm}String org.\-jgap.\-Base\-Gene.\-get\-Unique\-I\-D\-Template (
\begin{DoxyParamCaption}
\item[{int}]{a\-\_\-index}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_ad3c210e440f2b542391db5a909fda3b7}

\begin{DoxyParams}{Parameters}
{\em a\-\_\-index} & the index of the template to retrieve the key for \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
String
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-5 
\end{DoxySince}


Implements \hyperlink{interfaceorg_1_1jgap_1_1_i_unique_key_af26fbe03487632a16c84b62325489652}{org.\-jgap.\-I\-Unique\-Key}.



Definition at line 465 of file Base\-Gene.\-java.

\hypertarget{classorg_1_1jgap_1_1_base_gene_af842a106d90ca973956f560ae448464e}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!hash\-Code@{hash\-Code}}
\index{hash\-Code@{hash\-Code}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{hash\-Code}]{\setlength{\rightskip}{0pt plus 5cm}int org.\-jgap.\-Base\-Gene.\-hash\-Code (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_af842a106d90ca973956f560ae448464e}
Retrieves the hash code value for a \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene}. Override if another \hyperlink{classorg_1_1jgap_1_1_base_gene_af842a106d90ca973956f560ae448464e}{hash\-Code()} implementation is necessary or more appropriate than this default implementation.

\begin{DoxyReturn}{Returns}
this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene}'s hash code
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Neil Rotstan 

Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
1.\-0 
\end{DoxySince}


Definition at line 118 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-get\-Internal\-Value().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_base_gene_af842a106d90ca973956f560ae448464e_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_base_gene_acd6431c06c81bc03b4abbe32e4d9d8ef}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!is\-Compare\-Application\-Data@{is\-Compare\-Application\-Data}}
\index{is\-Compare\-Application\-Data@{is\-Compare\-Application\-Data}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{is\-Compare\-Application\-Data}]{\setlength{\rightskip}{0pt plus 5cm}boolean org.\-jgap.\-Base\-Gene.\-is\-Compare\-Application\-Data (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_acd6431c06c81bc03b4abbe32e4d9d8ef}


Implements \hyperlink{interfaceorg_1_1jgap_1_1_gene_a71de32c2c6bf2b71332449f20d604f81}{org.\-jgap.\-Gene}.



Definition at line 302 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-m\-\_\-compare\-App\-Data.



Referenced by org.\-jgap.\-impl.\-Number\-Gene.\-compare\-To(), org.\-jgap.\-impl.\-Boolean\-Gene.\-compare\-To(), org.\-jgap.\-impl.\-String\-Gene.\-compare\-To(), org.\-jgap.\-impl.\-Fixed\-Binary\-Gene.\-compare\-To(), org.\-jgap.\-impl.\-Composite\-Gene.\-compare\-To(), and org.\-jgap.\-Base\-Gene.\-equals().

\hypertarget{classorg_1_1jgap_1_1_base_gene_a40f8f3d249145ad028892cd706ba9dae}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!new\-Gene@{new\-Gene}}
\index{new\-Gene@{new\-Gene}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{new\-Gene}]{\setlength{\rightskip}{0pt plus 5cm}{\bf Gene} org.\-jgap.\-Base\-Gene.\-new\-Gene (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_a40f8f3d249145ad028892cd706ba9dae}
Provides implementation-\/independent means for creating new \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} instances. The new instance that is created and returned should be setup with any implementation-\/dependent configuration that this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} instance is setup with (aside from the actual value, of course). For example, if this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} were setup with bounds on its value, then the \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} instance returned from this method should also be setup with those same bounds. This is important, as the J\-G\-A\-P core will invoke this method on each \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} in the sample \hyperlink{classorg_1_1jgap_1_1_chromosome}{Chromosome} in order to create each new \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} in the same respective gene position for a new \hyperlink{classorg_1_1jgap_1_1_chromosome}{Chromosome}.

\begin{DoxyReturn}{Returns}
a new \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} instance of the same type and with the same setup as this concrete \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene}
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Neil Rostan 

Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-6 (since 1.\-0 in Integer\-Gene) 
\end{DoxySince}
\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000041}{Todo}]clone app.\-data \end{DoxyRefDesc}


Implements \hyperlink{interfaceorg_1_1jgap_1_1_gene_ae4ddccb7bf569c0ed6cdc19def590964}{org.\-jgap.\-Gene}.



Definition at line 392 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-get\-Application\-Data(), org.\-jgap.\-Base\-Gene.\-get\-Constraint\-Checker(), org.\-jgap.\-Base\-Gene.\-get\-Energy(), and org.\-jgap.\-Base\-Gene.\-new\-Gene\-Internal().



Referenced by org.\-jgap.\-supergenes.\-Supergene\-Persistent\-Representation\-Test.\-test\-New\-Gene\-\_\-0(), org.\-jgap.\-impl.\-Map\-Gene\-Test.\-test\-New\-Gene\-\_\-0(), org.\-jgap.\-impl.\-Integer\-Gene\-Test.\-test\-New\-Gene\-\_\-0(), org.\-jgap.\-impl.\-Double\-Gene\-Test.\-test\-New\-Gene\-\_\-0(), org.\-jgap.\-impl.\-Composite\-Gene\-Test.\-test\-New\-Gene\-\_\-0(), org.\-jgap.\-impl.\-String\-Gene\-Test.\-test\-New\-Gene\-\_\-0(), org.\-jgap.\-impl.\-Fixed\-Binary\-Gene\-Test.\-test\-New\-Gene\-\_\-0(), org.\-jgap.\-supergenes.\-Supergene\-Persistent\-Representation\-Test.\-test\-New\-Gene\-\_\-1(), org.\-jgap.\-impl.\-Map\-Gene\-Test.\-test\-New\-Gene\-\_\-1(), org.\-jgap.\-supergenes.\-Supergene\-Persistent\-Representation\-Test.\-test\-New\-Gene\-\_\-2(), org.\-jgap.\-impl.\-Map\-Gene\-Test.\-test\-New\-Gene\-\_\-2(), and org.\-jgap.\-impl.\-Map\-Gene\-Test.\-test\-New\-Gene\-\_\-3().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_base_gene_a40f8f3d249145ad028892cd706ba9dae_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1_base_gene_aa423e96ffac5a9589fb3a4fbda791b3c}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!new\-Gene\-Internal@{new\-Gene\-Internal}}
\index{new\-Gene\-Internal@{new\-Gene\-Internal}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{new\-Gene\-Internal}]{\setlength{\rightskip}{0pt plus 5cm}abstract {\bf Gene} org.\-jgap.\-Base\-Gene.\-new\-Gene\-Internal (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [protected]}, {\ttfamily [pure virtual]}}}\label{classorg_1_1jgap_1_1_base_gene_aa423e96ffac5a9589fb3a4fbda791b3c}


Implemented in \hyperlink{classorg_1_1jgap_1_1impl_1_1_number_gene_test_1_1_number_gene_impl_a4dcdc2431e0b48f6d53f4cdbdc605996}{org.\-jgap.\-impl.\-Number\-Gene\-Test.\-Number\-Gene\-Impl}, \hyperlink{classorg_1_1jgap_1_1impl_1_1_composite_gene_ab71f66feb3b88bb9bdcab5682abd4b03}{org.\-jgap.\-impl.\-Composite\-Gene}, \hyperlink{classorg_1_1jgap_1_1impl_1_1_string_gene_ab9957400b5eb890f04032b5ce498cbf4}{org.\-jgap.\-impl.\-String\-Gene}, \hyperlink{classorg_1_1jgap_1_1supergenes_1_1_abstract_supergene_a1e30b794dee0e8da4d5a4d213e9fb2c1}{org.\-jgap.\-supergenes.\-Abstract\-Supergene}, \hyperlink{classorg_1_1jgap_1_1impl_1_1_boolean_gene_a57df173aaf3d927451500893e5cccaf3}{org.\-jgap.\-impl.\-Boolean\-Gene}, \hyperlink{classorg_1_1jgap_1_1impl_1_1_mutiple_integer_gene_a0d94e1fe37cbc3e7cda6481117f2a45a}{org.\-jgap.\-impl.\-Mutiple\-Integer\-Gene}, \hyperlink{classorg_1_1jgap_1_1impl_1_1_integer_gene_a225ed6a7409f890ce21009051eb7392f}{org.\-jgap.\-impl.\-Integer\-Gene}, \hyperlink{classorg_1_1jgap_1_1impl_1_1_double_gene_af714cdfc2b8f7f3e4d397cf149c09b95}{org.\-jgap.\-impl.\-Double\-Gene}, \hyperlink{classorg_1_1jgap_1_1impl_1_1_map_gene_aea5223852eb46cf96ffb04b4b16165b7}{org.\-jgap.\-impl.\-Map\-Gene}, \hyperlink{classorg_1_1jgap_1_1impl_1_1_fixed_binary_gene_acf83c8bd879cbbcf3e091eaf718594be}{org.\-jgap.\-impl.\-Fixed\-Binary\-Gene}, and \hyperlink{classorg_1_1jgap_1_1impl_1_1_set_gene_abe5acebc3f0d099bdb825e6ef7a7c233}{org.\-jgap.\-impl.\-Set\-Gene}.



Referenced by org.\-jgap.\-Base\-Gene.\-new\-Gene().

\hypertarget{classorg_1_1jgap_1_1_base_gene_afeb9a5f42018c057c8197fb0e7d57354}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!set\-Application\-Data@{set\-Application\-Data}}
\index{set\-Application\-Data@{set\-Application\-Data}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{set\-Application\-Data}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Base\-Gene.\-set\-Application\-Data (
\begin{DoxyParamCaption}
\item[{final Object}]{a\-\_\-new\-Data}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_afeb9a5f42018c057c8197fb0e7d57354}
This sets the application-\/specific data that is attached to this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene}. Attaching application-\/specific data may be useful for some applications when it comes time to distinguish a \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} from another. J\-G\-A\-P ignores this data functionally.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-new\-Data} & the new application-\/specific data to attach to this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene}\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-4 
\end{DoxySince}


Definition at line 262 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-m\-\_\-application\-Data.

\hypertarget{classorg_1_1jgap_1_1_base_gene_adad4ccd46efd7b2a3dc81bd0ba0c410c}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!set\-Compare\-Application\-Data@{set\-Compare\-Application\-Data}}
\index{set\-Compare\-Application\-Data@{set\-Compare\-Application\-Data}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{set\-Compare\-Application\-Data}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Base\-Gene.\-set\-Compare\-Application\-Data (
\begin{DoxyParamCaption}
\item[{final boolean}]{a\-\_\-do\-Compare}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_adad4ccd46efd7b2a3dc81bd0ba0c410c}
Should we also consider the application data when comparing? Default is \char`\"{}false\char`\"{} as \char`\"{}true\char`\"{} means a \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} is losing its identity when application data is set differently!


\begin{DoxyParams}{Parameters}
{\em a\-\_\-do\-Compare} & true\-: consider application data in method compare\-To\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-4 
\end{DoxySince}


Definition at line 292 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-m\-\_\-compare\-App\-Data.

\hypertarget{classorg_1_1jgap_1_1_base_gene_a334db8a543f3008719d78ee219826b65}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!set\-Constraint\-Checker@{set\-Constraint\-Checker}}
\index{set\-Constraint\-Checker@{set\-Constraint\-Checker}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{set\-Constraint\-Checker}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Base\-Gene.\-set\-Constraint\-Checker (
\begin{DoxyParamCaption}
\item[{final {\bf I\-Gene\-Constraint\-Checker}}]{a\-\_\-constraint\-Checker}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_a334db8a543f3008719d78ee219826b65}
Sets the constraint checker to be used for this gene whenever method \hyperlink{interfaceorg_1_1jgap_1_1_gene_af46a0c31e0fecea483497e87ca278127}{set\-Allele(\-Object)} is called. 
\begin{DoxyParams}{Parameters}
{\em a\-\_\-constraint\-Checker} & the constraint checker to be set\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-5 (moved from Composite\-Gene, where it was since 2.\-0) 
\end{DoxySince}


Implements \hyperlink{interfaceorg_1_1jgap_1_1_gene_a1ef8e9cd912cc68ebe6cd246cf896633}{org.\-jgap.\-Gene}.



Definition at line 358 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-m\-\_\-gene\-Allele\-Checker.

\hypertarget{classorg_1_1jgap_1_1_base_gene_afb39fbd8997971a140a9a4380ceecc4a}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!set\-Energy@{set\-Energy}}
\index{set\-Energy@{set\-Energy}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{set\-Energy}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Base\-Gene.\-set\-Energy (
\begin{DoxyParamCaption}
\item[{final double}]{a\-\_\-energy}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_afb39fbd8997971a140a9a4380ceecc4a}
Sets the energy of the gene 
\begin{DoxyParams}{Parameters}
{\em a\-\_\-energy} & the energy to set\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
2.\-3 
\end{DoxySince}


Definition at line 246 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-m\-\_\-energy.

\hypertarget{classorg_1_1jgap_1_1_base_gene_aa7d9f1c3c2e727d496d66a85818c17d0}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!set\-Unique\-I\-D\-Template@{set\-Unique\-I\-D\-Template}}
\index{set\-Unique\-I\-D\-Template@{set\-Unique\-I\-D\-Template}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{set\-Unique\-I\-D\-Template}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-Base\-Gene.\-set\-Unique\-I\-D\-Template (
\begin{DoxyParamCaption}
\item[{String}]{a\-\_\-template\-I\-D, }
\item[{int}]{a\-\_\-index}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_aa7d9f1c3c2e727d496d66a85818c17d0}
A template is a gene that is the logical predecessor of the current gene. A template can occur in mutation or crossing over. In the latter case can be at least two template genes. This is why in this setter method the parameter a\-\_\-index exists.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-template\-I\-D} & the unique I\-D of the template \\
\hline
{\em a\-\_\-index} & the index of the template, e.\-g. in crossing over for the second candidate gene this is 2\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-5 
\end{DoxySince}


Implements \hyperlink{interfaceorg_1_1jgap_1_1_i_unique_key_aeac223774f93ba6640bc07cc97a9ef4c}{org.\-jgap.\-I\-Unique\-Key}.



Definition at line 454 of file Base\-Gene.\-java.

\hypertarget{classorg_1_1jgap_1_1_base_gene_a26c00775111b327b178288164ed9c667}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!size@{size}}
\index{size@{size}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{size}]{\setlength{\rightskip}{0pt plus 5cm}int org.\-jgap.\-Base\-Gene.\-size (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_a26c00775111b327b178288164ed9c667}
\begin{DoxyReturn}{Returns}
the size of the gene, i.\-e the number of atomic elements. Always 1 for non-\/composed \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} types. Override for composed \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene} types
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Neil Rotstan 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
1.\-0 
\end{DoxySince}


Implements \hyperlink{interfaceorg_1_1jgap_1_1_gene_a7a975e180dd0ac0fd33cdc182f64bea8}{org.\-jgap.\-Gene}.



Definition at line 178 of file Base\-Gene.\-java.

\hypertarget{classorg_1_1jgap_1_1_base_gene_a935dfec06cb77bff0d2a45859d6a35b7}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!to\-String@{to\-String}}
\index{to\-String@{to\-String}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{to\-String}]{\setlength{\rightskip}{0pt plus 5cm}String org.\-jgap.\-Base\-Gene.\-to\-String (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1_base_gene_a935dfec06cb77bff0d2a45859d6a35b7}
Retrieves a string representation of this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene}'s value that may be useful for display purposes.

\begin{DoxyReturn}{Returns}
a string representation of this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene}'s value
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
1.\-0 
\end{DoxySince}


Implements \hyperlink{interfaceorg_1_1jgap_1_1_gene_ac62239b4b7bf81179fd491a4c4bd1901}{org.\-jgap.\-Gene}.



Definition at line 152 of file Base\-Gene.\-java.



References org.\-jgap.\-Base\-Gene.\-get\-Application\-Data(), org.\-jgap.\-Base\-Gene.\-get\-Internal\-Value(), and org.\-jgap.\-Base\-Gene.\-S\-\_\-\-A\-P\-P\-L\-I\-C\-A\-T\-I\-O\-N\-\_\-\-D\-A\-T\-A.



Referenced by org.\-jgap.\-impl.\-Set\-Gene.\-get\-Persistent\-Representation(), org.\-jgap.\-impl.\-Number\-Gene\-Test.\-Number\-Gene\-Impl.\-get\-Persistent\-Representation(), org.\-jgap.\-impl.\-Set\-Gene\-Test.\-test\-To\-String\-\_\-0(), and org.\-jgap.\-impl.\-Set\-Gene\-Test.\-test\-To\-String\-\_\-1().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1_base_gene_a935dfec06cb77bff0d2a45859d6a35b7_cgraph}
\end{center}
\end{figure}




\subsection{Member Data Documentation}
\hypertarget{classorg_1_1jgap_1_1_base_gene_a7e83775c4b2b9391ec833869808d0154}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N@{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N}}
\index{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N@{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N}]{\setlength{\rightskip}{0pt plus 5cm}final String org.\-jgap.\-Base\-Gene.\-C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N = \char`\"{}\$Revision\-: 1.\-28 \$\char`\"{}\hspace{0.3cm}{\ttfamily [static]}, {\ttfamily [private]}}}\label{classorg_1_1jgap_1_1_base_gene_a7e83775c4b2b9391ec833869808d0154}
String containing the C\-V\-S revision. Read out via reflection! 

Definition at line 26 of file Base\-Gene.\-java.

\hypertarget{classorg_1_1jgap_1_1_base_gene_a86edb3d2c29e5064305fabc2dd59a3b9}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!D\-E\-L\-T\-A@{D\-E\-L\-T\-A}}
\index{D\-E\-L\-T\-A@{D\-E\-L\-T\-A}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{D\-E\-L\-T\-A}]{\setlength{\rightskip}{0pt plus 5cm}final double org.\-jgap.\-Base\-Gene.\-D\-E\-L\-T\-A = 0.\-0000001\hspace{0.3cm}{\ttfamily [static]}}}\label{classorg_1_1jgap_1_1_base_gene_a86edb3d2c29e5064305fabc2dd59a3b9}
Delta, useful for comparing doubles and floats. 

Definition at line 36 of file Base\-Gene.\-java.

\hypertarget{classorg_1_1jgap_1_1_base_gene_ab2e121e812be7a4dead5ca7e300ac3d7}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!m\-\_\-application\-Data@{m\-\_\-application\-Data}}
\index{m\-\_\-application\-Data@{m\-\_\-application\-Data}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{m\-\_\-application\-Data}]{\setlength{\rightskip}{0pt plus 5cm}Object org.\-jgap.\-Base\-Gene.\-m\-\_\-application\-Data\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1_base_gene_ab2e121e812be7a4dead5ca7e300ac3d7}
Application-\/specific data that is attached to the \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene}. This data may assist the application in labelling this \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene}. J\-G\-A\-P ignores the data, aside from allowing it to be set and retrieved and considering it in clone() and compare\-To().

\begin{DoxySince}{Since}
2.\-4 
\end{DoxySince}


Definition at line 49 of file Base\-Gene.\-java.



Referenced by org.\-jgap.\-Base\-Gene.\-get\-Application\-Data(), and org.\-jgap.\-Base\-Gene.\-set\-Application\-Data().

\hypertarget{classorg_1_1jgap_1_1_base_gene_a27bb3cd52e8cd52fffc4953f172709a2}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!m\-\_\-compare\-App\-Data@{m\-\_\-compare\-App\-Data}}
\index{m\-\_\-compare\-App\-Data@{m\-\_\-compare\-App\-Data}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{m\-\_\-compare\-App\-Data}]{\setlength{\rightskip}{0pt plus 5cm}boolean org.\-jgap.\-Base\-Gene.\-m\-\_\-compare\-App\-Data\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1_base_gene_a27bb3cd52e8cd52fffc4953f172709a2}
Method compare\-To()\-: Should we also consider the application data when comparing? Default is \char`\"{}false\char`\"{} as \char`\"{}true\char`\"{} means a \hyperlink{interfaceorg_1_1jgap_1_1_gene}{Gene}'s losing its identity when application data is set differently!

\begin{DoxySince}{Since}
2.\-4 
\end{DoxySince}


Definition at line 58 of file Base\-Gene.\-java.



Referenced by org.\-jgap.\-Base\-Gene.\-is\-Compare\-Application\-Data(), and org.\-jgap.\-Base\-Gene.\-set\-Compare\-Application\-Data().

\hypertarget{classorg_1_1jgap_1_1_base_gene_a4d41d885b45ce031c33c88214f98019e}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!m\-\_\-configuration@{m\-\_\-configuration}}
\index{m\-\_\-configuration@{m\-\_\-configuration}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{m\-\_\-configuration}]{\setlength{\rightskip}{0pt plus 5cm}{\bf Configuration} org.\-jgap.\-Base\-Gene.\-m\-\_\-configuration\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1_base_gene_a4d41d885b45ce031c33c88214f98019e}


Definition at line 60 of file Base\-Gene.\-java.



Referenced by org.\-jgap.\-Base\-Gene.\-Base\-Gene(), and org.\-jgap.\-Base\-Gene.\-get\-Configuration().

\hypertarget{classorg_1_1jgap_1_1_base_gene_a8553da900cf30fc4b0f5463404b86a8e}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!m\-\_\-energy@{m\-\_\-energy}}
\index{m\-\_\-energy@{m\-\_\-energy}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{m\-\_\-energy}]{\setlength{\rightskip}{0pt plus 5cm}double org.\-jgap.\-Base\-Gene.\-m\-\_\-energy\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1_base_gene_a8553da900cf30fc4b0f5463404b86a8e}
Energy of a gene, see R\-F\-E 1102206 

Definition at line 39 of file Base\-Gene.\-java.



Referenced by org.\-jgap.\-Base\-Gene.\-get\-Energy(), and org.\-jgap.\-Base\-Gene.\-set\-Energy().

\hypertarget{classorg_1_1jgap_1_1_base_gene_a846344a347601679d0935dcdba7f7e6f}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!m\-\_\-gene\-Allele\-Checker@{m\-\_\-gene\-Allele\-Checker}}
\index{m\-\_\-gene\-Allele\-Checker@{m\-\_\-gene\-Allele\-Checker}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{m\-\_\-gene\-Allele\-Checker}]{\setlength{\rightskip}{0pt plus 5cm}{\bf I\-Gene\-Constraint\-Checker} org.\-jgap.\-Base\-Gene.\-m\-\_\-gene\-Allele\-Checker\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1_base_gene_a846344a347601679d0935dcdba7f7e6f}
Optional helper class for checking if a given allele value to be set for a given gene is valid. If not, the allele value may not be set for the gene or the gene type (e.\-g. Integer\-Gene) is not allowed in general!

\begin{DoxySince}{Since}
2.\-5 (moved from Composite\-Gene, where it was since 2.\-0) 
\end{DoxySince}


Definition at line 348 of file Base\-Gene.\-java.



Referenced by org.\-jgap.\-Base\-Gene.\-get\-Constraint\-Checker(), and org.\-jgap.\-Base\-Gene.\-set\-Constraint\-Checker().

\hypertarget{classorg_1_1jgap_1_1_base_gene_a8cd57f3a57f5a892f590df9e06b82e47}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!m\-\_\-unique\-I\-D@{m\-\_\-unique\-I\-D}}
\index{m\-\_\-unique\-I\-D@{m\-\_\-unique\-I\-D}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{m\-\_\-unique\-I\-D}]{\setlength{\rightskip}{0pt plus 5cm}String org.\-jgap.\-Base\-Gene.\-m\-\_\-unique\-I\-D\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1_base_gene_a8cd57f3a57f5a892f590df9e06b82e47}
Unique I\-D of the gene that allows to distinct it from other genes. In the best case, this I\-D is unique worldwide. 

Definition at line 66 of file Base\-Gene.\-java.



Referenced by org.\-jgap.\-Base\-Gene.\-Base\-Gene(), and org.\-jgap.\-Base\-Gene.\-get\-Unique\-I\-D().

\hypertarget{classorg_1_1jgap_1_1_base_gene_a3f14fef3c7107830f24306cf7bb0abf0}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!m\-\_\-unique\-I\-D\-Templates@{m\-\_\-unique\-I\-D\-Templates}}
\index{m\-\_\-unique\-I\-D\-Templates@{m\-\_\-unique\-I\-D\-Templates}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{m\-\_\-unique\-I\-D\-Templates}]{\setlength{\rightskip}{0pt plus 5cm}Map$<$Integer,String$>$ org.\-jgap.\-Base\-Gene.\-m\-\_\-unique\-I\-D\-Templates\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1_base_gene_a3f14fef3c7107830f24306cf7bb0abf0}
In case mutation, crossing over etc. happened, this sequence gives evidence about the parent(s) of the current gene. 

Definition at line 72 of file Base\-Gene.\-java.



Referenced by org.\-jgap.\-Base\-Gene.\-Base\-Gene().

\hypertarget{classorg_1_1jgap_1_1_base_gene_acc4b1c46f3508b835bf95b9058eae3a0}{\index{org\-::jgap\-::\-Base\-Gene@{org\-::jgap\-::\-Base\-Gene}!S\-\_\-\-A\-P\-P\-L\-I\-C\-A\-T\-I\-O\-N\-\_\-\-D\-A\-T\-A@{S\-\_\-\-A\-P\-P\-L\-I\-C\-A\-T\-I\-O\-N\-\_\-\-D\-A\-T\-A}}
\index{S\-\_\-\-A\-P\-P\-L\-I\-C\-A\-T\-I\-O\-N\-\_\-\-D\-A\-T\-A@{S\-\_\-\-A\-P\-P\-L\-I\-C\-A\-T\-I\-O\-N\-\_\-\-D\-A\-T\-A}!org::jgap::BaseGene@{org\-::jgap\-::\-Base\-Gene}}
\subsubsection[{S\-\_\-\-A\-P\-P\-L\-I\-C\-A\-T\-I\-O\-N\-\_\-\-D\-A\-T\-A}]{\setlength{\rightskip}{0pt plus 5cm}final String org.\-jgap.\-Base\-Gene.\-S\-\_\-\-A\-P\-P\-L\-I\-C\-A\-T\-I\-O\-N\-\_\-\-D\-A\-T\-A = \char`\"{}Application data\char`\"{}\hspace{0.3cm}{\ttfamily [static]}}}\label{classorg_1_1jgap_1_1_base_gene_acc4b1c46f3508b835bf95b9058eae3a0}
Constants for \hyperlink{classorg_1_1jgap_1_1_base_gene_a935dfec06cb77bff0d2a45859d6a35b7}{to\-String()} 

Definition at line 31 of file Base\-Gene.\-java.



Referenced by org.\-jgap.\-Base\-Gene\-Test.\-test\-To\-String\-\_\-0(), org.\-jgap.\-impl.\-Number\-Gene\-Test.\-test\-To\-String\-\_\-0(), org.\-jgap.\-Base\-Gene\-Test.\-test\-To\-String\-\_\-1(), org.\-jgap.\-Base\-Gene\-Test.\-test\-To\-String\-\_\-2(), and org.\-jgap.\-Base\-Gene.\-to\-String().



The documentation for this class was generated from the following file\-:\begin{DoxyCompactItemize}
\item 
L\-D\-H\-\_\-\-Git/src/org/jgap/\hyperlink{_base_gene_8java}{Base\-Gene.\-java}\end{DoxyCompactItemize}
