\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype}{\section{org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype Class Reference}
\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype}\index{org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype@{org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype}}
}


Inheritance diagram for org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=309pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype__inherit__graph}
\end{center}
\end{figure}


Collaboration diagram for org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype__coll__graph}
\end{center}
\end{figure}
\subsection*{Classes}
\begin{DoxyCompactItemize}
\item 
class {\bfseries G\-P\-Fitness\-Comparator}
\end{DoxyCompactItemize}
\subsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a66b8b63a2379f9b4f4e94ee3cc6974ad}{G\-P\-Genotype} ()  throws Invalid\-Configuration\-Exception 
\item 
\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a470bc360f57da2d313df69c51f9e7c3e}{G\-P\-Genotype} (\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} a\-\_\-configuration, \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{G\-P\-Population} a\-\_\-population, Class\mbox{[}$\,$\mbox{]} a\-\_\-types, Class\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-arg\-Types, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-node\-Sets, int\mbox{[}$\,$\mbox{]} a\-\_\-min\-Depths, int\mbox{[}$\,$\mbox{]} a\-\_\-max\-Depths, int a\-\_\-max\-Nodes)  throws Invalid\-Configuration\-Exception 
\item 
\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a71854e2aa9fc4e178a0c768efe9c80ad}{G\-P\-Genotype} (\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} a\-\_\-configuration, \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{G\-P\-Population} a\-\_\-population, Class\mbox{[}$\,$\mbox{]} a\-\_\-types, Class\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-arg\-Types, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-node\-Sets, int\mbox{[}$\,$\mbox{]} a\-\_\-min\-Depths, int\mbox{[}$\,$\mbox{]} a\-\_\-max\-Depths, int a\-\_\-max\-Nodes, I\-G\-P\-Population\-Initializer a\-\_\-pop\-Creator)  throws Invalid\-Configuration\-Exception 
\item 
\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a318ae291daca64dc4ba0b1e5085b4b87}{get\-G\-P\-Configuration} ()
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a43707d524245f50c92094e9e97568089}{evolve} (int a\-\_\-evolutions)
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_aa465c7cbfb45d77644b3d195782a12c9}{calc\-Fitness} ()
\item 
\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_afb8e0283f47eb7c0a0b6e09c8aeccc96}{get\-All\-Time\-Best} ()
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_abeef3282114d1313c97045dea004afc4}{output\-Solution} (\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} a\-\_\-best)
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ad495c700786596290a63e3d88a902766}{evolve} ()
\item 
\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{G\-P\-Population} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a9a2316294bf28b413eb8b0e288cd150d}{get\-G\-P\-Population} ()
\item 
double \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a631996e8862da78bdb67b76687ad1873}{get\-Total\-Fitness} ()
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a184964a4fc96748de13cad57bee24ad0}{run} ()
\item 
synchronized \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a2cff99f216e87c7e4b25c217c5bdc5af}{get\-Fittest\-Program} ()
\item 
synchronized \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a3a1e955f3a6c035913e311570591c683}{get\-Fittest\-Program\-Computed} ()
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a2d46992095125ecad698dc9e1a72b513}{set\-G\-P\-Configuration} (\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} a\-\_\-configuration)
\item 
boolean \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a847cbf8f0b5e3e223881ac1dfe5e36d3}{equals} (Object a\-\_\-other)
\item 
int \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_aef87073769e83d93b52d8df73ebb35dc}{compare\-To} (Object a\-\_\-other)
\item 
int \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a9b8cb5e7395359743bbee7658be90a36}{hash\-Code} ()
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a2cda11ec40667a6cb061a4c87ace0b6c}{set\-Verbose\-Output} (boolean a\-\_\-verbose)
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a0b615fba7524275eb09b3d7929d73329}{put\-Variable} (Variable a\-\_\-var)
\item 
Variable \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a66770c2d4fb9645f33afec5dc37bb47c}{get\-Variable} (String a\-\_\-var\-Name)
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a8ada3482e8689629778fe77164afb271}{add\-Fittest\-Program} (final \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} a\-\_\-to\-Add)
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a4327e7a69fcbcfe051e00e46c933ea7a}{fill\-Population} (final int a\-\_\-num)  throws Invalid\-Configuration\-Exception 
\end{DoxyCompactItemize}
\subsection*{Static Public Member Functions}
\begin{DoxyCompactItemize}
\item 
static \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype}{G\-P\-Genotype} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a59354275c3ff7c6b47c4c676c13a4422}{random\-Initial\-Genotype} (final \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} a\-\_\-conf, Class\mbox{[}$\,$\mbox{]} a\-\_\-types, Class\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-arg\-Types, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-node\-Sets, int a\-\_\-max\-Nodes, boolean a\-\_\-verbose\-Output)  throws Invalid\-Configuration\-Exception 
\item 
static \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype}{G\-P\-Genotype} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a7e59aff645b8386adcad62e9ccdfd00d}{random\-Initial\-Genotype} (final \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} a\-\_\-conf, Class\mbox{[}$\,$\mbox{]} a\-\_\-types, Class\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-arg\-Types, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-node\-Sets, int\mbox{[}$\,$\mbox{]} a\-\_\-min\-Depths, int\mbox{[}$\,$\mbox{]} a\-\_\-max\-Depths, int a\-\_\-max\-Nodes, boolean a\-\_\-verbose\-Output)  throws Invalid\-Configuration\-Exception 
\item 
static \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype}{G\-P\-Genotype} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a47a8848949757df02df0b4503bd2bea7}{random\-Initial\-Genotype} (final \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} a\-\_\-conf, Class\mbox{[}$\,$\mbox{]} a\-\_\-types, Class\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-arg\-Types, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-node\-Sets, int\mbox{[}$\,$\mbox{]} a\-\_\-min\-Depths, int\mbox{[}$\,$\mbox{]} a\-\_\-max\-Depths, int a\-\_\-max\-Nodes, boolean\mbox{[}$\,$\mbox{]} a\-\_\-full\-Mode\-Allowed, boolean a\-\_\-verbose\-Output)  throws Invalid\-Configuration\-Exception 
\item 
static \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype}{G\-P\-Genotype} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af3246cb6a802ae1cde833013c130f6e9}{random\-Initial\-Genotype} (final \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} a\-\_\-conf, Class\mbox{[}$\,$\mbox{]} a\-\_\-types, Class\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-arg\-Types, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-node\-Sets, int\mbox{[}$\,$\mbox{]} a\-\_\-min\-Depths, int\mbox{[}$\,$\mbox{]} a\-\_\-max\-Depths, int a\-\_\-max\-Nodes, boolean\mbox{[}$\,$\mbox{]} a\-\_\-full\-Mode\-Allowed, boolean a\-\_\-verbose\-Output, \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_population_creator}{I\-Population\-Creator} a\-\_\-pop\-Creator)  throws Invalid\-Configuration\-Exception 
\item 
static \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a07e4b22e1cd10bbc198278efd83d27ce}{get\-Static\-G\-P\-Configuration} ()
\item 
static void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a7750ff9b7bd73efcfff522365f4650d7}{set\-Static\-G\-P\-Configuration} (\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} a\-\_\-configuration)
\item 
static void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a71cc6d10d81d132ee542767aca12f502}{check\-Erroneous\-Pop} (\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{G\-P\-Population} pop, String s)
\item 
static void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af886a583789e547400134f2ad533cf8b}{check\-Erroneous\-Pop} (\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{G\-P\-Population} a\-\_\-pop, String a\-\_\-s, boolean a\-\_\-clear\-Fitness)
\item 
static void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a6cab9ecf163e4eec850da3cc63df8665}{check\-Erroneous\-Pop} (\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{G\-P\-Population} a\-\_\-pop, String a\-\_\-s, boolean a\-\_\-clear\-Fitness, boolean a\-\_\-active)
\item 
static void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a4104bc4b934e717c15e3a7b5e50a0748}{check\-Erroneous\-Prog} (\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} prog, String s)
\item 
static void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a9d986d5030ce180fc402a2a73d22c42d}{check\-Erroneous\-Prog} (\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} a\-\_\-prog, String a\-\_\-s, boolean a\-\_\-clear\-Fitness)
\item 
static void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_aa3bf0b50a73684ff3fe1177017cf9bb8}{check\-Erroneous\-Prog} (\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} a\-\_\-prog, String s, boolean a\-\_\-clear\-Fitness, boolean a\-\_\-active)
\item 
static Map$<$ String, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene} $>$ \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a77909da0c746ca16cdcbf0af9b1c0927}{verify\-Child\-Nodes} (\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} a\-\_\-conf, Class\mbox{[}$\,$\mbox{]} a\-\_\-types, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-node\-Sets)
\item 
static Map$<$ \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}, int\mbox{[}$\,$\mbox{]}$>$ \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ad264330c888b133296578decc78f265e}{verify\-Depths\-For\-Nodes} (\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{G\-P\-Population} a\-\_\-pop, \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} a\-\_\-conf, Class\mbox{[}$\,$\mbox{]} a\-\_\-types, int\mbox{[}$\,$\mbox{]} a\-\_\-min\-Depths, int\mbox{[}$\,$\mbox{]} a\-\_\-max\-Depths, int a\-\_\-max\-Nodes, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} a\-\_\-node\-Sets)
\end{DoxyCompactItemize}
\subsection*{Protected Member Functions}
\begin{DoxyCompactItemize}
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a6b017f0f0e028cc1a9164782c5ce5095}{init} ()
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a11dce8ec844c102bc151e4ab81c52ac7}{set\-G\-P\-Population} (\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{G\-P\-Population} a\-\_\-pop)
\end{DoxyCompactItemize}
\subsection*{Static Protected Member Functions}
\begin{DoxyCompactItemize}
\item 
static boolean \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ae3828fb2c7d3c3cc4ee2c1562c9851d8}{node\-Exists} (\hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]} a\-\_\-function\-Set, Class a\-\_\-return\-Type, int a\-\_\-sub\-Return\-Type)
\item 
static void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ae5eb2b5f82dd89d6e8287eac90864cb1}{output\-Warning} (Map$<$ String, \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene} $>$ invalid\-Nodes)
\item 
static void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a95fd91fcd94500415c67e18388695a61}{output\-Depth\-Info} (Map$<$ \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}, int\mbox{[}$\,$\mbox{]}$>$ a\-\_\-invalid\-Depths)
\end{DoxyCompactItemize}
\subsection*{Private Member Functions}
\begin{DoxyCompactItemize}
\item 
\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_aa21cc2dde0ca8cbc6048b37e62dd1593}{clone\-Program} (\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} a\-\_\-original)
\item 
void \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ac726d5312c002c60c239a9be1bc7cfba}{write\-To\-File} (\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} i1, \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} i2, \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} inew, String header)
\item 
String\-Buffer \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a356baf22df15fe6c0c88aadb45a80a5e}{get\-Program\-String} (\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} i1)
\end{DoxyCompactItemize}
\subsection*{Private Attributes}
\begin{DoxyCompactItemize}
\item 
\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{G\-P\-Population} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a32bf2d50f238455154d3466fc1ef09a1}{m\-\_\-population}
\item 
\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a2156c601b24d1a7fdcad1c59ec05e147}{m\-\_\-configuration}
\item 
double \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a957b0de7013e1055b66e03223dc890f2}{m\-\_\-best\-Fitness}
\item 
double \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a590812317593272cd66908ef7b712a63}{m\-\_\-total\-Fitness}
\item 
\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a6bd9f97a0bb39ec138f6d8ca8ebd25e5}{m\-\_\-all\-Time\-Best}
\item 
double \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a869b5d4ab431717009978a9d870bf5d8}{m\-\_\-all\-Time\-Best\-Fitness}
\item 
boolean \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a3bfbce17cabd5ef77caf8427f5511407}{m\-\_\-full\-Mode\-Allowed} \mbox{[}$\,$\mbox{]}
\item 
Class\mbox{[}$\,$\mbox{]} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ae23e8520ee01ca7cb8f927a262db49c6}{m\-\_\-types}
\item 
Class\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a8e26c3d41a0e1922310e9d966f80238f}{m\-\_\-arg\-Types}
\item 
\hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a045ec05f97be18ddbd121eb709d3e450}{m\-\_\-node\-Sets}
\item 
int\mbox{[}$\,$\mbox{]} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af87aee9616f225ec99bf2f3db601e492}{m\-\_\-min\-Depths}
\item 
int\mbox{[}$\,$\mbox{]} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a7e95fd6aefeefa158ca044abccc0ffa9}{m\-\_\-max\-Depths}
\item 
int \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ab15e0e8d3ee0ba8c1e56727f2a2f32ed}{m\-\_\-max\-Nodes}
\item 
boolean \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ac5e56e43bba8e3368cacb3111a5fd60a}{m\-\_\-verbose}
\item 
Map \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a03d40b9502d5bbbe9855cef68c9c8123}{m\-\_\-variables}
\item 
\hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program}{I\-G\-P\-Program} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af4a221f34692a5b47147dbbd31d26d23}{m\-\_\-fittest\-To\-Add}
\item 
boolean \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af0ec08e1f1fdf88d76bae2ba6d0fe986}{m\-\_\-clone\-Warning\-G\-P\-Program\-Shown}
\end{DoxyCompactItemize}
\subsection*{Static Private Attributes}
\begin{DoxyCompactItemize}
\item 
static final String \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ab88a16d04e096c2c2e1bfc4477ea7310}{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N} = \char`\"{}\$Revision\-: 1.\-60 \$\char`\"{}
\item 
static transient Logger \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ada30eb741391730f266047d15cd40338}{L\-O\-G\-G\-E\-R} = Logger.\-get\-Logger(G\-P\-Genotype.\-class)
\item 
static transient \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a9e2b1a8103181690f4bb6e584463214c}{m\-\_\-static\-Configuration}
\end{DoxyCompactItemize}


\subsection{Detailed Description}
\hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} for G\-P Programs.

\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 29 of file G\-P\-Genotype.\-java.



\subsection{Constructor \& Destructor Documentation}
\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a66b8b63a2379f9b4f4e94ee3cc6974ad}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!G\-P\-Genotype@{G\-P\-Genotype}}
\index{G\-P\-Genotype@{G\-P\-Genotype}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{G\-P\-Genotype}]{\setlength{\rightskip}{0pt plus 5cm}org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a66b8b63a2379f9b4f4e94ee3cc6974ad}
Default constructor. Ony use with dynamic instantiation. 
\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 121 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-init().



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-compare\-To(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-random\-Initial\-Genotype().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a66b8b63a2379f9b4f4e94ee3cc6974ad_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a470bc360f57da2d313df69c51f9e7c3e}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!G\-P\-Genotype@{G\-P\-Genotype}}
\index{G\-P\-Genotype@{G\-P\-Genotype}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{G\-P\-Genotype}]{\setlength{\rightskip}{0pt plus 5cm}org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype (
\begin{DoxyParamCaption}
\item[{{\bf G\-P\-Configuration}}]{a\-\_\-configuration, }
\item[{{\bf G\-P\-Population}}]{a\-\_\-population, }
\item[{Class\mbox{[}$\,$\mbox{]}}]{a\-\_\-types, }
\item[{Class}]{a\-\_\-arg\-Types\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{{\bf Command\-Gene}}]{a\-\_\-node\-Sets\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-min\-Depths, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-max\-Depths, }
\item[{int}]{a\-\_\-max\-Nodes}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a470bc360f57da2d313df69c51f9e7c3e}
Preferred constructor to use, if not using the static method random\-Initial\-Genotype.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-configuration} & the configuration to use \\
\hline
{\em a\-\_\-population} & the initialized population to use \\
\hline
{\em a\-\_\-types} & the type for each chromosome, the length of the array represents the number of chromosomes \\
\hline
{\em a\-\_\-arg\-Types} & the types of the arguments to each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of arguments to the chromosome \\
\hline
{\em a\-\_\-node\-Sets} & the nodes which are allowed to be used by each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of nodes \\
\hline
{\em a\-\_\-min\-Depths} & contains the minimum depth allowed for each chromosome \\
\hline
{\em a\-\_\-max\-Depths} & contains the maximum depth allowed for each chromosome \\
\hline
{\em a\-\_\-max\-Nodes} & reserve space for a\-\_\-max\-Nodes number of nodes\\
\hline
\end{DoxyParams}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 150 of file G\-P\-Genotype.\-java.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a71854e2aa9fc4e178a0c768efe9c80ad}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!G\-P\-Genotype@{G\-P\-Genotype}}
\index{G\-P\-Genotype@{G\-P\-Genotype}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{G\-P\-Genotype}]{\setlength{\rightskip}{0pt plus 5cm}org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype (
\begin{DoxyParamCaption}
\item[{{\bf G\-P\-Configuration}}]{a\-\_\-configuration, }
\item[{{\bf G\-P\-Population}}]{a\-\_\-population, }
\item[{Class\mbox{[}$\,$\mbox{]}}]{a\-\_\-types, }
\item[{Class}]{a\-\_\-arg\-Types\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{{\bf Command\-Gene}}]{a\-\_\-node\-Sets\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-min\-Depths, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-max\-Depths, }
\item[{int}]{a\-\_\-max\-Nodes, }
\item[{I\-G\-P\-Population\-Initializer}]{a\-\_\-pop\-Creator}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a71854e2aa9fc4e178a0c768efe9c80ad}
See above constructor. Parameter a\-\_\-pop\-Creator is used in grid context only.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-configuration} & \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration}{G\-P\-Configuration} \\
\hline
{\em a\-\_\-population} & \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{G\-P\-Population} \\
\hline
{\em a\-\_\-types} & Class\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-arg\-Types} & Class\mbox{[}\mbox{]}\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-node\-Sets} & \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}\mbox{]}\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-min\-Depths} & int\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-max\-Depths} & int\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-max\-Nodes} & int \\
\hline
{\em a\-\_\-pop\-Creator} & \hyperlink{interfaceorg_1_1jgap_1_1gp_1_1_i_population_creator}{I\-Population\-Creator} \\
\hline
\end{DoxyParams}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-6 
\end{DoxySince}


Definition at line 176 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Configuration(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-init(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-all\-Time\-Best\-Fitness, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-arg\-Types, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-max\-Depths, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-max\-Nodes, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-min\-Depths, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-node\-Sets, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-types, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-variables, org.\-jgap.\-Fitness\-Function.\-N\-O\-\_\-\-F\-I\-T\-N\-E\-S\-S\-\_\-\-V\-A\-L\-U\-E, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-set\-G\-P\-Configuration(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-set\-G\-P\-Population().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a71854e2aa9fc4e178a0c768efe9c80ad_cgraph}
\end{center}
\end{figure}




\subsection{Member Function Documentation}
\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a8ada3482e8689629778fe77164afb271}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!add\-Fittest\-Program@{add\-Fittest\-Program}}
\index{add\-Fittest\-Program@{add\-Fittest\-Program}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{add\-Fittest\-Program}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-add\-Fittest\-Program (
\begin{DoxyParamCaption}
\item[{final {\bf I\-G\-P\-Program}}]{a\-\_\-to\-Add}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a8ada3482e8689629778fe77164afb271}
Adds a G\-P program to this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype}. Does nothing when given null. The injection is actually executed in method create(..) of \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population}{G\-P\-Population}.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-to\-Add} & the program to add\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2 
\end{DoxySince}


Definition at line 1180 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-fittest\-To\-Add.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-calc\-Fitness().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_aa465c7cbfb45d77644b3d195782a12c9}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!calc\-Fitness@{calc\-Fitness}}
\index{calc\-Fitness@{calc\-Fitness}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{calc\-Fitness}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-calc\-Fitness (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_aa465c7cbfb45d77644b3d195782a12c9}
Calculates the fitness value of all programs, of the best solution as well as the total fitness (sum of all fitness values).

\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}
\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000144}{Todo}]get information from fitness function how calculation happened. In case of Robocode\-: return the robot competed against, in case the -\/enemies option was used without -\/battle\-All \end{DoxyRefDesc}


\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000145}{Todo}]do something similar here as with Genotype.\-preserve\-Fittest\-Chromosome \end{DoxyRefDesc}


\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000146}{Todo}]introduce global value object to be passed to the listener \end{DoxyRefDesc}


\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000147}{Todo}]should not happen but does with ensure\-Uniqueness(..) \end{DoxyRefDesc}


Definition at line 557 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-add\-Fittest\-Program(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Configuration(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Population(), org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-G\-P\-Program(), org.\-jgap.\-gp.\-I\-G\-P\-Fitness\-Evaluator.\-is\-Fitter(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-all\-Time\-Best, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-all\-Time\-Best\-Fitness, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-best\-Fitness, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-clone\-Warning\-G\-P\-Program\-Shown, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-total\-Fitness, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-verbose, org.\-jgap.\-Fitness\-Function.\-N\-O\-\_\-\-F\-I\-T\-N\-E\-S\-S\-\_\-\-V\-A\-L\-U\-E, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-output\-Solution(), org.\-jgap.\-I\-Handler.\-perform(), and org.\-jgap.\-gp.\-I\-G\-P\-Program.\-to\-String\-Norm().



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-run().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_aa465c7cbfb45d77644b3d195782a12c9_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a71cc6d10d81d132ee542767aca12f502}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!check\-Erroneous\-Pop@{check\-Erroneous\-Pop}}
\index{check\-Erroneous\-Pop@{check\-Erroneous\-Pop}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{check\-Erroneous\-Pop}]{\setlength{\rightskip}{0pt plus 5cm}static void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Pop (
\begin{DoxyParamCaption}
\item[{{\bf G\-P\-Population}}]{pop, }
\item[{String}]{s}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a71cc6d10d81d132ee542767aca12f502}
Disabling a chromosome is equivalent to not declaring it. However, if you skip a declaration, indices will shift. With this method it is easier skipping a chromosome.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-index} & index of the chromosome to disable.\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2.\-2
\end{DoxySince}

\begin{DoxyParams}{Parameters}
{\em a\-\_\-index} & index of the chromosome to check. \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
true if chromosome disabled
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2.\-2 
\end{DoxySince}


Definition at line 1251 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Pop(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-random\-Initial\-Genotype().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af886a583789e547400134f2ad533cf8b}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!check\-Erroneous\-Pop@{check\-Erroneous\-Pop}}
\index{check\-Erroneous\-Pop@{check\-Erroneous\-Pop}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{check\-Erroneous\-Pop}]{\setlength{\rightskip}{0pt plus 5cm}static void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Pop (
\begin{DoxyParamCaption}
\item[{{\bf G\-P\-Population}}]{a\-\_\-pop, }
\item[{String}]{a\-\_\-s, }
\item[{boolean}]{a\-\_\-clear\-Fitness}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af886a583789e547400134f2ad533cf8b}


Definition at line 1255 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Pop(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Population.\-get\-G\-P\-Configuration().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af886a583789e547400134f2ad533cf8b_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a6cab9ecf163e4eec850da3cc63df8665}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!check\-Erroneous\-Pop@{check\-Erroneous\-Pop}}
\index{check\-Erroneous\-Pop@{check\-Erroneous\-Pop}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{check\-Erroneous\-Pop}]{\setlength{\rightskip}{0pt plus 5cm}static void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Pop (
\begin{DoxyParamCaption}
\item[{{\bf G\-P\-Population}}]{a\-\_\-pop, }
\item[{String}]{a\-\_\-s, }
\item[{boolean}]{a\-\_\-clear\-Fitness, }
\item[{boolean}]{a\-\_\-active}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a6cab9ecf163e4eec850da3cc63df8665}


Definition at line 1264 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Prog().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a6cab9ecf163e4eec850da3cc63df8665_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a4104bc4b934e717c15e3a7b5e50a0748}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!check\-Erroneous\-Prog@{check\-Erroneous\-Prog}}
\index{check\-Erroneous\-Prog@{check\-Erroneous\-Prog}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{check\-Erroneous\-Prog}]{\setlength{\rightskip}{0pt plus 5cm}static void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Prog (
\begin{DoxyParamCaption}
\item[{{\bf I\-G\-P\-Program}}]{prog, }
\item[{String}]{s}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a4104bc4b934e717c15e3a7b5e50a0748}


Definition at line 1282 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Pop(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Prog(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a9d986d5030ce180fc402a2a73d22c42d}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!check\-Erroneous\-Prog@{check\-Erroneous\-Prog}}
\index{check\-Erroneous\-Prog@{check\-Erroneous\-Prog}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{check\-Erroneous\-Prog}]{\setlength{\rightskip}{0pt plus 5cm}static void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Prog (
\begin{DoxyParamCaption}
\item[{{\bf I\-G\-P\-Program}}]{a\-\_\-prog, }
\item[{String}]{a\-\_\-s, }
\item[{boolean}]{a\-\_\-clear\-Fitness}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a9d986d5030ce180fc402a2a73d22c42d}


Definition at line 1286 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Prog(), and org.\-jgap.\-gp.\-I\-G\-P\-Program.\-get\-G\-P\-Configuration().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a9d986d5030ce180fc402a2a73d22c42d_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_aa3bf0b50a73684ff3fe1177017cf9bb8}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!check\-Erroneous\-Prog@{check\-Erroneous\-Prog}}
\index{check\-Erroneous\-Prog@{check\-Erroneous\-Prog}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{check\-Erroneous\-Prog}]{\setlength{\rightskip}{0pt plus 5cm}static void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Prog (
\begin{DoxyParamCaption}
\item[{{\bf I\-G\-P\-Program}}]{a\-\_\-prog, }
\item[{String}]{s, }
\item[{boolean}]{a\-\_\-clear\-Fitness, }
\item[{boolean}]{a\-\_\-active}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_aa3bf0b50a73684ff3fe1177017cf9bb8}
\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000152}{Todo}]impl. cache \end{DoxyRefDesc}


Definition at line 1295 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-G\-P\-Fitness\-Function.\-N\-O\-\_\-\-F\-I\-T\-N\-E\-S\-S\-\_\-\-V\-A\-L\-U\-E.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_aa21cc2dde0ca8cbc6048b37e62dd1593}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!clone\-Program@{clone\-Program}}
\index{clone\-Program@{clone\-Program}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{clone\-Program}]{\setlength{\rightskip}{0pt plus 5cm}{\bf I\-G\-P\-Program} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-clone\-Program (
\begin{DoxyParamCaption}
\item[{{\bf I\-G\-P\-Program}}]{a\-\_\-original}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_aa21cc2dde0ca8cbc6048b37e62dd1593}


Definition at line 1131 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Configuration(), and org.\-jgap.\-I\-Handler.\-perform().



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_aa21cc2dde0ca8cbc6048b37e62dd1593_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_aef87073769e83d93b52d8df73ebb35dc}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!compare\-To@{compare\-To}}
\index{compare\-To@{compare\-To}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{compare\-To}]{\setlength{\rightskip}{0pt plus 5cm}int org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-compare\-To (
\begin{DoxyParamCaption}
\item[{Object}]{a\-\_\-other}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_aef87073769e83d93b52d8df73ebb35dc}
Compares this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} against the specified object. The result is true if the argument is an instance of the \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype} class, has exactly the same number of programs as the given \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype}, and, for each \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program}{G\-P\-Program} in this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype}, there is an equal program in the given \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype}. The programs do not need to appear in the same order within the populations.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-other} & the object to compare against \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
a negative number if this genotype is \char`\"{}less than\char`\"{} the given genotype, zero if they are equal to each other, and a positive number if this genotype is \char`\"{}greater than\char`\"{} the given genotype
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 1045 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Population(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype().



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-equals().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_aef87073769e83d93b52d8df73ebb35dc_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a847cbf8f0b5e3e223881ac1dfe5e36d3}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!equals@{equals}}
\index{equals@{equals}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{equals}]{\setlength{\rightskip}{0pt plus 5cm}boolean org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-equals (
\begin{DoxyParamCaption}
\item[{Object}]{a\-\_\-other}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a847cbf8f0b5e3e223881ac1dfe5e36d3}
Compares this entity against the specified object.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-other} & the object to compare against \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
true\-: if the objects are the same, false otherwise
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 1021 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-compare\-To().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a847cbf8f0b5e3e223881ac1dfe5e36d3_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a43707d524245f50c92094e9e97568089}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!evolve@{evolve}}
\index{evolve@{evolve}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{evolve}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve (
\begin{DoxyParamCaption}
\item[{int}]{a\-\_\-evolutions}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a43707d524245f50c92094e9e97568089}
Evolves the population n times.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-evolutions} & number of evolution\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 524 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-calc\-Fitness(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Configuration(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-verbose.



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a43707d524245f50c92094e9e97568089_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ad495c700786596290a63e3d88a902766}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!evolve@{evolve}}
\index{evolve@{evolve}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{evolve}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ad495c700786596290a63e3d88a902766}
Evolve the population by one generation. Probabilistically reproduces and crosses individuals into a new population which then overwrites the original population.

\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}
\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000148}{Todo}]use program creator in case such is registered and returns a non-\/null program \end{DoxyRefDesc}


\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000149}{Todo}]instead of re-\/using prototype, create a program anyway (ignoring the validator) in case it is the last try. Or even better\-: Make the validator return a defect rate! \end{DoxyRefDesc}


Definition at line 696 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Pop(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Prog(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-clone\-Program(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Configuration(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Population(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-arg\-Types, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-fittest\-To\-Add, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-full\-Mode\-Allowed, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-max\-Depths, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-max\-Nodes, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-min\-Depths, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-node\-Sets, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-types, org.\-jgap.\-Random\-Generator.\-next\-Int(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-set\-G\-P\-Population(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-write\-To\-File().



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-run().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ad495c700786596290a63e3d88a902766_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a4327e7a69fcbcfe051e00e46c933ea7a}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!fill\-Population@{fill\-Population}}
\index{fill\-Population@{fill\-Population}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{fill\-Population}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-fill\-Population (
\begin{DoxyParamCaption}
\item[{final int}]{a\-\_\-num}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a4327e7a69fcbcfe051e00e46c933ea7a}
Fills up the population with random programs if necessary.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-num} & the number of programs to add \\
\hline
\end{DoxyParams}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2 
\end{DoxySince}
\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000150}{Todo}]care about \end{DoxyRefDesc}


\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000151}{Todo}]implement filling up population \end{DoxyRefDesc}


Definition at line 1195 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Configuration().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a4327e7a69fcbcfe051e00e46c933ea7a_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_afb8e0283f47eb7c0a0b6e09c8aeccc96}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!get\-All\-Time\-Best@{get\-All\-Time\-Best}}
\index{get\-All\-Time\-Best@{get\-All\-Time\-Best}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{get\-All\-Time\-Best}]{\setlength{\rightskip}{0pt plus 5cm}{\bf I\-G\-P\-Program} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-All\-Time\-Best (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_afb8e0283f47eb7c0a0b6e09c8aeccc96}
\begin{DoxyReturn}{Returns}
the all-\/time best solution found
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 647 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-all\-Time\-Best.



Referenced by examples.\-gp.\-Fibonacci.\-main().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a2cff99f216e87c7e4b25c217c5bdc5af}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!get\-Fittest\-Program@{get\-Fittest\-Program}}
\index{get\-Fittest\-Program@{get\-Fittest\-Program}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{get\-Fittest\-Program}]{\setlength{\rightskip}{0pt plus 5cm}synchronized {\bf I\-G\-P\-Program} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-Fittest\-Program (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a2cff99f216e87c7e4b25c217c5bdc5af}
Retrieves the \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program}{G\-P\-Program} in the population with the best fitness value.

\begin{DoxyReturn}{Returns}
the \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program}{G\-P\-Program} with the highest fitness value, or null if there are no programs in this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype}
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 959 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-I\-G\-P\-Program.\-get\-Fitness\-Value(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Configuration(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Population(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-all\-Time\-Best, and org.\-jgap.\-Fitness\-Function.\-N\-O\-\_\-\-F\-I\-T\-N\-E\-S\-S\-\_\-\-V\-A\-L\-U\-E.



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a2cff99f216e87c7e4b25c217c5bdc5af_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a3a1e955f3a6c035913e311570591c683}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!get\-Fittest\-Program\-Computed@{get\-Fittest\-Program\-Computed}}
\index{get\-Fittest\-Program\-Computed@{get\-Fittest\-Program\-Computed}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{get\-Fittest\-Program\-Computed}]{\setlength{\rightskip}{0pt plus 5cm}synchronized {\bf I\-G\-P\-Program} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-Fittest\-Program\-Computed (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a3a1e955f3a6c035913e311570591c683}
Retrieves the \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program}{G\-P\-Program} in the population with the highest fitness value. Only considers programs for which the fitness value has already been computed.

\begin{DoxyReturn}{Returns}
the \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program}{G\-P\-Program} with the best fitness value, or null if there are no programs with known fitness value in this \hyperlink{classorg_1_1jgap_1_1_genotype}{Genotype}
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2 
\end{DoxySince}


Definition at line 993 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Population().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a3a1e955f3a6c035913e311570591c683_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a318ae291daca64dc4ba0b1e5085b4b87}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!get\-G\-P\-Configuration@{get\-G\-P\-Configuration}}
\index{get\-G\-P\-Configuration@{get\-G\-P\-Configuration}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{get\-G\-P\-Configuration}]{\setlength{\rightskip}{0pt plus 5cm}{\bf G\-P\-Configuration} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Configuration (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a318ae291daca64dc4ba0b1e5085b4b87}


Definition at line 470 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-configuration.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-calc\-Fitness(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-clone\-Program(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-fill\-Population(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-Fittest\-Program(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype(), and examples.\-gp.\-monalisa.\-gui.\-Evolution\-Runnable.\-run().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a9a2316294bf28b413eb8b0e288cd150d}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!get\-G\-P\-Population@{get\-G\-P\-Population}}
\index{get\-G\-P\-Population@{get\-G\-P\-Population}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{get\-G\-P\-Population}]{\setlength{\rightskip}{0pt plus 5cm}{\bf G\-P\-Population} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Population (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a9a2316294bf28b413eb8b0e288cd150d}


Definition at line 914 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-population.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-calc\-Fitness(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-compare\-To(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-Fittest\-Program(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-Fittest\-Program\-Computed(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-hash\-Code().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a356baf22df15fe6c0c88aadb45a80a5e}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!get\-Program\-String@{get\-Program\-String}}
\index{get\-Program\-String@{get\-Program\-String}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{get\-Program\-String}]{\setlength{\rightskip}{0pt plus 5cm}String\-Buffer org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-Program\-String (
\begin{DoxyParamCaption}
\item[{{\bf I\-G\-P\-Program}}]{i1}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a356baf22df15fe6c0c88aadb45a80a5e}
Utility function.


\begin{DoxyParams}{Parameters}
{\em i1} & program to get textual representation for \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
textual representation of the given program
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}


Definition at line 1365 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-Command\-Gene.\-to\-String().



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-write\-To\-File().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a356baf22df15fe6c0c88aadb45a80a5e_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a07e4b22e1cd10bbc198278efd83d27ce}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!get\-Static\-G\-P\-Configuration@{get\-Static\-G\-P\-Configuration}}
\index{get\-Static\-G\-P\-Configuration@{get\-Static\-G\-P\-Configuration}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{get\-Static\-G\-P\-Configuration}]{\setlength{\rightskip}{0pt plus 5cm}static {\bf G\-P\-Configuration} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-Static\-G\-P\-Configuration (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a07e4b22e1cd10bbc198278efd83d27ce}
\begin{DoxyReturn}{Returns}
the static configuration to use with Genetic Programming
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2 
\end{DoxySince}


Definition at line 480 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-static\-Configuration.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a631996e8862da78bdb67b76687ad1873}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!get\-Total\-Fitness@{get\-Total\-Fitness}}
\index{get\-Total\-Fitness@{get\-Total\-Fitness}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{get\-Total\-Fitness}]{\setlength{\rightskip}{0pt plus 5cm}double org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-Total\-Fitness (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a631996e8862da78bdb67b76687ad1873}
\begin{DoxyReturn}{Returns}
the total fitness, that is the fitness over all chromosomes
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 924 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-total\-Fitness.



Referenced by org.\-jgap.\-gp.\-impl.\-Fitness\-Proportionate\-Selection.\-select().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a66770c2d4fb9645f33afec5dc37bb47c}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!get\-Variable@{get\-Variable}}
\index{get\-Variable@{get\-Variable}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{get\-Variable}]{\setlength{\rightskip}{0pt plus 5cm}Variable org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-Variable (
\begin{DoxyParamCaption}
\item[{String}]{a\-\_\-var\-Name}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a66770c2d4fb9645f33afec5dc37bb47c}

\begin{DoxyParams}{Parameters}
{\em a\-\_\-var\-Name} & name of variable to retriebe \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
Variable instance or null, if not found
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2 
\end{DoxySince}


Definition at line 1167 of file G\-P\-Genotype.\-java.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a9b8cb5e7395359743bbee7658be90a36}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!hash\-Code@{hash\-Code}}
\index{hash\-Code@{hash\-Code}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{hash\-Code}]{\setlength{\rightskip}{0pt plus 5cm}int org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-hash\-Code (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a9b8cb5e7395359743bbee7658be90a36}


Definition at line 1106 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Population().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a9b8cb5e7395359743bbee7658be90a36_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a6b017f0f0e028cc1a9164782c5ce5095}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!init@{init}}
\index{init@{init}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{init}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-init (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a6b017f0f0e028cc1a9164782c5ce5095}


Definition at line 230 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ae3828fb2c7d3c3cc4ee2c1562c9851d8}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!node\-Exists@{node\-Exists}}
\index{node\-Exists@{node\-Exists}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{node\-Exists}]{\setlength{\rightskip}{0pt plus 5cm}static boolean org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-node\-Exists (
\begin{DoxyParamCaption}
\item[{{\bf Command\-Gene}\mbox{[}$\,$\mbox{]}}]{a\-\_\-function\-Set, }
\item[{Class}]{a\-\_\-return\-Type, }
\item[{int}]{a\-\_\-sub\-Return\-Type}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ae3828fb2c7d3c3cc4ee2c1562c9851d8}
Is there a node with the needed return type and sub return type within the given function set?


\begin{DoxyParams}{Parameters}
{\em a\-\_\-function\-Set} & collection of available nodes \\
\hline
{\em a\-\_\-return\-Type} & needed return type \\
\hline
{\em a\-\_\-sub\-Return\-Type} & needed sub return type \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
true\-: node found
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-4.\-4 
\end{DoxySince}


Definition at line 1465 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-verify\-Child\-Nodes().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a95fd91fcd94500415c67e18388695a61}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!output\-Depth\-Info@{output\-Depth\-Info}}
\index{output\-Depth\-Info@{output\-Depth\-Info}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{output\-Depth\-Info}]{\setlength{\rightskip}{0pt plus 5cm}static void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-output\-Depth\-Info (
\begin{DoxyParamCaption}
\item[{Map$<$ {\bf Command\-Gene}, int\mbox{[}$\,$\mbox{]}$>$}]{a\-\_\-invalid\-Depths}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a95fd91fcd94500415c67e18388695a61}
Outputs the nodes that are never used because they would result in creating G\-P programs which would not satisfy the depth constraints given in configuration.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-invalid\-Depths} & impossible depths for nodes\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-4.\-4 
\end{DoxySince}


Definition at line 1511 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-random\-Initial\-Genotype().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_abeef3282114d1313c97045dea004afc4}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!output\-Solution@{output\-Solution}}
\index{output\-Solution@{output\-Solution}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{output\-Solution}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-output\-Solution (
\begin{DoxyParamCaption}
\item[{{\bf I\-G\-P\-Program}}]{a\-\_\-best}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_abeef3282114d1313c97045dea004afc4}
Outputs the best solution until now.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-best} & the fittest \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome}{Program\-Chromosome}\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 659 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-calc\-Fitness().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ae5eb2b5f82dd89d6e8287eac90864cb1}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!output\-Warning@{output\-Warning}}
\index{output\-Warning@{output\-Warning}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{output\-Warning}]{\setlength{\rightskip}{0pt plus 5cm}static void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-output\-Warning (
\begin{DoxyParamCaption}
\item[{Map$<$ String, {\bf Command\-Gene} $>$}]{invalid\-Nodes}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ae5eb2b5f82dd89d6e8287eac90864cb1}
Outputs the nodes that are never used.


\begin{DoxyParams}{Parameters}
{\em invalid\-Nodes} & never used nodes\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-4.\-4 
\end{DoxySince}


Definition at line 1486 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-random\-Initial\-Genotype().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a0b615fba7524275eb09b3d7929d73329}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!put\-Variable@{put\-Variable}}
\index{put\-Variable@{put\-Variable}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{put\-Variable}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-put\-Variable (
\begin{DoxyParamCaption}
\item[{Variable}]{a\-\_\-var}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a0b615fba7524275eb09b3d7929d73329}
Stores a Variable.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-var} & the Variable to store\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2 
\end{DoxySince}


Definition at line 1156 of file G\-P\-Genotype.\-java.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a59354275c3ff7c6b47c4c676c13a4422}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!random\-Initial\-Genotype@{random\-Initial\-Genotype}}
\index{random\-Initial\-Genotype@{random\-Initial\-Genotype}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{random\-Initial\-Genotype}]{\setlength{\rightskip}{0pt plus 5cm}static {\bf G\-P\-Genotype} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-random\-Initial\-Genotype (
\begin{DoxyParamCaption}
\item[{final {\bf G\-P\-Configuration}}]{a\-\_\-conf, }
\item[{Class\mbox{[}$\,$\mbox{]}}]{a\-\_\-types, }
\item[{Class}]{a\-\_\-arg\-Types\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{{\bf Command\-Gene}}]{a\-\_\-node\-Sets\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{int}]{a\-\_\-max\-Nodes, }
\item[{boolean}]{a\-\_\-verbose\-Output}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a59354275c3ff7c6b47c4c676c13a4422}
Creates a genotype with initial population for the world set.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-conf} & the configuration to use \\
\hline
{\em a\-\_\-types} & the type of each chromosome, the length is the number of chromosomes \\
\hline
{\em a\-\_\-arg\-Types} & the types of the arguments to each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of arguments to the chromosome \\
\hline
{\em a\-\_\-node\-Sets} & the nodes which are allowed to be used by each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of nodes. Note that it is not necessary to include the arguments of a chromosome as terminals in the chromosome's node set. This is done automatically \\
\hline
{\em a\-\_\-max\-Nodes} & reserve space for a\-\_\-max\-Nodes number of nodes \\
\hline
{\em a\-\_\-verbose\-Output} & true\-: output status information to console\\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
created genotype with initialized population
\end{DoxyReturn}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 260 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-random\-Initial\-Genotype().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a7e59aff645b8386adcad62e9ccdfd00d}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!random\-Initial\-Genotype@{random\-Initial\-Genotype}}
\index{random\-Initial\-Genotype@{random\-Initial\-Genotype}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{random\-Initial\-Genotype}]{\setlength{\rightskip}{0pt plus 5cm}static {\bf G\-P\-Genotype} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-random\-Initial\-Genotype (
\begin{DoxyParamCaption}
\item[{final {\bf G\-P\-Configuration}}]{a\-\_\-conf, }
\item[{Class\mbox{[}$\,$\mbox{]}}]{a\-\_\-types, }
\item[{Class}]{a\-\_\-arg\-Types\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{{\bf Command\-Gene}}]{a\-\_\-node\-Sets\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-min\-Depths, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-max\-Depths, }
\item[{int}]{a\-\_\-max\-Nodes, }
\item[{boolean}]{a\-\_\-verbose\-Output}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a7e59aff645b8386adcad62e9ccdfd00d}
Creates a genotype with initial population for the world set.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-conf} & the configuration to use \\
\hline
{\em a\-\_\-types} & the type of each chromosome, the length is the number of chromosomes \\
\hline
{\em a\-\_\-arg\-Types} & the types of the arguments to each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of arguments to the chromosome \\
\hline
{\em a\-\_\-node\-Sets} & the nodes which are allowed to be used by each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of nodes. Note that it is not necessary to include the arguments of a chromosome as terminals in the chromosome's node set. This is done automatically \\
\hline
{\em a\-\_\-min\-Depths} & array of minimum depths to use\-: for each chromosome one entry \\
\hline
{\em a\-\_\-max\-Depths} & array of maximum depths to use\-: for each chromosome one entry \\
\hline
{\em a\-\_\-max\-Nodes} & reserve space for a\-\_\-max\-Nodes number of nodes \\
\hline
{\em a\-\_\-verbose\-Output} & true\-: output status information to console\\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
created genotype with initialized population
\end{DoxyReturn}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 301 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-random\-Initial\-Genotype().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a7e59aff645b8386adcad62e9ccdfd00d_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a47a8848949757df02df0b4503bd2bea7}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!random\-Initial\-Genotype@{random\-Initial\-Genotype}}
\index{random\-Initial\-Genotype@{random\-Initial\-Genotype}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{random\-Initial\-Genotype}]{\setlength{\rightskip}{0pt plus 5cm}static {\bf G\-P\-Genotype} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-random\-Initial\-Genotype (
\begin{DoxyParamCaption}
\item[{final {\bf G\-P\-Configuration}}]{a\-\_\-conf, }
\item[{Class\mbox{[}$\,$\mbox{]}}]{a\-\_\-types, }
\item[{Class}]{a\-\_\-arg\-Types\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{{\bf Command\-Gene}}]{a\-\_\-node\-Sets\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-min\-Depths, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-max\-Depths, }
\item[{int}]{a\-\_\-max\-Nodes, }
\item[{boolean\mbox{[}$\,$\mbox{]}}]{a\-\_\-full\-Mode\-Allowed, }
\item[{boolean}]{a\-\_\-verbose\-Output}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a47a8848949757df02df0b4503bd2bea7}
Creates a genotype with a randomly created initial population.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-conf} & the configuration to use \\
\hline
{\em a\-\_\-types} & the type of each chromosome, the length is the number of chromosomes \\
\hline
{\em a\-\_\-arg\-Types} & the types of the arguments to each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of arguments to the chromosome \\
\hline
{\em a\-\_\-node\-Sets} & the nodes which are allowed to be used by each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of nodes. Note that it is not necessary to include the arguments of a chromosome as terminals in the chromosome's node set. This is done automatically \\
\hline
{\em a\-\_\-min\-Depths} & array of minimum depths to use\-: for each chromosome one entry \\
\hline
{\em a\-\_\-max\-Depths} & array of maximum depths to use\-: for each chromosome one entry \\
\hline
{\em a\-\_\-max\-Nodes} & reserve space for a\-\_\-max\-Nodes number of nodes \\
\hline
{\em a\-\_\-full\-Mode\-Allowed} & array of boolean values. For each chromosome there is one value indicating whether the full mode for creating chromosome generations during evolution is allowed (true) or not (false) \\
\hline
{\em a\-\_\-verbose\-Output} & true\-: output status information to console\\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
created genotype with initialized population
\end{DoxyReturn}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 352 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-random\-Initial\-Genotype().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a47a8848949757df02df0b4503bd2bea7_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af3246cb6a802ae1cde833013c130f6e9}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!random\-Initial\-Genotype@{random\-Initial\-Genotype}}
\index{random\-Initial\-Genotype@{random\-Initial\-Genotype}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{random\-Initial\-Genotype}]{\setlength{\rightskip}{0pt plus 5cm}static {\bf G\-P\-Genotype} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-random\-Initial\-Genotype (
\begin{DoxyParamCaption}
\item[{final {\bf G\-P\-Configuration}}]{a\-\_\-conf, }
\item[{Class\mbox{[}$\,$\mbox{]}}]{a\-\_\-types, }
\item[{Class}]{a\-\_\-arg\-Types\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{{\bf Command\-Gene}}]{a\-\_\-node\-Sets\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-min\-Depths, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-max\-Depths, }
\item[{int}]{a\-\_\-max\-Nodes, }
\item[{boolean\mbox{[}$\,$\mbox{]}}]{a\-\_\-full\-Mode\-Allowed, }
\item[{boolean}]{a\-\_\-verbose\-Output, }
\item[{{\bf I\-Population\-Creator}}]{a\-\_\-pop\-Creator}
\end{DoxyParamCaption}
) throws {\bf Invalid\-Configuration\-Exception}\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af3246cb6a802ae1cde833013c130f6e9}
Allows to use a custom mechanism for population creation.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-conf} & the configuration to use \\
\hline
{\em a\-\_\-types} & the type of each chromosome, the length is the number of chromosomes \\
\hline
{\em a\-\_\-arg\-Types} & the types of the arguments to each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of arguments to the chromosome \\
\hline
{\em a\-\_\-node\-Sets} & the nodes which are allowed to be used by each chromosome, must be an array of arrays, the first dimension of which is the number of chromosomes and the second dimension of which is the number of nodes. Note that it is not necessary to include the arguments of a chromosome as terminals in the chromosome's node set. This is done automatically \\
\hline
{\em a\-\_\-min\-Depths} & array of minimum depths to use\-: for each chromosome one entry \\
\hline
{\em a\-\_\-max\-Depths} & array of maximum depths to use\-: for each chromosome one entry \\
\hline
{\em a\-\_\-max\-Nodes} & reserve space for a\-\_\-max\-Nodes number of nodes \\
\hline
{\em a\-\_\-full\-Mode\-Allowed} & array of boolean values. For each chromosome there is one value indicating whether the full mode for creating chromosome generations during evolution is allowed (true) or not (false) \\
\hline
{\em a\-\_\-verbose\-Output} & true\-: output status information to console \\
\hline
{\em a\-\_\-pop\-Creator} & mechanism fior creating the population\\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
\hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype}{G\-P\-Genotype}
\end{DoxyReturn}

\begin{DoxyExceptions}{Exceptions}
{\em \hyperlink{classorg_1_1jgap_1_1_invalid_configuration_exception}{Invalid\-Configuration\-Exception}} & \\
\hline
\end{DoxyExceptions}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2.\-2 
\end{DoxySince}
\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000142}{Todo}]remove unused nodes from configuration \end{DoxyRefDesc}


\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000143}{Todo}]optimize access to map \end{DoxyRefDesc}


Definition at line 397 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-check\-Erroneous\-Pop(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-variables, org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-output\-Depth\-Info(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-output\-Warning(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-verify\-Child\-Nodes(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-verify\-Depths\-For\-Nodes().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af3246cb6a802ae1cde833013c130f6e9_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a184964a4fc96748de13cad57bee24ad0}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!run@{run}}
\index{run@{run}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{run}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-run (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a184964a4fc96748de13cad57bee24ad0}
Default implementation of method to run \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype}{G\-P\-Genotype} as a thread.

\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 934 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-calc\-Fitness(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a184964a4fc96748de13cad57bee24ad0_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a2d46992095125ecad698dc9e1a72b513}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!set\-G\-P\-Configuration@{set\-G\-P\-Configuration}}
\index{set\-G\-P\-Configuration@{set\-G\-P\-Configuration}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{set\-G\-P\-Configuration}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-set\-G\-P\-Configuration (
\begin{DoxyParamCaption}
\item[{{\bf G\-P\-Configuration}}]{a\-\_\-configuration}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a2d46992095125ecad698dc9e1a72b513}
Sets the configuration to use with the Genetic Algorithm. 
\begin{DoxyParams}{Parameters}
{\em a\-\_\-configuration} & the configuration to use\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 1008 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-configuration.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a11dce8ec844c102bc151e4ab81c52ac7}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!set\-G\-P\-Population@{set\-G\-P\-Population}}
\index{set\-G\-P\-Population@{set\-G\-P\-Population}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{set\-G\-P\-Population}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-set\-G\-P\-Population (
\begin{DoxyParamCaption}
\item[{{\bf G\-P\-Population}}]{a\-\_\-pop}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [protected]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a11dce8ec844c102bc151e4ab81c52ac7}


Definition at line 997 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-population.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a7750ff9b7bd73efcfff522365f4650d7}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!set\-Static\-G\-P\-Configuration@{set\-Static\-G\-P\-Configuration}}
\index{set\-Static\-G\-P\-Configuration@{set\-Static\-G\-P\-Configuration}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{set\-Static\-G\-P\-Configuration}]{\setlength{\rightskip}{0pt plus 5cm}static void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-set\-Static\-G\-P\-Configuration (
\begin{DoxyParamCaption}
\item[{{\bf G\-P\-Configuration}}]{a\-\_\-configuration}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a7750ff9b7bd73efcfff522365f4650d7}
Sets the static configuration to use with the Genetic Programming.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-configuration} & the static configuration to use\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-2 
\end{DoxySince}


Definition at line 492 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-static\-Configuration.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a2cda11ec40667a6cb061a4c87ace0b6c}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!set\-Verbose\-Output@{set\-Verbose\-Output}}
\index{set\-Verbose\-Output@{set\-Verbose\-Output}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{set\-Verbose\-Output}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-set\-Verbose\-Output (
\begin{DoxyParamCaption}
\item[{boolean}]{a\-\_\-verbose}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a2cda11ec40667a6cb061a4c87ace0b6c}

\begin{DoxyParams}{Parameters}
{\em a\-\_\-verbose} & true\-: output status information to console\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-0 
\end{DoxySince}


Definition at line 1127 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-verbose.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a77909da0c746ca16cdcbf0af9b1c0927}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!verify\-Child\-Nodes@{verify\-Child\-Nodes}}
\index{verify\-Child\-Nodes@{verify\-Child\-Nodes}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{verify\-Child\-Nodes}]{\setlength{\rightskip}{0pt plus 5cm}static Map$<$String, {\bf Command\-Gene}$>$ org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-verify\-Child\-Nodes (
\begin{DoxyParamCaption}
\item[{{\bf G\-P\-Configuration}}]{a\-\_\-conf, }
\item[{Class\mbox{[}$\,$\mbox{]}}]{a\-\_\-types, }
\item[{{\bf Command\-Gene}}]{a\-\_\-node\-Sets\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a77909da0c746ca16cdcbf0af9b1c0927}
Determine impossible functions and terminals to obtimize creation of G\-P programs.


\begin{DoxyParams}{Parameters}
{\em a\-\_\-conf} & G\-P configuration \\
\hline
{\em a\-\_\-types} & Class\mbox{[}\mbox{]} \\
\hline
{\em a\-\_\-node\-Sets} & \hyperlink{classorg_1_1jgap_1_1gp_1_1_command_gene}{Command\-Gene}\mbox{[}\mbox{]}\mbox{[}\mbox{]} \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
Map
\end{DoxyReturn}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}
\begin{DoxySince}{Since}
3.\-4.\-4 
\end{DoxySince}


Definition at line 1396 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-Command\-Gene.\-get\-Sub\-Child\-Types(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-node\-Exists().



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-random\-Initial\-Genotype().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a77909da0c746ca16cdcbf0af9b1c0927_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ad264330c888b133296578decc78f265e}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!verify\-Depths\-For\-Nodes@{verify\-Depths\-For\-Nodes}}
\index{verify\-Depths\-For\-Nodes@{verify\-Depths\-For\-Nodes}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{verify\-Depths\-For\-Nodes}]{\setlength{\rightskip}{0pt plus 5cm}static Map$<${\bf Command\-Gene}, int\mbox{[}$\,$\mbox{]}$>$ org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-verify\-Depths\-For\-Nodes (
\begin{DoxyParamCaption}
\item[{{\bf G\-P\-Population}}]{a\-\_\-pop, }
\item[{{\bf G\-P\-Configuration}}]{a\-\_\-conf, }
\item[{Class\mbox{[}$\,$\mbox{]}}]{a\-\_\-types, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-min\-Depths, }
\item[{int\mbox{[}$\,$\mbox{]}}]{a\-\_\-max\-Depths, }
\item[{int}]{a\-\_\-max\-Nodes, }
\item[{{\bf Command\-Gene}}]{a\-\_\-node\-Sets\mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]}}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [static]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ad264330c888b133296578decc78f265e}
\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000153}{Todo}]impl \end{DoxyRefDesc}


\begin{DoxyRefDesc}{Todo}
\item[\hyperlink{todo__todo000154}{Todo}]impl \end{DoxyRefDesc}


Definition at line 1532 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-Command\-Gene.\-get\-Return\-Type(), org.\-jgap.\-gp.\-Command\-Gene.\-get\-Sub\-Child\-Type(), and org.\-jgap.\-gp.\-Command\-Gene.\-get\-Sub\-Return\-Type().



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-random\-Initial\-Genotype().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ad264330c888b133296578decc78f265e_cgraph}
\end{center}
\end{figure}


\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ac726d5312c002c60c239a9be1bc7cfba}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!write\-To\-File@{write\-To\-File}}
\index{write\-To\-File@{write\-To\-File}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{write\-To\-File}]{\setlength{\rightskip}{0pt plus 5cm}void org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-write\-To\-File (
\begin{DoxyParamCaption}
\item[{{\bf I\-G\-P\-Program}}]{i1, }
\item[{{\bf I\-G\-P\-Program}}]{i2, }
\item[{{\bf I\-G\-P\-Program}}]{inew, }
\item[{String}]{header}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ac726d5312c002c60c239a9be1bc7cfba}
Write three G\-P programs being involved in crossover as a string to a file for debug purposes.


\begin{DoxyParams}{Parameters}
{\em i1} & first program \\
\hline
{\em i2} & second program \\
\hline
{\em inew} & resulting program \\
\hline
{\em header} & text for header text\\
\hline
\end{DoxyParams}
\begin{DoxyAuthor}{Author}
Klaus Meffert 
\end{DoxyAuthor}


Definition at line 1335 of file G\-P\-Genotype.\-java.



References org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-Program\-String().



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve().



Here is the call graph for this function\-:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ac726d5312c002c60c239a9be1bc7cfba_cgraph}
\end{center}
\end{figure}




\subsection{Member Data Documentation}
\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ab88a16d04e096c2c2e1bfc4477ea7310}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N@{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N}}
\index{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N@{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N}]{\setlength{\rightskip}{0pt plus 5cm}final String org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-C\-V\-S\-\_\-\-R\-E\-V\-I\-S\-I\-O\-N = \char`\"{}\$Revision\-: 1.\-60 \$\char`\"{}\hspace{0.3cm}{\ttfamily [static]}, {\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ab88a16d04e096c2c2e1bfc4477ea7310}
String containing the C\-V\-S revision. Read out via reflection! 

Definition at line 32 of file G\-P\-Genotype.\-java.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ada30eb741391730f266047d15cd40338}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!L\-O\-G\-G\-E\-R@{L\-O\-G\-G\-E\-R}}
\index{L\-O\-G\-G\-E\-R@{L\-O\-G\-G\-E\-R}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{L\-O\-G\-G\-E\-R}]{\setlength{\rightskip}{0pt plus 5cm}transient Logger org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-L\-O\-G\-G\-E\-R = Logger.\-get\-Logger(G\-P\-Genotype.\-class)\hspace{0.3cm}{\ttfamily [static]}, {\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ada30eb741391730f266047d15cd40338}


Definition at line 34 of file G\-P\-Genotype.\-java.

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a6bd9f97a0bb39ec138f6d8ca8ebd25e5}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-all\-Time\-Best@{m\-\_\-all\-Time\-Best}}
\index{m\-\_\-all\-Time\-Best@{m\-\_\-all\-Time\-Best}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-all\-Time\-Best}]{\setlength{\rightskip}{0pt plus 5cm}{\bf I\-G\-P\-Program} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-all\-Time\-Best\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a6bd9f97a0bb39ec138f6d8ca8ebd25e5}
Best solution found 

Definition at line 61 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-calc\-Fitness(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-All\-Time\-Best(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-Fittest\-Program().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a869b5d4ab431717009978a9d870bf5d8}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-all\-Time\-Best\-Fitness@{m\-\_\-all\-Time\-Best\-Fitness}}
\index{m\-\_\-all\-Time\-Best\-Fitness@{m\-\_\-all\-Time\-Best\-Fitness}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-all\-Time\-Best\-Fitness}]{\setlength{\rightskip}{0pt plus 5cm}double org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-all\-Time\-Best\-Fitness\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a869b5d4ab431717009978a9d870bf5d8}


Definition at line 63 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-calc\-Fitness(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a8e26c3d41a0e1922310e9d966f80238f}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-arg\-Types@{m\-\_\-arg\-Types}}
\index{m\-\_\-arg\-Types@{m\-\_\-arg\-Types}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-arg\-Types}]{\setlength{\rightskip}{0pt plus 5cm}Class \mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-arg\-Types\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a8e26c3d41a0e1922310e9d966f80238f}
Argument types for A\-D\-F's 

Definition at line 78 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a957b0de7013e1055b66e03223dc890f2}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-best\-Fitness@{m\-\_\-best\-Fitness}}
\index{m\-\_\-best\-Fitness@{m\-\_\-best\-Fitness}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-best\-Fitness}]{\setlength{\rightskip}{0pt plus 5cm}double org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-best\-Fitness\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a957b0de7013e1055b66e03223dc890f2}
Fitness value of the best solution 

Definition at line 51 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-calc\-Fitness().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af0ec08e1f1fdf88d76bae2ba6d0fe986}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-clone\-Warning\-G\-P\-Program\-Shown@{m\-\_\-clone\-Warning\-G\-P\-Program\-Shown}}
\index{m\-\_\-clone\-Warning\-G\-P\-Program\-Shown@{m\-\_\-clone\-Warning\-G\-P\-Program\-Shown}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-clone\-Warning\-G\-P\-Program\-Shown}]{\setlength{\rightskip}{0pt plus 5cm}boolean org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-clone\-Warning\-G\-P\-Program\-Shown\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af0ec08e1f1fdf88d76bae2ba6d0fe986}


Definition at line 110 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-calc\-Fitness().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a2156c601b24d1a7fdcad1c59ec05e147}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-configuration@{m\-\_\-configuration}}
\index{m\-\_\-configuration@{m\-\_\-configuration}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-configuration}]{\setlength{\rightskip}{0pt plus 5cm}{\bf G\-P\-Configuration} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-configuration\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a2156c601b24d1a7fdcad1c59ec05e147}
The current configuration instance 

Definition at line 44 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Configuration(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-set\-G\-P\-Configuration().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af4a221f34692a5b47147dbbd31d26d23}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-fittest\-To\-Add@{m\-\_\-fittest\-To\-Add}}
\index{m\-\_\-fittest\-To\-Add@{m\-\_\-fittest\-To\-Add}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-fittest\-To\-Add}]{\setlength{\rightskip}{0pt plus 5cm}{\bf I\-G\-P\-Program} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-fittest\-To\-Add\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af4a221f34692a5b47147dbbd31d26d23}


Definition at line 108 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-add\-Fittest\-Program(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a3bfbce17cabd5ef77caf8427f5511407}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-full\-Mode\-Allowed@{m\-\_\-full\-Mode\-Allowed}}
\index{m\-\_\-full\-Mode\-Allowed@{m\-\_\-full\-Mode\-Allowed}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-full\-Mode\-Allowed}]{\setlength{\rightskip}{0pt plus 5cm}boolean org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-full\-Mode\-Allowed\mbox{[}$\,$\mbox{]}\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a3bfbce17cabd5ef77caf8427f5511407}
Is full mode with program construction allowed? 

Definition at line 68 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a7e95fd6aefeefa158ca044abccc0ffa9}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-max\-Depths@{m\-\_\-max\-Depths}}
\index{m\-\_\-max\-Depths@{m\-\_\-max\-Depths}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-max\-Depths}]{\setlength{\rightskip}{0pt plus 5cm}int \mbox{[}$\,$\mbox{]} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-max\-Depths\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a7e95fd6aefeefa158ca044abccc0ffa9}
Maximum depth per each chromosome 

Definition at line 93 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ab15e0e8d3ee0ba8c1e56727f2a2f32ed}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-max\-Nodes@{m\-\_\-max\-Nodes}}
\index{m\-\_\-max\-Nodes@{m\-\_\-max\-Nodes}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-max\-Nodes}]{\setlength{\rightskip}{0pt plus 5cm}int org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-max\-Nodes\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ab15e0e8d3ee0ba8c1e56727f2a2f32ed}
Maximum number of nodes allowed per chromosome (when exceeded program aborts) 

Definition at line 99 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af87aee9616f225ec99bf2f3db601e492}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-min\-Depths@{m\-\_\-min\-Depths}}
\index{m\-\_\-min\-Depths@{m\-\_\-min\-Depths}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-min\-Depths}]{\setlength{\rightskip}{0pt plus 5cm}int \mbox{[}$\,$\mbox{]} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-min\-Depths\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_af87aee9616f225ec99bf2f3db601e492}
Minimum depth per each chromosome 

Definition at line 88 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a045ec05f97be18ddbd121eb709d3e450}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-node\-Sets@{m\-\_\-node\-Sets}}
\index{m\-\_\-node\-Sets@{m\-\_\-node\-Sets}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-node\-Sets}]{\setlength{\rightskip}{0pt plus 5cm}{\bf Command\-Gene} \mbox{[}$\,$\mbox{]}\mbox{[}$\,$\mbox{]} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-node\-Sets\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a045ec05f97be18ddbd121eb709d3e450}
Available G\-P-\/functions. 

Definition at line 83 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a32bf2d50f238455154d3466fc1ef09a1}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-population@{m\-\_\-population}}
\index{m\-\_\-population@{m\-\_\-population}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-population}]{\setlength{\rightskip}{0pt plus 5cm}{\bf G\-P\-Population} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-population\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a32bf2d50f238455154d3466fc1ef09a1}
The array of \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_program}{G\-P\-Program}'s that make-\/up this \hyperlink{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype}{G\-P\-Genotype}'s population 

Definition at line 39 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-G\-P\-Population(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-set\-G\-P\-Population().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a9e2b1a8103181690f4bb6e584463214c}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-static\-Configuration@{m\-\_\-static\-Configuration}}
\index{m\-\_\-static\-Configuration@{m\-\_\-static\-Configuration}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-static\-Configuration}]{\setlength{\rightskip}{0pt plus 5cm}transient {\bf G\-P\-Configuration} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-static\-Configuration\hspace{0.3cm}{\ttfamily [static]}, {\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a9e2b1a8103181690f4bb6e584463214c}


Definition at line 46 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-Static\-G\-P\-Configuration(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-set\-Static\-G\-P\-Configuration().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a590812317593272cd66908ef7b712a63}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-total\-Fitness@{m\-\_\-total\-Fitness}}
\index{m\-\_\-total\-Fitness@{m\-\_\-total\-Fitness}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-total\-Fitness}]{\setlength{\rightskip}{0pt plus 5cm}double org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-total\-Fitness\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a590812317593272cd66908ef7b712a63}
Sum of fitness values over all chromosomes 

Definition at line 56 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-calc\-Fitness(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-get\-Total\-Fitness().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ae23e8520ee01ca7cb8f927a262db49c6}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-types@{m\-\_\-types}}
\index{m\-\_\-types@{m\-\_\-types}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-types}]{\setlength{\rightskip}{0pt plus 5cm}Class \mbox{[}$\,$\mbox{]} org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-types\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ae23e8520ee01ca7cb8f927a262db49c6}
Return type per chromosome 

Definition at line 73 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a03d40b9502d5bbbe9855cef68c9c8123}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-variables@{m\-\_\-variables}}
\index{m\-\_\-variables@{m\-\_\-variables}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-variables}]{\setlength{\rightskip}{0pt plus 5cm}Map org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-variables\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_a03d40b9502d5bbbe9855cef68c9c8123}


Definition at line 106 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-G\-P\-Genotype(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-random\-Initial\-Genotype().

\hypertarget{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ac5e56e43bba8e3368cacb3111a5fd60a}{\index{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}!m\-\_\-verbose@{m\-\_\-verbose}}
\index{m\-\_\-verbose@{m\-\_\-verbose}!org::jgap::gp::impl::GPGenotype@{org\-::jgap\-::gp\-::impl\-::\-G\-P\-Genotype}}
\subsubsection[{m\-\_\-verbose}]{\setlength{\rightskip}{0pt plus 5cm}boolean org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-m\-\_\-verbose\hspace{0.3cm}{\ttfamily [private]}}}\label{classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype_ac5e56e43bba8e3368cacb3111a5fd60a}
True\-: Output status information to console 

Definition at line 104 of file G\-P\-Genotype.\-java.



Referenced by org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-calc\-Fitness(), org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-evolve(), and org.\-jgap.\-gp.\-impl.\-G\-P\-Genotype.\-set\-Verbose\-Output().



The documentation for this class was generated from the following file\-:\begin{DoxyCompactItemize}
\item 
L\-D\-H\-\_\-\-Git/src/org/jgap/gp/impl/\hyperlink{_g_p_genotype_8java}{G\-P\-Genotype.\-java}\end{DoxyCompactItemize}
