<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Práctica_5_LDH: org.jgap.impl.GreedyCrossover Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">Práctica_5_LDH
   &#160;<span id="projectnumber">1.2</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li><a href="inherits.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Classes</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Namespaces</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Files</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Functions</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(6)"><span class="SelectionMark">&#160;</span>Pages</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="namespaceorg.html">org</a></li><li class="navelem"><a class="el" href="namespaceorg_1_1jgap.html">jgap</a></li><li class="navelem"><a class="el" href="namespaceorg_1_1jgap_1_1impl.html">impl</a></li><li class="navelem"><a class="el" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html">GreedyCrossover</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pro-methods">Protected Member Functions</a> &#124;
<a href="#pri-attribs">Private Attributes</a> &#124;
<a href="#pri-static-attribs">Static Private Attributes</a> &#124;
<a href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">org.jgap.impl.GreedyCrossover Class Reference</div>  </div>
</div><!--header-->
<div class="contents">
<div class="dynheader">
Inheritance diagram for org.jgap.impl.GreedyCrossover:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1impl_1_1_greedy_crossover__inherit__graph.png" border="0" usemap="#org_8jgap_8impl_8_greedy_crossover_inherit__map" alt="Inheritance graph"/></div>
<map name="org_8jgap_8impl_8_greedy_crossover_inherit__map" id="org_8jgap_8impl_8_greedy_crossover_inherit__map">
<area shape="rect" id="node2" href="classorg_1_1jgap_1_1_base_genetic_operator.html" title="org.jgap.BaseGeneticOperator" alt="" coords="65,155,263,181"/><area shape="rect" id="node4" href="interfaceorg_1_1jgap_1_1_genetic_operator.html" title="org.jgap.GeneticOperator" alt="" coords="5,80,171,107"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<div class="dynheader">
Collaboration diagram for org.jgap.impl.GreedyCrossover:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1impl_1_1_greedy_crossover__coll__graph.png" border="0" usemap="#org_8jgap_8impl_8_greedy_crossover_coll__map" alt="Collaboration graph"/></div>
<map name="org_8jgap_8impl_8_greedy_crossover_coll__map" id="org_8jgap_8impl_8_greedy_crossover_coll__map">
<area shape="rect" id="node2" href="classorg_1_1jgap_1_1_base_genetic_operator.html" title="org.jgap.BaseGeneticOperator" alt="" coords="1085,406,1283,433"/><area shape="rect" id="node4" href="interfaceorg_1_1jgap_1_1_genetic_operator.html" title="org.jgap.GeneticOperator" alt="" coords="598,21,765,47"/><area shape="rect" id="node10" href="classorg_1_1jgap_1_1_configuration.html" title="org.jgap.Configuration" alt="" coords="789,406,939,433"/><area shape="rect" id="node20" href="interfaceorg_1_1jgap_1_1_i_breeder.html" title="org.jgap.IBreeder" alt="" coords="397,585,518,611"/><area shape="rect" id="node25" href="interfaceorg_1_1jgap_1_1_i_j_g_a_p_factory.html" title="org.jgap.IJGAPFactory" alt="" coords="380,163,535,190"/><area shape="rect" id="node30" href="interfaceorg_1_1jgap_1_1_fitness_evaluator.html" title="org.jgap.FitnessEvaluator" alt="" coords="372,214,543,241"/><area shape="rect" id="node33" href="interfaceorg_1_1jgap_1_1_i_chromosome.html" title="org.jgap.IChromosome" alt="" coords="380,737,535,763"/><area shape="rect" id="node40" href="interfaceorg_1_1jgap_1_1_random_generator.html" title="org.jgap.RandomGenerator" alt="" coords="368,315,547,342"/><area shape="rect" id="node43" href="classorg_1_1jgap_1_1_bulk_fitness_function.html" title="org.jgap.BulkFitnessFunction" alt="" coords="362,433,553,459"/><area shape="rect" id="node47" href="classorg_1_1jgap_1_1_configuration_1_1_configuration_configurable.html" title="org.jgap.Configuration.Configuration\lConfigurable" alt="" coords="342,367,573,408"/><area shape="rect" id="node50" href="classorg_1_1jgap_1_1_fitness_function.html" title="org.jgap.FitnessFunction" alt="" coords="375,483,540,510"/><area shape="rect" id="node15" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html" title="org.jgap.util.ICloneable" alt="" coords="136,582,291,609"/><area shape="rect" id="node28" href="interfaceorg_1_1jgap_1_1_i_chromosome_pool.html" title="org.jgap.IChromosomePool" alt="" coords="367,265,547,291"/><area shape="rect" id="node38" href="interfaceorg_1_1jgap_1_1_i_unique_key.html" title="org.jgap.IUniqueKey" alt="" coords="143,821,283,847"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a5777f6e13508c61d2c1e43ee0900cbf3"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#a5777f6e13508c61d2c1e43ee0900cbf3">GreedyCrossover</a> ()  throws InvalidConfigurationException </td></tr>
<tr class="separator:a5777f6e13508c61d2c1e43ee0900cbf3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2592b70d785e0344ac6cbc23d10c7864"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#a2592b70d785e0344ac6cbc23d10c7864">GreedyCrossover</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a2592b70d785e0344ac6cbc23d10c7864"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a41deed206eb483d92e5dee7d0bf537ef"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#a41deed206eb483d92e5dee7d0bf537ef">distance</a> (final Object a_from, final Object a_to)</td></tr>
<tr class="separator:a41deed206eb483d92e5dee7d0bf537ef"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a57adb8d442035ecae43e194f29dff43c"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#a57adb8d442035ecae43e194f29dff43c">operate</a> (final <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> a_population, final List a_candidateChromosomes)</td></tr>
<tr class="separator:a57adb8d442035ecae43e194f29dff43c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9fc2e11874fbc8b8349c1a0d93a8b5d9"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#a9fc2e11874fbc8b8349c1a0d93a8b5d9">operate</a> (final <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> a_firstMate, final <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> a_secondMate)</td></tr>
<tr class="separator:a9fc2e11874fbc8b8349c1a0d93a8b5d9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6f518b090d9a358f7d96b7b11e6bf785"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#a6f518b090d9a358f7d96b7b11e6bf785">setStartOffset</a> (int a_offset)</td></tr>
<tr class="separator:a6f518b090d9a358f7d96b7b11e6bf785"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6d4b71a26f742714520a6dfa931cc8c2"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#a6d4b71a26f742714520a6dfa931cc8c2">getStartOffset</a> ()</td></tr>
<tr class="separator:a6d4b71a26f742714520a6dfa931cc8c2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab1cb7c568f154dba5f43311e2c75ccfe"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#ab1cb7c568f154dba5f43311e2c75ccfe">compareTo</a> (final Object a_other)</td></tr>
<tr class="separator:ab1cb7c568f154dba5f43311e2c75ccfe"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_methods_classorg_1_1jgap_1_1_base_genetic_operator"><td colspan="2" onclick="javascript:toggleInherit('pub_methods_classorg_1_1jgap_1_1_base_genetic_operator')"><img src="closed.png" alt="-"/>&#160;Public Member Functions inherited from <a class="el" href="classorg_1_1jgap_1_1_base_genetic_operator.html">org.jgap.BaseGeneticOperator</a></td></tr>
<tr class="memitem:a061b47c0f891c0041d5ece5bc6e052f8 inherit pub_methods_classorg_1_1jgap_1_1_base_genetic_operator"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_genetic_operator.html#a061b47c0f891c0041d5ece5bc6e052f8">BaseGeneticOperator</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a061b47c0f891c0041d5ece5bc6e052f8 inherit pub_methods_classorg_1_1jgap_1_1_base_genetic_operator"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:addbc6b98f572331d47397227b7a49f6a inherit pub_methods_classorg_1_1jgap_1_1_base_genetic_operator"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_genetic_operator.html#addbc6b98f572331d47397227b7a49f6a">getConfiguration</a> ()</td></tr>
<tr class="separator:addbc6b98f572331d47397227b7a49f6a inherit pub_methods_classorg_1_1jgap_1_1_base_genetic_operator"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4db6fa1a99177114c0f19f02cd4a87ec inherit pub_methods_classorg_1_1jgap_1_1_base_genetic_operator"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_genetic_operator.html#a4db6fa1a99177114c0f19f02cd4a87ec">equals</a> (final Object a_other)</td></tr>
<tr class="separator:a4db6fa1a99177114c0f19f02cd4a87ec inherit pub_methods_classorg_1_1jgap_1_1_base_genetic_operator"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pro-methods"></a>
Protected Member Functions</h2></td></tr>
<tr class="memitem:a6a19fa44501e6bd2d3e5be8ba7424b3a"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#a6a19fa44501e6bd2d3e5be8ba7424b3a">operate</a> (final <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[] a_g1, final <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[] a_g2)</td></tr>
<tr class="separator:a6a19fa44501e6bd2d3e5be8ba7424b3a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abb3111b92c3d771a9a4d5320071259a6"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#abb3111b92c3d771a9a4d5320071259a6">findNext</a> (final <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[] a_g, final <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> a_x)</td></tr>
<tr class="separator:abb3111b92c3d771a9a4d5320071259a6"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-attribs"></a>
Private Attributes</h2></td></tr>
<tr class="memitem:a2c851e311ad5a263c915e01d8a30aa70"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#a2c851e311ad5a263c915e01d8a30aa70">m_startOffset</a> = 1</td></tr>
<tr class="separator:a2c851e311ad5a263c915e01d8a30aa70"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-static-attribs"></a>
Static Private Attributes</h2></td></tr>
<tr class="memitem:ab6ee35a26acab5f96032250a6a5d2696"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#ab6ee35a26acab5f96032250a6a5d2696">CVS_REVISION</a> = &quot;$Revision: 1.30 $&quot;</td></tr>
<tr class="separator:ab6ee35a26acab5f96032250a6a5d2696"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="inherited"></a>
Additional Inherited Members</h2></td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_genetic_operator"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_genetic_operator')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_genetic_operator.html">org.jgap.GeneticOperator</a></td></tr>
<tr class="memitem:ad0ddbf3fa1dfd7bd8639eba1d9e35267 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_genetic_operator"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_genetic_operator.html#ad0ddbf3fa1dfd7bd8639eba1d9e35267">CVS_REVISION</a> = &quot;$Revision: 1.16 $&quot;</td></tr>
<tr class="separator:ad0ddbf3fa1dfd7bd8639eba1d9e35267 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_genetic_operator"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pro_attribs_classorg_1_1jgap_1_1_base_genetic_operator"><td colspan="2" onclick="javascript:toggleInherit('pro_attribs_classorg_1_1jgap_1_1_base_genetic_operator')"><img src="closed.png" alt="-"/>&#160;Protected Attributes inherited from <a class="el" href="classorg_1_1jgap_1_1_base_genetic_operator.html">org.jgap.BaseGeneticOperator</a></td></tr>
<tr class="memitem:a516d89e8f56eb2e58768218c7f9cf3de inherit pro_attribs_classorg_1_1jgap_1_1_base_genetic_operator"><td class="memItemLeft" align="right" valign="top">IEvolutionMonitor&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_genetic_operator.html#a516d89e8f56eb2e58768218c7f9cf3de">m_monitor</a></td></tr>
<tr class="separator:a516d89e8f56eb2e58768218c7f9cf3de inherit pro_attribs_classorg_1_1jgap_1_1_base_genetic_operator"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5194e705ed206f99852bf158546646b5 inherit pro_attribs_classorg_1_1jgap_1_1_base_genetic_operator"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_genetic_operator.html#a5194e705ed206f99852bf158546646b5">m_monitorActive</a></td></tr>
<tr class="separator:a5194e705ed206f99852bf158546646b5 inherit pro_attribs_classorg_1_1jgap_1_1_base_genetic_operator"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>The Greedy Crossover is a specific type of crossover. It can only be is applied if </p>
<ul>
<li>
<ol type="1">
<li>All genes in the chromosome are different and  </li>
</ol>
</li>
<li>
<ol type="1">
<li>The set of genes for both chromosomes is identical and only their order in the chromosome can vary.  </li>
</ol>
</li>
</ul>
<p>After the <a class="el" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html">GreedyCrossover</a>, these two conditions always remain true, so it can be applied again and again.</p>
<p>The algorithm throws an assertion error if the two initial chromosomes does not satisfy these conditions.</p>
<p>Greedy crossover can be best explained in the terms of the Traveling Salesman Problem:</p>
<p>The algorithm selects the first city of one parent, compares the cities leaving that city in both parents, and chooses the closer one to extend the tour. If one city has already appeared in the tour, we choose the other city. If both cities have already appeared, we randomly select a non-selected city.</p>
<p>See J. Grefenstette, R. Gopal, R. Rosmaita, and D. Gucht. <em>Genetic algorithms for the traveling salesman problem</em>. In Proceedings of the Second International Conference on Genetic Algorithms. Lawrence Eribaum Associates, Mahwah, NJ, 1985. and also <a href="http://ecsl.cs.unr.edu/docs/techreports/gong/node3.html">Sushil J. Louis &amp; Gong Li</a>}</p>
<dl class="section author"><dt>Author</dt><dd>Audrius Meskauskas </dd>
<dd>
&lt;font size="-1"&gt;Neil Rotstan, Klaus Meffert (reused code from <a class="el" href="classorg_1_1jgap_1_1impl_1_1_crossover_operator.html">CrossoverOperator</a>)&lt;/font&gt; </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Definition at line <a class="el" href="_greedy_crossover_8java_source.html#l00055">55</a> of file <a class="el" href="_greedy_crossover_8java_source.html">GreedyCrossover.java</a>.</p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="a5777f6e13508c61d2c1e43ee0900cbf3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.impl.GreedyCrossover.GreedyCrossover </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Default constructor for dynamic instantiation.</p>
<p>Attention: The configuration used is the one set with the static method Genotype.setConfiguration.</p>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd>
<dd>
3.0 (since 2.0 without a_configuration) </dd></dl>

<p>Definition at line <a class="el" href="_greedy_crossover_8java_source.html#l00076">76</a> of file <a class="el" href="_greedy_crossover_8java_source.html">GreedyCrossover.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00520">org.jgap.Genotype.getStaticConfiguration()</a>.</p>

<p>Referenced by <a class="el" href="_greedy_crossover_8java_source.html#l00306">org.jgap.impl.GreedyCrossover.compareTo()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a5777f6e13508c61d2c1e43ee0900cbf3_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a5777f6e13508c61d2c1e43ee0900cbf3_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a5777f6e13508c61d2c1e43ee0900cbf3_cgraph" id="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a5777f6e13508c61d2c1e43ee0900cbf3_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#aaf3d4bf0b470aa4655e85262289443eb" title="org.jgap.Genotype.getStatic\lConfiguration" alt="" coords="261,5,445,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a2592b70d785e0344ac6cbc23d10c7864"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.impl.GreedyCrossover.GreedyCrossover </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Using the given configuration.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the configuration to use </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 (since 2.6 without a_configuration) </dd></dl>

<p>Definition at line <a class="el" href="_greedy_crossover_8java_source.html#l00090">90</a> of file <a class="el" href="_greedy_crossover_8java_source.html">GreedyCrossover.java</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="ab1cb7c568f154dba5f43311e2c75ccfe"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.impl.GreedyCrossover.compareTo </td>
          <td>(</td>
          <td class="paramtype">final Object&#160;</td>
          <td class="paramname"><em>a_other</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Compares the given <a class="el" href="interfaceorg_1_1jgap_1_1_genetic_operator.html">GeneticOperator</a> to this <a class="el" href="interfaceorg_1_1jgap_1_1_genetic_operator.html">GeneticOperator</a>.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_other</td><td>the instance against which to compare this instance </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>a negative number if this instance is "less than" the given instance, zero if they are equal to each other, and a positive number if this is "greater than" the given instance</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Definition at line <a class="el" href="_greedy_crossover_8java_source.html#l00306">306</a> of file <a class="el" href="_greedy_crossover_8java_source.html">GreedyCrossover.java</a>.</p>

<p>References <a class="el" href="_greedy_crossover_8java_source.html#l00291">org.jgap.impl.GreedyCrossover.getStartOffset()</a>, and <a class="el" href="_greedy_crossover_8java_source.html#l00076">org.jgap.impl.GreedyCrossover.GreedyCrossover()</a>.</p>

<p>Referenced by <a class="el" href="_greedy_crossover_test_8java_source.html#l00508">org.jgap.impl.GreedyCrossoverTest.testCompareTo_0()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_ab1cb7c568f154dba5f43311e2c75ccfe_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1impl_1_1_greedy_crossover_ab1cb7c568f154dba5f43311e2c75ccfe_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_ab1cb7c568f154dba5f43311e2c75ccfe_cgraph" id="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_ab1cb7c568f154dba5f43311e2c75ccfe_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#a6d4b71a26f742714520a6dfa931cc8c2" title="org.jgap.impl.GreedyCrossover.get\lStartOffset" alt="" coords="325,5,549,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#a5777f6e13508c61d2c1e43ee0900cbf3" title="org.jgap.impl.GreedyCrossover.\lGreedyCrossover" alt="" coords="334,70,541,111"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_genotype.html#aaf3d4bf0b470aa4655e85262289443eb" title="org.jgap.Genotype.getStatic\lConfiguration" alt="" coords="597,70,781,111"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a41deed206eb483d92e5dee7d0bf537ef"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double org.jgap.impl.GreedyCrossover.distance </td>
          <td>(</td>
          <td class="paramtype">final Object&#160;</td>
          <td class="paramname"><em>a_from</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final Object&#160;</td>
          <td class="paramname"><em>a_to</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Compute the distance between "cities", indicated by these two given genes. The default method expects the genes to be <a class="el" href="classorg_1_1jgap_1_1impl_1_1_integer_gene.html">IntegerGene</a>'s and returns their absolute difference, that makes sense only for tests.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_from</td><td>Object </td></tr>
    <tr><td class="paramname">a_to</td><td>Object </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>distance between the two given cities </dd></dl>

<p>Definition at line <a class="el" href="_greedy_crossover_8java_source.html#l00105">105</a> of file <a class="el" href="_greedy_crossover_8java_source.html">GreedyCrossover.java</a>.</p>

<p>References <a class="el" href="_integer_gene_8java_source.html#l00239">org.jgap.impl.IntegerGene.intValue()</a>.</p>

<p>Referenced by <a class="el" href="_greedy_crossover_8java_source.html#l00176">org.jgap.impl.GreedyCrossover.operate()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a41deed206eb483d92e5dee7d0bf537ef_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a41deed206eb483d92e5dee7d0bf537ef_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a41deed206eb483d92e5dee7d0bf537ef_cgraph" id="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a41deed206eb483d92e5dee7d0bf537ef_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1impl_1_1_integer_gene.html#a90c518c0ed7c08016323ea88b9fb4078" title="org.jgap.impl.IntegerGene.int\lValue" alt="" coords="309,5,499,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_gene.html#a35e754e8e217bdd26962e1c3c31d0169" title="org.jgap.BaseGene.getAllele" alt="" coords="547,12,733,39"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_gene.html#ad20da6b23ba524a95caa28f727c26dcd" title="org.jgap.BaseGene.getInternal\lValue" alt="" coords="782,5,981,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="abb3111b92c3d771a9a4d5320071259a6"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> org.jgap.impl.GreedyCrossover.findNext </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[]&#160;</td>
          <td class="paramname"><em>a_g</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>&#160;</td>
          <td class="paramname"><em>a_x</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_greedy_crossover_8java_source.html#l00262">262</a> of file <a class="el" href="_greedy_crossover_8java_source.html">GreedyCrossover.java</a>.</p>

<p>References <a class="el" href="_base_genetic_operator_8java_source.html#l00072">org.jgap.BaseGeneticOperator.equals()</a>, and <a class="el" href="_greedy_crossover_8java_source.html#l00063">org.jgap.impl.GreedyCrossover.m_startOffset</a>.</p>

<p>Referenced by <a class="el" href="_greedy_crossover_8java_source.html#l00176">org.jgap.impl.GreedyCrossover.operate()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_abb3111b92c3d771a9a4d5320071259a6_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1impl_1_1_greedy_crossover_abb3111b92c3d771a9a4d5320071259a6_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_abb3111b92c3d771a9a4d5320071259a6_cgraph" id="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_abb3111b92c3d771a9a4d5320071259a6_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_genetic_operator.html#a4db6fa1a99177114c0f19f02cd4a87ec" title="org.jgap.BaseGeneticOperator.equals" alt="" coords="310,5,549,32"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a6d4b71a26f742714520a6dfa931cc8c2"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.impl.GreedyCrossover.getStartOffset </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Gets a number of genes at the start of chromosome, that are excluded from the swapping. In the Salesman task, the first city in the list should (where the salesman leaves from) probably should not change as it is part of the list. The default value is 1.</p>
<dl class="section return"><dt>Returns</dt><dd>the start offset used </dd></dl>

<p>Definition at line <a class="el" href="_greedy_crossover_8java_source.html#l00291">291</a> of file <a class="el" href="_greedy_crossover_8java_source.html">GreedyCrossover.java</a>.</p>

<p>References <a class="el" href="_greedy_crossover_8java_source.html#l00063">org.jgap.impl.GreedyCrossover.m_startOffset</a>.</p>

<p>Referenced by <a class="el" href="_greedy_crossover_8java_source.html#l00306">org.jgap.impl.GreedyCrossover.compareTo()</a>, and <a class="el" href="_greedy_crossover_test_8java_source.html#l00449">org.jgap.impl.GreedyCrossoverTest.testStartoffset_0()</a>.</p>

</div>
</div>
<a class="anchor" id="a57adb8d442035ecae43e194f29dff43c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.impl.GreedyCrossover.operate </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>&#160;</td>
          <td class="paramname"><em>a_population</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final List&#160;</td>
          <td class="paramname"><em>a_candidateChromosomes</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>The operate method will be invoked on each of the genetic operators referenced by the current <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object during the evolution phase. Operators are given an opportunity to run in the order that they are added to the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>. Implementations of this method may reference the population of Chromosomes as it was at the beginning of the evolutionary phase and/or they may instead reference the candidate Chromosomes, which are the results of prior genetic operators. In either case, only Chromosomes added to the list of candidate chromosomes will be considered for natural selection.</p>
<p>The parameters a_population and a_candidateChromosomes may refer to the same list of chromosomes for performance issues. Thus would mean an in-place modification. In ealier JGAP versions it was suggested never modifying the input population. Please refer to implementations delivered with JGAP to get a picture of the way non-susceptible in-place modifications are possible. If wrongly done, ConcurrentModificationException could be risen when accessing the population by an iterator in a <a class="el" href="interfaceorg_1_1jgap_1_1_genetic_operator.html">GeneticOperator</a>. Or, if population.getChromosomes().size() was used inside a loop where chromosomes were added to the input population this could lead to an infinite loop in worst case.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_population</td><td>the population of chromosomes from the current evolution prior to exposure to any genetic operators. Chromosomes in this array should not be modified. Please, notice, that the call in <a class="el" href="classorg_1_1jgap_1_1_genotype.html#a54a6791fbb6e13920e79f4e2dd719eb4">Genotype.evolve()</a> to the implementations of <a class="el" href="interfaceorg_1_1jgap_1_1_genetic_operator.html">GeneticOperator</a> overgoes this due to performance issues </td></tr>
    <tr><td class="paramname">a_candidateChromosomes</td><td>the pool of chromosomes that have been selected for the next evolved population</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 (earlier versions referenced the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object) </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_genetic_operator.html#a0f9dfe925c4c21e07522be67d6c5d084">org.jgap.GeneticOperator</a>.</p>

<p>Definition at line <a class="el" href="_greedy_crossover_8java_source.html#l00111">111</a> of file <a class="el" href="_greedy_crossover_8java_source.html">GreedyCrossover.java</a>.</p>

<p>References <a class="el" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html#aa7e7d62077e6428ad7904932b1b4f7d5">org.jgap.util.ICloneable.clone()</a>, <a class="el" href="_base_genetic_operator_8java_source.html#l00057">org.jgap.BaseGeneticOperator.getConfiguration()</a>, <a class="el" href="_base_genetic_operator_8java_source.html#l00030">org.jgap.BaseGeneticOperator.m_monitorActive</a>, and <a class="el" href="_population_8java_source.html#l00267">org.jgap.Population.size()</a>.</p>

<p>Referenced by <a class="el" href="_greedy_crossover_8java_source.html#l00155">org.jgap.impl.GreedyCrossover.operate()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a57adb8d442035ecae43e194f29dff43c_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a57adb8d442035ecae43e194f29dff43c_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a57adb8d442035ecae43e194f29dff43c_cgraph" id="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a57adb8d442035ecae43e194f29dff43c_cgraph">
<area shape="rect" id="node3" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html#aa7e7d62077e6428ad7904932b1b4f7d5" title="org.jgap.util.ICloneable.clone" alt="" coords="318,5,509,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_genetic_operator.html#addbc6b98f572331d47397227b7a49f6a" title="org.jgap.BaseGeneticOperator.get\lConfiguration" alt="" coords="304,57,523,98"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_population.html#a2d09529f8e93d57a0546625362ca5bee" title="org.jgap.Population.size" alt="" coords="333,123,494,149"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a9fc2e11874fbc8b8349c1a0d93a8b5d9"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.impl.GreedyCrossover.operate </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td>
          <td class="paramname"><em>a_firstMate</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td>
          <td class="paramname"><em>a_secondMate</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Performs a greedy crossover for the two given chromosoms.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_firstMate</td><td>the first chromosome to crossover on </td></tr>
    <tr><td class="paramname">a_secondMate</td><td>the second chromosome to crossover on </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Error</td><td>if the gene set in the chromosomes is not identical</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Audrius Meskauskas </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.1 </dd></dl>

<p>Definition at line <a class="el" href="_greedy_crossover_8java_source.html#l00155">155</a> of file <a class="el" href="_greedy_crossover_8java_source.html">GreedyCrossover.java</a>.</p>

<p>References <a class="el" href="_greedy_crossover_8java_source.html#l00063">org.jgap.impl.GreedyCrossover.m_startOffset</a>, and <a class="el" href="_greedy_crossover_8java_source.html#l00111">org.jgap.impl.GreedyCrossover.operate()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a9fc2e11874fbc8b8349c1a0d93a8b5d9_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a9fc2e11874fbc8b8349c1a0d93a8b5d9_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a9fc2e11874fbc8b8349c1a0d93a8b5d9_cgraph" id="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a9fc2e11874fbc8b8349c1a0d93a8b5d9_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#a57adb8d442035ecae43e194f29dff43c" title="org.jgap.impl.GreedyCrossover.operate" alt="" coords="305,64,554,91"/><area shape="rect" id="node5" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html#aa7e7d62077e6428ad7904932b1b4f7d5" title="org.jgap.util.ICloneable.clone" alt="" coords="617,5,807,32"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_genetic_operator.html#addbc6b98f572331d47397227b7a49f6a" title="org.jgap.BaseGeneticOperator.get\lConfiguration" alt="" coords="603,57,821,98"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_population.html#a2d09529f8e93d57a0546625362ca5bee" title="org.jgap.Population.size" alt="" coords="631,123,793,149"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a6a19fa44501e6bd2d3e5be8ba7424b3a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> [] org.jgap.impl.GreedyCrossover.operate </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[]&#160;</td>
          <td class="paramname"><em>a_g1</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[]&#160;</td>
          <td class="paramname"><em>a_g2</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_greedy_crossover_8java_source.html#l00176">176</a> of file <a class="el" href="_greedy_crossover_8java_source.html">GreedyCrossover.java</a>.</p>

<p>References <a class="el" href="_greedy_crossover_8java_source.html#l00105">org.jgap.impl.GreedyCrossover.distance()</a>, <a class="el" href="_base_genetic_operator_8java_source.html#l00072">org.jgap.BaseGeneticOperator.equals()</a>, <a class="el" href="_greedy_crossover_8java_source.html#l00262">org.jgap.impl.GreedyCrossover.findNext()</a>, and <a class="el" href="_greedy_crossover_8java_source.html#l00063">org.jgap.impl.GreedyCrossover.m_startOffset</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a6a19fa44501e6bd2d3e5be8ba7424b3a_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a6a19fa44501e6bd2d3e5be8ba7424b3a_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a6a19fa44501e6bd2d3e5be8ba7424b3a_cgraph" id="classorg_1_1jgap_1_1impl_1_1_greedy_crossover_a6a19fa44501e6bd2d3e5be8ba7424b3a_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#a41deed206eb483d92e5dee7d0bf537ef" title="org.jgap.impl.GreedyCrossover.distance" alt="" coords="305,12,559,39"/><area shape="rect" id="node11" href="classorg_1_1jgap_1_1_base_genetic_operator.html#a4db6fa1a99177114c0f19f02cd4a87ec" title="org.jgap.BaseGeneticOperator.equals" alt="" coords="609,92,847,119"/><area shape="rect" id="node13" href="classorg_1_1jgap_1_1impl_1_1_greedy_crossover.html#abb3111b92c3d771a9a4d5320071259a6" title="org.jgap.impl.GreedyCrossover.findNext" alt="" coords="305,113,559,140"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1impl_1_1_integer_gene.html#a90c518c0ed7c08016323ea88b9fb4078" title="org.jgap.impl.IntegerGene.int\lValue" alt="" coords="633,5,823,46"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_gene.html#a35e754e8e217bdd26962e1c3c31d0169" title="org.jgap.BaseGene.getAllele" alt="" coords="896,12,1083,39"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_base_gene.html#ad20da6b23ba524a95caa28f727c26dcd" title="org.jgap.BaseGene.getInternal\lValue" alt="" coords="1131,5,1330,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a6f518b090d9a358f7d96b7b11e6bf785"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.impl.GreedyCrossover.setStartOffset </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_offset</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets a number of genes at the start of chromosome, that are excluded from the swapping. In the Salesman task, the first city in the list should (where the salesman leaves from) probably should not change as it is part of the list. The default value is 1.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_offset</td><td>the start offset to use </td></tr>
  </table>
  </dd>
</dl>

<p>Definition at line <a class="el" href="_greedy_crossover_8java_source.html#l00279">279</a> of file <a class="el" href="_greedy_crossover_8java_source.html">GreedyCrossover.java</a>.</p>

<p>References <a class="el" href="_greedy_crossover_8java_source.html#l00063">org.jgap.impl.GreedyCrossover.m_startOffset</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Data Documentation</h2>
<a class="anchor" id="ab6ee35a26acab5f96032250a6a5d2696"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final String org.jgap.impl.GreedyCrossover.CVS_REVISION = &quot;$Revision: 1.30 $&quot;</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>String containing the CVS revision. Read out via reflection! </p>

<p>Definition at line <a class="el" href="_greedy_crossover_8java_source.html#l00058">58</a> of file <a class="el" href="_greedy_crossover_8java_source.html">GreedyCrossover.java</a>.</p>

</div>
</div>
<a class="anchor" id="a2c851e311ad5a263c915e01d8a30aa70"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.impl.GreedyCrossover.m_startOffset = 1</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_greedy_crossover_8java_source.html#l00063">63</a> of file <a class="el" href="_greedy_crossover_8java_source.html">GreedyCrossover.java</a>.</p>

<p>Referenced by <a class="el" href="_greedy_crossover_8java_source.html#l00262">org.jgap.impl.GreedyCrossover.findNext()</a>, <a class="el" href="_greedy_crossover_8java_source.html#l00291">org.jgap.impl.GreedyCrossover.getStartOffset()</a>, <a class="el" href="_greedy_crossover_8java_source.html#l00155">org.jgap.impl.GreedyCrossover.operate()</a>, and <a class="el" href="_greedy_crossover_8java_source.html#l00279">org.jgap.impl.GreedyCrossover.setStartOffset()</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>LDH_Git/src/org/jgap/impl/<a class="el" href="_greedy_crossover_8java_source.html">GreedyCrossover.java</a></li>
</ul>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Sat Nov 23 2013 17:42:52 for Práctica_5_LDH by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
