<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Práctica_5_LDH: org.jgap.BaseChromosome Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">Práctica_5_LDH
   &#160;<span id="projectnumber">1.2</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li><a href="inherits.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Classes</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Namespaces</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Files</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Functions</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(6)"><span class="SelectionMark">&#160;</span>Pages</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="namespaceorg.html">org</a></li><li class="navelem"><a class="el" href="namespaceorg_1_1jgap.html">jgap</a></li><li class="navelem"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html">BaseChromosome</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-static-attribs">Static Public Attributes</a> &#124;
<a href="#pro-methods">Protected Member Functions</a> &#124;
<a href="#pro-static-methods">Static Protected Member Functions</a> &#124;
<a href="#pri-attribs">Private Attributes</a> &#124;
<a href="#pri-static-attribs">Static Private Attributes</a> &#124;
<a href="classorg_1_1jgap_1_1_base_chromosome-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">org.jgap.BaseChromosome Class Reference<span class="mlabels"><span class="mlabel">abstract</span></span></div>  </div>
</div><!--header-->
<div class="contents">
<div class="dynheader">
Inheritance diagram for org.jgap.BaseChromosome:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_chromosome__inherit__graph.png" border="0" usemap="#org_8jgap_8_base_chromosome_inherit__map" alt="Inheritance graph"/></div>
<map name="org_8jgap_8_base_chromosome_inherit__map" id="org_8jgap_8_base_chromosome_inherit__map">
<area shape="rect" id="node23" href="classorg_1_1jgap_1_1_chromosome.html" title="org.jgap.Chromosome" alt="" coords="838,208,989,235"/><area shape="rect" id="node2" href="interfaceorg_1_1jgap_1_1_i_chromosome.html" title="org.jgap.IChromosome" alt="" coords="371,107,525,133"/><area shape="rect" id="node6" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html" title="org.jgap.util.ICloneable" alt="" coords="133,56,288,83"/><area shape="rect" id="node12" href="interfaceorg_1_1jgap_1_1_i_unique_key.html" title="org.jgap.IUniqueKey" alt="" coords="141,157,281,184"/><area shape="rect" id="node14" href="interfaceorg_1_1jgap_1_1_i_initializer.html" title="org.jgap.IInitializer" alt="" coords="384,183,512,209"/><area shape="rect" id="node16" href="interfaceorg_1_1jgap_1_1_i_handler.html" title="org.jgap.IHandler" alt="" coords="150,208,271,235"/><area shape="rect" id="node18" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html" title="org.jgap.IPersistentRepresentation" alt="" coords="337,233,559,260"/><area shape="rect" id="node20" href="interfaceorg_1_1jgap_1_1_i_business_key.html" title="org.jgap.IBusinessKey" alt="" coords="371,284,525,311"/><area shape="rect" id="node25" href="classorg_1_1jgap_1_1_chromosome_for_testing.html" title="org.jgap.ChromosomeForTesting" alt="" coords="1041,157,1253,184"/><area shape="rect" id="node29" href="classorg_1_1jgap_1_1_chromosome_for_testing2.html" title="org.jgap.ChromosomeForTesting2" alt="" coords="1037,208,1256,235"/><area shape="rect" id="node27" href="classorg_1_1jgap_1_1_genotype_test_1_1_chromosome_for_test2.html" title="org.jgap.GenotypeTest.Chromosome\lForTest2" alt="" coords="1305,150,1541,191"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<div class="dynheader">
Collaboration diagram for org.jgap.BaseChromosome:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_chromosome__coll__graph.png" border="0" usemap="#org_8jgap_8_base_chromosome_coll__map" alt="Collaboration graph"/></div>
<map name="org_8jgap_8_base_chromosome_coll__map" id="org_8jgap_8_base_chromosome_coll__map">
<area shape="rect" id="node2" href="interfaceorg_1_1jgap_1_1_i_chromosome.html" title="org.jgap.IChromosome" alt="" coords="380,181,535,208"/><area shape="rect" id="node22" href="classorg_1_1jgap_1_1_configuration.html" title="org.jgap.Configuration" alt="" coords="811,444,960,471"/><area shape="rect" id="node29" href="interfaceorg_1_1jgap_1_1_i_breeder.html" title="org.jgap.IBreeder" alt="" coords="397,292,518,319"/><area shape="rect" id="node59" href="interfaceorg_1_1jgap_1_1_gene.html" title="org.jgap.Gene" alt="" coords="406,63,509,89"/><area shape="rect" id="node6" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html" title="org.jgap.util.ICloneable" alt="" coords="136,321,291,348"/><area shape="rect" id="node46" href="classorg_1_1jgap_1_1_bulk_fitness_function.html" title="org.jgap.BulkFitnessFunction" alt="" coords="362,561,553,588"/><area shape="rect" id="node53" href="classorg_1_1jgap_1_1_fitness_function.html" title="org.jgap.FitnessFunction" alt="" coords="375,343,540,369"/><area shape="rect" id="node34" href="interfaceorg_1_1jgap_1_1_i_j_g_a_p_factory.html" title="org.jgap.IJGAPFactory" alt="" coords="380,663,535,689"/><area shape="rect" id="node39" href="interfaceorg_1_1jgap_1_1_fitness_evaluator.html" title="org.jgap.FitnessEvaluator" alt="" coords="372,713,543,740"/><area shape="rect" id="node43" href="interfaceorg_1_1jgap_1_1_random_generator.html" title="org.jgap.RandomGenerator" alt="" coords="368,815,547,841"/><area shape="rect" id="node50" href="classorg_1_1jgap_1_1_configuration_1_1_configuration_configurable.html" title="org.jgap.Configuration.Configuration\lConfigurable" alt="" coords="342,495,573,537"/><area shape="rect" id="node12" href="interfaceorg_1_1jgap_1_1_i_unique_key.html" title="org.jgap.IUniqueKey" alt="" coords="143,5,283,32"/><area shape="rect" id="node14" href="interfaceorg_1_1jgap_1_1_i_initializer.html" title="org.jgap.IInitializer" alt="" coords="821,495,949,521"/><area shape="rect" id="node16" href="interfaceorg_1_1jgap_1_1_i_handler.html" title="org.jgap.IHandler" alt="" coords="397,865,518,892"/><area shape="rect" id="node18" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html" title="org.jgap.IPersistentRepresentation" alt="" coords="774,545,997,572"/><area shape="rect" id="node20" href="interfaceorg_1_1jgap_1_1_i_business_key.html" title="org.jgap.IBusinessKey" alt="" coords="809,596,962,623"/><area shape="rect" id="node37" href="interfaceorg_1_1jgap_1_1_i_chromosome_pool.html" title="org.jgap.IChromosomePool" alt="" coords="367,444,547,471"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a94e1af5ba804061833ed3883d42f9441"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a94e1af5ba804061833ed3883d42f9441">BaseChromosome</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a94e1af5ba804061833ed3883d42f9441"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8cf998c46fddd9ee2f8bfcd2a7b6bffa"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a8cf998c46fddd9ee2f8bfcd2a7b6bffa">getUniqueID</a> ()</td></tr>
<tr class="separator:a8cf998c46fddd9ee2f8bfcd2a7b6bffa"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a06156bb0f2c8512ca6a7f7f0fee305b9"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a06156bb0f2c8512ca6a7f7f0fee305b9">setUniqueIDTemplate</a> (String a_templateID, int a_index)</td></tr>
<tr class="separator:a06156bb0f2c8512ca6a7f7f0fee305b9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2d23cfd243723be4d5eb775532e36a6e"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a2d23cfd243723be4d5eb775532e36a6e">getUniqueIDTemplate</a> (int a_index)</td></tr>
<tr class="separator:a2d23cfd243723be4d5eb775532e36a6e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a104b33f29941ba6ba405889964fc2bf5"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a104b33f29941ba6ba405889964fc2bf5">getConfiguration</a> ()</td></tr>
<tr class="separator:a104b33f29941ba6ba405889964fc2bf5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a97071ef6dba552c1c1f2b2780d27eec3"><td class="memItemLeft" align="right" valign="top">abstract Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a97071ef6dba552c1c1f2b2780d27eec3">clone</a> ()</td></tr>
<tr class="separator:a97071ef6dba552c1c1f2b2780d27eec3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6dcae8b12cad267f6f91431281869e02"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a6dcae8b12cad267f6f91431281869e02">increaseAge</a> ()</td></tr>
<tr class="separator:a6dcae8b12cad267f6f91431281869e02"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7d4483ef679d0e1eb335256b72ab0731"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a7d4483ef679d0e1eb335256b72ab0731">resetAge</a> ()</td></tr>
<tr class="separator:a7d4483ef679d0e1eb335256b72ab0731"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af987f6a74486145a139985297c18ae80"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#af987f6a74486145a139985297c18ae80">getAge</a> ()</td></tr>
<tr class="separator:af987f6a74486145a139985297c18ae80"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aaf9e8ca50d99c27775ce186613320624"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#aaf9e8ca50d99c27775ce186613320624">setAge</a> (int a_age)</td></tr>
<tr class="separator:aaf9e8ca50d99c27775ce186613320624"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a89c0c0959ca4af0152520a2d6c512edd"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a89c0c0959ca4af0152520a2d6c512edd">increaseOperatedOn</a> ()</td></tr>
<tr class="separator:a89c0c0959ca4af0152520a2d6c512edd"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a93a541bdb76c316543a201e8718dd29c"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a93a541bdb76c316543a201e8718dd29c">resetOperatedOn</a> ()</td></tr>
<tr class="separator:a93a541bdb76c316543a201e8718dd29c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a86349c8e5dbe93c54580a283f70ea2d0"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a86349c8e5dbe93c54580a283f70ea2d0">operatedOn</a> ()</td></tr>
<tr class="separator:a86349c8e5dbe93c54580a283f70ea2d0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1f9e4971a34546a0d77fa24061e753a0"><td class="memItemLeft" align="right" valign="top">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a1f9e4971a34546a0d77fa24061e753a0">getGenes</a> ()</td></tr>
<tr class="separator:a1f9e4971a34546a0d77fa24061e753a0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac4de43d022e2cc7a7a8ac78ff47ad0a0"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#ac4de43d022e2cc7a7a8ac78ff47ad0a0">setGenes</a> (<a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[] a_genes)  throws InvalidConfigurationException </td></tr>
<tr class="separator:ac4de43d022e2cc7a7a8ac78ff47ad0a0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aef0e40ec9b5956389365f5c3144ab207"><td class="memItemLeft" align="right" valign="top">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207">getGene</a> (int a_desiredLocus)</td></tr>
<tr class="separator:aef0e40ec9b5956389365f5c3144ab207"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8177b71d092d3ce87909d827ec91492a"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a8177b71d092d3ce87909d827ec91492a">setGene</a> (int a_index, <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> a_gene)</td></tr>
<tr class="separator:a8177b71d092d3ce87909d827ec91492a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a78c8ae0471973855e19a9b79496f2bec"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a78c8ae0471973855e19a9b79496f2bec">size</a> ()</td></tr>
<tr class="separator:a78c8ae0471973855e19a9b79496f2bec"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab4a4ad55945458548488f87940aa852e"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#ab4a4ad55945458548488f87940aa852e">getPersistentRepresentation</a> ()</td></tr>
<tr class="separator:ab4a4ad55945458548488f87940aa852e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a124394f7e5c4651e59dd1475e0d2c3b0"><td class="memItemLeft" align="right" valign="top">StringBuffer&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a124394f7e5c4651e59dd1475e0d2c3b0">getGenesPersistentRepresentation</a> ()</td></tr>
<tr class="separator:a124394f7e5c4651e59dd1475e0d2c3b0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa35172e2b9535ace3b8fa31c1b2ef2fe"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#aa35172e2b9535ace3b8fa31c1b2ef2fe">getBusinessKey</a> ()</td></tr>
<tr class="separator:aa35172e2b9535ace3b8fa31c1b2ef2fe"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2bef76845dd6e92c09037dae72e7ca15"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a2bef76845dd6e92c09037dae72e7ca15">getGenesPersistentRepresentation</a> (StringBuffer a_buffer)</td></tr>
<tr class="separator:a2bef76845dd6e92c09037dae72e7ca15"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a30276dc8afb0bd632c288bf7d2e18b55"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a30276dc8afb0bd632c288bf7d2e18b55">setValueFromPersistentRepresentation</a> (String a_representation)  throws UnsupportedRepresentationException </td></tr>
<tr class="separator:a30276dc8afb0bd632c288bf7d2e18b55"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td colspan="2" onclick="javascript:toggleInherit('pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome')"><img src="closed.png" alt="-"/>&#160;Public Member Functions inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">org.jgap.IChromosome</a></td></tr>
<tr class="memitem:a56f82d43b102fc5da8e0a071e32dc13d inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a56f82d43b102fc5da8e0a071e32dc13d">setFitnessValue</a> (double a_newFitnessValue)</td></tr>
<tr class="separator:a56f82d43b102fc5da8e0a071e32dc13d inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac2acf33157b233e8c5553fc62861fa23 inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#ac2acf33157b233e8c5553fc62861fa23">setFitnessValueDirectly</a> (double a_newFitnessValue)</td></tr>
<tr class="separator:ac2acf33157b233e8c5553fc62861fa23 inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9b68991c832a2054176d522e49a19ebd inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a9b68991c832a2054176d522e49a19ebd">getFitnessValue</a> ()</td></tr>
<tr class="separator:a9b68991c832a2054176d522e49a19ebd inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a743303c75d2a001f88af579051862964 inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a743303c75d2a001f88af579051862964">getFitnessValueDirectly</a> ()</td></tr>
<tr class="separator:a743303c75d2a001f88af579051862964 inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afeeed3697d5f22154cab618432602ca6 inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#afeeed3697d5f22154cab618432602ca6">setIsSelectedForNextGeneration</a> (boolean a_isSelected)</td></tr>
<tr class="separator:afeeed3697d5f22154cab618432602ca6 inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af4aa1487bfa13ba0f038b628f44554be inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#af4aa1487bfa13ba0f038b628f44554be">isSelectedForNextGeneration</a> ()</td></tr>
<tr class="separator:af4aa1487bfa13ba0f038b628f44554be inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a05c61ae9fb2caf56455eb66405a164c4 inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a05c61ae9fb2caf56455eb66405a164c4">setConstraintChecker</a> (<a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a> a_constraintChecker)  throws InvalidConfigurationException</td></tr>
<tr class="separator:a05c61ae9fb2caf56455eb66405a164c4 inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a99da03ae9bca8f2ca884186aea82c771 inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a99da03ae9bca8f2ca884186aea82c771">setApplicationData</a> (Object a_newData)</td></tr>
<tr class="separator:a99da03ae9bca8f2ca884186aea82c771 inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1734f351c78c6d36bec48fdc8a334902 inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a1734f351c78c6d36bec48fdc8a334902">getApplicationData</a> ()</td></tr>
<tr class="separator:a1734f351c78c6d36bec48fdc8a334902 inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a184e942f9b44185236492ee7261b2b14 inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a184e942f9b44185236492ee7261b2b14">cleanup</a> ()</td></tr>
<tr class="separator:a184e942f9b44185236492ee7261b2b14 inherit pub_methods_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_methods_interfaceorg_1_1jgap_1_1_i_handler"><td colspan="2" onclick="javascript:toggleInherit('pub_methods_interfaceorg_1_1jgap_1_1_i_handler')"><img src="closed.png" alt="-"/>&#160;Public Member Functions inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_handler.html">org.jgap.IHandler</a></td></tr>
<tr class="memitem:a026d7a41944049a4f812ee0d1151d40e inherit pub_methods_interfaceorg_1_1jgap_1_1_i_handler"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_handler.html#a026d7a41944049a4f812ee0d1151d40e">isHandlerFor</a> (Object a_obj, Class a_class)</td></tr>
<tr class="separator:a026d7a41944049a4f812ee0d1151d40e inherit pub_methods_interfaceorg_1_1jgap_1_1_i_handler"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a60199862000df04a4b2491d6d08f2f2c inherit pub_methods_interfaceorg_1_1jgap_1_1_i_handler"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_handler.html#a60199862000df04a4b2491d6d08f2f2c">perform</a> (Object a_obj, Class a_class, Object a_params)  throws Exception</td></tr>
<tr class="separator:a60199862000df04a4b2491d6d08f2f2c inherit pub_methods_interfaceorg_1_1jgap_1_1_i_handler"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-static-attribs"></a>
Static Public Attributes</h2></td></tr>
<tr class="memitem:a5cb3ac5d20e1fccb7eacdd0bb38bd650"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a5cb3ac5d20e1fccb7eacdd0bb38bd650">GENE_DELIMITER</a> = &quot;#&quot;</td></tr>
<tr class="separator:a5cb3ac5d20e1fccb7eacdd0bb38bd650"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a52428dfc927129c8960a9d9ca3864346"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a52428dfc927129c8960a9d9ca3864346">GENE_DELIMITER_HEADING</a> = &quot;&lt;&quot;</td></tr>
<tr class="separator:a52428dfc927129c8960a9d9ca3864346"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a05f3c0c0caa0e1676345b2303c842be0"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a05f3c0c0caa0e1676345b2303c842be0">GENE_DELIMITER_CLOSING</a> = &quot;&gt;&quot;</td></tr>
<tr class="separator:a05f3c0c0caa0e1676345b2303c842be0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4efa73cb7ff2c5dcd0e16f1d7287dc78"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a4efa73cb7ff2c5dcd0e16f1d7287dc78">CHROM_DELIMITER</a> = &quot;#&quot;</td></tr>
<tr class="separator:a4efa73cb7ff2c5dcd0e16f1d7287dc78"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">org.jgap.IChromosome</a></td></tr>
<tr class="memitem:a4e7381b89963f2ee6f69839b38ea8dab inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a4e7381b89963f2ee6f69839b38ea8dab">CVS_REVISION</a> = &quot;$Revision: 1.17 $&quot;</td></tr>
<tr class="separator:a4e7381b89963f2ee6f69839b38ea8dab inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a34c8c0933688a10837692d1c0024f663 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a34c8c0933688a10837692d1c0024f663">S_FITNESS_VALUE</a> = &quot;Fitness value&quot;</td></tr>
<tr class="separator:a34c8c0933688a10837692d1c0024f663 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae93666d5923e449f45656ad5fed52f84 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#ae93666d5923e449f45656ad5fed52f84">S_ALLELES</a> = &quot;Alleles&quot;</td></tr>
<tr class="separator:ae93666d5923e449f45656ad5fed52f84 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a03218ba49e2c53108005cd6290252284 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a03218ba49e2c53108005cd6290252284">S_APPLICATION_DATA</a> = &quot;Application data&quot;</td></tr>
<tr class="separator:a03218ba49e2c53108005cd6290252284 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a50371491243a21f494112bf66d2e1a26 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a50371491243a21f494112bf66d2e1a26">S_SIZE</a> = &quot;Size&quot;</td></tr>
<tr class="separator:a50371491243a21f494112bf66d2e1a26 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html">org.jgap.util.ICloneable</a></td></tr>
<tr class="memitem:a4eaaa85432026f5bc54c3e520b22caa7 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html#a4eaaa85432026f5bc54c3e520b22caa7">CVS_REVISION</a> = &quot;$Revision: 1.3 $&quot;</td></tr>
<tr class="separator:a4eaaa85432026f5bc54c3e520b22caa7 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html">org.jgap.IUniqueKey</a></td></tr>
<tr class="memitem:a5b6472b1e40001ee262da59ee449a77b inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html#a5b6472b1e40001ee262da59ee449a77b">CVS_REVISION</a> = &quot;$Revision: 1.1 $&quot;</td></tr>
<tr class="separator:a5b6472b1e40001ee262da59ee449a77b inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_initializer"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_initializer')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_initializer.html">org.jgap.IInitializer</a></td></tr>
<tr class="memitem:af84c0edbc3fdf99bda88da8d4ed1d2ea inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_initializer"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_initializer.html#af84c0edbc3fdf99bda88da8d4ed1d2ea">CVS_REVISION</a> = &quot;$Revision: 1.2 $&quot;</td></tr>
<tr class="separator:af84c0edbc3fdf99bda88da8d4ed1d2ea inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_initializer"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_handler"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_handler')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_handler.html">org.jgap.IHandler</a></td></tr>
<tr class="memitem:ac87eeb82b5fb4d6e5ba1f8850db52118 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_handler"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_handler.html#ac87eeb82b5fb4d6e5ba1f8850db52118">CVS_REVISION</a> = &quot;$Revision: 1.3 $&quot;</td></tr>
<tr class="separator:ac87eeb82b5fb4d6e5ba1f8850db52118 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_handler"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_persistent_representation"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_persistent_representation')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html">org.jgap.IPersistentRepresentation</a></td></tr>
<tr class="memitem:ae48c5895f4af079337ffebe487b176f5 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_persistent_representation"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html#ae48c5895f4af079337ffebe487b176f5">CVS_REVISION</a> = &quot;$Revision: 1.2 $&quot;</td></tr>
<tr class="separator:ae48c5895f4af079337ffebe487b176f5 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_persistent_representation"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_business_key.html">org.jgap.IBusinessKey</a></td></tr>
<tr class="memitem:a6d43b687da0842739b8021f466ca451a inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_business_key.html#a6d43b687da0842739b8021f466ca451a">CVS_REVISION</a> = &quot;$Revision: 1.3 $&quot;</td></tr>
<tr class="separator:a6d43b687da0842739b8021f466ca451a inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pro-methods"></a>
Protected Member Functions</h2></td></tr>
<tr class="memitem:a2c339c9c70184d1a911878e1be39dc82"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a2c339c9c70184d1a911878e1be39dc82">encode</a> (String a_string)</td></tr>
<tr class="separator:a2c339c9c70184d1a911878e1be39dc82"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0ccfa10730437df1a23322886a6949e7"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a0ccfa10730437df1a23322886a6949e7">decode</a> (String a_string)  throws UnsupportedEncodingException </td></tr>
<tr class="separator:a0ccfa10730437df1a23322886a6949e7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a61d504512312fdc761af87fe0f3ca36d"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a61d504512312fdc761af87fe0f3ca36d">createGene</a> (String a_geneClassName, String a_persistentRepresentation)  throws Exception </td></tr>
<tr class="separator:a61d504512312fdc761af87fe0f3ca36d"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pro-static-methods"></a>
Static Protected Member Functions</h2></td></tr>
<tr class="memitem:a9237b2c8d40b9d466d5f68792ba04036"><td class="memItemLeft" align="right" valign="top">static final List&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a9237b2c8d40b9d466d5f68792ba04036">split</a> (String a_string)  throws UnsupportedRepresentationException </td></tr>
<tr class="separator:a9237b2c8d40b9d466d5f68792ba04036"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-attribs"></a>
Private Attributes</h2></td></tr>
<tr class="memitem:a2303f591b284ec1ad116c135806af7c3"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a2303f591b284ec1ad116c135806af7c3">m_configuration</a></td></tr>
<tr class="separator:a2303f591b284ec1ad116c135806af7c3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac1d06474d7d21f73fd602cee8f0d50b4"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#ac1d06474d7d21f73fd602cee8f0d50b4">m_genes</a></td></tr>
<tr class="separator:ac1d06474d7d21f73fd602cee8f0d50b4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8208f36f8feb650c1e72edf2244e44f3"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a8208f36f8feb650c1e72edf2244e44f3">m_age</a></td></tr>
<tr class="separator:a8208f36f8feb650c1e72edf2244e44f3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af0a5c64437918ed0e28233f39fa60a30"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#af0a5c64437918ed0e28233f39fa60a30">m_operatedOn</a></td></tr>
<tr class="separator:af0a5c64437918ed0e28233f39fa60a30"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a04dcfa7a4ba552f754ab440b21ebe558"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a04dcfa7a4ba552f754ab440b21ebe558">m_uniqueID</a></td></tr>
<tr class="separator:a04dcfa7a4ba552f754ab440b21ebe558"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae74c4eae0fc849d8210ec100bbb01042"><td class="memItemLeft" align="right" valign="top">Map&lt; Integer, String &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#ae74c4eae0fc849d8210ec100bbb01042">m_uniqueIDTemplates</a></td></tr>
<tr class="separator:ae74c4eae0fc849d8210ec100bbb01042"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-static-attribs"></a>
Static Private Attributes</h2></td></tr>
<tr class="memitem:a1d84ed799a8428ca04fdd53a59e68df0"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a1d84ed799a8428ca04fdd53a59e68df0">CVS_REVISION</a> = &quot;$Revision: 1.15 $&quot;</td></tr>
<tr class="separator:a1d84ed799a8428ca04fdd53a59e68df0"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Base class for any implementation of interface <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00025">25</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="a94e1af5ba804061833ed3883d42f9441"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.BaseChromosome.BaseChromosome </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>The only constructor in this class. Sets the immutable configuration.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the configuration to set </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td>if configuration is null</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00088">88</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_configuration_8java_source.html#l01500">org.jgap.Configuration.getJGAPFactory()</a>, <a class="el" href="_configuration_8java_source.html#l01911">org.jgap.Configuration.isUniqueKeysActive()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00056">org.jgap.BaseChromosome.m_configuration</a>, <a class="el" href="_base_chromosome_8java_source.html#l00071">org.jgap.BaseChromosome.m_uniqueID</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00077">org.jgap.BaseChromosome.m_uniqueIDTemplates</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_chromosome_a94e1af5ba804061833ed3883d42f9441_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_base_chromosome_a94e1af5ba804061833ed3883d42f9441_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_base_chromosome_a94e1af5ba804061833ed3883d42f9441_cgraph" id="classorg_1_1jgap_1_1_base_chromosome_a94e1af5ba804061833ed3883d42f9441_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_configuration.html#ad71c68a6972d6fd993197773439154a0" title="org.jgap.Configuration.get\lJGAPFactory" alt="" coords="237,5,408,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_configuration.html#a1cbe9f50b9017fe15ac4e27487cc549f" title="org.jgap.Configuration.is\lUniqueKeysActive" alt="" coords="241,70,404,111"/></map>
</div>
</p>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="a97071ef6dba552c1c1f2b2780d27eec3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">abstract Object org.jgap.BaseChromosome.clone </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">pure virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Creates and returns a copy of this object.</p>
<dl class="section return"><dt>Returns</dt><dd>a clone of this instance </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">IllegalStateException</td><td>instead of CloneNotSupportedException</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html#aa7e7d62077e6428ad7904932b1b4f7d5">org.jgap.util.ICloneable</a>.</p>

<p>Implemented in <a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a9f590d55a34d8b99c2a9a1b4aedd568b">org.jgap.Chromosome</a>, and <a class="el" href="classorg_1_1jgap_1_1_chromosome_for_testing.html#acfcef639c3443fde7d324164050e3e19">org.jgap.ChromosomeForTesting</a>.</p>

</div>
</div>
<a class="anchor" id="a61d504512312fdc761af87fe0f3ca36d"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> org.jgap.BaseChromosome.createGene </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_geneClassName</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_persistentRepresentation</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws Exception</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Creates a new <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> instance.</p>
<p>Taken from CompositeGene.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_geneClassName</td><td>name of the gene class </td></tr>
    <tr><td class="paramname">a_persistentRepresentation</td><td>persistent representation of the gene to create (could be obtained via getPersistentRepresentation)</td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>newly created gene </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Exception</td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00457">457</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00150">org.jgap.BaseChromosome.getConfiguration()</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00400">org.jgap.BaseChromosome.setValueFromPersistentRepresentation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_chromosome_a61d504512312fdc761af87fe0f3ca36d_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_base_chromosome_a61d504512312fdc761af87fe0f3ca36d_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_base_chromosome_a61d504512312fdc761af87fe0f3ca36d_cgraph" id="classorg_1_1jgap_1_1_base_chromosome_a61d504512312fdc761af87fe0f3ca36d_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_chromosome.html#a104b33f29941ba6ba405889964fc2bf5" title="org.jgap.BaseChromosome.get\lConfiguration" alt="" coords="304,5,507,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a0ccfa10730437df1a23322886a6949e7"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.BaseChromosome.decode </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_string</em></td><td>)</td>
          <td> throws UnsupportedEncodingException</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00385">385</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00400">org.jgap.BaseChromosome.setValueFromPersistentRepresentation()</a>.</p>

</div>
</div>
<a class="anchor" id="a2c339c9c70184d1a911878e1be39dc82"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.BaseChromosome.encode </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_string</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00381">381</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00368">org.jgap.BaseChromosome.getGenesPersistentRepresentation()</a>.</p>

</div>
</div>
<a class="anchor" id="af987f6a74486145a139985297c18ae80"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.BaseChromosome.getAge </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>0: <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> newly created in this generation. This means it does not need being crossed over with another newly created one</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a6186b5e6184d55dbc03adee6bacc8e9a">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00193">193</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00063">org.jgap.BaseChromosome.m_age</a>.</p>

</div>
</div>
<a class="anchor" id="aa35172e2b9535ace3b8fa31c1b2ef2fe"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.BaseChromosome.getBusinessKey </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>business key of the chromosome</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_business_key.html#ab1b12087d8726e9020441a1fed6be0f8">org.jgap.IBusinessKey</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00356">356</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00344">org.jgap.BaseChromosome.getGenesPersistentRepresentation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_chromosome_aa35172e2b9535ace3b8fa31c1b2ef2fe_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_base_chromosome_aa35172e2b9535ace3b8fa31c1b2ef2fe_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_base_chromosome_aa35172e2b9535ace3b8fa31c1b2ef2fe_cgraph" id="classorg_1_1jgap_1_1_base_chromosome_aa35172e2b9535ace3b8fa31c1b2ef2fe_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_chromosome.html#a124394f7e5c4651e59dd1475e0d2c3b0" title="org.jgap.BaseChromosome.get\lGenesPersistentRepresentation" alt="" coords="257,5,463,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a104b33f29941ba6ba405889964fc2bf5"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> org.jgap.BaseChromosome.getConfiguration </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>the configuration used</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a4426723f5d26695fd80a0b28b3acb9eb">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00150">150</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00056">org.jgap.BaseChromosome.m_configuration</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00560">org.jgap.Chromosome.calcFitnessValue()</a>, <a class="el" href="_chromosome_8java_source.html#l00928">org.jgap.Chromosome.cleanup()</a>, <a class="el" href="_chromosome_for_testing_8java_source.html#l00114">org.jgap.ChromosomeForTesting.clone()</a>, <a class="el" href="_chromosome_8java_source.html#l00334">org.jgap.Chromosome.clone()</a>, <a class="el" href="_chromosome_8java_source.html#l00478">org.jgap.Chromosome.cloneObject()</a>, <a class="el" href="_chromosome_8java_source.html#l00816">org.jgap.Chromosome.compareTo()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00457">org.jgap.BaseChromosome.createGene()</a>, <a class="el" href="_chromosome_8java_source.html#l01105">org.jgap.Chromosome.perform()</a>, and <a class="el" href="_chromosome_for_testing_8java_source.html#l00058">org.jgap.ChromosomeForTesting.randomInitialChromosome2()</a>.</p>

</div>
</div>
<a class="anchor" id="aef0e40ec9b5956389365f5c3144ab207"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> org.jgap.BaseChromosome.getGene </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_desiredLocus</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Returns the <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> at the given index (locus) within the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. The first gene is at index zero and the last gene is at the index equal to the size of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> - 1.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_desiredLocus</td><td>index of the gene value to be returned </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> at the given index</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#ac84a3ec514b365e25f023d9e78bfa382">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00283">283</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00061">org.jgap.BaseChromosome.m_genes</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00928">org.jgap.Chromosome.cleanup()</a>, <a class="el" href="_chromosome_8java_source.html#l00334">org.jgap.Chromosome.clone()</a>, <a class="el" href="_chromosome_8java_source.html#l00816">org.jgap.Chromosome.compareTo()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00368">org.jgap.BaseChromosome.getGenesPersistentRepresentation()</a>, <a class="el" href="_chromosome_8java_source.html#l00780">org.jgap.Chromosome.hashCode()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00355">org.jgap.ChromosomeTest.testConstruct_10()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00300">org.jgap.ChromosomeTest.testConstruct_7_3()</a>, <a class="el" href="_mutation_operator_test_8java_source.html#l00250">org.jgap.impl.MutationOperatorTest.testOperate_3_1()</a>, <a class="el" href="_two_way_mutation_operator_test_8java_source.html#l00255">org.jgap.impl.TwoWayMutationOperatorTest.testOperate_3_1()</a>, <a class="el" href="_chromosome_8java_source.html#l00614">org.jgap.Chromosome.toString()</a>, and <a class="el" href="_chromosome_8java_source.html#l01073">org.jgap.Chromosome.verify()</a>.</p>

</div>
</div>
<a class="anchor" id="a1f9e4971a34546a0d77fa24061e753a0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> [] org.jgap.BaseChromosome.getGenes </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves the set of genes that make up this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. This method exists primarily for the benefit of GeneticOperators that require the ability to manipulate Chromosomes at a low level.</p>
<dl class="section return"><dt>Returns</dt><dd>an array of the Genes contained within this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a></dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a995b4203af3bd22b8fd2028749736a03">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00252">252</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00061">org.jgap.BaseChromosome.m_genes</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00780">org.jgap.Chromosome.hashCode()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00476">org.jgap.ChromosomeTest.testClone_1()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00300">org.jgap.ChromosomeTest.testConstruct_7_3()</a>, and <a class="el" href="_chromosome_8java_source.html#l01073">org.jgap.Chromosome.verify()</a>.</p>

</div>
</div>
<a class="anchor" id="a124394f7e5c4651e59dd1475e0d2c3b0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">StringBuffer org.jgap.BaseChromosome.getGenesPersistentRepresentation </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>the persistent representation of the chromosome by considering its genes.</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00344">344</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00356">org.jgap.BaseChromosome.getBusinessKey()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00324">org.jgap.BaseChromosome.getPersistentRepresentation()</a>.</p>

</div>
</div>
<a class="anchor" id="a2bef76845dd6e92c09037dae72e7ca15"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseChromosome.getGenesPersistentRepresentation </td>
          <td>(</td>
          <td class="paramtype">StringBuffer&#160;</td>
          <td class="paramname"><em>a_buffer</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves the persistent representation of the chromosome by considering its genes.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_buffer</td><td>the variable to store the persistent representation in</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00368">368</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00381">org.jgap.BaseChromosome.encode()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00034">org.jgap.BaseChromosome.GENE_DELIMITER</a>, <a class="el" href="_base_chromosome_8java_source.html#l00046">org.jgap.BaseChromosome.GENE_DELIMITER_CLOSING</a>, <a class="el" href="_base_chromosome_8java_source.html#l00040">org.jgap.BaseChromosome.GENE_DELIMITER_HEADING</a>, <a class="el" href="_base_chromosome_8java_source.html#l00283">org.jgap.BaseChromosome.getGene()</a>, <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#a044aa8f0a04ee06bb04bbd35ad4c7f2f">org.jgap.Gene.getPersistentRepresentation()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00302">org.jgap.BaseChromosome.size()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_chromosome_a2bef76845dd6e92c09037dae72e7ca15_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_base_chromosome_a2bef76845dd6e92c09037dae72e7ca15_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_base_chromosome_a2bef76845dd6e92c09037dae72e7ca15_cgraph" id="classorg_1_1jgap_1_1_base_chromosome_a2bef76845dd6e92c09037dae72e7ca15_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_chromosome.html#a2c339c9c70184d1a911878e1be39dc82" title="org.jgap.BaseChromosome.encode" alt="" coords="265,5,492,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207" title="org.jgap.BaseChromosome.getGene" alt="" coords="262,56,495,83"/><area shape="rect" id="node7" href="interfaceorg_1_1jgap_1_1_gene.html#a044aa8f0a04ee06bb04bbd35ad4c7f2f" title="org.jgap.Gene.getPersistent\lRepresentation" alt="" coords="287,107,471,149"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_base_chromosome.html#a78c8ae0471973855e19a9b79496f2bec" title="org.jgap.BaseChromosome.size" alt="" coords="275,173,483,200"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ab4a4ad55945458548488f87940aa852e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.BaseChromosome.getPersistentRepresentation </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Returns a persistent representation of this chromosome, see interface <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> for description. Similar to CompositeGene's routine. But does not include all information of the chromosome (yet).</p>
<dl class="section return"><dt>Returns</dt><dd>string representation of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>'s relevant parts of its current state </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">UnsupportedOperationException</td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html#a07bf7dbaa8de08e268c9357a46e79dac">org.jgap.IPersistentRepresentation</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00324">324</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00051">org.jgap.BaseChromosome.CHROM_DELIMITER</a>, <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a743303c75d2a001f88af579051862964">org.jgap.IChromosome.getFitnessValueDirectly()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00344">org.jgap.BaseChromosome.getGenesPersistentRepresentation()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00302">org.jgap.BaseChromosome.size()</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_test_8java_source.html#l00336">org.jgap.ChromosomeTest.testConstruct_9()</a>, <a class="el" href="_chromosome_test_8java_source.html#l01613">org.jgap.ChromosomeTest.testPersistentRepresentation_0()</a>, <a class="el" href="_chromosome_test_8java_source.html#l01635">org.jgap.ChromosomeTest.testPersistentRepresentation_1()</a>, and <a class="el" href="_chromosome_test_8java_source.html#l01734">org.jgap.ChromosomeTest.testPersistentRepresentation_7()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_chromosome_ab4a4ad55945458548488f87940aa852e_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_base_chromosome_ab4a4ad55945458548488f87940aa852e_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_base_chromosome_ab4a4ad55945458548488f87940aa852e_cgraph" id="classorg_1_1jgap_1_1_base_chromosome_ab4a4ad55945458548488f87940aa852e_cgraph">
<area shape="rect" id="node3" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a743303c75d2a001f88af579051862964" title="org.jgap.IChromosome.get\lFitnessValueDirectly" alt="" coords="272,5,448,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_chromosome.html#a124394f7e5c4651e59dd1475e0d2c3b0" title="org.jgap.BaseChromosome.get\lGenesPersistentRepresentation" alt="" coords="257,70,463,111"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_chromosome.html#a78c8ae0471973855e19a9b79496f2bec" title="org.jgap.BaseChromosome.size" alt="" coords="256,136,464,163"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a8cf998c46fddd9ee2f8bfcd2a7b6bffa"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.BaseChromosome.getUniqueID </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>unique ID of the chromosome, which allows to distinct this instance from others, in the best case worldwide</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.5 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html#a1276412b558e169199771997a190f876">org.jgap.IUniqueKey</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00112">112</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00071">org.jgap.BaseChromosome.m_uniqueID</a>.</p>

</div>
</div>
<a class="anchor" id="a2d23cfd243723be4d5eb775532e36a6e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.BaseChromosome.getUniqueIDTemplate </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_index</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_index</td><td>the index of the template to retrieve the key for </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>String</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.5 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html#af26fbe03487632a16c84b62325489652">org.jgap.IUniqueKey</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00140">140</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

</div>
</div>
<a class="anchor" id="a6dcae8b12cad267f6f91431281869e02"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseChromosome.increaseAge </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Increases the number of evolutionary rounds of chromosome in which it has not been changed by one.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a651016ea5a204fc14c7c146e32fd76cf">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00172">172</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00063">org.jgap.BaseChromosome.m_age</a>.</p>

</div>
</div>
<a class="anchor" id="a89c0c0959ca4af0152520a2d6c512edd"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseChromosome.increaseOperatedOn </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Increase information of number of genetic operations performed on chromosome in current evolution round.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a8fda999d5a66640dd577bcff960b368b">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00215">215</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00065">org.jgap.BaseChromosome.m_operatedOn</a>.</p>

</div>
</div>
<a class="anchor" id="a86349c8e5dbe93c54580a283f70ea2d0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.BaseChromosome.operatedOn </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>number of genetic operations performed on chromosome in current evolution round</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a47f09dd85e4f9f0e3908c85b63394a7d">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00238">238</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00065">org.jgap.BaseChromosome.m_operatedOn</a>.</p>

</div>
</div>
<a class="anchor" id="a7d4483ef679d0e1eb335256b72ab0731"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseChromosome.resetAge </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Reset age of chromosome because it has been changed.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#aefc4884cc5bbbc511feb8d9ec827a31f">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00182">182</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00063">org.jgap.BaseChromosome.m_age</a>.</p>

</div>
</div>
<a class="anchor" id="a93a541bdb76c316543a201e8718dd29c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseChromosome.resetOperatedOn </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Resets the information of how many genetic operators have been performed on the chromosome in the current round of evolution.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a4fb2f78d39b20b7f5a9ff8d37402f7bd">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00227">227</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00065">org.jgap.BaseChromosome.m_operatedOn</a>.</p>

</div>
</div>
<a class="anchor" id="aaf9e8ca50d99c27775ce186613320624"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseChromosome.setAge </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_age</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_age</td><td>set the age of the chromosome, see BestChromosomesSelector for a use-case</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.3.3 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a9146b114972d86aac3cfb32b249b166f">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00204">204</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00063">org.jgap.BaseChromosome.m_age</a>.</p>

</div>
</div>
<a class="anchor" id="a8177b71d092d3ce87909d827ec91492a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseChromosome.setGene </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_index</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>&#160;</td>
          <td class="paramname"><em>a_gene</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00287">287</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00061">org.jgap.BaseChromosome.m_genes</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00235">org.jgap.Chromosome.initFromGene()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00400">org.jgap.BaseChromosome.setValueFromPersistentRepresentation()</a>.</p>

</div>
</div>
<a class="anchor" id="ac4de43d022e2cc7a7a8ac78ff47ad0a0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseChromosome.setGenes </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[]&#160;</td>
          <td class="paramname"><em>a_genes</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets the genes for the chromosome.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_genes</td><td>the genes to set for the chromosome</td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td>in case constraint checker is provided</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 (previously in class <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>) </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a0ff42f1935dab73296301b29ad0a1716">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00267">267</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00061">org.jgap.BaseChromosome.m_genes</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00400">org.jgap.BaseChromosome.setValueFromPersistentRepresentation()</a>.</p>

</div>
</div>
<a class="anchor" id="a06156bb0f2c8512ca6a7f7f0fee305b9"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseChromosome.setUniqueIDTemplate </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_templateID</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_index</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>A template is a chromosome that is the logical predecessor of the current chromosome. A template can occur in mutation or crossing over. In the latter case can be at least two template chromosomes. This is why in this setter method the parameter a_index exists.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_templateID</td><td>the unique ID of the template </td></tr>
    <tr><td class="paramname">a_index</td><td>the index of the template, e.g. in crossing over for the second candidate chromosome this is 2</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.5 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html#aeac223774f93ba6640bc07cc97a9ef4c">org.jgap.IUniqueKey</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00129">129</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

</div>
</div>
<a class="anchor" id="a30276dc8afb0bd632c288bf7d2e18b55"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseChromosome.setValueFromPersistentRepresentation </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_representation</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_unsupported_representation_exception.html">UnsupportedRepresentationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Counterpart of getPersistentRepresentation.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_representation</td><td>the string representation retrieved from a prior call to the <a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#ab4a4ad55945458548488f87940aa852e">getPersistentRepresentation()</a> method</td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_unsupported_representation_exception.html">UnsupportedRepresentationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000039">Todo:</a></b></dt><dd>we can do this faster! </dd></dl>
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000040">Todo:</a></b></dt><dd>we can do this faster! </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html#a94a345f1919c4840dd0b9eecf7afc6a3">org.jgap.IPersistentRepresentation</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00400">400</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00457">org.jgap.BaseChromosome.createGene()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00385">org.jgap.BaseChromosome.decode()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00034">org.jgap.BaseChromosome.GENE_DELIMITER</a>, <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a56f82d43b102fc5da8e0a071e32dc13d">org.jgap.IChromosome.setFitnessValue()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00287">org.jgap.BaseChromosome.setGene()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00267">org.jgap.BaseChromosome.setGenes()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00479">org.jgap.BaseChromosome.split()</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00173">org.jgap.Chromosome.Chromosome()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_chromosome_a30276dc8afb0bd632c288bf7d2e18b55_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_base_chromosome_a30276dc8afb0bd632c288bf7d2e18b55_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_base_chromosome_a30276dc8afb0bd632c288bf7d2e18b55_cgraph" id="classorg_1_1jgap_1_1_base_chromosome_a30276dc8afb0bd632c288bf7d2e18b55_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_chromosome.html#a61d504512312fdc761af87fe0f3ca36d" title="org.jgap.BaseChromosome.createGene" alt="" coords="285,12,536,39"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_chromosome.html#a0ccfa10730437df1a23322886a6949e7" title="org.jgap.BaseChromosome.decode" alt="" coords="297,63,524,89"/><area shape="rect" id="node9" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a56f82d43b102fc5da8e0a071e32dc13d" title="org.jgap.IChromosome.set\lFitnessValue" alt="" coords="323,114,499,155"/><area shape="rect" id="node11" href="classorg_1_1jgap_1_1_base_chromosome.html#a8177b71d092d3ce87909d827ec91492a" title="org.jgap.BaseChromosome.setGene" alt="" coords="294,180,527,207"/><area shape="rect" id="node13" href="classorg_1_1jgap_1_1_base_chromosome.html#ac4de43d022e2cc7a7a8ac78ff47ad0a0" title="org.jgap.BaseChromosome.set\lGenes" alt="" coords="309,231,512,273"/><area shape="rect" id="node15" href="classorg_1_1jgap_1_1_base_chromosome.html#a9237b2c8d40b9d466d5f68792ba04036" title="org.jgap.BaseChromosome.split" alt="" coords="307,297,515,324"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_chromosome.html#a104b33f29941ba6ba405889964fc2bf5" title="org.jgap.BaseChromosome.get\lConfiguration" alt="" coords="584,5,787,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a78c8ae0471973855e19a9b79496f2bec"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.BaseChromosome.size </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Returns the size of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> (the number of genes it contains). A <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>'s size is constant and will not change, until setGenes(...) is used.</p>
<dl class="section return"><dt>Returns</dt><dd>number of genes contained within this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> instance</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a9b460ad665ea1ab4e789061233fc6fa3">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00302">302</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00061">org.jgap.BaseChromosome.m_genes</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00928">org.jgap.Chromosome.cleanup()</a>, <a class="el" href="_chromosome_8java_source.html#l00334">org.jgap.Chromosome.clone()</a>, <a class="el" href="_chromosome_8java_source.html#l00816">org.jgap.Chromosome.compareTo()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00368">org.jgap.BaseChromosome.getGenesPersistentRepresentation()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00324">org.jgap.BaseChromosome.getPersistentRepresentation()</a>, <a class="el" href="_chromosome_8java_source.html#l00780">org.jgap.Chromosome.hashCode()</a>, <a class="el" href="_chromosome_8java_source.html#l00235">org.jgap.Chromosome.initFromGene()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00453">org.jgap.ChromosomeTest.testClone_6()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00355">org.jgap.ChromosomeTest.testConstruct_10()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00184">org.jgap.ChromosomeTest.testConstruct_3()</a>, <a class="el" href="_chromosome_test_8java_source.html#l01656">org.jgap.ChromosomeTest.testPersistentRepresentation_2()</a>, <a class="el" href="_chromosome_test_8java_source.html#l01757">org.jgap.ChromosomeTest.testPersistentRepresentation_5()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00869">org.jgap.ChromosomeTest.testSize_0()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00884">org.jgap.ChromosomeTest.testSize_1()</a>, <a class="el" href="_chromosome_test_8java_source.html#l01301">org.jgap.ChromosomeTest.testToString_0()</a>, <a class="el" href="_chromosome_test_8java_source.html#l01349">org.jgap.ChromosomeTest.testToString_2()</a>, and <a class="el" href="_chromosome_8java_source.html#l00614">org.jgap.Chromosome.toString()</a>.</p>

</div>
</div>
<a class="anchor" id="a9237b2c8d40b9d466d5f68792ba04036"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">static final List org.jgap.BaseChromosome.split </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_string</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_unsupported_representation_exception.html">UnsupportedRepresentationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">static</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Splits the input a_string into individual gene representations.</p>
<p>Taken and adapted from CompositeGene.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_string</td><td>the string to split </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the elements of the returned array are the persistent representation strings of the chromosome's components </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_unsupported_representation_exception.html">UnsupportedRepresentationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00479">479</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00051">org.jgap.BaseChromosome.CHROM_DELIMITER</a>, <a class="el" href="_base_chromosome_8java_source.html#l00046">org.jgap.BaseChromosome.GENE_DELIMITER_CLOSING</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00040">org.jgap.BaseChromosome.GENE_DELIMITER_HEADING</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00400">org.jgap.BaseChromosome.setValueFromPersistentRepresentation()</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Data Documentation</h2>
<a class="anchor" id="a4efa73cb7ff2c5dcd0e16f1d7287dc78"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final String org.jgap.BaseChromosome.CHROM_DELIMITER = &quot;#&quot;</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Separates chromosome-related information. </p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00051">51</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00324">org.jgap.BaseChromosome.getPersistentRepresentation()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00479">org.jgap.BaseChromosome.split()</a>, <a class="el" href="_chromosome_test_8java_source.html#l01688">org.jgap.ChromosomeTest.testPersistentRepresentation_4()</a>, <a class="el" href="_population_test_8java_source.html#l00954">org.jgap.PopulationTest.testPersistentRepresentation_5()</a>, and <a class="el" href="_population_test_8java_source.html#l00981">org.jgap.PopulationTest.testPersistentRepresentation_6()</a>.</p>

</div>
</div>
<a class="anchor" id="a1d84ed799a8428ca04fdd53a59e68df0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final String org.jgap.BaseChromosome.CVS_REVISION = &quot;$Revision: 1.15 $&quot;</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>String containing the CVS revision. Read out via reflection! </p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00028">28</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

</div>
</div>
<a class="anchor" id="a5cb3ac5d20e1fccb7eacdd0bb38bd650"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final String org.jgap.BaseChromosome.GENE_DELIMITER = &quot;#&quot;</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>This field separates gene class name from the gene persistent representation string. '*' does not work properly with URLEncoder! </p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00034">34</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00368">org.jgap.BaseChromosome.getGenesPersistentRepresentation()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00400">org.jgap.BaseChromosome.setValueFromPersistentRepresentation()</a>, <a class="el" href="_chromosome_test_8java_source.html#l01757">org.jgap.ChromosomeTest.testPersistentRepresentation_5()</a>, and <a class="el" href="_chromosome_test_8java_source.html#l01710">org.jgap.ChromosomeTest.testPersistentRepresentation_6()</a>.</p>

</div>
</div>
<a class="anchor" id="a05f3c0c0caa0e1676345b2303c842be0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final String org.jgap.BaseChromosome.GENE_DELIMITER_CLOSING = &quot;&gt;&quot;</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Represents the closing delimiter that is used to separate genes in the persistent representation of <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> instances. </p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00046">46</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00368">org.jgap.BaseChromosome.getGenesPersistentRepresentation()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00479">org.jgap.BaseChromosome.split()</a>.</p>

</div>
</div>
<a class="anchor" id="a52428dfc927129c8960a9d9ca3864346"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final String org.jgap.BaseChromosome.GENE_DELIMITER_HEADING = &quot;&lt;&quot;</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Represents the heading delimiter that is used to separate genes in the persistent representation of <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> instances. </p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00040">40</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00368">org.jgap.BaseChromosome.getGenesPersistentRepresentation()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00479">org.jgap.BaseChromosome.split()</a>.</p>

</div>
</div>
<a class="anchor" id="a8208f36f8feb650c1e72edf2244e44f3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.BaseChromosome.m_age</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00063">63</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00193">org.jgap.BaseChromosome.getAge()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00172">org.jgap.BaseChromosome.increaseAge()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00182">org.jgap.BaseChromosome.resetAge()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00204">org.jgap.BaseChromosome.setAge()</a>.</p>

</div>
</div>
<a class="anchor" id="a2303f591b284ec1ad116c135806af7c3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> org.jgap.BaseChromosome.m_configuration</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>The configuration object to use </p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00056">56</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00088">org.jgap.BaseChromosome.BaseChromosome()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00150">org.jgap.BaseChromosome.getConfiguration()</a>.</p>

</div>
</div>
<a class="anchor" id="ac1d06474d7d21f73fd602cee8f0d50b4"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> [] org.jgap.BaseChromosome.m_genes</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>The array of Genes contained in this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. </p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00061">61</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00283">org.jgap.BaseChromosome.getGene()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00252">org.jgap.BaseChromosome.getGenes()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00287">org.jgap.BaseChromosome.setGene()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00267">org.jgap.BaseChromosome.setGenes()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00302">org.jgap.BaseChromosome.size()</a>.</p>

</div>
</div>
<a class="anchor" id="af0a5c64437918ed0e28233f39fa60a30"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.BaseChromosome.m_operatedOn</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00065">65</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00215">org.jgap.BaseChromosome.increaseOperatedOn()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00238">org.jgap.BaseChromosome.operatedOn()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00227">org.jgap.BaseChromosome.resetOperatedOn()</a>.</p>

</div>
</div>
<a class="anchor" id="a04dcfa7a4ba552f754ab440b21ebe558"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.BaseChromosome.m_uniqueID</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Unique ID of the chromosome that allows to distinct it from other chromosomes. In the best case, this ID is unique worldwide. </p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00071">71</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00088">org.jgap.BaseChromosome.BaseChromosome()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00112">org.jgap.BaseChromosome.getUniqueID()</a>.</p>

</div>
</div>
<a class="anchor" id="ae74c4eae0fc849d8210ec100bbb01042"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Map&lt;Integer,String&gt; org.jgap.BaseChromosome.m_uniqueIDTemplates</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>In case mutation, crossing over etc. happened, this sequence gives evidence about the parent(s) of the current chromosome. </p>

<p>Definition at line <a class="el" href="_base_chromosome_8java_source.html#l00077">77</a> of file <a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_base_chromosome_8java_source.html#l00088">org.jgap.BaseChromosome.BaseChromosome()</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>LDH_Git/src/org/jgap/<a class="el" href="_base_chromosome_8java_source.html">BaseChromosome.java</a></li>
</ul>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Sat Nov 23 2013 17:42:10 for Práctica_5_LDH by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
