<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Práctica_5_LDH: org.jgap.Chromosome Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">Práctica_5_LDH
   &#160;<span id="projectnumber">1.2</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li><a href="inherits.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Classes</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Namespaces</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Files</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Functions</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(6)"><span class="SelectionMark">&#160;</span>Pages</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="namespaceorg.html">org</a></li><li class="navelem"><a class="el" href="namespaceorg_1_1jgap.html">jgap</a></li><li class="navelem"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-static-methods">Static Public Member Functions</a> &#124;
<a href="#pro-methods">Protected Member Functions</a> &#124;
<a href="#pro-attribs">Protected Attributes</a> &#124;
<a href="#pri-attribs">Private Attributes</a> &#124;
<a href="#pri-static-attribs">Static Private Attributes</a> &#124;
<a href="classorg_1_1jgap_1_1_chromosome-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">org.jgap.Chromosome Class Reference</div>  </div>
</div><!--header-->
<div class="contents">
<div class="dynheader">
Inheritance diagram for org.jgap.Chromosome:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome__inherit__graph.png" border="0" usemap="#org_8jgap_8_chromosome_inherit__map" alt="Inheritance graph"/></div>
<map name="org_8jgap_8_chromosome_inherit__map" id="org_8jgap_8_chromosome_inherit__map">
<area shape="rect" id="node25" href="classorg_1_1jgap_1_1_chromosome_for_testing.html" title="org.jgap.ChromosomeForTesting" alt="" coords="1041,157,1253,184"/><area shape="rect" id="node29" href="classorg_1_1jgap_1_1_chromosome_for_testing2.html" title="org.jgap.ChromosomeForTesting2" alt="" coords="1037,208,1256,235"/><area shape="rect" id="node2" href="classorg_1_1jgap_1_1_base_chromosome.html" title="org.jgap.BaseChromosome" alt="" coords="609,208,789,235"/><area shape="rect" id="node4" href="interfaceorg_1_1jgap_1_1_i_chromosome.html" title="org.jgap.IChromosome" alt="" coords="371,107,525,133"/><area shape="rect" id="node8" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html" title="org.jgap.util.ICloneable" alt="" coords="133,56,288,83"/><area shape="rect" id="node14" href="interfaceorg_1_1jgap_1_1_i_unique_key.html" title="org.jgap.IUniqueKey" alt="" coords="141,157,281,184"/><area shape="rect" id="node16" href="interfaceorg_1_1jgap_1_1_i_initializer.html" title="org.jgap.IInitializer" alt="" coords="384,183,512,209"/><area shape="rect" id="node18" href="interfaceorg_1_1jgap_1_1_i_handler.html" title="org.jgap.IHandler" alt="" coords="150,208,271,235"/><area shape="rect" id="node20" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html" title="org.jgap.IPersistentRepresentation" alt="" coords="337,233,559,260"/><area shape="rect" id="node22" href="interfaceorg_1_1jgap_1_1_i_business_key.html" title="org.jgap.IBusinessKey" alt="" coords="371,284,525,311"/><area shape="rect" id="node27" href="classorg_1_1jgap_1_1_genotype_test_1_1_chromosome_for_test2.html" title="org.jgap.GenotypeTest.Chromosome\lForTest2" alt="" coords="1305,150,1541,191"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<div class="dynheader">
Collaboration diagram for org.jgap.Chromosome:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome__coll__graph.png" border="0" usemap="#org_8jgap_8_chromosome_coll__map" alt="Collaboration graph"/></div>
<map name="org_8jgap_8_chromosome_coll__map" id="org_8jgap_8_chromosome_coll__map">
<area shape="rect" id="node2" href="classorg_1_1jgap_1_1_base_chromosome.html" title="org.jgap.BaseChromosome" alt="" coords="1195,440,1375,467"/><area shape="rect" id="node4" href="interfaceorg_1_1jgap_1_1_i_chromosome.html" title="org.jgap.IChromosome" alt="" coords="380,187,535,213"/><area shape="rect" id="node24" href="classorg_1_1jgap_1_1_configuration.html" title="org.jgap.Configuration" alt="" coords="827,440,976,467"/><area shape="rect" id="node31" href="interfaceorg_1_1jgap_1_1_i_breeder.html" title="org.jgap.IBreeder" alt="" coords="397,288,518,315"/><area shape="rect" id="node61" href="interfaceorg_1_1jgap_1_1_gene.html" title="org.jgap.Gene" alt="" coords="406,63,509,89"/><area shape="rect" id="node8" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html" title="org.jgap.util.ICloneable" alt="" coords="136,321,291,348"/><area shape="rect" id="node48" href="classorg_1_1jgap_1_1_bulk_fitness_function.html" title="org.jgap.BulkFitnessFunction" alt="" coords="362,339,553,365"/><area shape="rect" id="node55" href="classorg_1_1jgap_1_1_fitness_function.html" title="org.jgap.FitnessFunction" alt="" coords="375,440,540,467"/><area shape="rect" id="node36" href="interfaceorg_1_1jgap_1_1_i_j_g_a_p_factory.html" title="org.jgap.IJGAPFactory" alt="" coords="380,592,535,619"/><area shape="rect" id="node41" href="interfaceorg_1_1jgap_1_1_fitness_evaluator.html" title="org.jgap.FitnessEvaluator" alt="" coords="372,643,543,669"/><area shape="rect" id="node45" href="interfaceorg_1_1jgap_1_1_random_generator.html" title="org.jgap.RandomGenerator" alt="" coords="368,693,547,720"/><area shape="rect" id="node52" href="classorg_1_1jgap_1_1_configuration_1_1_configuration_configurable.html" title="org.jgap.Configuration.Configuration\lConfigurable" alt="" coords="342,745,573,786"/><area shape="rect" id="node66" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html" title="org.jgap.IGeneConstraint\lChecker" alt="" coords="598,854,765,895"/><area shape="rect" id="node14" href="interfaceorg_1_1jgap_1_1_i_unique_key.html" title="org.jgap.IUniqueKey" alt="" coords="143,5,283,32"/><area shape="rect" id="node16" href="interfaceorg_1_1jgap_1_1_i_initializer.html" title="org.jgap.IInitializer" alt="" coords="837,657,965,684"/><area shape="rect" id="node18" href="interfaceorg_1_1jgap_1_1_i_handler.html" title="org.jgap.IHandler" alt="" coords="397,811,518,837"/><area shape="rect" id="node20" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html" title="org.jgap.IPersistentRepresentation" alt="" coords="790,136,1013,163"/><area shape="rect" id="node22" href="interfaceorg_1_1jgap_1_1_i_business_key.html" title="org.jgap.IBusinessKey" alt="" coords="825,491,978,517"/><area shape="rect" id="node39" href="interfaceorg_1_1jgap_1_1_i_chromosome_pool.html" title="org.jgap.IChromosomePool" alt="" coords="367,491,547,517"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a14c96eb2e38fe28b05526633e4677ec6"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a14c96eb2e38fe28b05526633e4677ec6">Chromosome</a> ()  throws InvalidConfigurationException </td></tr>
<tr class="separator:a14c96eb2e38fe28b05526633e4677ec6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a62510703fe45c4a8c453801eaa148bb7"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a62510703fe45c4a8c453801eaa148bb7">Chromosome</a> (final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a62510703fe45c4a8c453801eaa148bb7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7290755db8269b79ebdb00b0cd0a9935"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a7290755db8269b79ebdb00b0cd0a9935">Chromosome</a> (final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration, String a_persistentRepresentatuion)  throws InvalidConfigurationException, UnsupportedRepresentationException </td></tr>
<tr class="separator:a7290755db8269b79ebdb00b0cd0a9935"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6e5e72f40a4e2f2b7b72ce6678235f40"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a6e5e72f40a4e2f2b7b72ce6678235f40">Chromosome</a> (final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration, final int a_desiredSize)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a6e5e72f40a4e2f2b7b72ce6678235f40"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2872ac725d9182b66352057850f89176"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a2872ac725d9182b66352057850f89176">Chromosome</a> (final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration, final <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> a_sampleGene, final int a_desiredSize)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a2872ac725d9182b66352057850f89176"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4f6e9e1bf122719fe420c3bc034fb4b3"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a4f6e9e1bf122719fe420c3bc034fb4b3">Chromosome</a> (final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration, <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> a_sampleGene, int a_desiredSize, <a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a> a_constraintChecker)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a4f6e9e1bf122719fe420c3bc034fb4b3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9156d92adcdf9c918bb37c771805b497"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a9156d92adcdf9c918bb37c771805b497">Chromosome</a> (final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration, <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[] a_initialGenes)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a9156d92adcdf9c918bb37c771805b497"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab836c132fb64477cad4d0a2e3b997d6f"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#ab836c132fb64477cad4d0a2e3b997d6f">Chromosome</a> (final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration, <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[] a_initialGenes, <a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a> a_constraintChecker)  throws InvalidConfigurationException </td></tr>
<tr class="separator:ab836c132fb64477cad4d0a2e3b997d6f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9f590d55a34d8b99c2a9a1b4aedd568b"><td class="memItemLeft" align="right" valign="top">synchronized Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a9f590d55a34d8b99c2a9a1b4aedd568b">clone</a> ()</td></tr>
<tr class="separator:a9f590d55a34d8b99c2a9a1b4aedd568b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9ccc6c05aa9c785d1cbae40acc90c80c"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a9ccc6c05aa9c785d1cbae40acc90c80c">getFitnessValue</a> ()</td></tr>
<tr class="separator:a9ccc6c05aa9c785d1cbae40acc90c80c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad0c53486ddb583ddbee2bede065ae068"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#ad0c53486ddb583ddbee2bede065ae068">getFitnessValueDirectly</a> ()</td></tr>
<tr class="separator:ad0c53486ddb583ddbee2bede065ae068"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3715ce380f59756a954566a315514ed0"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a3715ce380f59756a954566a315514ed0">setFitnessValue</a> (double a_newFitnessValue)</td></tr>
<tr class="separator:a3715ce380f59756a954566a315514ed0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4fb0934f7919084f0b5e9ca132e6aaa6"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a4fb0934f7919084f0b5e9ca132e6aaa6">setFitnessValueDirectly</a> (double a_newFitnessValue)</td></tr>
<tr class="separator:a4fb0934f7919084f0b5e9ca132e6aaa6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9cedcb4abe7b9e5f5cfd6cfea0cd949d"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a9cedcb4abe7b9e5f5cfd6cfea0cd949d">toString</a> ()</td></tr>
<tr class="separator:a9cedcb4abe7b9e5f5cfd6cfea0cd949d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac75a21fd23483ff275422e0791289ac8"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#ac75a21fd23483ff275422e0791289ac8">equals</a> (Object other)</td></tr>
<tr class="separator:ac75a21fd23483ff275422e0791289ac8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9f3e1fc14f84b25f09386a2dd1fc69d9"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a9f3e1fc14f84b25f09386a2dd1fc69d9">hashCode</a> ()</td></tr>
<tr class="separator:a9f3e1fc14f84b25f09386a2dd1fc69d9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2be0507944e2ca98bef8372402b33e13"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a2be0507944e2ca98bef8372402b33e13">compareTo</a> (Object other)</td></tr>
<tr class="separator:a2be0507944e2ca98bef8372402b33e13"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1249d045c9830c9dc76e14e4dbe20f5f"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a1249d045c9830c9dc76e14e4dbe20f5f">setIsSelectedForNextGeneration</a> (boolean a_isSelected)</td></tr>
<tr class="separator:a1249d045c9830c9dc76e14e4dbe20f5f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a61d28f792a1389214f06a88dba42260d"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a61d28f792a1389214f06a88dba42260d">isSelectedForNextGeneration</a> ()</td></tr>
<tr class="separator:a61d28f792a1389214f06a88dba42260d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab608acb21ff95b3fe2ff9667a207fd72"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#ab608acb21ff95b3fe2ff9667a207fd72">cleanup</a> ()</td></tr>
<tr class="separator:ab608acb21ff95b3fe2ff9667a207fd72"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a03e0d17b0df1765a442f76645aa4e5b8"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a03e0d17b0df1765a442f76645aa4e5b8">setApplicationData</a> (Object a_newData)</td></tr>
<tr class="separator:a03e0d17b0df1765a442f76645aa4e5b8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aae49bf34668621de1d691cd759ff50b9"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#aae49bf34668621de1d691cd759ff50b9">getApplicationData</a> ()</td></tr>
<tr class="separator:aae49bf34668621de1d691cd759ff50b9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a96692e7942a035e9b661360ca758c7cb"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a96692e7942a035e9b661360ca758c7cb">setGenes</a> (<a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[] a_genes)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a96692e7942a035e9b661360ca758c7cb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa82fc94cbbfeb5b6b5b89a97f9089e9e"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#aa82fc94cbbfeb5b6b5b89a97f9089e9e">setCompareApplicationData</a> (boolean a_doCompare)</td></tr>
<tr class="separator:aa82fc94cbbfeb5b6b5b89a97f9089e9e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a284d75bb2919718325b376e2c91870d0"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a284d75bb2919718325b376e2c91870d0">isCompareApplicationData</a> ()</td></tr>
<tr class="separator:a284d75bb2919718325b376e2c91870d0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae99251567d68bd079e8c4dfaf15dc513"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#ae99251567d68bd079e8c4dfaf15dc513">setConstraintChecker</a> (<a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a> a_constraintChecker)  throws InvalidConfigurationException </td></tr>
<tr class="separator:ae99251567d68bd079e8c4dfaf15dc513"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6c56caed7a052c678daa6af668cad388"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a6c56caed7a052c678daa6af668cad388">getConstraintChecker</a> ()</td></tr>
<tr class="separator:a6c56caed7a052c678daa6af668cad388"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a45bc47533c183ecf3297d7507299cc85"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a45bc47533c183ecf3297d7507299cc85">isHandlerFor</a> (Object a_obj, Class a_class)</td></tr>
<tr class="separator:a45bc47533c183ecf3297d7507299cc85"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abee38007a07ba7a1442dd96571ed7cc1"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#abee38007a07ba7a1442dd96571ed7cc1">perform</a> (Object a_obj, Class a_class, Object a_params)  throws Exception </td></tr>
<tr class="separator:abee38007a07ba7a1442dd96571ed7cc1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aaeae2078063f38074853b06d5c913fa6"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#aaeae2078063f38074853b06d5c913fa6">setMultiObjectives</a> (List a_values)</td></tr>
<tr class="separator:aaeae2078063f38074853b06d5c913fa6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6129d035292cd1d271b5ef5dbc3e3161"><td class="memItemLeft" align="right" valign="top">List&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a6129d035292cd1d271b5ef5dbc3e3161">getMultiObjectives</a> ()</td></tr>
<tr class="separator:a6129d035292cd1d271b5ef5dbc3e3161"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td colspan="2" onclick="javascript:toggleInherit('pub_methods_classorg_1_1jgap_1_1_base_chromosome')"><img src="closed.png" alt="-"/>&#160;Public Member Functions inherited from <a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html">org.jgap.BaseChromosome</a></td></tr>
<tr class="memitem:a94e1af5ba804061833ed3883d42f9441 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a94e1af5ba804061833ed3883d42f9441">BaseChromosome</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a94e1af5ba804061833ed3883d42f9441 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8cf998c46fddd9ee2f8bfcd2a7b6bffa inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a8cf998c46fddd9ee2f8bfcd2a7b6bffa">getUniqueID</a> ()</td></tr>
<tr class="separator:a8cf998c46fddd9ee2f8bfcd2a7b6bffa inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a06156bb0f2c8512ca6a7f7f0fee305b9 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a06156bb0f2c8512ca6a7f7f0fee305b9">setUniqueIDTemplate</a> (String a_templateID, int a_index)</td></tr>
<tr class="separator:a06156bb0f2c8512ca6a7f7f0fee305b9 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2d23cfd243723be4d5eb775532e36a6e inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a2d23cfd243723be4d5eb775532e36a6e">getUniqueIDTemplate</a> (int a_index)</td></tr>
<tr class="separator:a2d23cfd243723be4d5eb775532e36a6e inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a104b33f29941ba6ba405889964fc2bf5 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a104b33f29941ba6ba405889964fc2bf5">getConfiguration</a> ()</td></tr>
<tr class="separator:a104b33f29941ba6ba405889964fc2bf5 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6dcae8b12cad267f6f91431281869e02 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a6dcae8b12cad267f6f91431281869e02">increaseAge</a> ()</td></tr>
<tr class="separator:a6dcae8b12cad267f6f91431281869e02 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7d4483ef679d0e1eb335256b72ab0731 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a7d4483ef679d0e1eb335256b72ab0731">resetAge</a> ()</td></tr>
<tr class="separator:a7d4483ef679d0e1eb335256b72ab0731 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af987f6a74486145a139985297c18ae80 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#af987f6a74486145a139985297c18ae80">getAge</a> ()</td></tr>
<tr class="separator:af987f6a74486145a139985297c18ae80 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aaf9e8ca50d99c27775ce186613320624 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#aaf9e8ca50d99c27775ce186613320624">setAge</a> (int a_age)</td></tr>
<tr class="separator:aaf9e8ca50d99c27775ce186613320624 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a89c0c0959ca4af0152520a2d6c512edd inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a89c0c0959ca4af0152520a2d6c512edd">increaseOperatedOn</a> ()</td></tr>
<tr class="separator:a89c0c0959ca4af0152520a2d6c512edd inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a93a541bdb76c316543a201e8718dd29c inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a93a541bdb76c316543a201e8718dd29c">resetOperatedOn</a> ()</td></tr>
<tr class="separator:a93a541bdb76c316543a201e8718dd29c inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a86349c8e5dbe93c54580a283f70ea2d0 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a86349c8e5dbe93c54580a283f70ea2d0">operatedOn</a> ()</td></tr>
<tr class="separator:a86349c8e5dbe93c54580a283f70ea2d0 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1f9e4971a34546a0d77fa24061e753a0 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a1f9e4971a34546a0d77fa24061e753a0">getGenes</a> ()</td></tr>
<tr class="separator:a1f9e4971a34546a0d77fa24061e753a0 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac4de43d022e2cc7a7a8ac78ff47ad0a0 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#ac4de43d022e2cc7a7a8ac78ff47ad0a0">setGenes</a> (<a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[] a_genes)  throws InvalidConfigurationException </td></tr>
<tr class="separator:ac4de43d022e2cc7a7a8ac78ff47ad0a0 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aef0e40ec9b5956389365f5c3144ab207 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207">getGene</a> (int a_desiredLocus)</td></tr>
<tr class="separator:aef0e40ec9b5956389365f5c3144ab207 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8177b71d092d3ce87909d827ec91492a inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a8177b71d092d3ce87909d827ec91492a">setGene</a> (int a_index, <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> a_gene)</td></tr>
<tr class="separator:a8177b71d092d3ce87909d827ec91492a inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a78c8ae0471973855e19a9b79496f2bec inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a78c8ae0471973855e19a9b79496f2bec">size</a> ()</td></tr>
<tr class="separator:a78c8ae0471973855e19a9b79496f2bec inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab4a4ad55945458548488f87940aa852e inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#ab4a4ad55945458548488f87940aa852e">getPersistentRepresentation</a> ()</td></tr>
<tr class="separator:ab4a4ad55945458548488f87940aa852e inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a124394f7e5c4651e59dd1475e0d2c3b0 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">StringBuffer&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a124394f7e5c4651e59dd1475e0d2c3b0">getGenesPersistentRepresentation</a> ()</td></tr>
<tr class="separator:a124394f7e5c4651e59dd1475e0d2c3b0 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa35172e2b9535ace3b8fa31c1b2ef2fe inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#aa35172e2b9535ace3b8fa31c1b2ef2fe">getBusinessKey</a> ()</td></tr>
<tr class="separator:aa35172e2b9535ace3b8fa31c1b2ef2fe inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2bef76845dd6e92c09037dae72e7ca15 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a2bef76845dd6e92c09037dae72e7ca15">getGenesPersistentRepresentation</a> (StringBuffer a_buffer)</td></tr>
<tr class="separator:a2bef76845dd6e92c09037dae72e7ca15 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a30276dc8afb0bd632c288bf7d2e18b55 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a30276dc8afb0bd632c288bf7d2e18b55">setValueFromPersistentRepresentation</a> (String a_representation)  throws UnsupportedRepresentationException </td></tr>
<tr class="separator:a30276dc8afb0bd632c288bf7d2e18b55 inherit pub_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-static-methods"></a>
Static Public Member Functions</h2></td></tr>
<tr class="memitem:ab77c73b07816a5f3b99a3459fbe0208c"><td class="memItemLeft" align="right" valign="top">static Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#ab77c73b07816a5f3b99a3459fbe0208c">cloneObject</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_config, Object a_object, Object a_master)  throws Exception </td></tr>
<tr class="separator:ab77c73b07816a5f3b99a3459fbe0208c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6b6fb5c7c5b532813a4661de5dbd9ab3"><td class="memItemLeft" align="right" valign="top">static <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a6b6fb5c7c5b532813a4661de5dbd9ab3">randomInitialChromosome</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a6b6fb5c7c5b532813a4661de5dbd9ab3"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pro-methods"></a>
Protected Member Functions</h2></td></tr>
<tr class="memitem:aa6b7f021f64700b1926d5819f5fde5aa"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#aa6b7f021f64700b1926d5819f5fde5aa">initFromGene</a> (<a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> a_sampleGene)</td></tr>
<tr class="separator:aa6b7f021f64700b1926d5819f5fde5aa"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a74df7a8e7986a3ed2a18c9715fae7b90"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a74df7a8e7986a3ed2a18c9715fae7b90">checkGenes</a> (<a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[] a_initialGenes)</td></tr>
<tr class="separator:a74df7a8e7986a3ed2a18c9715fae7b90"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:adf4d7c5113d0c70283ffb2323b7e0cda"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#adf4d7c5113d0c70283ffb2323b7e0cda">cloneObject</a> (Object a_object)  throws Exception </td></tr>
<tr class="separator:adf4d7c5113d0c70283ffb2323b7e0cda"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aea063618aebe23aa8579baf20063db24"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#aea063618aebe23aa8579baf20063db24">calcFitnessValue</a> ()</td></tr>
<tr class="separator:aea063618aebe23aa8579baf20063db24"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af472c1bacfaf6e5e5c1d5ac3af75ddef"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#af472c1bacfaf6e5e5c1d5ac3af75ddef">verify</a> (<a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a> a_constraintChecker)  throws InvalidConfigurationException </td></tr>
<tr class="separator:af472c1bacfaf6e5e5c1d5ac3af75ddef"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pro_methods_classorg_1_1jgap_1_1_base_chromosome"><td colspan="2" onclick="javascript:toggleInherit('pro_methods_classorg_1_1jgap_1_1_base_chromosome')"><img src="closed.png" alt="-"/>&#160;Protected Member Functions inherited from <a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html">org.jgap.BaseChromosome</a></td></tr>
<tr class="memitem:a2c339c9c70184d1a911878e1be39dc82 inherit pro_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a2c339c9c70184d1a911878e1be39dc82">encode</a> (String a_string)</td></tr>
<tr class="separator:a2c339c9c70184d1a911878e1be39dc82 inherit pro_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0ccfa10730437df1a23322886a6949e7 inherit pro_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a0ccfa10730437df1a23322886a6949e7">decode</a> (String a_string)  throws UnsupportedEncodingException </td></tr>
<tr class="separator:a0ccfa10730437df1a23322886a6949e7 inherit pro_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a61d504512312fdc761af87fe0f3ca36d inherit pro_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a61d504512312fdc761af87fe0f3ca36d">createGene</a> (String a_geneClassName, String a_persistentRepresentation)  throws Exception </td></tr>
<tr class="separator:a61d504512312fdc761af87fe0f3ca36d inherit pro_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pro-attribs"></a>
Protected Attributes</h2></td></tr>
<tr class="memitem:a905236819a9904ca3d096fa1c325aa1b"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a905236819a9904ca3d096fa1c325aa1b">m_fitnessValue</a> = <a class="el" href="classorg_1_1jgap_1_1_fitness_function.html#a5f1f23e6f8a0f9606aee5d52fac6a06b">FitnessFunction.NO_FITNESS_VALUE</a></td></tr>
<tr class="separator:a905236819a9904ca3d096fa1c325aa1b"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-attribs"></a>
Private Attributes</h2></td></tr>
<tr class="memitem:af2b4e1abcdf5995b45fb7d5cf93a559c"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#af2b4e1abcdf5995b45fb7d5cf93a559c">m_applicationData</a></td></tr>
<tr class="separator:af2b4e1abcdf5995b45fb7d5cf93a559c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6d73c5e40fce0219e80021211d4a2d57"><td class="memItemLeft" align="right" valign="top">List&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a6d73c5e40fce0219e80021211d4a2d57">m_multiObjective</a></td></tr>
<tr class="separator:a6d73c5e40fce0219e80021211d4a2d57"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad8bb6734b245d112a8e6d66fb38914c9"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#ad8bb6734b245d112a8e6d66fb38914c9">m_isSelectedForNextGeneration</a></td></tr>
<tr class="separator:ad8bb6734b245d112a8e6d66fb38914c9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7c696f550a4f4458ed4115833abf02fb"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a7c696f550a4f4458ed4115833abf02fb">m_compareAppData</a></td></tr>
<tr class="separator:a7c696f550a4f4458ed4115833abf02fb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af890b1160c9b97defdf107e6a4e53329"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#af890b1160c9b97defdf107e6a4e53329">m_geneAlleleChecker</a></td></tr>
<tr class="separator:af890b1160c9b97defdf107e6a4e53329"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a678ed92a2a1a47343638de24f40b66e3"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a678ed92a2a1a47343638de24f40b66e3">m_alwaysCalculate</a></td></tr>
<tr class="separator:a678ed92a2a1a47343638de24f40b66e3"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-static-attribs"></a>
Static Private Attributes</h2></td></tr>
<tr class="memitem:aa8e104e9bc3f5f2bfdfe67b604c10d9c"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_chromosome.html#aa8e104e9bc3f5f2bfdfe67b604c10d9c">CVS_REVISION</a> = &quot;$Revision: 1.105 $&quot;</td></tr>
<tr class="separator:aa8e104e9bc3f5f2bfdfe67b604c10d9c"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="inherited"></a>
Additional Inherited Members</h2></td></tr>
<tr class="inherit_header pub_static_attribs_classorg_1_1jgap_1_1_base_chromosome"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_classorg_1_1jgap_1_1_base_chromosome')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html">org.jgap.BaseChromosome</a></td></tr>
<tr class="memitem:a5cb3ac5d20e1fccb7eacdd0bb38bd650 inherit pub_static_attribs_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a5cb3ac5d20e1fccb7eacdd0bb38bd650">GENE_DELIMITER</a> = &quot;#&quot;</td></tr>
<tr class="separator:a5cb3ac5d20e1fccb7eacdd0bb38bd650 inherit pub_static_attribs_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a52428dfc927129c8960a9d9ca3864346 inherit pub_static_attribs_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a52428dfc927129c8960a9d9ca3864346">GENE_DELIMITER_HEADING</a> = &quot;&lt;&quot;</td></tr>
<tr class="separator:a52428dfc927129c8960a9d9ca3864346 inherit pub_static_attribs_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a05f3c0c0caa0e1676345b2303c842be0 inherit pub_static_attribs_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a05f3c0c0caa0e1676345b2303c842be0">GENE_DELIMITER_CLOSING</a> = &quot;&gt;&quot;</td></tr>
<tr class="separator:a05f3c0c0caa0e1676345b2303c842be0 inherit pub_static_attribs_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4efa73cb7ff2c5dcd0e16f1d7287dc78 inherit pub_static_attribs_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a4efa73cb7ff2c5dcd0e16f1d7287dc78">CHROM_DELIMITER</a> = &quot;#&quot;</td></tr>
<tr class="separator:a4efa73cb7ff2c5dcd0e16f1d7287dc78 inherit pub_static_attribs_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">org.jgap.IChromosome</a></td></tr>
<tr class="memitem:a4e7381b89963f2ee6f69839b38ea8dab inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a4e7381b89963f2ee6f69839b38ea8dab">CVS_REVISION</a> = &quot;$Revision: 1.17 $&quot;</td></tr>
<tr class="separator:a4e7381b89963f2ee6f69839b38ea8dab inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a34c8c0933688a10837692d1c0024f663 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a34c8c0933688a10837692d1c0024f663">S_FITNESS_VALUE</a> = &quot;Fitness value&quot;</td></tr>
<tr class="separator:a34c8c0933688a10837692d1c0024f663 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae93666d5923e449f45656ad5fed52f84 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#ae93666d5923e449f45656ad5fed52f84">S_ALLELES</a> = &quot;Alleles&quot;</td></tr>
<tr class="separator:ae93666d5923e449f45656ad5fed52f84 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a03218ba49e2c53108005cd6290252284 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a03218ba49e2c53108005cd6290252284">S_APPLICATION_DATA</a> = &quot;Application data&quot;</td></tr>
<tr class="separator:a03218ba49e2c53108005cd6290252284 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a50371491243a21f494112bf66d2e1a26 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a50371491243a21f494112bf66d2e1a26">S_SIZE</a> = &quot;Size&quot;</td></tr>
<tr class="separator:a50371491243a21f494112bf66d2e1a26 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html">org.jgap.util.ICloneable</a></td></tr>
<tr class="memitem:a4eaaa85432026f5bc54c3e520b22caa7 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html#a4eaaa85432026f5bc54c3e520b22caa7">CVS_REVISION</a> = &quot;$Revision: 1.3 $&quot;</td></tr>
<tr class="separator:a4eaaa85432026f5bc54c3e520b22caa7 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html">org.jgap.IUniqueKey</a></td></tr>
<tr class="memitem:a5b6472b1e40001ee262da59ee449a77b inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html#a5b6472b1e40001ee262da59ee449a77b">CVS_REVISION</a> = &quot;$Revision: 1.1 $&quot;</td></tr>
<tr class="separator:a5b6472b1e40001ee262da59ee449a77b inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_initializer"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_initializer')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_initializer.html">org.jgap.IInitializer</a></td></tr>
<tr class="memitem:af84c0edbc3fdf99bda88da8d4ed1d2ea inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_initializer"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_initializer.html#af84c0edbc3fdf99bda88da8d4ed1d2ea">CVS_REVISION</a> = &quot;$Revision: 1.2 $&quot;</td></tr>
<tr class="separator:af84c0edbc3fdf99bda88da8d4ed1d2ea inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_initializer"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_handler"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_handler')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_handler.html">org.jgap.IHandler</a></td></tr>
<tr class="memitem:ac87eeb82b5fb4d6e5ba1f8850db52118 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_handler"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_handler.html#ac87eeb82b5fb4d6e5ba1f8850db52118">CVS_REVISION</a> = &quot;$Revision: 1.3 $&quot;</td></tr>
<tr class="separator:ac87eeb82b5fb4d6e5ba1f8850db52118 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_handler"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_persistent_representation"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_persistent_representation')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html">org.jgap.IPersistentRepresentation</a></td></tr>
<tr class="memitem:ae48c5895f4af079337ffebe487b176f5 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_persistent_representation"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html#ae48c5895f4af079337ffebe487b176f5">CVS_REVISION</a> = &quot;$Revision: 1.2 $&quot;</td></tr>
<tr class="separator:ae48c5895f4af079337ffebe487b176f5 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_persistent_representation"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_business_key.html">org.jgap.IBusinessKey</a></td></tr>
<tr class="memitem:a6d43b687da0842739b8021f466ca451a inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_business_key.html#a6d43b687da0842739b8021f466ca451a">CVS_REVISION</a> = &quot;$Revision: 1.3 $&quot;</td></tr>
<tr class="separator:a6d43b687da0842739b8021f466ca451a inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pro_static_methods_classorg_1_1jgap_1_1_base_chromosome"><td colspan="2" onclick="javascript:toggleInherit('pro_static_methods_classorg_1_1jgap_1_1_base_chromosome')"><img src="closed.png" alt="-"/>&#160;Static Protected Member Functions inherited from <a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html">org.jgap.BaseChromosome</a></td></tr>
<tr class="memitem:a9237b2c8d40b9d466d5f68792ba04036 inherit pro_static_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memItemLeft" align="right" valign="top">static final List&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a9237b2c8d40b9d466d5f68792ba04036">split</a> (String a_string)  throws UnsupportedRepresentationException </td></tr>
<tr class="separator:a9237b2c8d40b9d466d5f68792ba04036 inherit pro_static_methods_classorg_1_1jgap_1_1_base_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Chromosomes represent potential solutions and consist of a fixed-length collection of genes. Each gene represents a discrete part of the solution. Each gene in the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> may be backed by a different concrete implementation of the <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> interface, but all genes in a respective position (locus) must share the same concrete implementation across Chromosomes within a single population (genotype). In other words, gene 1 in a chromosome must share the same concrete implementation as gene 1 in all other chromosomes in the population.</p>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00065">65</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="a14c96eb2e38fe28b05526633e4677ec6"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Chromosome.Chromosome </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Default constructor, only provided for dynamic instantiation.</p>
<p>Attention: The configuration used is the one set with the static method Genotype.setConfiguration.</p>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.4 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00141">141</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00334">org.jgap.Chromosome.clone()</a>, and <a class="el" href="_chromosome_8java_source.html#l00668">org.jgap.Chromosome.randomInitialChromosome()</a>.</p>

</div>
</div>
<a class="anchor" id="a62510703fe45c4a8c453801eaa148bb7"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Chromosome.Chromosome </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Constructor, provided for dynamic or minimal instantiation.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the configuration to use </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00155">155</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00129">org.jgap.Chromosome.m_alwaysCalculate</a>.</p>

</div>
</div>
<a class="anchor" id="a7290755db8269b79ebdb00b0cd0a9935"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Chromosome.Chromosome </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_persistentRepresentatuion</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a>, <a class="el" href="classorg_1_1jgap_1_1_unsupported_representation_exception.html">UnsupportedRepresentationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Constructor, provided for instantiation via persistent representation.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the configuration to use </td></tr>
    <tr><td class="paramname">a_persistentRepresentatuion</td><td>valid persistent representation that was most likely obtained via <a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#ab4a4ad55945458548488f87940aa852e">getPersistentRepresentation()</a> </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_unsupported_representation_exception.html">UnsupportedRepresentationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00173">173</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00400">org.jgap.BaseChromosome.setValueFromPersistentRepresentation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_a7290755db8269b79ebdb00b0cd0a9935_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_a7290755db8269b79ebdb00b0cd0a9935_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_a7290755db8269b79ebdb00b0cd0a9935_cgraph" id="classorg_1_1jgap_1_1_chromosome_a7290755db8269b79ebdb00b0cd0a9935_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_chromosome.html#a30276dc8afb0bd632c288bf7d2e18b55" title="org.jgap.BaseChromosome.set\lValueFromPersistentRepresentation" alt="" coords="285,143,517,185"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_chromosome.html#a61d504512312fdc761af87fe0f3ca36d" title="org.jgap.BaseChromosome.createGene" alt="" coords="565,12,816,39"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_base_chromosome.html#a0ccfa10730437df1a23322886a6949e7" title="org.jgap.BaseChromosome.decode" alt="" coords="577,63,804,89"/><area shape="rect" id="node11" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a56f82d43b102fc5da8e0a071e32dc13d" title="org.jgap.IChromosome.set\lFitnessValue" alt="" coords="603,114,779,155"/><area shape="rect" id="node13" href="classorg_1_1jgap_1_1_base_chromosome.html#a8177b71d092d3ce87909d827ec91492a" title="org.jgap.BaseChromosome.setGene" alt="" coords="574,180,807,207"/><area shape="rect" id="node15" href="classorg_1_1jgap_1_1_base_chromosome.html#ac4de43d022e2cc7a7a8ac78ff47ad0a0" title="org.jgap.BaseChromosome.set\lGenes" alt="" coords="589,231,792,273"/><area shape="rect" id="node17" href="classorg_1_1jgap_1_1_base_chromosome.html#a9237b2c8d40b9d466d5f68792ba04036" title="org.jgap.BaseChromosome.split" alt="" coords="587,297,795,324"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_chromosome.html#a104b33f29941ba6ba405889964fc2bf5" title="org.jgap.BaseChromosome.get\lConfiguration" alt="" coords="864,5,1067,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a6e5e72f40a4e2f2b7b72ce6678235f40"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Chromosome.Chromosome </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final int&#160;</td>
          <td class="paramname"><em>a_desiredSize</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Constructor for specifying the number of genes.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the configuration to use </td></tr>
    <tr><td class="paramname">a_desiredSize</td><td>number of genes the chromosome contains of </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.2 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00190">190</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l01006">org.jgap.Chromosome.setGenes()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_a6e5e72f40a4e2f2b7b72ce6678235f40_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_a6e5e72f40a4e2f2b7b72ce6678235f40_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_a6e5e72f40a4e2f2b7b72ce6678235f40_cgraph" id="classorg_1_1jgap_1_1_chromosome_a6e5e72f40a4e2f2b7b72ce6678235f40_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_chromosome.html#a96692e7942a035e9b661360ca758c7cb" title="org.jgap.Chromosome.setGenes" alt="" coords="285,71,496,97"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_chromosome.html#af472c1bacfaf6e5e5c1d5ac3af75ddef" title="org.jgap.Chromosome.verify" alt="" coords="567,41,753,68"/><area shape="rect" id="node11" href="classorg_1_1jgap_1_1_chromosome.html#a6c56caed7a052c678daa6af668cad388" title="org.jgap.Chromosome.getConstraint\lChecker" alt="" coords="544,93,776,134"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_chromosome.html#a1f9e4971a34546a0d77fa24061e753a0" title="org.jgap.BaseChromosome.get\lGenes" alt="" coords="840,5,1043,46"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207" title="org.jgap.BaseChromosome.getGene" alt="" coords="825,71,1058,97"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a2872ac725d9182b66352057850f89176"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Chromosome.Chromosome </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>&#160;</td>
          <td class="paramname"><em>a_sampleGene</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final int&#160;</td>
          <td class="paramname"><em>a_desiredSize</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Constructs a <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> of the given size separate from any specific <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>. This constructor will use the given sample <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> to construct a new <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> instance containing genes all of the same type as the sample <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>. This can be useful for constructing sample chromosomes that use the same <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> type for all of their genes and that are to be used to setup a <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the configuration to use </td></tr>
    <tr><td class="paramname">a_sampleGene</td><td>a concrete sampleGene instance that will be used as a template for all of the genes in this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> </td></tr>
    <tr><td class="paramname">a_desiredSize</td><td>the desired size (number of genes) of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00219">219</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00235">org.jgap.Chromosome.initFromGene()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_a2872ac725d9182b66352057850f89176_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_a2872ac725d9182b66352057850f89176_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_a2872ac725d9182b66352057850f89176_cgraph" id="classorg_1_1jgap_1_1_chromosome_a2872ac725d9182b66352057850f89176_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_chromosome.html#aa6b7f021f64700b1926d5819f5fde5aa" title="org.jgap.Chromosome.init\lFromGene" alt="" coords="285,49,456,90"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_chromosome.html#a78c8ae0471973855e19a9b79496f2bec" title="org.jgap.BaseChromosome.size" alt="" coords="517,5,725,32"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_chromosome.html#a8177b71d092d3ce87909d827ec91492a" title="org.jgap.BaseChromosome.setGene" alt="" coords="505,56,738,83"/><area shape="rect" id="node9" href="interfaceorg_1_1jgap_1_1_gene.html#ae4ddccb7bf569c0ed6cdc19def590964" title="org.jgap.Gene.newGene" alt="" coords="540,107,703,133"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a4f6e9e1bf122719fe420c3bc034fb4b3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Chromosome.Chromosome </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>&#160;</td>
          <td class="paramname"><em>a_sampleGene</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_desiredSize</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a>&#160;</td>
          <td class="paramname"><em>a_constraintChecker</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00226">226</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00235">org.jgap.Chromosome.initFromGene()</a>, and <a class="el" href="_chromosome_8java_source.html#l01046">org.jgap.Chromosome.setConstraintChecker()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_a4f6e9e1bf122719fe420c3bc034fb4b3_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_a4f6e9e1bf122719fe420c3bc034fb4b3_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_a4f6e9e1bf122719fe420c3bc034fb4b3_cgraph" id="classorg_1_1jgap_1_1_chromosome_a4f6e9e1bf122719fe420c3bc034fb4b3_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_chromosome.html#aa6b7f021f64700b1926d5819f5fde5aa" title="org.jgap.Chromosome.init\lFromGene" alt="" coords="316,58,487,99"/><area shape="rect" id="node11" href="classorg_1_1jgap_1_1_chromosome.html#ae99251567d68bd079e8c4dfaf15dc513" title="org.jgap.Chromosome.setConstraint\lChecker" alt="" coords="285,133,517,174"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_chromosome.html#a78c8ae0471973855e19a9b79496f2bec" title="org.jgap.BaseChromosome.size" alt="" coords="579,5,787,32"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_chromosome.html#a8177b71d092d3ce87909d827ec91492a" title="org.jgap.BaseChromosome.setGene" alt="" coords="566,56,799,83"/><area shape="rect" id="node9" href="interfaceorg_1_1jgap_1_1_gene.html#ae4ddccb7bf569c0ed6cdc19def590964" title="org.jgap.Gene.newGene" alt="" coords="601,107,764,133"/><area shape="rect" id="node13" href="classorg_1_1jgap_1_1_chromosome.html#af472c1bacfaf6e5e5c1d5ac3af75ddef" title="org.jgap.Chromosome.verify" alt="" coords="589,157,776,184"/><area shape="rect" id="node15" href="classorg_1_1jgap_1_1_base_chromosome.html#a1f9e4971a34546a0d77fa24061e753a0" title="org.jgap.BaseChromosome.get\lGenes" alt="" coords="864,121,1067,162"/><area shape="rect" id="node17" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207" title="org.jgap.BaseChromosome.getGene" alt="" coords="849,187,1082,213"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a9156d92adcdf9c918bb37c771805b497"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Chromosome.Chromosome </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[]&#160;</td>
          <td class="paramname"><em>a_initialGenes</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Constructs a <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> separate from any specific <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>. This can be useful for constructing sample chromosomes that are to be used to setup a <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the configuration to use </td></tr>
    <tr><td class="paramname">a_initialGenes</td><td>the initial genes of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00264">264</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00304">org.jgap.Chromosome.checkGenes()</a>, and <a class="el" href="_chromosome_8java_source.html#l01006">org.jgap.Chromosome.setGenes()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_a9156d92adcdf9c918bb37c771805b497_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_a9156d92adcdf9c918bb37c771805b497_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_a9156d92adcdf9c918bb37c771805b497_cgraph" id="classorg_1_1jgap_1_1_chromosome_a9156d92adcdf9c918bb37c771805b497_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_chromosome.html#a74df7a8e7986a3ed2a18c9715fae7b90" title="org.jgap.Chromosome.check\lGenes" alt="" coords="296,5,485,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_chromosome.html#a96692e7942a035e9b661360ca758c7cb" title="org.jgap.Chromosome.setGenes" alt="" coords="285,71,496,97"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_chromosome.html#af472c1bacfaf6e5e5c1d5ac3af75ddef" title="org.jgap.Chromosome.verify" alt="" coords="567,41,753,68"/><area shape="rect" id="node13" href="classorg_1_1jgap_1_1_chromosome.html#a6c56caed7a052c678daa6af668cad388" title="org.jgap.Chromosome.getConstraint\lChecker" alt="" coords="544,93,776,134"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_base_chromosome.html#a1f9e4971a34546a0d77fa24061e753a0" title="org.jgap.BaseChromosome.get\lGenes" alt="" coords="840,5,1043,46"/><area shape="rect" id="node11" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207" title="org.jgap.BaseChromosome.getGene" alt="" coords="825,71,1058,97"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ab836c132fb64477cad4d0a2e3b997d6f"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Chromosome.Chromosome </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[]&#160;</td>
          <td class="paramname"><em>a_initialGenes</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a>&#160;</td>
          <td class="paramname"><em>a_constraintChecker</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Constructs a <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> separate from any specific <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>. This can be useful for constructing sample chromosomes that are to be used to setup a <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object. Additionally, a constraint checker can be specified. It is used right here to verify the validity of the gene types supplied.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the configuration to use </td></tr>
    <tr><td class="paramname">a_initialGenes</td><td>the initial genes of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> </td></tr>
    <tr><td class="paramname">a_constraintChecker</td><td>constraint checker to use </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td>in case the constraint checker reports a configuration error</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.5 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00287">287</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00304">org.jgap.Chromosome.checkGenes()</a>, <a class="el" href="_chromosome_8java_source.html#l01046">org.jgap.Chromosome.setConstraintChecker()</a>, and <a class="el" href="_chromosome_8java_source.html#l01006">org.jgap.Chromosome.setGenes()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_ab836c132fb64477cad4d0a2e3b997d6f_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_ab836c132fb64477cad4d0a2e3b997d6f_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_ab836c132fb64477cad4d0a2e3b997d6f_cgraph" id="classorg_1_1jgap_1_1_chromosome_ab836c132fb64477cad4d0a2e3b997d6f_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_chromosome.html#a74df7a8e7986a3ed2a18c9715fae7b90" title="org.jgap.Chromosome.check\lGenes" alt="" coords="307,5,496,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_chromosome.html#ae99251567d68bd079e8c4dfaf15dc513" title="org.jgap.Chromosome.setConstraint\lChecker" alt="" coords="285,70,517,111"/><area shape="rect" id="node13" href="classorg_1_1jgap_1_1_chromosome.html#a96692e7942a035e9b661360ca758c7cb" title="org.jgap.Chromosome.setGenes" alt="" coords="296,136,507,163"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_chromosome.html#af472c1bacfaf6e5e5c1d5ac3af75ddef" title="org.jgap.Chromosome.verify" alt="" coords="588,77,775,104"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_base_chromosome.html#a1f9e4971a34546a0d77fa24061e753a0" title="org.jgap.BaseChromosome.get\lGenes" alt="" coords="861,41,1064,82"/><area shape="rect" id="node11" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207" title="org.jgap.BaseChromosome.getGene" alt="" coords="846,107,1079,133"/><area shape="rect" id="node16" href="classorg_1_1jgap_1_1_chromosome.html#a6c56caed7a052c678daa6af668cad388" title="org.jgap.Chromosome.getConstraint\lChecker" alt="" coords="565,129,797,170"/></map>
</div>
</p>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="aea063618aebe23aa8579baf20063db24"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double org.jgap.Chromosome.calcFitnessValue </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>fitness value of this chromosome determined via the registered fitness function</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.4 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00560">560</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00150">org.jgap.BaseChromosome.getConfiguration()</a>, and <a class="el" href="_chromosome_8java_source.html#l00102">org.jgap.Chromosome.m_fitnessValue</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00534">org.jgap.Chromosome.getFitnessValue()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_aea063618aebe23aa8579baf20063db24_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_aea063618aebe23aa8579baf20063db24_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_aea063618aebe23aa8579baf20063db24_cgraph" id="classorg_1_1jgap_1_1_chromosome_aea063618aebe23aa8579baf20063db24_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_chromosome.html#a104b33f29941ba6ba405889964fc2bf5" title="org.jgap.BaseChromosome.get\lConfiguration" alt="" coords="232,5,435,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a74df7a8e7986a3ed2a18c9715fae7b90"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Chromosome.checkGenes </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[]&#160;</td>
          <td class="paramname"><em>a_initialGenes</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Helper: called by constructors only to verify the initial genes.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_initialGenes</td><td>the initial genes of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> to verify</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.5 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00304">304</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00264">org.jgap.Chromosome.Chromosome()</a>.</p>

</div>
</div>
<a class="anchor" id="ab608acb21ff95b3fe2ff9667a207fd72"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Chromosome.cleanup </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Invoked when this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> is no longer needed and should perform any necessary cleanup. Note that this method will attempt to release this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> instance to the active ChromosomePool, if any.</p>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a184e942f9b44185236492ee7261b2b14">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00928">928</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00150">org.jgap.BaseChromosome.getConfiguration()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00283">org.jgap.BaseChromosome.getGene()</a>, <a class="el" href="_chromosome_8java_source.html#l00102">org.jgap.Chromosome.m_fitnessValue</a>, <a class="el" href="_chromosome_8java_source.html#l00092">org.jgap.Chromosome.m_isSelectedForNextGeneration</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00302">org.jgap.BaseChromosome.size()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_ab608acb21ff95b3fe2ff9667a207fd72_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_ab608acb21ff95b3fe2ff9667a207fd72_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_ab608acb21ff95b3fe2ff9667a207fd72_cgraph" id="classorg_1_1jgap_1_1_chromosome_ab608acb21ff95b3fe2ff9667a207fd72_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_chromosome.html#a104b33f29941ba6ba405889964fc2bf5" title="org.jgap.BaseChromosome.get\lConfiguration" alt="" coords="269,5,472,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207" title="org.jgap.BaseChromosome.getGene" alt="" coords="254,71,487,97"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_chromosome.html#a78c8ae0471973855e19a9b79496f2bec" title="org.jgap.BaseChromosome.size" alt="" coords="267,121,475,148"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a9f590d55a34d8b99c2a9a1b4aedd568b"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">synchronized Object org.jgap.Chromosome.clone </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Returns a copy of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. The returned instance can evolve independently of this instance. Note that, if possible, this method will first attempt to acquire a <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> instance from the active ChromosomePool (if any) and set its value appropriately before returning it. If that is not possible, then a new <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> instance will be constructed and its value set appropriately before returning.</p>
<dl class="section return"><dt>Returns</dt><dd>copy of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">IllegalStateException</td><td>instead of CloneNotSupportedException</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000049">Todo:</a></b></dt><dd>once output a warning: allele should be cloneable! </dd></dl>

<p>Implements <a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a97071ef6dba552c1c1f2b2780d27eec3">org.jgap.BaseChromosome</a>.</p>

<p>Reimplemented in <a class="el" href="classorg_1_1jgap_1_1_chromosome_for_testing.html#acfcef639c3443fde7d324164050e3e19">org.jgap.ChromosomeForTesting</a>.</p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00334">334</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00141">org.jgap.Chromosome.Chromosome()</a>, <a class="el" href="_chromosome_8java_source.html#l00478">org.jgap.Chromosome.cloneObject()</a>, <a class="el" href="_chromosome_8java_source.html#l00993">org.jgap.Chromosome.getApplicationData()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00150">org.jgap.BaseChromosome.getConfiguration()</a>, <a class="el" href="_chromosome_8java_source.html#l01059">org.jgap.Chromosome.getConstraintChecker()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00283">org.jgap.BaseChromosome.getGene()</a>, <a class="el" href="_chromosome_8java_source.html#l00102">org.jgap.Chromosome.m_fitnessValue</a>, <a class="el" href="_chromosome_8java_source.html#l00086">org.jgap.Chromosome.m_multiObjective</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00302">org.jgap.BaseChromosome.size()</a>.</p>

<p>Referenced by <a class="el" href="_g_a_breeder_8java_source.html#l00349">org.jgap.impl.GABreeder.bulkFunctionEvaluation()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00476">org.jgap.ChromosomeTest.testClone_1()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00514">org.jgap.ChromosomeTest.testClone_2()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00567">org.jgap.ChromosomeTest.testClone_4()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00431">org.jgap.ChromosomeTest.testClone_5()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00453">org.jgap.ChromosomeTest.testClone_6()</a>, <a class="el" href="_averaging_crossover_operator_test_8java_source.html#l00093">org.jgap.impl.AveragingCrossoverOperatorTest.testOperate_1()</a>, <a class="el" href="_gaussian_mutation_operator_test_8java_source.html#l00094">org.jgap.impl.GaussianMutationOperatorTest.testOperate_1()</a>, <a class="el" href="_inversion_operator_test_8java_source.html#l00124">org.jgap.impl.InversionOperatorTest.testOperate_1()</a>, and <a class="el" href="_crossover_operator_test_8java_source.html#l00306">org.jgap.impl.CrossoverOperatorTest.testOperate_1()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_a9f590d55a34d8b99c2a9a1b4aedd568b_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_a9f590d55a34d8b99c2a9a1b4aedd568b_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_a9f590d55a34d8b99c2a9a1b4aedd568b_cgraph" id="classorg_1_1jgap_1_1_chromosome_a9f590d55a34d8b99c2a9a1b4aedd568b_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_chromosome.html#a14c96eb2e38fe28b05526633e4677ec6" title="org.jgap.Chromosome.Chromosome" alt="" coords="256,5,488,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_chromosome.html#adf4d7c5113d0c70283ffb2323b7e0cda" title="org.jgap.Chromosome.clone\lObject" alt="" coords="279,57,465,98"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_chromosome.html#a104b33f29941ba6ba405889964fc2bf5" title="org.jgap.BaseChromosome.get\lConfiguration" alt="" coords="552,86,755,127"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_chromosome.html#aae49bf34668621de1d691cd759ff50b9" title="org.jgap.Chromosome.getApplicationData" alt="" coords="240,173,504,200"/><area shape="rect" id="node12" href="classorg_1_1jgap_1_1_chromosome.html#a6c56caed7a052c678daa6af668cad388" title="org.jgap.Chromosome.getConstraint\lChecker" alt="" coords="256,225,488,266"/><area shape="rect" id="node14" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207" title="org.jgap.BaseChromosome.getGene" alt="" coords="255,291,489,317"/><area shape="rect" id="node16" href="classorg_1_1jgap_1_1_base_chromosome.html#a78c8ae0471973855e19a9b79496f2bec" title="org.jgap.BaseChromosome.size" alt="" coords="268,341,476,368"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="adf4d7c5113d0c70283ffb2323b7e0cda"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.Chromosome.cloneObject </td>
          <td>(</td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>a_object</em></td><td>)</td>
          <td> throws Exception</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Clones an object by using clone handlers. If no deep cloning possible, then return the reference.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_object</td><td>the object to clone </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the cloned object, or the object itself if no coning supported </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Exception</td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00478">478</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00150">org.jgap.BaseChromosome.getConfiguration()</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00334">org.jgap.Chromosome.clone()</a>, and <a class="el" href="_chromosome_8java_source.html#l00668">org.jgap.Chromosome.randomInitialChromosome()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_adf4d7c5113d0c70283ffb2323b7e0cda_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_adf4d7c5113d0c70283ffb2323b7e0cda_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_adf4d7c5113d0c70283ffb2323b7e0cda_cgraph" id="classorg_1_1jgap_1_1_chromosome_adf4d7c5113d0c70283ffb2323b7e0cda_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_chromosome.html#a104b33f29941ba6ba405889964fc2bf5" title="org.jgap.BaseChromosome.get\lConfiguration" alt="" coords="240,5,443,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ab77c73b07816a5f3b99a3459fbe0208c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">static Object org.jgap.Chromosome.cloneObject </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_config</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>a_object</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>a_master</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws Exception</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Static convenience method. Clones an object by using clone handlers. If no deep cloning possible, then return the reference.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_config</td><td>a valid configuration to obtain the JGAPFactory from </td></tr>
    <tr><td class="paramname">a_object</td><td>the object to clone </td></tr>
    <tr><td class="paramname">a_master</td><td>the super object of a_object, e.g. a chromosome in case of application to be cloned </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the cloned object, or the object itself if no coning supported </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Exception</td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.5 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00498">498</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

</div>
</div>
<a class="anchor" id="a2be0507944e2ca98bef8372402b33e13"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.Chromosome.compareTo </td>
          <td>(</td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>other</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Compares the given <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. This chromosome is considered to be "less than" the given chromosome if it has a fewer number of genes or if any of its gene values (alleles) are less than their corresponding gene values in the other chromosome.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">other</td><td>the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> against which to compare this chromosome </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>a negative number if this chromosome is "less than" the given chromosome, zero if they are equal to each other, and a positive number if this chromosome is "greater than" the given chromosome</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000052">Todo:</a></b></dt><dd>improve </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00816">816</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a1734f351c78c6d36bec48fdc8a334902">org.jgap.IChromosome.getApplicationData()</a>, <a class="el" href="_chromosome_8java_source.html#l00993">org.jgap.Chromosome.getApplicationData()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00150">org.jgap.BaseChromosome.getConfiguration()</a>, <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a743303c75d2a001f88af579051862964">org.jgap.IChromosome.getFitnessValueDirectly()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00283">org.jgap.BaseChromosome.getGene()</a>, <a class="el" href="interfaceorg_1_1jgap_1_1_fitness_evaluator.html#aad9309b75a437b32641480be173550eb">org.jgap.FitnessEvaluator.isFitter()</a>, <a class="el" href="_chromosome_8java_source.html#l00111">org.jgap.Chromosome.m_compareAppData</a>, <a class="el" href="_chromosome_8java_source.html#l00102">org.jgap.Chromosome.m_fitnessValue</a>, <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a9b460ad665ea1ab4e789061233fc6fa3">org.jgap.IChromosome.size()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00302">org.jgap.BaseChromosome.size()</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00751">org.jgap.Chromosome.equals()</a>, <a class="el" href="_chromosome_test_8java_source.html#l01009">org.jgap.ChromosomeTest.testCompareTo_1()</a>, <a class="el" href="_chromosome_test_8java_source.html#l01038">org.jgap.ChromosomeTest.testCompareTo_2()</a>, <a class="el" href="_chromosome_test_8java_source.html#l01063">org.jgap.ChromosomeTest.testCompareTo_3()</a>, and <a class="el" href="_chromosome_test_8java_source.html#l01077">org.jgap.ChromosomeTest.testCompareTo_4()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_a2be0507944e2ca98bef8372402b33e13_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_a2be0507944e2ca98bef8372402b33e13_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_a2be0507944e2ca98bef8372402b33e13_cgraph" id="classorg_1_1jgap_1_1_chromosome_a2be0507944e2ca98bef8372402b33e13_cgraph">
<area shape="rect" id="node3" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a1734f351c78c6d36bec48fdc8a334902" title="org.jgap.IChromosome.get\lApplicationData" alt="" coords="316,5,492,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_chromosome.html#aae49bf34668621de1d691cd759ff50b9" title="org.jgap.Chromosome.getApplicationData" alt="" coords="272,71,536,97"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_chromosome.html#a104b33f29941ba6ba405889964fc2bf5" title="org.jgap.BaseChromosome.get\lConfiguration" alt="" coords="303,122,505,163"/><area shape="rect" id="node9" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a743303c75d2a001f88af579051862964" title="org.jgap.IChromosome.get\lFitnessValueDirectly" alt="" coords="316,187,492,229"/><area shape="rect" id="node11" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207" title="org.jgap.BaseChromosome.getGene" alt="" coords="287,253,521,280"/><area shape="rect" id="node13" href="interfaceorg_1_1jgap_1_1_fitness_evaluator.html#aad9309b75a437b32641480be173550eb" title="org.jgap.FitnessEvaluator.is\lFitter" alt="" coords="312,305,496,346"/><area shape="rect" id="node15" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a9b460ad665ea1ab4e789061233fc6fa3" title="org.jgap.IChromosome.size" alt="" coords="313,371,495,397"/><area shape="rect" id="node17" href="classorg_1_1jgap_1_1_base_chromosome.html#a78c8ae0471973855e19a9b79496f2bec" title="org.jgap.BaseChromosome.size" alt="" coords="300,421,508,448"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ac75a21fd23483ff275422e0791289ac8"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Chromosome.equals </td>
          <td>(</td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>other</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Compares this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> against the specified object. The result is true if and the argument is an instance of the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> class and has a set of genes equal to this one.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">other</td><td>the object to compare against </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>true: if the objects are the same, false otherwise</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00751">751</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00816">org.jgap.Chromosome.compareTo()</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_test_8java_source.html#l00731">org.jgap.ChromosomeTest.testEquals_0()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00748">org.jgap.ChromosomeTest.testEquals_1()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00768">org.jgap.ChromosomeTest.testEquals_2()</a>, <a class="el" href="_chromosome_test_8java_source.html#l00802">org.jgap.ChromosomeTest.testEquals_3()</a>, <a class="el" href="_x_m_l_manager_test_8java_source.html#l00096">org.jgap.xml.XMLManagerTest.testGetChromosomeFromDocument_1()</a>, and <a class="el" href="_population_test_8java_source.html#l00126">org.jgap.PopulationTest.testSetChromosome_1()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_ac75a21fd23483ff275422e0791289ac8_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_ac75a21fd23483ff275422e0791289ac8_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_ac75a21fd23483ff275422e0791289ac8_cgraph" id="classorg_1_1jgap_1_1_chromosome_ac75a21fd23483ff275422e0791289ac8_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_chromosome.html#a2be0507944e2ca98bef8372402b33e13" title="org.jgap.Chromosome.compareTo" alt="" coords="245,224,464,251"/><area shape="rect" id="node5" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a1734f351c78c6d36bec48fdc8a334902" title="org.jgap.IChromosome.get\lApplicationData" alt="" coords="556,5,732,46"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_chromosome.html#aae49bf34668621de1d691cd759ff50b9" title="org.jgap.Chromosome.getApplicationData" alt="" coords="512,71,776,97"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_base_chromosome.html#a104b33f29941ba6ba405889964fc2bf5" title="org.jgap.BaseChromosome.get\lConfiguration" alt="" coords="543,122,745,163"/><area shape="rect" id="node11" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a743303c75d2a001f88af579051862964" title="org.jgap.IChromosome.get\lFitnessValueDirectly" alt="" coords="556,187,732,229"/><area shape="rect" id="node13" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207" title="org.jgap.BaseChromosome.getGene" alt="" coords="527,253,761,280"/><area shape="rect" id="node15" href="interfaceorg_1_1jgap_1_1_fitness_evaluator.html#aad9309b75a437b32641480be173550eb" title="org.jgap.FitnessEvaluator.is\lFitter" alt="" coords="552,305,736,346"/><area shape="rect" id="node17" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a9b460ad665ea1ab4e789061233fc6fa3" title="org.jgap.IChromosome.size" alt="" coords="553,371,735,397"/><area shape="rect" id="node19" href="classorg_1_1jgap_1_1_base_chromosome.html#a78c8ae0471973855e19a9b79496f2bec" title="org.jgap.BaseChromosome.size" alt="" coords="540,421,748,448"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="aae49bf34668621de1d691cd759ff50b9"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.Chromosome.getApplicationData </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves the application-specific data that is attached to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. Attaching application-specific data may be useful for some applications when it comes time to evaluate this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in the fitness function. JGAP ignores this data functionally.</p>
<dl class="section return"><dt>Returns</dt><dd>the application-specific data previously attached to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>, or null if there is no data attached</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.1 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a1734f351c78c6d36bec48fdc8a334902">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00993">993</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00077">org.jgap.Chromosome.m_applicationData</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00334">org.jgap.Chromosome.clone()</a>, <a class="el" href="_chromosome_8java_source.html#l00816">org.jgap.Chromosome.compareTo()</a>, and <a class="el" href="_chromosome_8java_source.html#l00614">org.jgap.Chromosome.toString()</a>.</p>

</div>
</div>
<a class="anchor" id="a6c56caed7a052c678daa6af668cad388"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a> org.jgap.Chromosome.getConstraintChecker </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd><a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a> the constraint checker to be used whenever method <a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a96692e7942a035e9b661360ca758c7cb">setGenes(Gene[])</a> is called.</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.5 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l01059">1059</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00120">org.jgap.Chromosome.m_geneAlleleChecker</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00334">org.jgap.Chromosome.clone()</a>, <a class="el" href="_chromosome_8java_source.html#l01006">org.jgap.Chromosome.setGenes()</a>, <a class="el" href="_chromosome_test_8java_source.html#l01370">org.jgap.ChromosomeTest.testSetConstraintChecker_0()</a>, <a class="el" href="_chromosome_test_8java_source.html#l01389">org.jgap.ChromosomeTest.testSetConstraintChecker_1()</a>, and <a class="el" href="_chromosome_test_8java_source.html#l01412">org.jgap.ChromosomeTest.testSetConstraintChecker_2()</a>.</p>

</div>
</div>
<a class="anchor" id="a9ccc6c05aa9c785d1cbae40acc90c80c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double org.jgap.Chromosome.getFitnessValue </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves the fitness value of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>, as determined by the active fitness function. If a bulk fitness function is in use and has not yet assigned a fitness value to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>, then -1 is returned.</p>
<p>Attention: should not be called from <a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a9cedcb4abe7b9e5f5cfd6cfea0cd949d">toString()</a> as the fitness value would be computed if it was initial!</p>
<dl class="section return"><dt>Returns</dt><dd>a positive double value representing the fitness of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>, or -1 if a bulk fitness function is in use and has not yet assigned a fitness value to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a></dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a9b68991c832a2054176d522e49a19ebd">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00534">534</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00560">org.jgap.Chromosome.calcFitnessValue()</a>, <a class="el" href="_chromosome_8java_source.html#l00129">org.jgap.Chromosome.m_alwaysCalculate</a>, and <a class="el" href="_chromosome_8java_source.html#l00102">org.jgap.Chromosome.m_fitnessValue</a>.</p>

<p>Referenced by <a class="el" href="_bulk_fitness_offset_remover_test_8java_source.html#l00229">org.jgap.impl.BulkFitnessOffsetRemoverTest.testEvaluate_3()</a>, and <a class="el" href="_bulk_fitness_offset_remover_test_8java_source.html#l00284">org.jgap.impl.BulkFitnessOffsetRemoverTest.testGetAbsoluteFitness_0()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_a9ccc6c05aa9c785d1cbae40acc90c80c_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_a9ccc6c05aa9c785d1cbae40acc90c80c_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_a9ccc6c05aa9c785d1cbae40acc90c80c_cgraph" id="classorg_1_1jgap_1_1_chromosome_a9ccc6c05aa9c785d1cbae40acc90c80c_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_chromosome.html#aea063618aebe23aa8579baf20063db24" title="org.jgap.Chromosome.calc\lFitnessValue" alt="" coords="269,5,448,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_chromosome.html#a104b33f29941ba6ba405889964fc2bf5" title="org.jgap.BaseChromosome.get\lConfiguration" alt="" coords="496,5,699,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ad0c53486ddb583ddbee2bede065ae068"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double org.jgap.Chromosome.getFitnessValueDirectly </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>the lastly computed fitness value, or <a class="el" href="classorg_1_1jgap_1_1_fitness_function.html#a5f1f23e6f8a0f9606aee5d52fac6a06b">FitnessFunction.NO_FITNESS_VALUE</a> in case no value has been computed yet.</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a743303c75d2a001f88af579051862964">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00549">549</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00102">org.jgap.Chromosome.m_fitnessValue</a>.</p>

</div>
</div>
<a class="anchor" id="a6129d035292cd1d271b5ef5dbc3e3161"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">List org.jgap.Chromosome.getMultiObjectives </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l01121">1121</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00086">org.jgap.Chromosome.m_multiObjective</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_test_8java_source.html#l01590">org.jgap.ChromosomeTest.testSetMultiObjectives_0()</a>.</p>

</div>
</div>
<a class="anchor" id="a9f3e1fc14f84b25f09386a2dd1fc69d9"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.Chromosome.hashCode </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieve a hash code for this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. Does not considers the order of the Genes for all cases (especially when gene is empty).</p>
<dl class="section return"><dt>Returns</dt><dd>the hash code of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a></dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00780">780</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00283">org.jgap.BaseChromosome.getGene()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00252">org.jgap.BaseChromosome.getGenes()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00302">org.jgap.BaseChromosome.size()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_a9f3e1fc14f84b25f09386a2dd1fc69d9_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_a9f3e1fc14f84b25f09386a2dd1fc69d9_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_a9f3e1fc14f84b25f09386a2dd1fc69d9_cgraph" id="classorg_1_1jgap_1_1_chromosome_a9f3e1fc14f84b25f09386a2dd1fc69d9_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207" title="org.jgap.BaseChromosome.getGene" alt="" coords="267,5,501,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_chromosome.html#a1f9e4971a34546a0d77fa24061e753a0" title="org.jgap.BaseChromosome.get\lGenes" alt="" coords="283,57,485,98"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_chromosome.html#a78c8ae0471973855e19a9b79496f2bec" title="org.jgap.BaseChromosome.size" alt="" coords="280,123,488,149"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="aa6b7f021f64700b1926d5819f5fde5aa"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Chromosome.initFromGene </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>&#160;</td>
          <td class="paramname"><em>a_sampleGene</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00235">235</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#ae4ddccb7bf569c0ed6cdc19def590964">org.jgap.Gene.newGene()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00287">org.jgap.BaseChromosome.setGene()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00302">org.jgap.BaseChromosome.size()</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00219">org.jgap.Chromosome.Chromosome()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_aa6b7f021f64700b1926d5819f5fde5aa_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_aa6b7f021f64700b1926d5819f5fde5aa_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_aa6b7f021f64700b1926d5819f5fde5aa_cgraph" id="classorg_1_1jgap_1_1_chromosome_aa6b7f021f64700b1926d5819f5fde5aa_cgraph">
<area shape="rect" id="node3" href="interfaceorg_1_1jgap_1_1_gene.html#ae4ddccb7bf569c0ed6cdc19def590964" title="org.jgap.Gene.newGene" alt="" coords="260,5,423,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_chromosome.html#a8177b71d092d3ce87909d827ec91492a" title="org.jgap.BaseChromosome.setGene" alt="" coords="225,56,458,83"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_chromosome.html#a78c8ae0471973855e19a9b79496f2bec" title="org.jgap.BaseChromosome.size" alt="" coords="237,107,445,133"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a284d75bb2919718325b376e2c91870d0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Chromosome.isCompareApplicationData </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l01032">1032</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00111">org.jgap.Chromosome.m_compareAppData</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_test_8java_source.html#l01284">org.jgap.ChromosomeTest.testSetCompareApplicationData_0()</a>.</p>

</div>
</div>
<a class="anchor" id="a45bc47533c183ecf3297d7507299cc85"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Chromosome.isHandlerFor </td>
          <td>(</td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>a_obj</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Class&#160;</td>
          <td class="paramname"><em>a_class</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Determines whether the handler is suitable for the given object instance or class.<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_obj</td><td>the object instance to check </td></tr>
    <tr><td class="paramname">a_class</td><td>the class to check, alternatively if no object instance is available (some handlers could need to get an object and would always return false if only a class is provided) </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>true: handler suitable for given object or class</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6</dd></dl>
 </p>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_handler.html#a026d7a41944049a4f812ee0d1151d40e">org.jgap.IHandler</a>.</p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l01095">1095</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_test_8java_source.html#l01574">org.jgap.ChromosomeTest.testIsHandlerFor_0()</a>.</p>

</div>
</div>
<a class="anchor" id="a61d28f792a1389214f06a88dba42260d"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Chromosome.isSelectedForNextGeneration </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves whether this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> has been selected by the natural selector to continue to the next generation.</p>
<dl class="section return"><dt>Returns</dt><dd>true if this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> has been selected, false otherwise</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#af4aa1487bfa13ba0f038b628f44554be">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00916">916</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00092">org.jgap.Chromosome.m_isSelectedForNextGeneration</a>.</p>

</div>
</div>
<a class="anchor" id="abee38007a07ba7a1442dd96571ed7cc1"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.Chromosome.perform </td>
          <td>(</td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>a_obj</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Class&#160;</td>
          <td class="paramname"><em>a_class</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>a_params</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws Exception</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Performs a task for the given object or class. For some handlers, additional parameters are necessary, which could be provided.<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_obj</td><td>the object instance to perform the handler task for </td></tr>
    <tr><td class="paramname">a_class</td><td>the class to perform the handler task for, in case no object instance is available (some handlers could need to get an object and would always throw an exception if only a class is provided) </td></tr>
    <tr><td class="paramname">a_params</td><td>optional parameters needed for the handler to perform its task </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Exception</td><td>in case of any error </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>Object</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6</dd></dl>
 </p>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_handler.html#a60199862000df04a4b2491d6d08f2f2c">org.jgap.IHandler</a>.</p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l01105">1105</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00150">org.jgap.BaseChromosome.getConfiguration()</a>, and <a class="el" href="_chromosome_8java_source.html#l00668">org.jgap.Chromosome.randomInitialChromosome()</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_test_8java_source.html#l01574">org.jgap.ChromosomeTest.testIsHandlerFor_0()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_abee38007a07ba7a1442dd96571ed7cc1_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_abee38007a07ba7a1442dd96571ed7cc1_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_abee38007a07ba7a1442dd96571ed7cc1_cgraph" id="classorg_1_1jgap_1_1_chromosome_abee38007a07ba7a1442dd96571ed7cc1_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_chromosome.html#a104b33f29941ba6ba405889964fc2bf5" title="org.jgap.BaseChromosome.get\lConfiguration" alt="" coords="779,5,981,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_chromosome.html#a6b6fb5c7c5b532813a4661de5dbd9ab3" title="org.jgap.Chromosome.random\lInitialChromosome" alt="" coords="253,34,451,75"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_chromosome.html#adf4d7c5113d0c70283ffb2323b7e0cda" title="org.jgap.Chromosome.clone\lObject" alt="" coords="522,34,707,75"/><area shape="rect" id="node10" href="classorg_1_1jgap_1_1_chromosome.html#a14c96eb2e38fe28b05526633e4677ec6" title="org.jgap.Chromosome.Chromosome" alt="" coords="499,100,731,127"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a6b6fb5c7c5b532813a4661de5dbd9ab3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">static <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> org.jgap.Chromosome.randomInitialChromosome </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Convenience method that returns a new <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> instance with its genes values (alleles) randomized. Note that, if possible, this method will acquire a <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> instance from the active ChromosomePool (if any) and then randomize its gene values before returning it. If a <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> cannot be acquired from the pool, then a new instance will be constructed and its gene values randomized before returning it.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the configuration to use </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>randomly initialized <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td>if the given <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> instance is invalid </td></tr>
    <tr><td class="paramname">IllegalArgumentException</td><td>if the given <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> instance is null</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000050">Todo:</a></b></dt><dd>what about <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>'s energy? </dd></dl>
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000051">Todo:</a></b></dt><dd>what about <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>'s energy? </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00668">668</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00141">org.jgap.Chromosome.Chromosome()</a>, <a class="el" href="_chromosome_8java_source.html#l00478">org.jgap.Chromosome.cloneObject()</a>, and <a class="el" href="_fitness_function_8java_source.html#l00040">org.jgap.FitnessFunction.NO_FITNESS_VALUE</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l01105">org.jgap.Chromosome.perform()</a>, and <a class="el" href="_chromosome_test_8java_source.html#l01284">org.jgap.ChromosomeTest.testSetCompareApplicationData_0()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_a6b6fb5c7c5b532813a4661de5dbd9ab3_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_a6b6fb5c7c5b532813a4661de5dbd9ab3_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_a6b6fb5c7c5b532813a4661de5dbd9ab3_cgraph" id="classorg_1_1jgap_1_1_chromosome_a6b6fb5c7c5b532813a4661de5dbd9ab3_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_chromosome.html#a14c96eb2e38fe28b05526633e4677ec6" title="org.jgap.Chromosome.Chromosome" alt="" coords="251,5,483,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_chromosome.html#adf4d7c5113d0c70283ffb2323b7e0cda" title="org.jgap.Chromosome.clone\lObject" alt="" coords="274,57,459,98"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_chromosome.html#a104b33f29941ba6ba405889964fc2bf5" title="org.jgap.BaseChromosome.get\lConfiguration" alt="" coords="531,57,733,98"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a03e0d17b0df1765a442f76645aa4e5b8"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Chromosome.setApplicationData </td>
          <td>(</td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>a_newData</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>This sets the application-specific data that is attached to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. Attaching application-specific data may be useful for some applications when it comes time to evaluate this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in the fitness function. JGAP ignores this data, except for cloning and comparison (latter only if opted in via setCompareApplicationData(..))</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_newData</td><td>the new application-specific data to attach to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. Should be an instance of <a class="el" href="interfaceorg_1_1jgap_1_1_i_application_data.html">IApplicationData</a></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.1 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a99da03ae9bca8f2ca884186aea82c771">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00977">977</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00077">org.jgap.Chromosome.m_applicationData</a>.</p>

</div>
</div>
<a class="anchor" id="aa82fc94cbbfeb5b6b5b89a97f9089e9e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Chromosome.setCompareApplicationData </td>
          <td>(</td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_doCompare</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Should we also consider the application data when comparing? Default is "false" as "true" means a <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> is losing its identity when application data is set differently!</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_doCompare</td><td>true: consider application data in method compareTo</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.2 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l01022">1022</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00111">org.jgap.Chromosome.m_compareAppData</a>.</p>

</div>
</div>
<a class="anchor" id="ae99251567d68bd079e8c4dfaf15dc513"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Chromosome.setConstraintChecker </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a>&#160;</td>
          <td class="paramname"><em>a_constraintChecker</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets the constraint checker to be used for this gene whenever method setAllele(Object) is called.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_constraintChecker</td><td>the constraint checker to be set </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.5 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a05c61ae9fb2caf56455eb66405a164c4">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l01046">1046</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00120">org.jgap.Chromosome.m_geneAlleleChecker</a>, and <a class="el" href="_chromosome_8java_source.html#l01073">org.jgap.Chromosome.verify()</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00226">org.jgap.Chromosome.Chromosome()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_ae99251567d68bd079e8c4dfaf15dc513_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_ae99251567d68bd079e8c4dfaf15dc513_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_ae99251567d68bd079e8c4dfaf15dc513_cgraph" id="classorg_1_1jgap_1_1_chromosome_ae99251567d68bd079e8c4dfaf15dc513_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_chromosome.html#af472c1bacfaf6e5e5c1d5ac3af75ddef" title="org.jgap.Chromosome.verify" alt="" coords="285,41,472,68"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_chromosome.html#a1f9e4971a34546a0d77fa24061e753a0" title="org.jgap.BaseChromosome.get\lGenes" alt="" coords="536,5,739,46"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207" title="org.jgap.BaseChromosome.getGene" alt="" coords="521,71,754,97"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a3715ce380f59756a954566a315514ed0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Chromosome.setFitnessValue </td>
          <td>(</td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>a_newFitnessValue</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets the fitness value of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. This method is for use by bulk fitness functions and should not be invoked from anything else (except test cases).</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_newFitnessValue</td><td>a positive integer representing the fitness of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a56f82d43b102fc5da8e0a071e32dc13d">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00585">585</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00102">org.jgap.Chromosome.m_fitnessValue</a>.</p>

</div>
</div>
<a class="anchor" id="a4fb0934f7919084f0b5e9ca132e6aaa6"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Chromosome.setFitnessValueDirectly </td>
          <td>(</td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>a_newFitnessValue</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets the fitness value of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> directly without any constraint checks, conversions or checks. Only use if you know what you do.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_newFitnessValue</td><td>a positive integer representing the fitness of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#ac2acf33157b233e8c5553fc62861fa23">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00602">602</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00102">org.jgap.Chromosome.m_fitnessValue</a>.</p>

</div>
</div>
<a class="anchor" id="a96692e7942a035e9b661360ca758c7cb"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Chromosome.setGenes </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[]&#160;</td>
          <td class="paramname"><em>a_genes</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets the genes for the chromosome. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_genes</td><td>the genes to set for the chromosome</td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td>in case constraint checker is provided</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a0ff42f1935dab73296301b29ad0a1716">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l01006">1006</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l01059">org.jgap.Chromosome.getConstraintChecker()</a>, and <a class="el" href="_chromosome_8java_source.html#l01073">org.jgap.Chromosome.verify()</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00190">org.jgap.Chromosome.Chromosome()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_a96692e7942a035e9b661360ca758c7cb_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_a96692e7942a035e9b661360ca758c7cb_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_a96692e7942a035e9b661360ca758c7cb_cgraph" id="classorg_1_1jgap_1_1_chromosome_a96692e7942a035e9b661360ca758c7cb_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_chromosome.html#a6c56caed7a052c678daa6af668cad388" title="org.jgap.Chromosome.getConstraint\lChecker" alt="" coords="264,5,496,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_chromosome.html#af472c1bacfaf6e5e5c1d5ac3af75ddef" title="org.jgap.Chromosome.verify" alt="" coords="287,71,473,97"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_chromosome.html#a1f9e4971a34546a0d77fa24061e753a0" title="org.jgap.BaseChromosome.get\lGenes" alt="" coords="560,34,763,75"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207" title="org.jgap.BaseChromosome.getGene" alt="" coords="545,100,778,127"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a1249d045c9830c9dc76e14e4dbe20f5f"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Chromosome.setIsSelectedForNextGeneration </td>
          <td>(</td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_isSelected</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets whether this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> has been selected by the natural selector to continue to the next generation or manually (e.g. via an add-method).</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_isSelected</td><td>true if this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> has been selected, false otherwise</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#afeeed3697d5f22154cab618432602ca6">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00903">903</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00092">org.jgap.Chromosome.m_isSelectedForNextGeneration</a>.</p>

</div>
</div>
<a class="anchor" id="aaeae2078063f38074853b06d5c913fa6"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Chromosome.setMultiObjectives </td>
          <td>(</td>
          <td class="paramtype">List&#160;</td>
          <td class="paramname"><em>a_values</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l01113">1113</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00086">org.jgap.Chromosome.m_multiObjective</a>.</p>

</div>
</div>
<a class="anchor" id="a9cedcb4abe7b9e5f5cfd6cfea0cd949d"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.Chromosome.toString </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>a string representation of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>, useful for display purposes</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00614">614</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_chromosome_8java_source.html#l00993">org.jgap.Chromosome.getApplicationData()</a>, <a class="el" href="_base_chromosome_8java_source.html#l00283">org.jgap.BaseChromosome.getGene()</a>, <a class="el" href="_chromosome_8java_source.html#l00102">org.jgap.Chromosome.m_fitnessValue</a>, <a class="el" href="_i_chromosome_8java_source.html#l00033">org.jgap.IChromosome.S_ALLELES</a>, <a class="el" href="_i_chromosome_8java_source.html#l00035">org.jgap.IChromosome.S_APPLICATION_DATA</a>, <a class="el" href="_i_chromosome_8java_source.html#l00031">org.jgap.IChromosome.S_FITNESS_VALUE</a>, <a class="el" href="_i_chromosome_8java_source.html#l00037">org.jgap.IChromosome.S_SIZE</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00302">org.jgap.BaseChromosome.size()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_a9cedcb4abe7b9e5f5cfd6cfea0cd949d_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_a9cedcb4abe7b9e5f5cfd6cfea0cd949d_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_a9cedcb4abe7b9e5f5cfd6cfea0cd949d_cgraph" id="classorg_1_1jgap_1_1_chromosome_a9cedcb4abe7b9e5f5cfd6cfea0cd949d_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_chromosome.html#aae49bf34668621de1d691cd759ff50b9" title="org.jgap.Chromosome.getApplicationData" alt="" coords="253,5,517,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207" title="org.jgap.BaseChromosome.getGene" alt="" coords="269,56,502,83"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_chromosome.html#a78c8ae0471973855e19a9b79496f2bec" title="org.jgap.BaseChromosome.size" alt="" coords="281,107,489,133"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="af472c1bacfaf6e5e5c1d5ac3af75ddef"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Chromosome.verify </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a>&#160;</td>
          <td class="paramname"><em>a_constraintChecker</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Verifies the state of the chromosome. Especially takes care of the given constraint checker. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_constraintChecker</td><td>the constraint checker to verify</td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.5 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l01073">1073</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>References <a class="el" href="_base_chromosome_8java_source.html#l00283">org.jgap.BaseChromosome.getGene()</a>, and <a class="el" href="_base_chromosome_8java_source.html#l00252">org.jgap.BaseChromosome.getGenes()</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l01046">org.jgap.Chromosome.setConstraintChecker()</a>, and <a class="el" href="_chromosome_8java_source.html#l01006">org.jgap.Chromosome.setGenes()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_chromosome_af472c1bacfaf6e5e5c1d5ac3af75ddef_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_chromosome_af472c1bacfaf6e5e5c1d5ac3af75ddef_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_chromosome_af472c1bacfaf6e5e5c1d5ac3af75ddef_cgraph" id="classorg_1_1jgap_1_1_chromosome_af472c1bacfaf6e5e5c1d5ac3af75ddef_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_chromosome.html#aef0e40ec9b5956389365f5c3144ab207" title="org.jgap.BaseChromosome.getGene" alt="" coords="241,5,474,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_chromosome.html#a1f9e4971a34546a0d77fa24061e753a0" title="org.jgap.BaseChromosome.get\lGenes" alt="" coords="256,57,459,98"/></map>
</div>
</p>

</div>
</div>
<h2 class="groupheader">Member Data Documentation</h2>
<a class="anchor" id="aa8e104e9bc3f5f2bfdfe67b604c10d9c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final String org.jgap.Chromosome.CVS_REVISION = &quot;$Revision: 1.105 $&quot;</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>String containing the CVS revision. Read out via reflection! </p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00068">68</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

</div>
</div>
<a class="anchor" id="a678ed92a2a1a47343638de24f40b66e3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Chromosome.m_alwaysCalculate</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>If set to true the method <a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a9ccc6c05aa9c785d1cbae40acc90c80c">getFitnessValue()</a> will always (re-)calculate the fitness value. This may be necessary in case of environments where the state changes without the chromosome to notice</p>
<dl class="section since"><dt>Since</dt><dd>3.2.2 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00129">129</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00155">org.jgap.Chromosome.Chromosome()</a>, and <a class="el" href="_chromosome_8java_source.html#l00534">org.jgap.Chromosome.getFitnessValue()</a>.</p>

</div>
</div>
<a class="anchor" id="af2b4e1abcdf5995b45fb7d5cf93a559c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.Chromosome.m_applicationData</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Application-specific data that is attached to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. This data may assist the application in evaluating this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in the fitness function. JGAP does not operate on the data, aside from allowing it to be set and retrieved, and considering it with comparations (if user opted in to do so). </p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00077">77</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00993">org.jgap.Chromosome.getApplicationData()</a>, and <a class="el" href="_chromosome_8java_source.html#l00977">org.jgap.Chromosome.setApplicationData()</a>.</p>

</div>
</div>
<a class="anchor" id="a7c696f550a4f4458ed4115833abf02fb"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Chromosome.m_compareAppData</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Method <a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a2be0507944e2ca98bef8372402b33e13">compareTo()</a>: Should we also consider the application data when comparing? Default is "false", as "true" means a <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>'s losing its identity when application data is set differently!</p>
<dl class="section since"><dt>Since</dt><dd>2.2 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00111">111</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00816">org.jgap.Chromosome.compareTo()</a>, <a class="el" href="_chromosome_8java_source.html#l01032">org.jgap.Chromosome.isCompareApplicationData()</a>, and <a class="el" href="_chromosome_8java_source.html#l01022">org.jgap.Chromosome.setCompareApplicationData()</a>.</p>

</div>
</div>
<a class="anchor" id="a905236819a9904ca3d096fa1c325aa1b"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double org.jgap.Chromosome.m_fitnessValue = <a class="el" href="classorg_1_1jgap_1_1_fitness_function.html#a5f1f23e6f8a0f9606aee5d52fac6a06b">FitnessFunction.NO_FITNESS_VALUE</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Stores the fitness value of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> as determined by the active fitness function. A value of -1 indicates that this field has not yet been set with this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>'s fitness values (valid fitness values are always positive).</p>
<dl class="section since"><dt>Since</dt><dd>2.0 (until 1.1: type int) </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00102">102</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00560">org.jgap.Chromosome.calcFitnessValue()</a>, <a class="el" href="_chromosome_8java_source.html#l00928">org.jgap.Chromosome.cleanup()</a>, <a class="el" href="_chromosome_8java_source.html#l00334">org.jgap.Chromosome.clone()</a>, <a class="el" href="_chromosome_8java_source.html#l00816">org.jgap.Chromosome.compareTo()</a>, <a class="el" href="_chromosome_for_testing_8java_source.html#l00046">org.jgap.ChromosomeForTesting.getFitnessValue()</a>, <a class="el" href="_chromosome_8java_source.html#l00534">org.jgap.Chromosome.getFitnessValue()</a>, <a class="el" href="_chromosome_8java_source.html#l00549">org.jgap.Chromosome.getFitnessValueDirectly()</a>, <a class="el" href="_chromosome_8java_source.html#l00585">org.jgap.Chromosome.setFitnessValue()</a>, <a class="el" href="_chromosome_8java_source.html#l00602">org.jgap.Chromosome.setFitnessValueDirectly()</a>, and <a class="el" href="_chromosome_8java_source.html#l00614">org.jgap.Chromosome.toString()</a>.</p>

</div>
</div>
<a class="anchor" id="af890b1160c9b97defdf107e6a4e53329"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a> org.jgap.Chromosome.m_geneAlleleChecker</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Optional helper class for checking if a given allele value to be set for a given gene is valid. If not, the allele value may not be set for the gene or the gene type (e.g. IntegerGene) is not allowed in general!</p>
<dl class="section since"><dt>Since</dt><dd>2.5 </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00120">120</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l01059">org.jgap.Chromosome.getConstraintChecker()</a>, and <a class="el" href="_chromosome_8java_source.html#l01046">org.jgap.Chromosome.setConstraintChecker()</a>.</p>

</div>
</div>
<a class="anchor" id="ad8bb6734b245d112a8e6d66fb38914c9"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Chromosome.m_isSelectedForNextGeneration</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Keeps track of whether or not this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> has been selected by the natural selector to move on to the next generation. </p>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00092">92</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00928">org.jgap.Chromosome.cleanup()</a>, <a class="el" href="_chromosome_8java_source.html#l00916">org.jgap.Chromosome.isSelectedForNextGeneration()</a>, and <a class="el" href="_chromosome_8java_source.html#l00903">org.jgap.Chromosome.setIsSelectedForNextGeneration()</a>.</p>

</div>
</div>
<a class="anchor" id="a6d73c5e40fce0219e80021211d4a2d57"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">List org.jgap.Chromosome.m_multiObjective</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Holds multiobjective values.</p>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000048">Todo:</a></b></dt><dd>move to new subclass of <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> (and introduce new interface IMultiObjective with that) </dd></dl>

<p>Definition at line <a class="el" href="_chromosome_8java_source.html#l00086">86</a> of file <a class="el" href="_chromosome_8java_source.html">Chromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_8java_source.html#l00334">org.jgap.Chromosome.clone()</a>, <a class="el" href="_chromosome_8java_source.html#l01121">org.jgap.Chromosome.getMultiObjectives()</a>, and <a class="el" href="_chromosome_8java_source.html#l01113">org.jgap.Chromosome.setMultiObjectives()</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>LDH_Git/src/org/jgap/<a class="el" href="_chromosome_8java_source.html">Chromosome.java</a></li>
</ul>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Sat Nov 23 2013 17:42:11 for Práctica_5_LDH by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
