<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Práctica_5_LDH: org.jgap.Population Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">Práctica_5_LDH
   &#160;<span id="projectnumber">1.2</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li><a href="inherits.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Classes</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Namespaces</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Files</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Functions</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(6)"><span class="SelectionMark">&#160;</span>Pages</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="namespaceorg.html">org</a></li><li class="navelem"><a class="el" href="namespaceorg_1_1jgap.html">jgap</a></li><li class="navelem"><a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-static-attribs">Static Public Attributes</a> &#124;
<a href="#pro-methods">Protected Member Functions</a> &#124;
<a href="#pro-static-methods">Static Protected Member Functions</a> &#124;
<a href="#pri-methods">Private Member Functions</a> &#124;
<a href="#pri-attribs">Private Attributes</a> &#124;
<a href="#pri-static-attribs">Static Private Attributes</a> &#124;
<a href="classorg_1_1jgap_1_1_population-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">org.jgap.Population Class Reference</div>  </div>
</div><!--header-->
<div class="contents">
<div class="dynheader">
Inheritance diagram for org.jgap.Population:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population__inherit__graph.png" border="0" usemap="#org_8jgap_8_population_inherit__map" alt="Inheritance graph"/></div>
<map name="org_8jgap_8_population_inherit__map" id="org_8jgap_8_population_inherit__map">
<area shape="rect" id="node4" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html" title="org.jgap.util.ICloneable" alt="" coords="119,80,272,107"/><area shape="rect" id="node8" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html" title="org.jgap.IPersistentRepresentation" alt="" coords="297,80,519,107"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<div class="dynheader">
Collaboration diagram for org.jgap.Population:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population__coll__graph.png" border="0" usemap="#org_8jgap_8_population_coll__map" alt="Collaboration graph"/></div>
<map name="org_8jgap_8_population_coll__map" id="org_8jgap_8_population_coll__map">
<area shape="rect" id="node12" href="classorg_1_1jgap_1_1_configuration.html" title="org.jgap.Configuration" alt="" coords="811,437,960,463"/><area shape="rect" id="node20" href="interfaceorg_1_1jgap_1_1_i_breeder.html" title="org.jgap.IBreeder" alt="" coords="397,589,518,615"/><area shape="rect" id="node25" href="interfaceorg_1_1jgap_1_1_i_j_g_a_p_factory.html" title="org.jgap.IJGAPFactory" alt="" coords="380,117,535,143"/><area shape="rect" id="node30" href="interfaceorg_1_1jgap_1_1_fitness_evaluator.html" title="org.jgap.FitnessEvaluator" alt="" coords="372,167,543,194"/><area shape="rect" id="node33" href="interfaceorg_1_1jgap_1_1_i_chromosome.html" title="org.jgap.IChromosome" alt="" coords="380,690,535,717"/><area shape="rect" id="node40" href="interfaceorg_1_1jgap_1_1_random_generator.html" title="org.jgap.RandomGenerator" alt="" coords="368,269,547,295"/><area shape="rect" id="node43" href="classorg_1_1jgap_1_1_bulk_fitness_function.html" title="org.jgap.BulkFitnessFunction" alt="" coords="362,386,553,413"/><area shape="rect" id="node47" href="classorg_1_1jgap_1_1_configuration_1_1_configuration_configurable.html" title="org.jgap.Configuration.Configuration\lConfigurable" alt="" coords="342,320,573,361"/><area shape="rect" id="node50" href="classorg_1_1jgap_1_1_fitness_function.html" title="org.jgap.FitnessFunction" alt="" coords="375,487,540,514"/><area shape="rect" id="node4" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html" title="org.jgap.util.ICloneable" alt="" coords="136,555,291,582"/><area shape="rect" id="node8" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html" title="org.jgap.IPersistentRepresentation" alt="" coords="774,515,997,542"/><area shape="rect" id="node28" href="interfaceorg_1_1jgap_1_1_i_chromosome_pool.html" title="org.jgap.IChromosomePool" alt="" coords="367,437,547,463"/><area shape="rect" id="node38" href="interfaceorg_1_1jgap_1_1_i_unique_key.html" title="org.jgap.IUniqueKey" alt="" coords="143,774,283,801"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a525a931a8578edb1cda3fda3f34b21a8"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a525a931a8578edb1cda3fda3f34b21a8">Population</a> (final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_config)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a525a931a8578edb1cda3fda3f34b21a8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:acfba30f7cc1412d1c7eb57f11b454b4f"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#acfba30f7cc1412d1c7eb57f11b454b4f">Population</a> (final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_config, final <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>[] a_chromosomes)  throws InvalidConfigurationException </td></tr>
<tr class="separator:acfba30f7cc1412d1c7eb57f11b454b4f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afafcae85feefb2cf3f0e1bb04ee98af6"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#afafcae85feefb2cf3f0e1bb04ee98af6">Population</a> (final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_config, final <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> a_chromosome)  throws InvalidConfigurationException </td></tr>
<tr class="separator:afafcae85feefb2cf3f0e1bb04ee98af6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8a9a1e7790506e3e3ce43c55246c0331"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a8a9a1e7790506e3e3ce43c55246c0331">Population</a> (final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_config, final int a_size)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a8a9a1e7790506e3e3ce43c55246c0331"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a85b83c32f89f499a8ecd15b3293202ed"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a85b83c32f89f499a8ecd15b3293202ed">Population</a> ()  throws InvalidConfigurationException </td></tr>
<tr class="separator:a85b83c32f89f499a8ecd15b3293202ed"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a43613c7bc90d413e2a31bd459f49dfed"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a43613c7bc90d413e2a31bd459f49dfed">getConfiguration</a> ()</td></tr>
<tr class="separator:a43613c7bc90d413e2a31bd459f49dfed"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aab7e3575e2a5979ed22b081f2a7400d4"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#aab7e3575e2a5979ed22b081f2a7400d4">addChromosome</a> (final <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> a_toAdd)</td></tr>
<tr class="separator:aab7e3575e2a5979ed22b081f2a7400d4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a893fbff437dde3d07a4f38f889f5a32d"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a893fbff437dde3d07a4f38f889f5a32d">addChromosomes</a> (final <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> a_population)</td></tr>
<tr class="separator:a893fbff437dde3d07a4f38f889f5a32d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a86b38d39f48f6db65148cdbbd2b5d866"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a86b38d39f48f6db65148cdbbd2b5d866">setChromosomes</a> (final List a_chromosomes)</td></tr>
<tr class="separator:a86b38d39f48f6db65148cdbbd2b5d866"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aaa80c42ce70b695356b338b11d7b0364"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#aaa80c42ce70b695356b338b11d7b0364">setChromosome</a> (final int a_index, final <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> a_chromosome)</td></tr>
<tr class="separator:aaa80c42ce70b695356b338b11d7b0364"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa9deff3f371931e7151070af03465e83"><td class="memItemLeft" align="right" valign="top">List&lt; <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#aa9deff3f371931e7151070af03465e83">getChromosomes</a> ()</td></tr>
<tr class="separator:aa9deff3f371931e7151070af03465e83"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8e095445f8d0395d3676974349ad0c13"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a8e095445f8d0395d3676974349ad0c13">getChromosome</a> (final int a_index)</td></tr>
<tr class="separator:a8e095445f8d0395d3676974349ad0c13"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2d09529f8e93d57a0546625362ca5bee"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a2d09529f8e93d57a0546625362ca5bee">size</a> ()</td></tr>
<tr class="separator:a2d09529f8e93d57a0546625362ca5bee"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a644b52f341ebe005d1dc1c2e84f2591e"><td class="memItemLeft" align="right" valign="top">Iterator&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a644b52f341ebe005d1dc1c2e84f2591e">iterator</a> ()</td></tr>
<tr class="separator:a644b52f341ebe005d1dc1c2e84f2591e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aecabe6047fb1b1a7ec80d9b13fb4e7a0"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#aecabe6047fb1b1a7ec80d9b13fb4e7a0">toChromosomes</a> ()</td></tr>
<tr class="separator:aecabe6047fb1b1a7ec80d9b13fb4e7a0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab152be01c73200bf39e61518d0146379"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#ab152be01c73200bf39e61518d0146379">determineFittestChromosome</a> ()</td></tr>
<tr class="separator:ab152be01c73200bf39e61518d0146379"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a010cf1cbf6a1f771d34e5c3ce9420f25"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a010cf1cbf6a1f771d34e5c3ce9420f25">determineFittestChromosome</a> (int a_startIndex, int a_endIndex)</td></tr>
<tr class="separator:a010cf1cbf6a1f771d34e5c3ce9420f25"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab35d09d90bc1262d12bc22df23689df4"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#ab35d09d90bc1262d12bc22df23689df4">isChanged</a> ()</td></tr>
<tr class="separator:ab35d09d90bc1262d12bc22df23689df4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1c07a36de9973bee29421a44f99b8319"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a1c07a36de9973bee29421a44f99b8319">contains</a> (final <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> a_chromosome)</td></tr>
<tr class="separator:a1c07a36de9973bee29421a44f99b8319"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7dd9dbd2fecb4e2c4841787fd78ea64c"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a7dd9dbd2fecb4e2c4841787fd78ea64c">keepPopSizeConstant</a> ()  throws InvalidConfigurationException </td></tr>
<tr class="separator:a7dd9dbd2fecb4e2c4841787fd78ea64c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9ec112af7542cb12b4a891d259a31038"><td class="memItemLeft" align="right" valign="top">List&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a9ec112af7542cb12b4a891d259a31038">determineFittestChromosomes</a> (final int a_numberOfChromosomes)</td></tr>
<tr class="separator:a9ec112af7542cb12b4a891d259a31038"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac66aac96dfb67e94a3de2284cd9c9066"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#ac66aac96dfb67e94a3de2284cd9c9066">sortByFitness</a> ()</td></tr>
<tr class="separator:ac66aac96dfb67e94a3de2284cd9c9066"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1236240ac1bfdcd7848a268f099ede8e"><td class="memItemLeft" align="right" valign="top">List&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a1236240ac1bfdcd7848a268f099ede8e">getGenome</a> (final boolean a_resolveCompositeGenes)</td></tr>
<tr class="separator:a1236240ac1bfdcd7848a268f099ede8e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a959e9940968abe5de202542a7a26c990"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a959e9940968abe5de202542a7a26c990">isSorted</a> ()</td></tr>
<tr class="separator:a959e9940968abe5de202542a7a26c990"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab415c85cc397ce6d54924c4d14aa8cf6"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#ab415c85cc397ce6d54924c4d14aa8cf6">equals</a> (Object a_pop)</td></tr>
<tr class="separator:ab415c85cc397ce6d54924c4d14aa8cf6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7671499966b4ffb937d45e373c6e95d9"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a7671499966b4ffb937d45e373c6e95d9">compareTo</a> (Object a_pop)</td></tr>
<tr class="separator:a7671499966b4ffb937d45e373c6e95d9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5c3b2ec3c218d3699dd4d452a2449748"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a5c3b2ec3c218d3699dd4d452a2449748">clone</a> ()</td></tr>
<tr class="separator:a5c3b2ec3c218d3699dd4d452a2449748"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4e4c5b7e98055340e9f0b073be13d24b"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a4e4c5b7e98055340e9f0b073be13d24b">clear</a> ()</td></tr>
<tr class="separator:a4e4c5b7e98055340e9f0b073be13d24b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afeedd380ea603cdefb5ef871f2bdf99e"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#afeedd380ea603cdefb5ef871f2bdf99e">getPersistentRepresentation</a> ()</td></tr>
<tr class="separator:afeedd380ea603cdefb5ef871f2bdf99e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5cb0aa594a5d3d089ea0cadbe9a002e8"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a5cb0aa594a5d3d089ea0cadbe9a002e8">setValueFromPersistentRepresentation</a> (String a_representation)  throws UnsupportedRepresentationException </td></tr>
<tr class="separator:a5cb0aa594a5d3d089ea0cadbe9a002e8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad6e08d20ae2bfc6d277c84f5309cac49"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#ad6e08d20ae2bfc6d277c84f5309cac49">hashCode</a> ()</td></tr>
<tr class="separator:ad6e08d20ae2bfc6d277c84f5309cac49"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-static-attribs"></a>
Static Public Attributes</h2></td></tr>
<tr class="memitem:a8fd43176ca6c601491bb9cf2674d9c42"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a8fd43176ca6c601491bb9cf2674d9c42">CHROM_DELIMITER</a> = &quot;~&quot;</td></tr>
<tr class="separator:a8fd43176ca6c601491bb9cf2674d9c42"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af6282eaa6a1d7a0218d85db1b36960a4"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#af6282eaa6a1d7a0218d85db1b36960a4">CHROM_DELIMITER_HEADING</a> = &quot;[&quot;</td></tr>
<tr class="separator:af6282eaa6a1d7a0218d85db1b36960a4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a35f548ce2820e079590565c66bae85aa"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a35f548ce2820e079590565c66bae85aa">CHROM_DELIMITER_CLOSING</a> = &quot;]&quot;</td></tr>
<tr class="separator:a35f548ce2820e079590565c66bae85aa"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html">org.jgap.util.ICloneable</a></td></tr>
<tr class="memitem:a4eaaa85432026f5bc54c3e520b22caa7 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html#a4eaaa85432026f5bc54c3e520b22caa7">CVS_REVISION</a> = &quot;$Revision: 1.3 $&quot;</td></tr>
<tr class="separator:a4eaaa85432026f5bc54c3e520b22caa7 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_persistent_representation"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_persistent_representation')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html">org.jgap.IPersistentRepresentation</a></td></tr>
<tr class="memitem:ae48c5895f4af079337ffebe487b176f5 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_persistent_representation"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html#ae48c5895f4af079337ffebe487b176f5">CVS_REVISION</a> = &quot;$Revision: 1.2 $&quot;</td></tr>
<tr class="separator:ae48c5895f4af079337ffebe487b176f5 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_persistent_representation"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pro-methods"></a>
Protected Member Functions</h2></td></tr>
<tr class="memitem:adf75c170902a54737483e3e845bcfacc"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#adf75c170902a54737483e3e845bcfacc">setChanged</a> (final boolean a_changed)</td></tr>
<tr class="separator:adf75c170902a54737483e3e845bcfacc"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a252f05034adae08bf72577e853adee88"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a252f05034adae08bf72577e853adee88">setSorted</a> (final boolean a_sorted)</td></tr>
<tr class="separator:a252f05034adae08bf72577e853adee88"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae97e76f0bf965b1741a1f98efe05b482"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#ae97e76f0bf965b1741a1f98efe05b482">sort</a> (Comparator a_comparator)</td></tr>
<tr class="separator:ae97e76f0bf965b1741a1f98efe05b482"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a62348bfaa285efe11845574496d7a14c"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a62348bfaa285efe11845574496d7a14c">createChromosome</a> (String a_chromClassName, String a_persistentRepresentation)  throws Exception </td></tr>
<tr class="separator:a62348bfaa285efe11845574496d7a14c"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pro-static-methods"></a>
Static Protected Member Functions</h2></td></tr>
<tr class="memitem:a8379b53e6b976e681ae6d57c3db6a4ef"><td class="memItemLeft" align="right" valign="top">static final List&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a8379b53e6b976e681ae6d57c3db6a4ef">split</a> (String a_string)  throws UnsupportedRepresentationException </td></tr>
<tr class="separator:a8379b53e6b976e681ae6d57c3db6a4ef"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-methods"></a>
Private Member Functions</h2></td></tr>
<tr class="memitem:a64e81fff98bc39fcc05071b304b3c167"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a64e81fff98bc39fcc05071b304b3c167">addCompositeGene</a> (final List a_result, final <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> a_gene)</td></tr>
<tr class="separator:a64e81fff98bc39fcc05071b304b3c167"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a18ae527f4110c5d0c525e2a7f61c0553"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a18ae527f4110c5d0c525e2a7f61c0553">addAtomicGene</a> (final List a_result, final <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> a_gene)</td></tr>
<tr class="separator:a18ae527f4110c5d0c525e2a7f61c0553"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-attribs"></a>
Private Attributes</h2></td></tr>
<tr class="memitem:abc9fbe0eb7dc8ea169782159ca69ad70"><td class="memItemLeft" align="right" valign="top">List&lt; <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#abc9fbe0eb7dc8ea169782159ca69ad70">m_chromosomes</a></td></tr>
<tr class="separator:abc9fbe0eb7dc8ea169782159ca69ad70"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a58811aa95b2363c1078d54320fe6a252"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a58811aa95b2363c1078d54320fe6a252">m_fittestChromosome</a></td></tr>
<tr class="separator:a58811aa95b2363c1078d54320fe6a252"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ade9e10d21203f65a3cf4c6385fd4eb76"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#ade9e10d21203f65a3cf4c6385fd4eb76">m_changed</a></td></tr>
<tr class="separator:ade9e10d21203f65a3cf4c6385fd4eb76"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abb65c9029b8b947f25cbad4d2a5e855e"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#abb65c9029b8b947f25cbad4d2a5e855e">m_sorted</a></td></tr>
<tr class="separator:abb65c9029b8b947f25cbad4d2a5e855e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a84643e202f73135b3fc3a31b19ed29e2"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a84643e202f73135b3fc3a31b19ed29e2">m_config</a></td></tr>
<tr class="separator:a84643e202f73135b3fc3a31b19ed29e2"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-static-attribs"></a>
Static Private Attributes</h2></td></tr>
<tr class="memitem:a66404ad18ee170cbd1d4423aa0acec14"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_population.html#a66404ad18ee170cbd1d4423aa0acec14">CVS_REVISION</a> = &quot;$Revision: 1.66 $&quot;</td></tr>
<tr class="separator:a66404ad18ee170cbd1d4423aa0acec14"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Makes up the population of a generation during evolution. Represented by a list of chromosomes held in the <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a>.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00026">26</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="a525a931a8578edb1cda3fda3f34b21a8"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Population.Population </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_config</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_population_8java_source.html#l00073">73</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

</div>
</div>
<a class="anchor" id="acfba30f7cc1412d1c7eb57f11b454b4f"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Population.Population </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_config</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>[]&#160;</td>
          <td class="paramname"><em>a_chromosomes</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_population_8java_source.html#l00087">87</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00034">org.jgap.Population.m_chromosomes</a>, and <a class="el" href="_population_8java_source.html#l00371">org.jgap.Population.setChanged()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_acfba30f7cc1412d1c7eb57f11b454b4f_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_acfba30f7cc1412d1c7eb57f11b454b4f_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_acfba30f7cc1412d1c7eb57f11b454b4f_cgraph" id="classorg_1_1jgap_1_1_population_acfba30f7cc1412d1c7eb57f11b454b4f_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#adf75c170902a54737483e3e845bcfacc" title="org.jgap.Population.setChanged" alt="" coords="254,5,461,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#a252f05034adae08bf72577e853adee88" title="org.jgap.Population.setSorted" alt="" coords="510,5,703,32"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="afafcae85feefb2cf3f0e1bb04ee98af6"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Population.Population </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_config</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td>
          <td class="paramname"><em>a_chromosome</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_population_8java_source.html#l00109">109</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00034">org.jgap.Population.m_chromosomes</a>, and <a class="el" href="_population_8java_source.html#l00371">org.jgap.Population.setChanged()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_afafcae85feefb2cf3f0e1bb04ee98af6_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_afafcae85feefb2cf3f0e1bb04ee98af6_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_afafcae85feefb2cf3f0e1bb04ee98af6_cgraph" id="classorg_1_1jgap_1_1_population_afafcae85feefb2cf3f0e1bb04ee98af6_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#adf75c170902a54737483e3e845bcfacc" title="org.jgap.Population.setChanged" alt="" coords="254,5,461,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#a252f05034adae08bf72577e853adee88" title="org.jgap.Population.setSorted" alt="" coords="510,5,703,32"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a8a9a1e7790506e3e3ce43c55246c0331"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Population.Population </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_config</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final int&#160;</td>
          <td class="paramname"><em>a_size</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_population_8java_source.html#l00131">131</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00034">org.jgap.Population.m_chromosomes</a>, <a class="el" href="_population_8java_source.html#l00052">org.jgap.Population.m_config</a>, and <a class="el" href="_population_8java_source.html#l00371">org.jgap.Population.setChanged()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_a8a9a1e7790506e3e3ce43c55246c0331_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_a8a9a1e7790506e3e3ce43c55246c0331_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_a8a9a1e7790506e3e3ce43c55246c0331_cgraph" id="classorg_1_1jgap_1_1_population_a8a9a1e7790506e3e3ce43c55246c0331_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#adf75c170902a54737483e3e845bcfacc" title="org.jgap.Population.setChanged" alt="" coords="254,5,461,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#a252f05034adae08bf72577e853adee88" title="org.jgap.Population.setSorted" alt="" coords="510,5,703,32"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a85b83c32f89f499a8ecd15b3293202ed"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Population.Population </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_population_8java_source.html#l00148">148</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00659">org.jgap.Population.clone()</a>, and <a class="el" href="_population_8java_source.html#l00629">org.jgap.Population.compareTo()</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="a18ae527f4110c5d0c525e2a7f61c0553"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Population.addAtomicGene </td>
          <td>(</td>
          <td class="paramtype">final List&#160;</td>
          <td class="paramname"><em>a_result</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>&#160;</td>
          <td class="paramname"><em>a_gene</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Helper method for addCompositeGene.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_result</td><td>List </td></tr>
    <tr><td class="paramname">a_gene</td><td><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.3 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00588">588</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00567">org.jgap.Population.addCompositeGene()</a>, and <a class="el" href="_population_8java_source.html#l00536">org.jgap.Population.getGenome()</a>.</p>

</div>
</div>
<a class="anchor" id="aab7e3575e2a5979ed22b081f2a7400d4"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Population.addChromosome </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td>
          <td class="paramname"><em>a_toAdd</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Adds a <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> to this <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>. Does nothing when given null.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_toAdd</td><td>the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> to add</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00165">165</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00034">org.jgap.Population.m_chromosomes</a>, and <a class="el" href="_population_8java_source.html#l00371">org.jgap.Population.setChanged()</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00226">org.jgap.Population.setChromosome()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_aab7e3575e2a5979ed22b081f2a7400d4_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_aab7e3575e2a5979ed22b081f2a7400d4_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_aab7e3575e2a5979ed22b081f2a7400d4_cgraph" id="classorg_1_1jgap_1_1_population_aab7e3575e2a5979ed22b081f2a7400d4_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#adf75c170902a54737483e3e845bcfacc" title="org.jgap.Population.setChanged" alt="" coords="291,5,498,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#a252f05034adae08bf72577e853adee88" title="org.jgap.Population.setSorted" alt="" coords="547,5,741,32"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a893fbff437dde3d07a4f38f889f5a32d"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Population.addChromosomes </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>&#160;</td>
          <td class="paramname"><em>a_population</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Adds all the Chromosomes in the given <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>. Does nothing on null or an empty <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_population</td><td>the <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> to add</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00183">183</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00034">org.jgap.Population.m_chromosomes</a>, and <a class="el" href="_population_8java_source.html#l00371">org.jgap.Population.setChanged()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_a893fbff437dde3d07a4f38f889f5a32d_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_a893fbff437dde3d07a4f38f889f5a32d_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_a893fbff437dde3d07a4f38f889f5a32d_cgraph" id="classorg_1_1jgap_1_1_population_a893fbff437dde3d07a4f38f889f5a32d_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#adf75c170902a54737483e3e845bcfacc" title="org.jgap.Population.setChanged" alt="" coords="297,5,503,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#a252f05034adae08bf72577e853adee88" title="org.jgap.Population.setSorted" alt="" coords="553,5,746,32"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a64e81fff98bc39fcc05071b304b3c167"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Population.addCompositeGene </td>
          <td>(</td>
          <td class="paramtype">final List&#160;</td>
          <td class="paramname"><em>a_result</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>&#160;</td>
          <td class="paramname"><em>a_gene</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Adds all the genes of a CompositeGene to a result list.</p>
<p>Note: Method calls itself recursively.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_result</td><td>the list to add to </td></tr>
    <tr><td class="paramname">a_gene</td><td>the gene to start with</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.3 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00567">567</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00588">org.jgap.Population.addAtomicGene()</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00536">org.jgap.Population.getGenome()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_a64e81fff98bc39fcc05071b304b3c167_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_a64e81fff98bc39fcc05071b304b3c167_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_a64e81fff98bc39fcc05071b304b3c167_cgraph" id="classorg_1_1jgap_1_1_population_a64e81fff98bc39fcc05071b304b3c167_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#a18ae527f4110c5d0c525e2a7f61c0553" title="org.jgap.Population.addAtomicGene" alt="" coords="307,5,538,32"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a4e4c5b7e98055340e9f0b073be13d24b"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Population.clear </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Clears the list of chromosomes. Normally, this should not be necessary. But especially in distributed computing, a fresh population has to be provided sometimes.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00686">686</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00045">org.jgap.Population.m_changed</a>, <a class="el" href="_population_8java_source.html#l00039">org.jgap.Population.m_fittestChromosome</a>, and <a class="el" href="_population_8java_source.html#l00050">org.jgap.Population.m_sorted</a>.</p>

</div>
</div>
<a class="anchor" id="a5c3b2ec3c218d3699dd4d452a2449748"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.Population.clone </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>deeply cloned population instance</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html#aa7e7d62077e6428ad7904932b1b4f7d5">org.jgap.util.ICloneable</a>.</p>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00659">659</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html#aa7e7d62077e6428ad7904932b1b4f7d5">org.jgap.util.ICloneable.clone()</a>, <a class="el" href="_population_8java_source.html#l00034">org.jgap.Population.m_chromosomes</a>, <a class="el" href="_population_8java_source.html#l00052">org.jgap.Population.m_config</a>, <a class="el" href="_population_8java_source.html#l00039">org.jgap.Population.m_fittestChromosome</a>, <a class="el" href="_population_8java_source.html#l00148">org.jgap.Population.Population()</a>, and <a class="el" href="_population_8java_source.html#l00267">org.jgap.Population.size()</a>.</p>

<p>Referenced by <a class="el" href="_evolution_monitor_8java_source.html#l00164">org.jgap.audit.EvolutionMonitor.nextCycle()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_a5c3b2ec3c218d3699dd4d452a2449748_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_a5c3b2ec3c218d3699dd4d452a2449748_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_a5c3b2ec3c218d3699dd4d452a2449748_cgraph" id="classorg_1_1jgap_1_1_population_a5c3b2ec3c218d3699dd4d452a2449748_cgraph">
<area shape="rect" id="node3" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html#aa7e7d62077e6428ad7904932b1b4f7d5" title="org.jgap.util.ICloneable.clone" alt="" coords="226,5,417,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#a85b83c32f89f499a8ecd15b3293202ed" title="org.jgap.Population.Population" alt="" coords="222,56,421,83"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_population.html#a2d09529f8e93d57a0546625362ca5bee" title="org.jgap.Population.size" alt="" coords="241,107,402,133"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a7671499966b4ffb937d45e373c6e95d9"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.Population.compareTo </td>
          <td>(</td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>a_pop</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>This method is not producing symmetric results as -1 is more often returned than 1 (see description of return value).</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_pop</td><td>the other population to compare </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>1: a_pop is null or having fewer chromosomes or equal number of chromosomes but at least one not contained. 0: both populations containing exactly the same chromosomes. -1: this population contains fewer chromosomes than a_pop</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00629">629</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00034">org.jgap.Population.m_chromosomes</a>, <a class="el" href="_population_8java_source.html#l00148">org.jgap.Population.Population()</a>, and <a class="el" href="_population_8java_source.html#l00267">org.jgap.Population.size()</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00605">org.jgap.Population.equals()</a>, and <a class="el" href="_population_test_8java_source.html#l00841">org.jgap.PopulationTest.testCompareTo_0()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_a7671499966b4ffb937d45e373c6e95d9_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_a7671499966b4ffb937d45e373c6e95d9_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_a7671499966b4ffb937d45e373c6e95d9_cgraph" id="classorg_1_1jgap_1_1_population_a7671499966b4ffb937d45e373c6e95d9_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#a85b83c32f89f499a8ecd15b3293202ed" title="org.jgap.Population.Population" alt="" coords="257,5,455,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#a2d09529f8e93d57a0546625362ca5bee" title="org.jgap.Population.size" alt="" coords="275,56,437,83"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a1c07a36de9973bee29421a44f99b8319"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Population.contains </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td>
          <td class="paramname"><em>a_chromosome</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Determines whether the given chromosome is contained within the population. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_chromosome</td><td>the chromosome to check </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>true: chromosome contained within population</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.1 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00406">406</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>Referenced by <a class="el" href="_g_a_breeder_8java_source.html#l00293">org.jgap.impl.GABreeder.reAddFittest()</a>, <a class="el" href="_natural_selector_ext_8java_source.html#l00156">org.jgap.NaturalSelectorExt.selectChromosome()</a>, and <a class="el" href="_weighted_roulette_selector_8java_source.html#l00158">org.jgap.impl.WeightedRouletteSelector.selectChromosomes()</a>.</p>

</div>
</div>
<a class="anchor" id="a62348bfaa285efe11845574496d7a14c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> org.jgap.Population.createChromosome </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_chromClassName</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_persistentRepresentation</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws Exception</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Creates a new <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> instance.</p>
<p>Taken and adapted from CompositeGene.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_chromClassName</td><td>name of the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> class </td></tr>
    <tr><td class="paramname">a_persistentRepresentation</td><td>persistent representation of the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> to create (could be obtained via getPersistentRepresentation)</td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>newly created <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Exception</td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00785">785</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00153">org.jgap.Population.getConfiguration()</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00739">org.jgap.Population.setValueFromPersistentRepresentation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_a62348bfaa285efe11845574496d7a14c_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_a62348bfaa285efe11845574496d7a14c_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_a62348bfaa285efe11845574496d7a14c_cgraph" id="classorg_1_1jgap_1_1_population_a62348bfaa285efe11845574496d7a14c_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#a43613c7bc90d413e2a31bd459f49dfed" title="org.jgap.Population.getConfiguration" alt="" coords="227,12,459,39"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ab152be01c73200bf39e61518d0146379"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> org.jgap.Population.determineFittestChromosome </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Determines the fittest <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in the <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> (the one with the highest fitness value) and memorizes it. This is an optimized version compared to calling determineFittesChromosomes(1).</p>
<dl class="section return"><dt>Returns</dt><dd>the fittest <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> of the <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a></dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00303">303</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00153">org.jgap.Population.getConfiguration()</a>, <a class="el" href="_population_8java_source.html#l00045">org.jgap.Population.m_changed</a>, <a class="el" href="_population_8java_source.html#l00039">org.jgap.Population.m_fittestChromosome</a>, and <a class="el" href="_population_8java_source.html#l00371">org.jgap.Population.setChanged()</a>.</p>

<p>Referenced by <a class="el" href="_evaluator_test_8java_source.html#l00072">org.jgap.audit.EvaluatorTest.testCalcPerformance_0()</a>, <a class="el" href="_evaluator_test_8java_source.html#l00183">org.jgap.audit.EvaluatorTest.testCalcPerformance_2()</a>, <a class="el" href="_population_test_8java_source.html#l00232">org.jgap.PopulationTest.testDetermineFittestChromosome_0()</a>, <a class="el" href="_population_test_8java_source.html#l00246">org.jgap.PopulationTest.testDetermineFittestChromosome_1()</a>, <a class="el" href="_population_test_8java_source.html#l00279">org.jgap.PopulationTest.testDetermineFittestChromosome_2()</a>, <a class="el" href="_population_test_8java_source.html#l00387">org.jgap.PopulationTest.testDetermineFittestChromosome_2_0()</a>, <a class="el" href="_population_test_8java_source.html#l00307">org.jgap.PopulationTest.testDetermineFittestChromosome_3()</a>, <a class="el" href="_population_test_8java_source.html#l00353">org.jgap.PopulationTest.testDetermineFittestChromosome_5()</a>, <a class="el" href="_population_test_8java_source.html#l00445">org.jgap.PopulationTest.testDetermineFittestChromosomes_2()</a>, and <a class="el" href="_population_test_8java_source.html#l00467">org.jgap.PopulationTest.testDetermineFittestChromosomes_3()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_ab152be01c73200bf39e61518d0146379_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_ab152be01c73200bf39e61518d0146379_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_ab152be01c73200bf39e61518d0146379_cgraph" id="classorg_1_1jgap_1_1_population_ab152be01c73200bf39e61518d0146379_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#a43613c7bc90d413e2a31bd459f49dfed" title="org.jgap.Population.getConfiguration" alt="" coords="248,5,480,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#adf75c170902a54737483e3e845bcfacc" title="org.jgap.Population.setChanged" alt="" coords="261,56,467,83"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_population.html#a252f05034adae08bf72577e853adee88" title="org.jgap.Population.setSorted" alt="" coords="529,56,722,83"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a010cf1cbf6a1f771d34e5c3ce9420f25"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> org.jgap.Population.determineFittestChromosome </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_startIndex</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_endIndex</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Determines the fittest <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in the population (the one with the highest fitness value) within the given indices and memorizes it. This is an optimized version compared to calling determineFittesChromosomes(1).</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_startIndex</td><td>index to begin the evaluation with </td></tr>
    <tr><td class="paramname">a_endIndex</td><td>index to end the evaluation with </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the fittest <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> of the population within the given indices</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00342">342</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00153">org.jgap.Population.getConfiguration()</a>, <a class="el" href="interfaceorg_1_1jgap_1_1_fitness_evaluator.html#aad9309b75a437b32641480be173550eb">org.jgap.FitnessEvaluator.isFitter()</a>, <a class="el" href="_population_8java_source.html#l00034">org.jgap.Population.m_chromosomes</a>, and <a class="el" href="_population_8java_source.html#l00039">org.jgap.Population.m_fittestChromosome</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_a010cf1cbf6a1f771d34e5c3ce9420f25_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_a010cf1cbf6a1f771d34e5c3ce9420f25_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_a010cf1cbf6a1f771d34e5c3ce9420f25_cgraph" id="classorg_1_1jgap_1_1_population_a010cf1cbf6a1f771d34e5c3ce9420f25_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#a43613c7bc90d413e2a31bd459f49dfed" title="org.jgap.Population.getConfiguration" alt="" coords="248,5,480,32"/><area shape="rect" id="node5" href="interfaceorg_1_1jgap_1_1_fitness_evaluator.html#aad9309b75a437b32641480be173550eb" title="org.jgap.FitnessEvaluator.is\lFitter" alt="" coords="272,57,456,98"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a9ec112af7542cb12b4a891d259a31038"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">List org.jgap.Population.determineFittestChromosomes </td>
          <td>(</td>
          <td class="paramtype">final int&#160;</td>
          <td class="paramname"><em>a_numberOfChromosomes</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sorts the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> list and returns the fittest n Chromosomes in the population.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_numberOfChromosomes</td><td>number of top performer chromosomes to be returned </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>list of the fittest n Chromosomes of the population, or the fittest x Chromosomes with x = number of chromosomes in case n &gt; x.</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Charles Kevin Hill </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.4 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00479">479</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00246">org.jgap.Population.getChromosomes()</a>, <a class="el" href="_population_8java_source.html#l00045">org.jgap.Population.m_changed</a>, <a class="el" href="_population_8java_source.html#l00050">org.jgap.Population.m_sorted</a>, and <a class="el" href="_population_8java_source.html#l00502">org.jgap.Population.sortByFitness()</a>.</p>

<p>Referenced by <a class="el" href="_population_test_8java_source.html#l00334">org.jgap.PopulationTest.testDetermineFittestChromosome_4()</a>, and <a class="el" href="_population_test_8java_source.html#l00423">org.jgap.PopulationTest.testDetermineFittestChromosomes_1()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_a9ec112af7542cb12b4a891d259a31038_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_a9ec112af7542cb12b4a891d259a31038_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_a9ec112af7542cb12b4a891d259a31038_cgraph" id="classorg_1_1jgap_1_1_population_a9ec112af7542cb12b4a891d259a31038_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#aa9deff3f371931e7151070af03465e83" title="org.jgap.Population.getChromosomes" alt="" coords="736,5,976,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#ac66aac96dfb67e94a3de2284cd9c9066" title="org.jgap.Population.sort\lByFitness" alt="" coords="249,74,407,115"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_population.html#ae97e76f0bf965b1741a1f98efe05b482" title="org.jgap.Population.sort" alt="" coords="493,31,651,57"/><area shape="rect" id="node10" href="classorg_1_1jgap_1_1_population.html#a43613c7bc90d413e2a31bd459f49dfed" title="org.jgap.Population.getConfiguration" alt="" coords="456,81,688,108"/><area shape="rect" id="node12" href="classorg_1_1jgap_1_1_population.html#adf75c170902a54737483e3e845bcfacc" title="org.jgap.Population.setChanged" alt="" coords="469,132,675,159"/><area shape="rect" id="node14" href="classorg_1_1jgap_1_1_population.html#a252f05034adae08bf72577e853adee88" title="org.jgap.Population.setSorted" alt="" coords="759,157,953,184"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ab415c85cc397ce6d54924c4d14aa8cf6"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Population.equals </td>
          <td>(</td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>a_pop</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>The equals-method.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_pop</td><td>the population instance to compare with </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>true: given object equal to comparing one</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00605">605</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00629">org.jgap.Population.compareTo()</a>.</p>

<p>Referenced by <a class="el" href="_population_history_8java_source.html#l00120">org.jgap.eval.PopulationHistory.equals()</a>, <a class="el" href="_standard_post_selector_8java_source.html#l00201">org.jgap.impl.StandardPostSelector.equals()</a>, <a class="el" href="_best_chromosomes_selector_8java_source.html#l00275">org.jgap.impl.BestChromosomesSelector.equals()</a>, and <a class="el" href="_population_history_test_8java_source.html#l00076">org.jgap.eval.PopulationHistoryTest.testGetPopulation_1()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_ab415c85cc397ce6d54924c4d14aa8cf6_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_ab415c85cc397ce6d54924c4d14aa8cf6_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_ab415c85cc397ce6d54924c4d14aa8cf6_cgraph" id="classorg_1_1jgap_1_1_population_ab415c85cc397ce6d54924c4d14aa8cf6_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#a7671499966b4ffb937d45e373c6e95d9" title="org.jgap.Population.compareTo" alt="" coords="229,31,432,57"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#a85b83c32f89f499a8ecd15b3293202ed" title="org.jgap.Population.Population" alt="" coords="481,5,679,32"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_population.html#a2d09529f8e93d57a0546625362ca5bee" title="org.jgap.Population.size" alt="" coords="499,56,661,83"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a8e095445f8d0395d3676974349ad0c13"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> org.jgap.Population.getChromosome </td>
          <td>(</td>
          <td class="paramtype">final int&#160;</td>
          <td class="paramname"><em>a_index</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_index</td><td>the index of the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> to be returned </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd><a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> at given index in the <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a></dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00257">257</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00854">org.jgap.Population.hashCode()</a>, <a class="el" href="_data_tree_builder_8java_source.html#l00132">org.jgap.data.DataTreeBuilder.representGenotypeAsElement()</a>, <a class="el" href="_x_m_l_manager_8java_source.html#l00274">org.jgap.xml.XMLManager.representGenotypeAsElement()</a>, <a class="el" href="_standard_post_selector_8java_source.html#l00108">org.jgap.impl.StandardPostSelector.select()</a>, <a class="el" href="_natural_selector_ext_8java_source.html#l00127">org.jgap.NaturalSelectorExt.select()</a>, <a class="el" href="_population_test_8java_source.html#l00171">org.jgap.PopulationTest.testGetChromosome_0()</a>, <a class="el" href="_fittest_population_merger_test_8java_source.html#l00048">org.jgap.impl.FittestPopulationMergerTest.testMergePopulations_0()</a>, and <a class="el" href="_population_test_8java_source.html#l00126">org.jgap.PopulationTest.testSetChromosome_1()</a>.</p>

</div>
</div>
<a class="anchor" id="aa9deff3f371931e7151070af03465e83"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">List&lt;<a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&gt; org.jgap.Population.getChromosomes </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>the list of <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>'s in the <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>. Don't modify the retrieved list by using <a class="el" href="classorg_1_1jgap_1_1_population.html#a4e4c5b7e98055340e9f0b073be13d24b">clear()</a>, remove(int) etc. If you do so, you need to call setChanged(true)</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00246">246</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00034">org.jgap.Population.m_chromosomes</a>.</p>

<p>Referenced by <a class="el" href="_best_chromosomes_selector_8java_source.html#l00094">org.jgap.impl.BestChromosomesSelector.add()</a>, <a class="el" href="_population_8java_source.html#l00479">org.jgap.Population.determineFittestChromosomes()</a>, <a class="el" href="_population_8java_source.html#l00536">org.jgap.Population.getGenome()</a>, <a class="el" href="_population_8java_source.html#l00521">org.jgap.Population.sort()</a>, and <a class="el" href="_population_test_8java_source.html#l00155">org.jgap.PopulationTest.testGetChromosomes_0()</a>.</p>

</div>
</div>
<a class="anchor" id="a43613c7bc90d413e2a31bd459f49dfed"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> org.jgap.Population.getConfiguration </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_population_8java_source.html#l00153">153</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00052">org.jgap.Population.m_config</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00785">org.jgap.Population.createChromosome()</a>, <a class="el" href="_population_8java_source.html#l00303">org.jgap.Population.determineFittestChromosome()</a>, <a class="el" href="_population_8java_source.html#l00438">org.jgap.Population.keepPopSizeConstant()</a>, <a class="el" href="_fittest_population_merger_8java_source.html#l00030">org.jgap.impl.FittestPopulationMerger.mergePopulations()</a>, and <a class="el" href="_population_8java_source.html#l00502">org.jgap.Population.sortByFitness()</a>.</p>

</div>
</div>
<a class="anchor" id="a1236240ac1bfdcd7848a268f099ede8e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">List org.jgap.Population.getGenome </td>
          <td>(</td>
          <td class="paramtype">final boolean&#160;</td>
          <td class="paramname"><em>a_resolveCompositeGenes</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Returns the genotype of the population, i.e. the list of genes in the <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_resolveCompositeGenes</td><td>true: split encountered CompositeGenes into their single (atomic) genes </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>genotype of the population</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.3 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00536">536</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00588">org.jgap.Population.addAtomicGene()</a>, <a class="el" href="_population_8java_source.html#l00567">org.jgap.Population.addCompositeGene()</a>, and <a class="el" href="_population_8java_source.html#l00246">org.jgap.Population.getChromosomes()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_a1236240ac1bfdcd7848a268f099ede8e_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_a1236240ac1bfdcd7848a268f099ede8e_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_a1236240ac1bfdcd7848a268f099ede8e_cgraph" id="classorg_1_1jgap_1_1_population_a1236240ac1bfdcd7848a268f099ede8e_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#a18ae527f4110c5d0c525e2a7f61c0553" title="org.jgap.Population.addAtomicGene" alt="" coords="561,5,791,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#a64e81fff98bc39fcc05071b304b3c167" title="org.jgap.Population.addCompositeGene" alt="" coords="259,31,511,57"/><area shape="rect" id="node8" href="classorg_1_1jgap_1_1_population.html#aa9deff3f371931e7151070af03465e83" title="org.jgap.Population.getChromosomes" alt="" coords="265,81,505,108"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="afeedd380ea603cdefb5ef871f2bdf99e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.Population.getPersistentRepresentation </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Returns a persistent representation of this chromosome, see interface <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> for description. Similar to CompositeGene's routine. But does not include all information of the chromosome (yet).</p>
<dl class="section return"><dt>Returns</dt><dd>string representation of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>'s relevant parts of its current state </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">UnsupportedOperationException</td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html#a07bf7dbaa8de08e268c9357a46e79dac">org.jgap.IPersistentRepresentation</a>.</p>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00705">705</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00054">org.jgap.Population.CHROM_DELIMITER</a>, <a class="el" href="_population_8java_source.html#l00066">org.jgap.Population.CHROM_DELIMITER_CLOSING</a>, <a class="el" href="_population_8java_source.html#l00060">org.jgap.Population.CHROM_DELIMITER_HEADING</a>, and <a class="el" href="_population_8java_source.html#l00034">org.jgap.Population.m_chromosomes</a>.</p>

<p>Referenced by <a class="el" href="_population_test_8java_source.html#l00879">org.jgap.PopulationTest.testPersistentRepresentation_1()</a>.</p>

</div>
</div>
<a class="anchor" id="ad6e08d20ae2bfc6d277c84f5309cac49"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.Population.hashCode </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_population_8java_source.html#l00854">854</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00257">org.jgap.Population.getChromosome()</a>, and <a class="el" href="_population_8java_source.html#l00267">org.jgap.Population.size()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_ad6e08d20ae2bfc6d277c84f5309cac49_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_ad6e08d20ae2bfc6d277c84f5309cac49_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_ad6e08d20ae2bfc6d277c84f5309cac49_cgraph" id="classorg_1_1jgap_1_1_population_ad6e08d20ae2bfc6d277c84f5309cac49_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#a8e095445f8d0395d3676974349ad0c13" title="org.jgap.Population.getChromosome" alt="" coords="251,5,485,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#a2d09529f8e93d57a0546625362ca5bee" title="org.jgap.Population.size" alt="" coords="287,56,449,83"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ab35d09d90bc1262d12bc22df23689df4"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Population.isChanged </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>true: population's chromosomes (maybe) were changed, false: not changed for sure</dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00382">382</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00045">org.jgap.Population.m_changed</a>.</p>

<p>Referenced by <a class="el" href="_population_test_8java_source.html#l00467">org.jgap.PopulationTest.testDetermineFittestChromosomes_3()</a>, <a class="el" href="_population_test_8java_source.html#l00587">org.jgap.PopulationTest.testGetGenome_1()</a>, <a class="el" href="_population_test_8java_source.html#l00623">org.jgap.PopulationTest.testGetGenome_2()</a>, and <a class="el" href="_population_test_8java_source.html#l00807">org.jgap.PopulationTest.testSortByFitness_1()</a>.</p>

</div>
</div>
<a class="anchor" id="a959e9940968abe5de202542a7a26c990"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Population.isSorted </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_population_8java_source.html#l00592">592</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00050">org.jgap.Population.m_sorted</a>.</p>

<p>Referenced by <a class="el" href="_population_test_8java_source.html#l00807">org.jgap.PopulationTest.testSortByFitness_1()</a>.</p>

</div>
</div>
<a class="anchor" id="a644b52f341ebe005d1dc1c2e84f2591e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Iterator org.jgap.Population.iterator </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>Iterator for the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> list in the <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>. Please be aware that using remove() forces you to call setChanged(true)</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00278">278</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

</div>
</div>
<a class="anchor" id="a7dd9dbd2fecb4e2c4841787fd78ea64c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Population.keepPopSizeConstant </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Cares that the population size does not exceed the maximum size given in the configuration.</p>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000177">Todo:</a></b></dt><dd>use dedicated selector for that </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00438">438</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00153">org.jgap.Population.getConfiguration()</a>, and <a class="el" href="_population_8java_source.html#l00267">org.jgap.Population.size()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_a7dd9dbd2fecb4e2c4841787fd78ea64c_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_a7dd9dbd2fecb4e2c4841787fd78ea64c_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_a7dd9dbd2fecb4e2c4841787fd78ea64c_cgraph" id="classorg_1_1jgap_1_1_population_a7dd9dbd2fecb4e2c4841787fd78ea64c_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#a43613c7bc90d413e2a31bd459f49dfed" title="org.jgap.Population.getConfiguration" alt="" coords="219,5,451,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#a2d09529f8e93d57a0546625362ca5bee" title="org.jgap.Population.size" alt="" coords="254,56,415,83"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="adf75c170902a54737483e3e845bcfacc"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Population.setChanged </td>
          <td>(</td>
          <td class="paramtype">final boolean&#160;</td>
          <td class="paramname"><em>a_changed</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Mark that for the population the fittest chromosome may have changed.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_changed</td><td>true: population's fittest chromosome may have changed, false: fittest chromosome evaluated earlier is still valid</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.2 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00371">371</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00045">org.jgap.Population.m_changed</a>, and <a class="el" href="_population_8java_source.html#l00394">org.jgap.Population.setSorted()</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00165">org.jgap.Population.addChromosome()</a>, <a class="el" href="_population_8java_source.html#l00183">org.jgap.Population.addChromosomes()</a>, <a class="el" href="_population_8java_source.html#l00303">org.jgap.Population.determineFittestChromosome()</a>, <a class="el" href="_population_8java_source.html#l00087">org.jgap.Population.Population()</a>, <a class="el" href="_population_8java_source.html#l00226">org.jgap.Population.setChromosome()</a>, <a class="el" href="_population_8java_source.html#l00208">org.jgap.Population.setChromosomes()</a>, <a class="el" href="_population_8java_source.html#l00739">org.jgap.Population.setValueFromPersistentRepresentation()</a>, and <a class="el" href="_population_8java_source.html#l00502">org.jgap.Population.sortByFitness()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_adf75c170902a54737483e3e845bcfacc_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_adf75c170902a54737483e3e845bcfacc_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_adf75c170902a54737483e3e845bcfacc_cgraph" id="classorg_1_1jgap_1_1_population_adf75c170902a54737483e3e845bcfacc_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#a252f05034adae08bf72577e853adee88" title="org.jgap.Population.setSorted" alt="" coords="262,5,455,32"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="aaa80c42ce70b695356b338b11d7b0364"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Population.setChromosome </td>
          <td>(</td>
          <td class="paramtype">final int&#160;</td>
          <td class="paramname"><em>a_index</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td>
          <td class="paramname"><em>a_chromosome</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets in the given <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> on the given index in the list of chromosomes. If the given index is exceeding the list by one, the chromosome is appended.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_index</td><td>the index to set the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in </td></tr>
    <tr><td class="paramname">a_chromosome</td><td>the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> to be set</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00226">226</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00165">org.jgap.Population.addChromosome()</a>, <a class="el" href="_population_8java_source.html#l00034">org.jgap.Population.m_chromosomes</a>, and <a class="el" href="_population_8java_source.html#l00371">org.jgap.Population.setChanged()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_aaa80c42ce70b695356b338b11d7b0364_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_aaa80c42ce70b695356b338b11d7b0364_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_aaa80c42ce70b695356b338b11d7b0364_cgraph" id="classorg_1_1jgap_1_1_population_aaa80c42ce70b695356b338b11d7b0364_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#aab7e3575e2a5979ed22b081f2a7400d4" title="org.jgap.Population.addChromosome" alt="" coords="289,5,525,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#adf75c170902a54737483e3e845bcfacc" title="org.jgap.Population.setChanged" alt="" coords="574,31,781,57"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_population.html#a252f05034adae08bf72577e853adee88" title="org.jgap.Population.setSorted" alt="" coords="830,31,1023,57"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a86b38d39f48f6db65148cdbbd2b5d866"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Population.setChromosomes </td>
          <td>(</td>
          <td class="paramtype">final List&#160;</td>
          <td class="paramname"><em>a_chromosomes</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Replaces all chromosomes in the population with the give list of chromosomes.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_chromosomes</td><td>the chromosomes to make the population up from</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00208">208</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00034">org.jgap.Population.m_chromosomes</a>, and <a class="el" href="_population_8java_source.html#l00371">org.jgap.Population.setChanged()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_a86b38d39f48f6db65148cdbbd2b5d866_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_a86b38d39f48f6db65148cdbbd2b5d866_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_a86b38d39f48f6db65148cdbbd2b5d866_cgraph" id="classorg_1_1jgap_1_1_population_a86b38d39f48f6db65148cdbbd2b5d866_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#adf75c170902a54737483e3e845bcfacc" title="org.jgap.Population.setChanged" alt="" coords="294,5,501,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#a252f05034adae08bf72577e853adee88" title="org.jgap.Population.setSorted" alt="" coords="550,5,743,32"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a252f05034adae08bf72577e853adee88"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Population.setSorted </td>
          <td>(</td>
          <td class="paramtype">final boolean&#160;</td>
          <td class="paramname"><em>a_sorted</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Mark the population as sorted.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_sorted</td><td>true: mark population as sorted</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00394">394</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00050">org.jgap.Population.m_sorted</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00371">org.jgap.Population.setChanged()</a>, and <a class="el" href="_population_8java_source.html#l00502">org.jgap.Population.sortByFitness()</a>.</p>

</div>
</div>
<a class="anchor" id="a5cb0aa594a5d3d089ea0cadbe9a002e8"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Population.setValueFromPersistentRepresentation </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_representation</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_unsupported_representation_exception.html">UnsupportedRepresentationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Counterpart of getPersistentRepresentation.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_representation</td><td>the string representation retrieved from a prior call to the <a class="el" href="classorg_1_1jgap_1_1_population.html#afeedd380ea603cdefb5ef871f2bdf99e">getPersistentRepresentation()</a> method</td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_unsupported_representation_exception.html">UnsupportedRepresentationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html#a94a345f1919c4840dd0b9eecf7afc6a3">org.jgap.IPersistentRepresentation</a>.</p>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00739">739</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00054">org.jgap.Population.CHROM_DELIMITER</a>, <a class="el" href="_population_8java_source.html#l00785">org.jgap.Population.createChromosome()</a>, <a class="el" href="_population_8java_source.html#l00034">org.jgap.Population.m_chromosomes</a>, <a class="el" href="_population_8java_source.html#l00371">org.jgap.Population.setChanged()</a>, and <a class="el" href="_population_8java_source.html#l00809">org.jgap.Population.split()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_a5cb0aa594a5d3d089ea0cadbe9a002e8_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_a5cb0aa594a5d3d089ea0cadbe9a002e8_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_a5cb0aa594a5d3d089ea0cadbe9a002e8_cgraph" id="classorg_1_1jgap_1_1_population_a5cb0aa594a5d3d089ea0cadbe9a002e8_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#a62348bfaa285efe11845574496d7a14c" title="org.jgap.Population.create\lChromosome" alt="" coords="271,5,444,46"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_population.html#adf75c170902a54737483e3e845bcfacc" title="org.jgap.Population.setChanged" alt="" coords="254,71,461,97"/><area shape="rect" id="node11" href="classorg_1_1jgap_1_1_population.html#a8379b53e6b976e681ae6d57c3db6a4ef" title="org.jgap.Population.split" alt="" coords="277,121,438,148"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#a43613c7bc90d413e2a31bd459f49dfed" title="org.jgap.Population.getConfiguration" alt="" coords="509,12,741,39"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_population.html#a252f05034adae08bf72577e853adee88" title="org.jgap.Population.setSorted" alt="" coords="529,71,722,97"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a2d09529f8e93d57a0546625362ca5bee"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.Population.size </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>number of <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>'s in the <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a></dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00267">267</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>Referenced by <a class="el" href="_g_a_breeder_8java_source.html#l00349">org.jgap.impl.GABreeder.bulkFunctionEvaluation()</a>, <a class="el" href="_population_8java_source.html#l00659">org.jgap.Population.clone()</a>, <a class="el" href="_population_8java_source.html#l00629">org.jgap.Population.compareTo()</a>, <a class="el" href="_g_a_breeder_8java_source.html#l00217">org.jgap.impl.GABreeder.fillPopulationRandomlyToOriginalSize()</a>, <a class="el" href="_population_8java_source.html#l00854">org.jgap.Population.hashCode()</a>, <a class="el" href="_population_8java_source.html#l00438">org.jgap.Population.keepPopSizeConstant()</a>, <a class="el" href="_inversion_operator_8java_source.html#l00065">org.jgap.impl.InversionOperator.operate()</a>, <a class="el" href="_gaussian_mutation_operator_8java_source.html#l00087">org.jgap.impl.GaussianMutationOperator.operate()</a>, <a class="el" href="_greedy_crossover_8java_source.html#l00111">org.jgap.impl.GreedyCrossover.operate()</a>, <a class="el" href="_two_way_mutation_operator_8java_source.html#l00134">org.jgap.impl.TwoWayMutationOperator.operate()</a>, <a class="el" href="_mutation_operator_8java_source.html#l00136">org.jgap.impl.MutationOperator.operate()</a>, <a class="el" href="_averaging_crossover_operator_8java_source.html#l00164">org.jgap.impl.AveragingCrossoverOperator.operate()</a>, <a class="el" href="_crossover_operator_8java_source.html#l00323">org.jgap.impl.CrossoverOperator.operate()</a>, <a class="el" href="_population_test_8java_source.html#l00084">org.jgap.PopulationTest.testAddChromosome_1()</a>, <a class="el" href="_population_test_8java_source.html#l00103">org.jgap.PopulationTest.testAddChromosomes_1()</a>, <a class="el" href="_population_test_8java_source.html#l00064">org.jgap.PopulationTest.testConstruct_3()</a>, <a class="el" href="_population_test_8java_source.html#l00387">org.jgap.PopulationTest.testDetermineFittestChromosome_2_0()</a>, <a class="el" href="_population_test_8java_source.html#l00527">org.jgap.PopulationTest.testIterator_0()</a>, <a class="el" href="_fittest_population_merger_test_8java_source.html#l00048">org.jgap.impl.FittestPopulationMergerTest.testMergePopulations_0()</a>, <a class="el" href="_population_test_8java_source.html#l00900">org.jgap.PopulationTest.testPersistentRepresentation_2()</a>, <a class="el" href="_population_test_8java_source.html#l00954">org.jgap.PopulationTest.testPersistentRepresentation_5()</a>, <a class="el" href="_population_test_8java_source.html#l00110">org.jgap.PopulationTest.testSetChromosome_0()</a>, <a class="el" href="_population_test_8java_source.html#l00126">org.jgap.PopulationTest.testSetChromosome_1()</a>, <a class="el" href="_population_test_8java_source.html#l00139">org.jgap.PopulationTest.testSetChromosomes_0()</a>, and <a class="el" href="_population_test_8java_source.html#l00513">org.jgap.PopulationTest.testSize_0()</a>.</p>

</div>
</div>
<a class="anchor" id="ae97e76f0bf965b1741a1f98efe05b482"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Population.sort </td>
          <td>(</td>
          <td class="paramtype">Comparator&#160;</td>
          <td class="paramname"><em>a_comparator</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sorts the chromosomes within the population utilzing the given comparator.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_comparator</td><td>the comparator to utilize for sorting</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00521">521</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00246">org.jgap.Population.getChromosomes()</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00502">org.jgap.Population.sortByFitness()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_ae97e76f0bf965b1741a1f98efe05b482_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_ae97e76f0bf965b1741a1f98efe05b482_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_ae97e76f0bf965b1741a1f98efe05b482_cgraph" id="classorg_1_1jgap_1_1_population_ae97e76f0bf965b1741a1f98efe05b482_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#aa9deff3f371931e7151070af03465e83" title="org.jgap.Population.getChromosomes" alt="" coords="213,5,453,32"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ac66aac96dfb67e94a3de2284cd9c9066"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Population.sortByFitness </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sorts the chromosomes within the population according to their fitness value using ChromosomFitnessComparator. The fittest chromosome is then at index 0.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00502">502</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00153">org.jgap.Population.getConfiguration()</a>, <a class="el" href="_population_8java_source.html#l00034">org.jgap.Population.m_chromosomes</a>, <a class="el" href="_population_8java_source.html#l00039">org.jgap.Population.m_fittestChromosome</a>, <a class="el" href="_population_8java_source.html#l00371">org.jgap.Population.setChanged()</a>, <a class="el" href="_population_8java_source.html#l00394">org.jgap.Population.setSorted()</a>, and <a class="el" href="_population_8java_source.html#l00521">org.jgap.Population.sort()</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00479">org.jgap.Population.determineFittestChromosomes()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_population_ac66aac96dfb67e94a3de2284cd9c9066_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_population_ac66aac96dfb67e94a3de2284cd9c9066_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_population_ac66aac96dfb67e94a3de2284cd9c9066_cgraph" id="classorg_1_1jgap_1_1_population_ac66aac96dfb67e94a3de2284cd9c9066_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_population.html#a43613c7bc90d413e2a31bd459f49dfed" title="org.jgap.Population.getConfiguration" alt="" coords="213,5,445,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_population.html#adf75c170902a54737483e3e845bcfacc" title="org.jgap.Population.setChanged" alt="" coords="226,56,433,83"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_population.html#a252f05034adae08bf72577e853adee88" title="org.jgap.Population.setSorted" alt="" coords="517,81,710,108"/><area shape="rect" id="node10" href="classorg_1_1jgap_1_1_population.html#ae97e76f0bf965b1741a1f98efe05b482" title="org.jgap.Population.sort" alt="" coords="250,157,409,184"/><area shape="rect" id="node12" href="classorg_1_1jgap_1_1_population.html#aa9deff3f371931e7151070af03465e83" title="org.jgap.Population.getChromosomes" alt="" coords="493,157,733,184"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a8379b53e6b976e681ae6d57c3db6a4ef"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">static final List org.jgap.Population.split </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_string</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_unsupported_representation_exception.html">UnsupportedRepresentationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">static</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Splits the input a_string into individual chromosome representations.</p>
<p>Taken and adapted from CompositeGene.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_string</td><td>the string to split </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the elements of the returned array are the persistent representation strings of the population's components </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_unsupported_representation_exception.html">UnsupportedRepresentationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00809">809</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>References <a class="el" href="_population_8java_source.html#l00066">org.jgap.Population.CHROM_DELIMITER_CLOSING</a>, and <a class="el" href="_population_8java_source.html#l00060">org.jgap.Population.CHROM_DELIMITER_HEADING</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00739">org.jgap.Population.setValueFromPersistentRepresentation()</a>.</p>

</div>
</div>
<a class="anchor" id="aecabe6047fb1b1a7ec80d9b13fb4e7a0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> [] org.jgap.Population.toChromosomes </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>the <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> converted into a list of <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>'s</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert, Dan Clark </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00288">288</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Data Documentation</h2>
<a class="anchor" id="a8fd43176ca6c601491bb9cf2674d9c42"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final String org.jgap.Population.CHROM_DELIMITER = &quot;~&quot;</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_population_8java_source.html#l00054">54</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00705">org.jgap.Population.getPersistentRepresentation()</a>, <a class="el" href="_population_8java_source.html#l00739">org.jgap.Population.setValueFromPersistentRepresentation()</a>, <a class="el" href="_population_test_8java_source.html#l00932">org.jgap.PopulationTest.testPersistentRepresentation_4()</a>, <a class="el" href="_population_test_8java_source.html#l00954">org.jgap.PopulationTest.testPersistentRepresentation_5()</a>, and <a class="el" href="_population_test_8java_source.html#l00981">org.jgap.PopulationTest.testPersistentRepresentation_6()</a>.</p>

</div>
</div>
<a class="anchor" id="a35f548ce2820e079590565c66bae85aa"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final String org.jgap.Population.CHROM_DELIMITER_CLOSING = &quot;]&quot;</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Represents the closing delimiter that is used to separate chromosomes in the persistent representation. </p>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00066">66</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00705">org.jgap.Population.getPersistentRepresentation()</a>, <a class="el" href="_population_8java_source.html#l00809">org.jgap.Population.split()</a>, and <a class="el" href="_population_test_8java_source.html#l00954">org.jgap.PopulationTest.testPersistentRepresentation_5()</a>.</p>

</div>
</div>
<a class="anchor" id="af6282eaa6a1d7a0218d85db1b36960a4"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final String org.jgap.Population.CHROM_DELIMITER_HEADING = &quot;[&quot;</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Represents the heading delimiter that is used to separate chromosomes in the persistent representation. </p>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00060">60</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00705">org.jgap.Population.getPersistentRepresentation()</a>, and <a class="el" href="_population_8java_source.html#l00809">org.jgap.Population.split()</a>.</p>

</div>
</div>
<a class="anchor" id="a66404ad18ee170cbd1d4423aa0acec14"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final String org.jgap.Population.CVS_REVISION = &quot;$Revision: 1.66 $&quot;</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>String containing the CVS revision. Read out via reflection! </p>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00029">29</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

</div>
</div>
<a class="anchor" id="ade9e10d21203f65a3cf4c6385fd4eb76"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Population.m_changed</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Indicates whether at least one of the chromosomes has been changed (deleted, added, modified). </p>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00045">45</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00686">org.jgap.Population.clear()</a>, <a class="el" href="_population_8java_source.html#l00303">org.jgap.Population.determineFittestChromosome()</a>, <a class="el" href="_population_8java_source.html#l00479">org.jgap.Population.determineFittestChromosomes()</a>, <a class="el" href="_population_8java_source.html#l00382">org.jgap.Population.isChanged()</a>, and <a class="el" href="_population_8java_source.html#l00371">org.jgap.Population.setChanged()</a>.</p>

</div>
</div>
<a class="anchor" id="abc9fbe0eb7dc8ea169782159ca69ad70"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">List&lt;<a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&gt; org.jgap.Population.m_chromosomes</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>The array of Chromosomes that makeup the <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a>'s population. </p>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00034">34</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00165">org.jgap.Population.addChromosome()</a>, <a class="el" href="_population_8java_source.html#l00183">org.jgap.Population.addChromosomes()</a>, <a class="el" href="_population_8java_source.html#l00659">org.jgap.Population.clone()</a>, <a class="el" href="_population_8java_source.html#l00629">org.jgap.Population.compareTo()</a>, <a class="el" href="_population_8java_source.html#l00342">org.jgap.Population.determineFittestChromosome()</a>, <a class="el" href="_population_8java_source.html#l00246">org.jgap.Population.getChromosomes()</a>, <a class="el" href="_population_8java_source.html#l00705">org.jgap.Population.getPersistentRepresentation()</a>, <a class="el" href="_population_8java_source.html#l00087">org.jgap.Population.Population()</a>, <a class="el" href="_population_8java_source.html#l00226">org.jgap.Population.setChromosome()</a>, <a class="el" href="_population_8java_source.html#l00208">org.jgap.Population.setChromosomes()</a>, <a class="el" href="_population_8java_source.html#l00739">org.jgap.Population.setValueFromPersistentRepresentation()</a>, and <a class="el" href="_population_8java_source.html#l00502">org.jgap.Population.sortByFitness()</a>.</p>

</div>
</div>
<a class="anchor" id="a84643e202f73135b3fc3a31b19ed29e2"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> org.jgap.Population.m_config</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_population_8java_source.html#l00052">52</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00659">org.jgap.Population.clone()</a>, <a class="el" href="_population_8java_source.html#l00153">org.jgap.Population.getConfiguration()</a>, and <a class="el" href="_population_8java_source.html#l00131">org.jgap.Population.Population()</a>.</p>

</div>
</div>
<a class="anchor" id="a58811aa95b2363c1078d54320fe6a252"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> org.jgap.Population.m_fittestChromosome</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>The fittest <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> of the population. </p>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00039">39</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00686">org.jgap.Population.clear()</a>, <a class="el" href="_population_8java_source.html#l00659">org.jgap.Population.clone()</a>, <a class="el" href="_population_8java_source.html#l00303">org.jgap.Population.determineFittestChromosome()</a>, and <a class="el" href="_population_8java_source.html#l00502">org.jgap.Population.sortByFitness()</a>.</p>

</div>
</div>
<a class="anchor" id="abb65c9029b8b947f25cbad4d2a5e855e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Population.m_sorted</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Indicates that the list of Chromosomes has been sorted. </p>

<p>Definition at line <a class="el" href="_population_8java_source.html#l00050">50</a> of file <a class="el" href="_population_8java_source.html">Population.java</a>.</p>

<p>Referenced by <a class="el" href="_population_8java_source.html#l00686">org.jgap.Population.clear()</a>, <a class="el" href="_population_8java_source.html#l00479">org.jgap.Population.determineFittestChromosomes()</a>, <a class="el" href="_population_8java_source.html#l00592">org.jgap.Population.isSorted()</a>, and <a class="el" href="_population_8java_source.html#l00394">org.jgap.Population.setSorted()</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>LDH_Git/src/org/jgap/<a class="el" href="_population_8java_source.html">Population.java</a></li>
</ul>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Sat Nov 23 2013 17:42:13 for Práctica_5_LDH by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
