<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Práctica_5_LDH: org.jgap.gp.impl.ProgramChromosome Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">Práctica_5_LDH
   &#160;<span id="projectnumber">1.2</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li><a href="inherits.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Classes</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Namespaces</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Files</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Functions</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(6)"><span class="SelectionMark">&#160;</span>Pages</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="namespaceorg.html">org</a></li><li class="navelem"><a class="el" href="namespaceorg_1_1jgap.html">jgap</a></li><li class="navelem"><a class="el" href="namespaceorg_1_1jgap_1_1gp.html">gp</a></li><li class="navelem"><a class="el" href="namespaceorg_1_1jgap_1_1gp_1_1impl.html">impl</a></li><li class="navelem"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html">ProgramChromosome</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#nested-classes">Classes</a> &#124;
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pro-methods">Protected Member Functions</a> &#124;
<a href="#pro-static-methods">Static Protected Member Functions</a> &#124;
<a href="#pri-methods">Private Member Functions</a> &#124;
<a href="#pri-attribs">Private Attributes</a> &#124;
<a href="#pri-static-attribs">Static Private Attributes</a> &#124;
<a href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">org.jgap.gp.impl.ProgramChromosome Class Reference</div>  </div>
</div><!--header-->
<div class="contents">
<div class="dynheader">
Inheritance diagram for org.jgap.gp.impl.ProgramChromosome:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome__inherit__graph.png" border="0" usemap="#org_8jgap_8gp_8impl_8_program_chromosome_inherit__map" alt="Inheritance graph"/></div>
<map name="org_8jgap_8gp_8impl_8_program_chromosome_inherit__map" id="org_8jgap_8gp_8impl_8_program_chromosome_inherit__map">
<area shape="rect" id="node2" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html" title="org.jgap.gp.BaseGPChromosome" alt="" coords="5,155,221,181"/><area shape="rect" id="node4" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome.html" title="org.jgap.gp.IGPChromosome" alt="" coords="18,80,208,107"/><area shape="rect" id="node12" href="interfaceorg_1_1jgap_1_1_i_business_key.html" title="org.jgap.IBusinessKey" alt="" coords="466,155,619,181"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<div class="dynheader">
Collaboration diagram for org.jgap.gp.impl.ProgramChromosome:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome__coll__graph.png" border="0" usemap="#org_8jgap_8gp_8impl_8_program_chromosome_coll__map" alt="Collaboration graph"/></div>
<map name="org_8jgap_8gp_8impl_8_program_chromosome_coll__map" id="org_8jgap_8gp_8impl_8_program_chromosome_coll__map">
<area shape="rect" id="node2" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html" title="org.jgap.gp.BaseGPChromosome" alt="" coords="622,69,839,96"/><area shape="rect" id="node4" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome.html" title="org.jgap.gp.IGPChromosome" alt="" coords="342,76,533,103"/><area shape="rect" id="node8" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program.html" title="org.jgap.gp.IGPProgram" alt="" coords="356,25,519,52"/><area shape="rect" id="node23" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html" title="org.jgap.gp.CommandGene" alt="" coords="347,136,528,163"/><area shape="rect" id="node13" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html" title="org.jgap.util.ICloneable" alt="" coords="136,5,291,32"/><area shape="rect" id="node19" href="interfaceorg_1_1jgap_1_1_i_business_key.html" title="org.jgap.IBusinessKey" alt="" coords="654,248,807,275"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="nested-classes"></a>
Classes</h2></td></tr>
<tr class="memitem:"><td class="memItemLeft" align="right" valign="top">class &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_1_1_node_info.html">NodeInfo</a></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:aefbb2bea6ceab1a79c6bd444962b5400"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aefbb2bea6ceab1a79c6bd444962b5400">ProgramChromosome</a> (<a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html">GPConfiguration</a> a_conf, int a_size)  throws InvalidConfigurationException </td></tr>
<tr class="separator:aefbb2bea6ceab1a79c6bd444962b5400"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad014d2d525e7210edd3fa35146432f67"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ad014d2d525e7210edd3fa35146432f67">ProgramChromosome</a> (<a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html">GPConfiguration</a> a_conf, int a_size, <a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program.html">IGPProgram</a> a_ind)  throws InvalidConfigurationException </td></tr>
<tr class="separator:ad014d2d525e7210edd3fa35146432f67"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6895103538a0e05c6693fd2f6711b50f"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a6895103538a0e05c6693fd2f6711b50f">ProgramChromosome</a> (<a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html">GPConfiguration</a> a_conf, int a_size, <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[] a_functionSet, Class[] a_argTypes, <a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program.html">IGPProgram</a> a_ind)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a6895103538a0e05c6693fd2f6711b50f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac4ca2284886dd13bfe27e4802a40d5fd"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ac4ca2284886dd13bfe27e4802a40d5fd">ProgramChromosome</a> (<a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html">GPConfiguration</a> a_conf, <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[] a_initialGenes)  throws InvalidConfigurationException </td></tr>
<tr class="separator:ac4ca2284886dd13bfe27e4802a40d5fd"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aafd1f551ada36c7835737f0e3daeead4"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aafd1f551ada36c7835737f0e3daeead4">ProgramChromosome</a> (final <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html">GPConfiguration</a> a_conf)  throws InvalidConfigurationException </td></tr>
<tr class="separator:aafd1f551ada36c7835737f0e3daeead4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a49e32c4ee7a42b4e7fbddae454b325c2"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a49e32c4ee7a42b4e7fbddae454b325c2">ProgramChromosome</a> ()  throws InvalidConfigurationException </td></tr>
<tr class="separator:a49e32c4ee7a42b4e7fbddae454b325c2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a714709ffa46266e370becc723467c8bb"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a714709ffa46266e370becc723467c8bb">setArgTypes</a> (Class[] a_argTypes)</td></tr>
<tr class="separator:a714709ffa46266e370becc723467c8bb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af25451cf53447cdec834e5be9ac77f79"><td class="memItemLeft" align="right" valign="top">synchronized Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#af25451cf53447cdec834e5be9ac77f79">clone</a> ()</td></tr>
<tr class="separator:af25451cf53447cdec834e5be9ac77f79"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a274fc584475b32253ee2ced3897e7547"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a274fc584475b32253ee2ced3897e7547">cleanup</a> ()</td></tr>
<tr class="separator:a274fc584475b32253ee2ced3897e7547"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6b798831e4e59f1dbcd58381e090a9d0"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a6b798831e4e59f1dbcd58381e090a9d0">growOrFull</a> (final int a_num, final int a_depth, final Class a_type, final Class[] a_argTypes, final <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[] a_functionSet, boolean a_grow, int a_tries)</td></tr>
<tr class="separator:a6b798831e4e59f1dbcd58381e090a9d0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:addf578c58866aaf237a11d91308d0c04"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#addf578c58866aaf237a11d91308d0c04">toString</a> (final int a_startNode)</td></tr>
<tr class="separator:addf578c58866aaf237a11d91308d0c04"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a56c22c94166729d37888a6daaef76bb3"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a56c22c94166729d37888a6daaef76bb3">toStringNorm</a> (final int a_startNode)</td></tr>
<tr class="separator:a56c22c94166729d37888a6daaef76bb3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a88f1eb9dac4272ec7f7274f0cd66d763"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a88f1eb9dac4272ec7f7274f0cd66d763">getBusinessKey</a> ()</td></tr>
<tr class="separator:a88f1eb9dac4272ec7f7274f0cd66d763"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a145081f4837b9a7b50af35f8256c0eea"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a145081f4837b9a7b50af35f8256c0eea">toStringDebug</a> ()</td></tr>
<tr class="separator:a145081f4837b9a7b50af35f8256c0eea"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a171447ec4020a6ca9f4e089836147993"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a171447ec4020a6ca9f4e089836147993">isPossible</a> (Class a_returnType, int a_subReturnType, <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[] a_nodeSet, boolean a_function, boolean a_growing)</td></tr>
<tr class="separator:a171447ec4020a6ca9f4e089836147993"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a72e5e8ab3c253f4ec7dc10fb17a45180"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a72e5e8ab3c253f4ec7dc10fb17a45180">redepth</a> ()</td></tr>
<tr class="separator:a72e5e8ab3c253f4ec7dc10fb17a45180"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aae34fa625e78f8be2de585462d3bbcf8"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aae34fa625e78f8be2de585462d3bbcf8">getChild</a> (int a_index, int a_child)</td></tr>
<tr class="separator:aae34fa625e78f8be2de585462d3bbcf8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a885c955a0bb1f22ad404c242d51cba45"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a885c955a0bb1f22ad404c242d51cba45">getChild</a> (<a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> a_node, int a_child)</td></tr>
<tr class="separator:a885c955a0bb1f22ad404c242d51cba45"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8640ae6ff64b6b0fe5d0972a554104f0"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a8640ae6ff64b6b0fe5d0972a554104f0">getFunctionSet</a> ()</td></tr>
<tr class="separator:a8640ae6ff64b6b0fe5d0972a554104f0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a81171ec1774c2605dc2080c476768f19"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a81171ec1774c2605dc2080c476768f19">setFunctionSet</a> (<a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[] a_functionSet)</td></tr>
<tr class="separator:a81171ec1774c2605dc2080c476768f19"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9814ee68c4893d979b6d6c702f754ed6"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9814ee68c4893d979b6d6c702f754ed6">getFunctions</a> ()</td></tr>
<tr class="separator:a9814ee68c4893d979b6d6c702f754ed6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9b47c5e0e01b774eed4f0be74a94764e"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9b47c5e0e01b774eed4f0be74a94764e">setFunctions</a> (<a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[] a_functions)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a9b47c5e0e01b774eed4f0be74a94764e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a14b7aa6ec69495ad519f64dbc362488e"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a14b7aa6ec69495ad519f64dbc362488e">getSize</a> (int a_index)</td></tr>
<tr class="separator:a14b7aa6ec69495ad519f64dbc362488e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac6c8388003452adf7ebd2b0e70c916ac"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ac6c8388003452adf7ebd2b0e70c916ac">getDepth</a> (int a_index)</td></tr>
<tr class="separator:ac6c8388003452adf7ebd2b0e70c916ac"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a94c73ab9b25ed666e4fabe83e70dd8c4"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a94c73ab9b25ed666e4fabe83e70dd8c4">getParentNode</a> (int a_child)</td></tr>
<tr class="separator:a94c73ab9b25ed666e4fabe83e70dd8c4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9dcb6045fafdd72b8241f3f1634ec433"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9dcb6045fafdd72b8241f3f1634ec433">getNode</a> (Class a_type, boolean a_exactMatch)</td></tr>
<tr class="separator:a9dcb6045fafdd72b8241f3f1634ec433"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2e3b39bebbe5a3f6bb46494341d8d39d"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a2e3b39bebbe5a3f6bb46494341d8d39d">getNode</a> (Class a_type, boolean a_exactMatch, int a_startIndex)</td></tr>
<tr class="separator:a2e3b39bebbe5a3f6bb46494341d8d39d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac02462405fa19e5b91d8b7c99f91c872"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ac02462405fa19e5b91d8b7c99f91c872">execute_boolean</a> (Object[] args)</td></tr>
<tr class="separator:ac02462405fa19e5b91d8b7c99f91c872"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a500cdb04cca5286bb9e10d41abba324f"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a500cdb04cca5286bb9e10d41abba324f">execute_boolean</a> (int n, int child, Object[] args)</td></tr>
<tr class="separator:a500cdb04cca5286bb9e10d41abba324f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa493171a9ea6d5a0973337d5394cf973"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aa493171a9ea6d5a0973337d5394cf973">execute_void</a> (Object[] args)</td></tr>
<tr class="separator:aa493171a9ea6d5a0973337d5394cf973"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a92fd1beeb17969fc404237251003e388"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a92fd1beeb17969fc404237251003e388">execute_void</a> (int n, int child, Object[] args)</td></tr>
<tr class="separator:a92fd1beeb17969fc404237251003e388"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3d0cb8958612150faaab36f418a8777e"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a3d0cb8958612150faaab36f418a8777e">execute_int</a> (Object[] args)</td></tr>
<tr class="separator:a3d0cb8958612150faaab36f418a8777e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:affb613bc931c4b3be35fc7cca9ed1feb"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#affb613bc931c4b3be35fc7cca9ed1feb">execute_int</a> (int n, int child, Object[] args)</td></tr>
<tr class="separator:affb613bc931c4b3be35fc7cca9ed1feb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8296c9d5187129c039ad4053d4d70ea4"><td class="memItemLeft" align="right" valign="top">long&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a8296c9d5187129c039ad4053d4d70ea4">execute_long</a> (Object[] args)</td></tr>
<tr class="separator:a8296c9d5187129c039ad4053d4d70ea4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a63c4c4c86434ae1a0c9f4febe21e6572"><td class="memItemLeft" align="right" valign="top">long&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a63c4c4c86434ae1a0c9f4febe21e6572">execute_long</a> (int n, int child, Object[] args)</td></tr>
<tr class="separator:a63c4c4c86434ae1a0c9f4febe21e6572"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:acbd9e0bc12ba6077893ffc9d1ab3d085"><td class="memItemLeft" align="right" valign="top">float&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#acbd9e0bc12ba6077893ffc9d1ab3d085">execute_float</a> (Object[] args)</td></tr>
<tr class="separator:acbd9e0bc12ba6077893ffc9d1ab3d085"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8b8e3b3cdda9a453f7854418edfbe0c9"><td class="memItemLeft" align="right" valign="top">float&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a8b8e3b3cdda9a453f7854418edfbe0c9">execute_float</a> (int n, int child, Object[] args)</td></tr>
<tr class="separator:a8b8e3b3cdda9a453f7854418edfbe0c9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9462046182cf58ed4d59729fa584586a"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9462046182cf58ed4d59729fa584586a">execute_double</a> (Object[] args)</td></tr>
<tr class="separator:a9462046182cf58ed4d59729fa584586a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aca61c643bfcfc2e3fbbba2d6ef55e170"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aca61c643bfcfc2e3fbbba2d6ef55e170">execute_double</a> (int n, int child, Object[] args)</td></tr>
<tr class="separator:aca61c643bfcfc2e3fbbba2d6ef55e170"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a49bbf47534a08674eefdbd40c4103534"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a49bbf47534a08674eefdbd40c4103534">execute_object</a> (Object[] args)</td></tr>
<tr class="separator:a49bbf47534a08674eefdbd40c4103534"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aef96a42e1986888e049781e8202b8df3"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aef96a42e1986888e049781e8202b8df3">execute_object</a> (int n, int child, Object[] args)</td></tr>
<tr class="separator:aef96a42e1986888e049781e8202b8df3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2aac4f2020dfe3929473d163929b1971"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a2aac4f2020dfe3929473d163929b1971">execute</a> (Object[] args)</td></tr>
<tr class="separator:a2aac4f2020dfe3929473d163929b1971"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a376d8beedfc62a18b9d105d280cc141d"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a376d8beedfc62a18b9d105d280cc141d">execute</a> (int n, int child, Object[] args)</td></tr>
<tr class="separator:a376d8beedfc62a18b9d105d280cc141d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a705e4148b5d42c772bab87068287b8bf"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a705e4148b5d42c772bab87068287b8bf">setGene</a> (int index, <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> a_gene)</td></tr>
<tr class="separator:a705e4148b5d42c772bab87068287b8bf"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a96b8adcf50530c6bfbadda604b57314f"><td class="memItemLeft" align="right" valign="top">Class[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a96b8adcf50530c6bfbadda604b57314f">getArgTypes</a> ()</td></tr>
<tr class="separator:a96b8adcf50530c6bfbadda604b57314f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1873ffcdf7eaaf635b85b70f5a4ac59a"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1873ffcdf7eaaf635b85b70f5a4ac59a">getArity</a> ()</td></tr>
<tr class="separator:a1873ffcdf7eaaf635b85b70f5a4ac59a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ade35d12aa45f98f2acea90b52bda3375"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ade35d12aa45f98f2acea90b52bda3375">size</a> ()</td></tr>
<tr class="separator:ade35d12aa45f98f2acea90b52bda3375"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac9896e142c28c71711133698041fb72f"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ac9896e142c28c71711133698041fb72f">compareTo</a> (Object a_other)</td></tr>
<tr class="separator:ac9896e142c28c71711133698041fb72f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a39b343e4b3d1e851adef34ec081814ee"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a39b343e4b3d1e851adef34ec081814ee">equals</a> (Object a_other)</td></tr>
<tr class="separator:a39b343e4b3d1e851adef34ec081814ee"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afdecb0a33331b2ecef9be73815f72ef1"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#afdecb0a33331b2ecef9be73815f72ef1">setCompareApplicationData</a> (boolean a_doCompare)</td></tr>
<tr class="separator:afdecb0a33331b2ecef9be73815f72ef1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a24c91c7e53c8ad7be9d7badee0d1aad5"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a24c91c7e53c8ad7be9d7badee0d1aad5">isCompareApplicationData</a> ()</td></tr>
<tr class="separator:a24c91c7e53c8ad7be9d7badee0d1aad5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a34507350def1243e665819c4914738f9"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a34507350def1243e665819c4914738f9">getApplicationData</a> ()</td></tr>
<tr class="separator:a34507350def1243e665819c4914738f9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0204f8ead0682a3210428c6291456be7"><td class="memItemLeft" align="right" valign="top">synchronized <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a0204f8ead0682a3210428c6291456be7">getGene</a> (int a_locus)</td></tr>
<tr class="separator:a0204f8ead0682a3210428c6291456be7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6536e99f35877cb53ef5d01ea0c786be"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a6536e99f35877cb53ef5d01ea0c786be">getPersistentRepresentation</a> ()</td></tr>
<tr class="separator:a6536e99f35877cb53ef5d01ea0c786be"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2ea8008dc2f1662945564f8d0ee67c82"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a2ea8008dc2f1662945564f8d0ee67c82">setValueFromPersistentRepresentation</a> (final String a_representation)  throws UnsupportedRepresentationException </td></tr>
<tr class="separator:a2ea8008dc2f1662945564f8d0ee67c82"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a32591e6bcdec0c22383f482ca7c9b39a"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a32591e6bcdec0c22383f482ca7c9b39a">contains</a> (<a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[] a_functionSet, <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> a_function)</td></tr>
<tr class="separator:a32591e6bcdec0c22383f482ca7c9b39a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af2a24457cb0fedb950f59a919117cdab"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#af2a24457cb0fedb950f59a919117cdab">contains</a> (<a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[] a_functionSet, Class a_function)</td></tr>
<tr class="separator:af2a24457cb0fedb950f59a919117cdab"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6b4c3936426388aeb249ec73ee5131ed"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a6b4c3936426388aeb249ec73ee5131ed">contains</a> (<a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[] a_functionSet, <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> a_function, int a_limit)</td></tr>
<tr class="separator:a6b4c3936426388aeb249ec73ee5131ed"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td colspan="2" onclick="javascript:toggleInherit('pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome')"><img src="closed.png" alt="-"/>&#160;Public Member Functions inherited from <a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html">org.jgap.gp.BaseGPChromosome</a></td></tr>
<tr class="memitem:a3a61509cd85185506992df4e6a34eee9 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a3a61509cd85185506992df4e6a34eee9">BaseGPChromosome</a> (GPConfiguration a_configuration)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a3a61509cd85185506992df4e6a34eee9 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae675ec47db0ce8fd88b6ac3f2235ba6c inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#ae675ec47db0ce8fd88b6ac3f2235ba6c">BaseGPChromosome</a> (GPConfiguration a_configuration, <a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program.html">IGPProgram</a> a_ind)  throws InvalidConfigurationException </td></tr>
<tr class="separator:ae675ec47db0ce8fd88b6ac3f2235ba6c inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a13d8f2e469a316fd02a863a1d924ad5d inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program.html">IGPProgram</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a13d8f2e469a316fd02a863a1d924ad5d">getIndividual</a> ()</td></tr>
<tr class="separator:a13d8f2e469a316fd02a863a1d924ad5d inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:adef0faa3a73f86c1fe91af2f15c2d514 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#adef0faa3a73f86c1fe91af2f15c2d514">setIndividual</a> (<a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program.html">IGPProgram</a> a_ind)</td></tr>
<tr class="separator:adef0faa3a73f86c1fe91af2f15c2d514 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0c2122b764f1957c28e418a27a062f08 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a0c2122b764f1957c28e418a27a062f08">getTerminal</a> (int a_index)</td></tr>
<tr class="separator:a0c2122b764f1957c28e418a27a062f08 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af1de58df592374bed5f5765988e37f6e inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#af1de58df592374bed5f5765988e37f6e">getFunction</a> (int a_index)</td></tr>
<tr class="separator:af1de58df592374bed5f5765988e37f6e inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9bc14b556db8f5e4c05e1a4af5424828 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a9bc14b556db8f5e4c05e1a4af5424828">getTerminal</a> (int a_index, Class a_type, int a_subType)</td></tr>
<tr class="separator:a9bc14b556db8f5e4c05e1a4af5424828 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1d10424ef765a2f738593b06a476a193 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a1d10424ef765a2f738593b06a476a193">getFunction</a> (int a_index, Class a_type, int a_subType)</td></tr>
<tr class="separator:a1d10424ef765a2f738593b06a476a193 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af454a483e125909d9a4f9b06c5ec8e98 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#af454a483e125909d9a4f9b06c5ec8e98">numTerminals</a> ()</td></tr>
<tr class="separator:af454a483e125909d9a4f9b06c5ec8e98 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab46c095e367ecc9d40b3a7646571e61c inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#ab46c095e367ecc9d40b3a7646571e61c">numFunctions</a> ()</td></tr>
<tr class="separator:ab46c095e367ecc9d40b3a7646571e61c inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abd24b5ee2a8e7552a724303706c5b978 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#abd24b5ee2a8e7552a724303706c5b978">numTerminals</a> (Class a_type, int a_subType)</td></tr>
<tr class="separator:abd24b5ee2a8e7552a724303706c5b978 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a88ccabd79eb3647f41d073b48b10696d inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a88ccabd79eb3647f41d073b48b10696d">numFunctions</a> (Class a_type, int a_subType)</td></tr>
<tr class="separator:a88ccabd79eb3647f41d073b48b10696d inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abb9145c6fb3601ff2b579f9789408258 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#abb9145c6fb3601ff2b579f9789408258">getNode</a> (int a_index)</td></tr>
<tr class="separator:abb9145c6fb3601ff2b579f9789408258 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a92b35be05ab42231ab0842197332fde8 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a92b35be05ab42231ab0842197332fde8">getCommandOfClass</a> (int a_n, Class a_class)</td></tr>
<tr class="separator:a92b35be05ab42231ab0842197332fde8 inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:adbeb3b252e3e345b2f08b0377030179d inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#adbeb3b252e3e345b2f08b0377030179d">getAssignableFromClass</a> (int a_n, Class a_class)</td></tr>
<tr class="separator:adbeb3b252e3e345b2f08b0377030179d inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a166411347ff6092fd2ba1c957893846d inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a166411347ff6092fd2ba1c957893846d">getVariableWithReturnType</a> (int a_n, Class a_returnType)</td></tr>
<tr class="separator:a166411347ff6092fd2ba1c957893846d inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a57f6a4989aa1e01bda2c1f77cc086c4a inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">GPConfiguration&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a57f6a4989aa1e01bda2c1f77cc086c4a">getGPConfiguration</a> ()</td></tr>
<tr class="separator:a57f6a4989aa1e01bda2c1f77cc086c4a inherit pub_methods_classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pro-methods"></a>
Protected Member Functions</h2></td></tr>
<tr class="memitem:a1e11a46dcd815ed2778c03800abd8e56"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1e11a46dcd815ed2778c03800abd8e56">selectNode</a> (int a_chromIndex, Class a_returnType, int a_subReturnType, <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[] a_functionSet, boolean a_function, boolean a_growing)</td></tr>
<tr class="separator:a1e11a46dcd815ed2778c03800abd8e56"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1ca14815335297629dc3bd4812445405"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1ca14815335297629dc3bd4812445405">growOrFullNode</a> (int a_num, int a_depth, Class a_returnType, int a_subReturnType, <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[] a_functionSet, <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> a_rootNode, int a_recurseLevel, boolean a_grow, int a_childNum, boolean a_validateNode)</td></tr>
<tr class="separator:a1ca14815335297629dc3bd4812445405"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af63010758c5c507c9e58889cc37e985e"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#af63010758c5c507c9e58889cc37e985e">redepth</a> (int a_index)</td></tr>
<tr class="separator:af63010758c5c507c9e58889cc37e985e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1ceca8b2f7a033bf9ab63f70b438c211"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1ceca8b2f7a033bf9ab63f70b438c211">encode</a> (String a_string)</td></tr>
<tr class="separator:a1ceca8b2f7a033bf9ab63f70b438c211"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7a41513bb42bc79e7708c46afd61b4a4"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a7a41513bb42bc79e7708c46afd61b4a4">decode</a> (String a_string)</td></tr>
<tr class="separator:a7a41513bb42bc79e7708c46afd61b4a4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5a09ecbda2b198791e979d3dae3b0f01"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a5a09ecbda2b198791e979d3dae3b0f01">createGene</a> (String a_geneClassName, String a_persistentRepresentation)  throws Exception </td></tr>
<tr class="separator:a5a09ecbda2b198791e979d3dae3b0f01"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pro-static-methods"></a>
Static Protected Member Functions</h2></td></tr>
<tr class="memitem:a5db9d5ee745620591027ff0572e2cff3"><td class="memItemLeft" align="right" valign="top">static final List&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a5db9d5ee745620591027ff0572e2cff3">split</a> (String a_string)  throws UnsupportedRepresentationException </td></tr>
<tr class="separator:a5db9d5ee745620591027ff0572e2cff3"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-methods"></a>
Private Member Functions</h2></td></tr>
<tr class="memitem:a1d821adfada0d4ba257d1ac1cf15969a"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1d821adfada0d4ba257d1ac1cf15969a">init</a> ()  throws InvalidConfigurationException </td></tr>
<tr class="separator:a1d821adfada0d4ba257d1ac1cf15969a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac082a0f6a404ae98a32289ca561b765f"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ac082a0f6a404ae98a32289ca561b765f">init</a> (final int a_size)  throws InvalidConfigurationException </td></tr>
<tr class="separator:ac082a0f6a404ae98a32289ca561b765f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abb3f952c7eb74808cf6dc8dda7bfd94a"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#abb3f952c7eb74808cf6dc8dda7bfd94a">remove</a> (<a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[] a_functionSet, <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> node)</td></tr>
<tr class="separator:abb3f952c7eb74808cf6dc8dda7bfd94a"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-attribs"></a>
Private Attributes</h2></td></tr>
<tr class="memitem:a9d7f024ebd3113416251c968d06dce6d"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9d7f024ebd3113416251c968d06dce6d">m_functionSet</a></td></tr>
<tr class="separator:a9d7f024ebd3113416251c968d06dce6d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a81e14f6750b10a0331646f6b1891f023"><td class="memItemLeft" align="right" valign="top">int[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a81e14f6750b10a0331646f6b1891f023">m_depth</a></td></tr>
<tr class="separator:a81e14f6750b10a0331646f6b1891f023"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a770064f02147733f5077b5b0bfa125fb"><td class="memItemLeft" align="right" valign="top">Class[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a770064f02147733f5077b5b0bfa125fb">argTypes</a></td></tr>
<tr class="separator:a770064f02147733f5077b5b0bfa125fb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a05a7b989fa9e7a7526b3c796bf033fe3"><td class="memItemLeft" align="right" valign="top">transient int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a05a7b989fa9e7a7526b3c796bf033fe3">m_index</a></td></tr>
<tr class="separator:a05a7b989fa9e7a7526b3c796bf033fe3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:add5812f1aa24a114aa44923db580b919"><td class="memItemLeft" align="right" valign="top">transient int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#add5812f1aa24a114aa44923db580b919">m_maxDepth</a></td></tr>
<tr class="separator:add5812f1aa24a114aa44923db580b919"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad3903410ad37a12a87ee51b4a3f4858a"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ad3903410ad37a12a87ee51b4a3f4858a">m_genes</a></td></tr>
<tr class="separator:ad3903410ad37a12a87ee51b4a3f4858a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a918187c443c8cfb61eaa33a32801df64"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a918187c443c8cfb61eaa33a32801df64">m_applicationData</a></td></tr>
<tr class="separator:a918187c443c8cfb61eaa33a32801df64"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad4256d733bf0188c9530d7826f1320d3"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ad4256d733bf0188c9530d7826f1320d3">m_compareAppData</a></td></tr>
<tr class="separator:ad4256d733bf0188c9530d7826f1320d3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a42c376b973ad38ab97754447406a5480"><td class="memItemLeft" align="right" valign="top">Map&lt; <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_1_1_node_info.html">NodeInfo</a>, Boolean &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a42c376b973ad38ab97754447406a5480">m_possibleNodes</a> = new HashMap()</td></tr>
<tr class="separator:a42c376b973ad38ab97754447406a5480"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-static-attribs"></a>
Static Private Attributes</h2></td></tr>
<tr class="memitem:a1710d7febbb2e909bb1a9d41bba3aa17"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1710d7febbb2e909bb1a9d41bba3aa17">CVS_REVISION</a> = &quot;$Revision: 1.52 $&quot;</td></tr>
<tr class="separator:a1710d7febbb2e909bb1a9d41bba3aa17"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0c09c5995d7929662288b0fc4d0b4858"><td class="memItemLeft" align="right" valign="top">static transient Logger&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a0c09c5995d7929662288b0fc4d0b4858">LOGGER</a> = Logger.getLogger(ProgramChromosome.class)</td></tr>
<tr class="separator:a0c09c5995d7929662288b0fc4d0b4858"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="inherited"></a>
Additional Inherited Members</h2></td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome.html">org.jgap.gp.IGPChromosome</a></td></tr>
<tr class="memitem:a81a454d1fba54dbb9fb0850143e65484 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome.html#a81a454d1fba54dbb9fb0850143e65484">CVS_REVISION</a> = &quot;$Revision: 1.7 $&quot;</td></tr>
<tr class="separator:a81a454d1fba54dbb9fb0850143e65484 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_business_key.html">org.jgap.IBusinessKey</a></td></tr>
<tr class="memitem:a6d43b687da0842739b8021f466ca451a inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_business_key.html#a6d43b687da0842739b8021f466ca451a">CVS_REVISION</a> = &quot;$Revision: 1.3 $&quot;</td></tr>
<tr class="separator:a6d43b687da0842739b8021f466ca451a inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p><a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> representing a single GP Program.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00027">27</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="aefbb2bea6ceab1a79c6bd444962b5400"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.gp.impl.ProgramChromosome.ProgramChromosome </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html">GPConfiguration</a>&#160;</td>
          <td class="paramname"><em>a_conf</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_size</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00084">84</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00158">org.jgap.gp.impl.ProgramChromosome.init()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aefbb2bea6ceab1a79c6bd444962b5400_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aefbb2bea6ceab1a79c6bd444962b5400_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aefbb2bea6ceab1a79c6bd444962b5400_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aefbb2bea6ceab1a79c6bd444962b5400_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1d821adfada0d4ba257d1ac1cf15969a" title="org.jgap.gp.impl.Program\lChromosome.init" alt="" coords="286,5,453,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a57f6a4989aa1e01bda2c1f77cc086c4a" title="org.jgap.gp.BaseGPChromosome.get\lGPConfiguration" alt="" coords="502,5,741,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ad014d2d525e7210edd3fa35146432f67"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.gp.impl.ProgramChromosome.ProgramChromosome </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html">GPConfiguration</a>&#160;</td>
          <td class="paramname"><em>a_conf</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_size</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program.html">IGPProgram</a>&#160;</td>
          <td class="paramname"><em>a_ind</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00094">94</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00158">org.jgap.gp.impl.ProgramChromosome.init()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ad014d2d525e7210edd3fa35146432f67_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ad014d2d525e7210edd3fa35146432f67_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ad014d2d525e7210edd3fa35146432f67_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ad014d2d525e7210edd3fa35146432f67_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1d821adfada0d4ba257d1ac1cf15969a" title="org.jgap.gp.impl.Program\lChromosome.init" alt="" coords="286,5,453,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a57f6a4989aa1e01bda2c1f77cc086c4a" title="org.jgap.gp.BaseGPChromosome.get\lGPConfiguration" alt="" coords="502,5,741,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a6895103538a0e05c6693fd2f6711b50f"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.gp.impl.ProgramChromosome.ProgramChromosome </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html">GPConfiguration</a>&#160;</td>
          <td class="paramname"><em>a_conf</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_size</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td>
          <td class="paramname"><em>a_functionSet</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Class[]&#160;</td>
          <td class="paramname"><em>a_argTypes</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_program.html">IGPProgram</a>&#160;</td>
          <td class="paramname"><em>a_ind</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00108">108</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00055">org.jgap.gp.impl.ProgramChromosome.argTypes</a>, <a class="el" href="_program_chromosome_8java_source.html#l00158">org.jgap.gp.impl.ProgramChromosome.init()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00045">org.jgap.gp.impl.ProgramChromosome.m_functionSet</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a6895103538a0e05c6693fd2f6711b50f_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a6895103538a0e05c6693fd2f6711b50f_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a6895103538a0e05c6693fd2f6711b50f_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a6895103538a0e05c6693fd2f6711b50f_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1d821adfada0d4ba257d1ac1cf15969a" title="org.jgap.gp.impl.Program\lChromosome.init" alt="" coords="286,5,453,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a57f6a4989aa1e01bda2c1f77cc086c4a" title="org.jgap.gp.BaseGPChromosome.get\lGPConfiguration" alt="" coords="502,5,741,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ac4ca2284886dd13bfe27e4802a40d5fd"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.gp.impl.ProgramChromosome.ProgramChromosome </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html">GPConfiguration</a>&#160;</td>
          <td class="paramname"><em>a_conf</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td>
          <td class="paramname"><em>a_initialGenes</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00126">126</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00158">org.jgap.gp.impl.ProgramChromosome.init()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ac4ca2284886dd13bfe27e4802a40d5fd_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ac4ca2284886dd13bfe27e4802a40d5fd_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ac4ca2284886dd13bfe27e4802a40d5fd_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ac4ca2284886dd13bfe27e4802a40d5fd_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1d821adfada0d4ba257d1ac1cf15969a" title="org.jgap.gp.impl.Program\lChromosome.init" alt="" coords="286,5,453,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a57f6a4989aa1e01bda2c1f77cc086c4a" title="org.jgap.gp.BaseGPChromosome.get\lGPConfiguration" alt="" coords="502,5,741,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="aafd1f551ada36c7835737f0e3daeead4"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.gp.impl.ProgramChromosome.ProgramChromosome </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html">GPConfiguration</a>&#160;</td>
          <td class="paramname"><em>a_conf</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00139">139</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00158">org.jgap.gp.impl.ProgramChromosome.init()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aafd1f551ada36c7835737f0e3daeead4_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aafd1f551ada36c7835737f0e3daeead4_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aafd1f551ada36c7835737f0e3daeead4_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aafd1f551ada36c7835737f0e3daeead4_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1d821adfada0d4ba257d1ac1cf15969a" title="org.jgap.gp.impl.Program\lChromosome.init" alt="" coords="286,5,453,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a57f6a4989aa1e01bda2c1f77cc086c4a" title="org.jgap.gp.BaseGPChromosome.get\lGPConfiguration" alt="" coords="502,5,741,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a49e32c4ee7a42b4e7fbddae454b325c2"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.gp.impl.ProgramChromosome.ProgramChromosome </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Default constructor. Only use for dynamic instantiation.</p>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00153">153</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00173">org.jgap.gp.impl.ProgramChromosome.clone()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l01200">org.jgap.gp.impl.ProgramChromosome.compareTo()</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="a274fc584475b32253ee2ced3897e7547"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.gp.impl.ProgramChromosome.cleanup </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Clean up the chromosome.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome.html#a4aab1f31cdecde983111b94e5ecc4193">org.jgap.gp.IGPChromosome</a>.</p>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00216">216</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00964">org.jgap.gp.impl.ProgramChromosome.execute_boolean()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01100">org.jgap.gp.impl.ProgramChromosome.execute_double()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01076">org.jgap.gp.impl.ProgramChromosome.execute_float()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01026">org.jgap.gp.impl.ProgramChromosome.execute_int()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01052">org.jgap.gp.impl.ProgramChromosome.execute_long()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01125">org.jgap.gp.impl.ProgramChromosome.execute_object()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l01000">org.jgap.gp.impl.ProgramChromosome.execute_void()</a>.</p>

</div>
</div>
<a class="anchor" id="af25451cf53447cdec834e5be9ac77f79"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">synchronized Object org.jgap.gp.impl.ProgramChromosome.clone </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00173">173</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00055">org.jgap.gp.impl.ProgramChromosome.argTypes</a>, <a class="el" href="_program_chromosome_8java_source.html#l00830">org.jgap.gp.impl.ProgramChromosome.getFunctionSet()</a>, <a class="el" href="_base_g_p_chromosome_8java_source.html#l00357">org.jgap.gp.BaseGPChromosome.getGPConfiguration()</a>, <a class="el" href="_base_g_p_chromosome_8java_source.html#l00059">org.jgap.gp.BaseGPChromosome.getIndividual()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00050">org.jgap.gp.impl.ProgramChromosome.m_depth</a>, <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>, <a class="el" href="_program_chromosome_8java_source.html#l00153">org.jgap.gp.impl.ProgramChromosome.ProgramChromosome()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l01178">org.jgap.gp.impl.ProgramChromosome.size()</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00591">org.jgap.gp.impl.ProgramChromosome.growOrFullNode()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_af25451cf53447cdec834e5be9ac77f79_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_af25451cf53447cdec834e5be9ac77f79_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_af25451cf53447cdec834e5be9ac77f79_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_af25451cf53447cdec834e5be9ac77f79_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a8640ae6ff64b6b0fe5d0972a554104f0" title="org.jgap.gp.impl.Program\lChromosome.getFunctionSet" alt="" coords="245,5,437,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a57f6a4989aa1e01bda2c1f77cc086c4a" title="org.jgap.gp.BaseGPChromosome.get\lGPConfiguration" alt="" coords="222,70,461,111"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a13d8f2e469a316fd02a863a1d924ad5d" title="org.jgap.gp.BaseGPChromosome.get\lIndividual" alt="" coords="222,135,461,177"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a49e32c4ee7a42b4e7fbddae454b325c2" title="org.jgap.gp.impl.Program\lChromosome.ProgramChromosome" alt="" coords="226,201,457,242"/><area shape="rect" id="node11" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ade35d12aa45f98f2acea90b52bda3375" title="org.jgap.gp.impl.Program\lChromosome.size" alt="" coords="258,266,425,307"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ac9896e142c28c71711133698041fb72f"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.gp.impl.ProgramChromosome.compareTo </td>
          <td>(</td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>a_other</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Compares the given chromosome to this chromosome. This chromosome is considered to be "less than" the given chromosome if it has a fewer number of genes or if any of its gene values (alleles) are less than their corresponding gene values in the other chromosome.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_other</td><td>the chromosome against which to compare this chromosome </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>a negative number if this chromosome is "less than" the given chromosome, zero if they are equal to each other, and a positive number if this chromosome is "greater than" the given chromosome</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000158">Todo:</a></b></dt><dd>compare m_functionSet </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01200">1200</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l01311">org.jgap.gp.impl.ProgramChromosome.getApplicationData()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01295">org.jgap.gp.impl.ProgramChromosome.isCompareApplicationData()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>, <a class="el" href="_program_chromosome_8java_source.html#l00153">org.jgap.gp.impl.ProgramChromosome.ProgramChromosome()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l01178">org.jgap.gp.impl.ProgramChromosome.size()</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l01267">org.jgap.gp.impl.ProgramChromosome.equals()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ac9896e142c28c71711133698041fb72f_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ac9896e142c28c71711133698041fb72f_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ac9896e142c28c71711133698041fb72f_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ac9896e142c28c71711133698041fb72f_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a34507350def1243e665819c4914738f9" title="org.jgap.gp.impl.Program\lChromosome.getApplicationData" alt="" coords="246,5,461,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a24c91c7e53c8ad7be9d7badee0d1aad5" title="org.jgap.gp.impl.Program\lChromosome.isCompareApplicationData" alt="" coords="224,70,483,111"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a49e32c4ee7a42b4e7fbddae454b325c2" title="org.jgap.gp.impl.Program\lChromosome.ProgramChromosome" alt="" coords="238,135,469,177"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ade35d12aa45f98f2acea90b52bda3375" title="org.jgap.gp.impl.Program\lChromosome.size" alt="" coords="270,201,437,242"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a32591e6bcdec0c22383f482ca7c9b39a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.gp.impl.ProgramChromosome.contains </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td>
          <td class="paramname"><em>a_functionSet</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>&#160;</td>
          <td class="paramname"><em>a_function</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Checks if a function or terminal of the class of the given instance exists within a_functionSet</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_functionSet</td><td>the set of functions to check </td></tr>
    <tr><td class="paramname">a_function</td><td>the commandgene to check for via its class </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>true if a_function's class exists within a_functionSet</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.6 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01496">1496</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_single_node_validator_8java_source.html#l00035">examples.gp.tictactoe.SingleNodeValidator.isAllowed()</a>.</p>

</div>
</div>
<a class="anchor" id="af2a24457cb0fedb950f59a919117cdab"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.gp.impl.ProgramChromosome.contains </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td>
          <td class="paramname"><em>a_functionSet</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Class&#160;</td>
          <td class="paramname"><em>a_function</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Checks if a function or terminal of the class of the given instance exists within a_functionSet </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_functionSet</td><td>the set of functions to check </td></tr>
    <tr><td class="paramname">a_function</td><td>the commandgene to check for via its class </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>true if a_function's class exists within a_functionSet</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.6 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01511">1511</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

</div>
</div>
<a class="anchor" id="a6b4c3936426388aeb249ec73ee5131ed"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.gp.impl.ProgramChromosome.contains </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td>
          <td class="paramname"><em>a_functionSet</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>&#160;</td>
          <td class="paramname"><em>a_function</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_limit</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Checks if a function or terminal of the class of the given instance exists within a_functionSet at least a_limit times</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_functionSet</td><td>the set of functions to check </td></tr>
    <tr><td class="paramname">a_function</td><td>the commandgene to check for via its class </td></tr>
    <tr><td class="paramname">a_limit</td><td>if a_limit occurrences of a_function were detected, exit </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>true if a_function's class exists at least a_limit times within a_functionSet</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.6 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01534">1534</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

</div>
</div>
<a class="anchor" id="a5a09ecbda2b198791e979d3dae3b0f01"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> org.jgap.gp.impl.ProgramChromosome.createGene </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_geneClassName</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_persistentRepresentation</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws Exception</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Creates a new instance of gene.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_geneClassName</td><td>name of the gene class </td></tr>
    <tr><td class="paramname">a_persistentRepresentation</td><td>persistent representation of the gene to create (could be obtained via getPersistentRepresentation)</td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>newly created gene </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Exception</td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.3 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01437">1437</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_base_g_p_chromosome_8java_source.html#l00357">org.jgap.gp.BaseGPChromosome.getGPConfiguration()</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l01393">org.jgap.gp.impl.ProgramChromosome.setValueFromPersistentRepresentation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a5a09ecbda2b198791e979d3dae3b0f01_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a5a09ecbda2b198791e979d3dae3b0f01_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a5a09ecbda2b198791e979d3dae3b0f01_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a5a09ecbda2b198791e979d3dae3b0f01_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a57f6a4989aa1e01bda2c1f77cc086c4a" title="org.jgap.gp.BaseGPChromosome.get\lGPConfiguration" alt="" coords="225,5,463,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a7a41513bb42bc79e7708c46afd61b4a4"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.gp.impl.ProgramChromosome.decode </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_string</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01352">1352</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l01393">org.jgap.gp.impl.ProgramChromosome.setValueFromPersistentRepresentation()</a>.</p>

</div>
</div>
<a class="anchor" id="a1ceca8b2f7a033bf9ab63f70b438c211"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.gp.impl.ProgramChromosome.encode </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_string</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01348">1348</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l01363">org.jgap.gp.impl.ProgramChromosome.getPersistentRepresentation()</a>.</p>

</div>
</div>
<a class="anchor" id="a39b343e4b3d1e851adef34ec081814ee"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.gp.impl.ProgramChromosome.equals </td>
          <td>(</td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>a_other</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Compares this chromosome against the specified object.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_other</td><td>the object to compare against </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>true: if the objects are the same, false otherwise</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01267">1267</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l01200">org.jgap.gp.impl.ProgramChromosome.compareTo()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a39b343e4b3d1e851adef34ec081814ee_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a39b343e4b3d1e851adef34ec081814ee_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a39b343e4b3d1e851adef34ec081814ee_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a39b343e4b3d1e851adef34ec081814ee_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ac9896e142c28c71711133698041fb72f" title="org.jgap.gp.impl.Program\lChromosome.compareTo" alt="" coords="222,103,391,145"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a34507350def1243e665819c4914738f9" title="org.jgap.gp.impl.Program\lChromosome.getApplicationData" alt="" coords="462,5,677,46"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a24c91c7e53c8ad7be9d7badee0d1aad5" title="org.jgap.gp.impl.Program\lChromosome.isCompareApplicationData" alt="" coords="440,70,699,111"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a49e32c4ee7a42b4e7fbddae454b325c2" title="org.jgap.gp.impl.Program\lChromosome.ProgramChromosome" alt="" coords="454,135,685,177"/><area shape="rect" id="node11" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ade35d12aa45f98f2acea90b52bda3375" title="org.jgap.gp.impl.Program\lChromosome.size" alt="" coords="486,201,653,242"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a2aac4f2020dfe3929473d163929b1971"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.gp.impl.ProgramChromosome.execute </td>
          <td>(</td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Executes this node without knowing its return type.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">args</td><td>the arguments for execution </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the Object which wraps the return value of this node, or null if the return type is null or unknown</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01149">1149</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

</div>
</div>
<a class="anchor" id="a376d8beedfc62a18b9d105d280cc141d"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.gp.impl.ProgramChromosome.execute </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>n</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>child</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01153">1153</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l01125">org.jgap.gp.impl.ProgramChromosome.execute_object()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a376d8beedfc62a18b9d105d280cc141d_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a376d8beedfc62a18b9d105d280cc141d_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a376d8beedfc62a18b9d105d280cc141d_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a376d8beedfc62a18b9d105d280cc141d_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a49bbf47534a08674eefdbd40c4103534" title="org.jgap.gp.impl.Program\lChromosome.execute_object" alt="" coords="221,5,413,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a274fc584475b32253ee2ced3897e7547" title="org.jgap.gp.impl.Program\lChromosome.cleanup" alt="" coords="462,5,629,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ac02462405fa19e5b91d8b7c99f91c872"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.gp.impl.ProgramChromosome.execute_boolean </td>
          <td>(</td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Executes this node as a boolean.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">args</td><td>the arguments for execution </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the boolean return value of this node </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">UnsupportedOperationException</td><td>if the type of this node is not boolean</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00964">964</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00216">org.jgap.gp.impl.ProgramChromosome.cleanup()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ac02462405fa19e5b91d8b7c99f91c872_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ac02462405fa19e5b91d8b7c99f91c872_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ac02462405fa19e5b91d8b7c99f91c872_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_ac02462405fa19e5b91d8b7c99f91c872_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a274fc584475b32253ee2ced3897e7547" title="org.jgap.gp.impl.Program\lChromosome.cleanup" alt="" coords="257,5,423,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a500cdb04cca5286bb9e10d41abba324f"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.gp.impl.ProgramChromosome.execute_boolean </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>n</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>child</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Executes this node as a boolean.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">n</td><td>the index of the parent node </td></tr>
    <tr><td class="paramname">child</td><td>the child number of the node to execute </td></tr>
    <tr><td class="paramname">args</td><td>the arguments for execution </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the boolean return value of this node </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">UnsupportedOperationException</td><td>if the type of this node is not boolean</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00983">983</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00786">org.jgap.gp.impl.ProgramChromosome.getChild()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a500cdb04cca5286bb9e10d41abba324f_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a500cdb04cca5286bb9e10d41abba324f_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a500cdb04cca5286bb9e10d41abba324f_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a500cdb04cca5286bb9e10d41abba324f_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aae34fa625e78f8be2de585462d3bbcf8" title="org.jgap.gp.impl.Program\lChromosome.getChild" alt="" coords="257,5,423,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9814ee68c4893d979b6d6c702f754ed6" title="org.jgap.gp.impl.Program\lChromosome.getFunctions" alt="" coords="472,5,651,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a9462046182cf58ed4d59729fa584586a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double org.jgap.gp.impl.ProgramChromosome.execute_double </td>
          <td>(</td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Executes this node as a double.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">args</td><td>the arguments for execution </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the double return value of this node </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">UnsupportedOperationException</td><td>if this node's type is not double</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01100">1100</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00216">org.jgap.gp.impl.ProgramChromosome.cleanup()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a9462046182cf58ed4d59729fa584586a_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a9462046182cf58ed4d59729fa584586a_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a9462046182cf58ed4d59729fa584586a_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a9462046182cf58ed4d59729fa584586a_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a274fc584475b32253ee2ced3897e7547" title="org.jgap.gp.impl.Program\lChromosome.cleanup" alt="" coords="249,5,415,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="aca61c643bfcfc2e3fbbba2d6ef55e170"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double org.jgap.gp.impl.ProgramChromosome.execute_double </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>n</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>child</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01106">1106</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00786">org.jgap.gp.impl.ProgramChromosome.getChild()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aca61c643bfcfc2e3fbbba2d6ef55e170_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aca61c643bfcfc2e3fbbba2d6ef55e170_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aca61c643bfcfc2e3fbbba2d6ef55e170_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aca61c643bfcfc2e3fbbba2d6ef55e170_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aae34fa625e78f8be2de585462d3bbcf8" title="org.jgap.gp.impl.Program\lChromosome.getChild" alt="" coords="249,5,415,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9814ee68c4893d979b6d6c702f754ed6" title="org.jgap.gp.impl.Program\lChromosome.getFunctions" alt="" coords="464,5,643,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="acbd9e0bc12ba6077893ffc9d1ab3d085"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">float org.jgap.gp.impl.ProgramChromosome.execute_float </td>
          <td>(</td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Executes this node as a float.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">args</td><td>the arguments for execution </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the float return value of this node </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">UnsupportedOperationException</td><td>if the type of this node is not float</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01076">1076</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00216">org.jgap.gp.impl.ProgramChromosome.cleanup()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_acbd9e0bc12ba6077893ffc9d1ab3d085_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_acbd9e0bc12ba6077893ffc9d1ab3d085_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_acbd9e0bc12ba6077893ffc9d1ab3d085_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_acbd9e0bc12ba6077893ffc9d1ab3d085_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a274fc584475b32253ee2ced3897e7547" title="org.jgap.gp.impl.Program\lChromosome.cleanup" alt="" coords="238,5,405,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a8b8e3b3cdda9a453f7854418edfbe0c9"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">float org.jgap.gp.impl.ProgramChromosome.execute_float </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>n</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>child</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01082">1082</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00786">org.jgap.gp.impl.ProgramChromosome.getChild()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a8b8e3b3cdda9a453f7854418edfbe0c9_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a8b8e3b3cdda9a453f7854418edfbe0c9_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a8b8e3b3cdda9a453f7854418edfbe0c9_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a8b8e3b3cdda9a453f7854418edfbe0c9_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aae34fa625e78f8be2de585462d3bbcf8" title="org.jgap.gp.impl.Program\lChromosome.getChild" alt="" coords="238,5,405,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9814ee68c4893d979b6d6c702f754ed6" title="org.jgap.gp.impl.Program\lChromosome.getFunctions" alt="" coords="453,5,632,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a3d0cb8958612150faaab36f418a8777e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.gp.impl.ProgramChromosome.execute_int </td>
          <td>(</td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Executes this node as an integer.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">args</td><td>the arguments for execution </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the integer return value of this node </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">UnsupportedOperationException</td><td>if the type of this node is not integer</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01026">1026</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00216">org.jgap.gp.impl.ProgramChromosome.cleanup()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a3d0cb8958612150faaab36f418a8777e_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a3d0cb8958612150faaab36f418a8777e_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a3d0cb8958612150faaab36f418a8777e_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a3d0cb8958612150faaab36f418a8777e_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a274fc584475b32253ee2ced3897e7547" title="org.jgap.gp.impl.Program\lChromosome.cleanup" alt="" coords="225,5,391,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="affb613bc931c4b3be35fc7cca9ed1feb"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.gp.impl.ProgramChromosome.execute_int </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>n</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>child</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01032">1032</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00786">org.jgap.gp.impl.ProgramChromosome.getChild()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_affb613bc931c4b3be35fc7cca9ed1feb_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_affb613bc931c4b3be35fc7cca9ed1feb_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_affb613bc931c4b3be35fc7cca9ed1feb_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_affb613bc931c4b3be35fc7cca9ed1feb_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aae34fa625e78f8be2de585462d3bbcf8" title="org.jgap.gp.impl.Program\lChromosome.getChild" alt="" coords="225,5,391,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9814ee68c4893d979b6d6c702f754ed6" title="org.jgap.gp.impl.Program\lChromosome.getFunctions" alt="" coords="440,5,619,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a8296c9d5187129c039ad4053d4d70ea4"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">long org.jgap.gp.impl.ProgramChromosome.execute_long </td>
          <td>(</td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Executes this node as a long.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">args</td><td>the arguments for execution </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the long return value of this node </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">UnsupportedOperationException</td><td>if the type of this node is not long</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01052">1052</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00216">org.jgap.gp.impl.ProgramChromosome.cleanup()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a8296c9d5187129c039ad4053d4d70ea4_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a8296c9d5187129c039ad4053d4d70ea4_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a8296c9d5187129c039ad4053d4d70ea4_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a8296c9d5187129c039ad4053d4d70ea4_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a274fc584475b32253ee2ced3897e7547" title="org.jgap.gp.impl.Program\lChromosome.cleanup" alt="" coords="235,5,402,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a63c4c4c86434ae1a0c9f4febe21e6572"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">long org.jgap.gp.impl.ProgramChromosome.execute_long </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>n</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>child</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01058">1058</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00786">org.jgap.gp.impl.ProgramChromosome.getChild()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a63c4c4c86434ae1a0c9f4febe21e6572_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a63c4c4c86434ae1a0c9f4febe21e6572_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a63c4c4c86434ae1a0c9f4febe21e6572_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a63c4c4c86434ae1a0c9f4febe21e6572_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aae34fa625e78f8be2de585462d3bbcf8" title="org.jgap.gp.impl.Program\lChromosome.getChild" alt="" coords="235,5,402,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9814ee68c4893d979b6d6c702f754ed6" title="org.jgap.gp.impl.Program\lChromosome.getFunctions" alt="" coords="451,5,629,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a49bbf47534a08674eefdbd40c4103534"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.gp.impl.ProgramChromosome.execute_object </td>
          <td>(</td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Executes this node as an object.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">args</td><td>the arguments for execution </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the object return value of this node </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">UnsupportedOperationException</td><td>if the type of this node is not of type Object</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01125">1125</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00216">org.jgap.gp.impl.ProgramChromosome.cleanup()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l01153">org.jgap.gp.impl.ProgramChromosome.execute()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a49bbf47534a08674eefdbd40c4103534_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a49bbf47534a08674eefdbd40c4103534_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a49bbf47534a08674eefdbd40c4103534_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a49bbf47534a08674eefdbd40c4103534_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a274fc584475b32253ee2ced3897e7547" title="org.jgap.gp.impl.Program\lChromosome.cleanup" alt="" coords="246,5,413,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="aef96a42e1986888e049781e8202b8df3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.gp.impl.ProgramChromosome.execute_object </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>n</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>child</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01131">1131</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00786">org.jgap.gp.impl.ProgramChromosome.getChild()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aef96a42e1986888e049781e8202b8df3_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aef96a42e1986888e049781e8202b8df3_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aef96a42e1986888e049781e8202b8df3_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aef96a42e1986888e049781e8202b8df3_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aae34fa625e78f8be2de585462d3bbcf8" title="org.jgap.gp.impl.Program\lChromosome.getChild" alt="" coords="246,5,413,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9814ee68c4893d979b6d6c702f754ed6" title="org.jgap.gp.impl.Program\lChromosome.getFunctions" alt="" coords="461,5,640,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="aa493171a9ea6d5a0973337d5394cf973"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.gp.impl.ProgramChromosome.execute_void </td>
          <td>(</td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Executes this node, returning nothing.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">args</td><td>the arguments for execution </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">UnsupportedOperationException</td><td>if the type of this node is not void</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01000">1000</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00216">org.jgap.gp.impl.ProgramChromosome.cleanup()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aa493171a9ea6d5a0973337d5394cf973_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aa493171a9ea6d5a0973337d5394cf973_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aa493171a9ea6d5a0973337d5394cf973_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aa493171a9ea6d5a0973337d5394cf973_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a274fc584475b32253ee2ced3897e7547" title="org.jgap.gp.impl.Program\lChromosome.cleanup" alt="" coords="235,5,402,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a92fd1beeb17969fc404237251003e388"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.gp.impl.ProgramChromosome.execute_void </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>n</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>child</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Object[]&#160;</td>
          <td class="paramname"><em>args</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01005">1005</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00786">org.jgap.gp.impl.ProgramChromosome.getChild()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a92fd1beeb17969fc404237251003e388_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a92fd1beeb17969fc404237251003e388_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a92fd1beeb17969fc404237251003e388_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a92fd1beeb17969fc404237251003e388_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aae34fa625e78f8be2de585462d3bbcf8" title="org.jgap.gp.impl.Program\lChromosome.getChild" alt="" coords="235,5,402,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9814ee68c4893d979b6d6c702f754ed6" title="org.jgap.gp.impl.Program\lChromosome.getFunctions" alt="" coords="451,5,629,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a34507350def1243e665819c4914738f9"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.gp.impl.ProgramChromosome.getApplicationData </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves the application-specific data that is attached to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. Attaching application-specific data may be useful for some applications when it comes time to evaluate this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in the fitness function. JGAP ignores this data functionally.</p>
<dl class="section return"><dt>Returns</dt><dd>the application-specific data previously attached to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>, or null if there is no data attached</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01311">1311</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00073">org.jgap.gp.impl.ProgramChromosome.m_applicationData</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l01200">org.jgap.gp.impl.ProgramChromosome.compareTo()</a>.</p>

</div>
</div>
<a class="anchor" id="a96b8adcf50530c6bfbadda604b57314f"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Class [] org.jgap.gp.impl.ProgramChromosome.getArgTypes </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01164">1164</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00055">org.jgap.gp.impl.ProgramChromosome.argTypes</a>.</p>

</div>
</div>
<a class="anchor" id="a1873ffcdf7eaaf635b85b70f5a4ac59a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.gp.impl.ProgramChromosome.getArity </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01168">1168</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00440">org.jgap.gp.impl.ProgramChromosome.isPossible()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00490">org.jgap.gp.impl.ProgramChromosome.selectNode()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00298">org.jgap.gp.impl.ProgramChromosome.toString()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00409">org.jgap.gp.impl.ProgramChromosome.toStringDebug()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00350">org.jgap.gp.impl.ProgramChromosome.toStringNorm()</a>.</p>

</div>
</div>
<a class="anchor" id="a88f1eb9dac4272ec7f7274f0cd66d763"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.gp.impl.ProgramChromosome.getBusinessKey </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>business key of the chromosome</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.4 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_business_key.html#ab1b12087d8726e9020441a1fed6be0f8">org.jgap.IBusinessKey</a>.</p>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00398">398</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00350">org.jgap.gp.impl.ProgramChromosome.toStringNorm()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a88f1eb9dac4272ec7f7274f0cd66d763_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a88f1eb9dac4272ec7f7274f0cd66d763_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a88f1eb9dac4272ec7f7274f0cd66d763_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a88f1eb9dac4272ec7f7274f0cd66d763_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a56c22c94166729d37888a6daaef76bb3" title="org.jgap.gp.impl.Program\lChromosome.toStringNorm" alt="" coords="253,129,435,170"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a13d8f2e469a316fd02a863a1d924ad5d" title="org.jgap.gp.BaseGPChromosome.get\lIndividual" alt="" coords="699,5,938,46"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1873ffcdf7eaaf635b85b70f5a4ac59a" title="org.jgap.gp.impl.Program\lChromosome.getArity" alt="" coords="735,70,902,111"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9814ee68c4893d979b6d6c702f754ed6" title="org.jgap.gp.impl.Program\lChromosome.getFunctions" alt="" coords="987,187,1165,229"/><area shape="rect" id="node11" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#addf578c58866aaf237a11d91308d0c04" title="org.jgap.gp.impl.Program\lChromosome.toString" alt="" coords="483,129,650,170"/><area shape="rect" id="node16" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aae34fa625e78f8be2de585462d3bbcf8" title="org.jgap.gp.impl.Program\lChromosome.getChild" alt="" coords="735,187,902,229"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="aae34fa625e78f8be2de585462d3bbcf8"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.gp.impl.ProgramChromosome.getChild </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_index</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_child</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Gets the a_child'th child of the a_index'th node in this chromosome. This is the same as the a_child'th node whose depth is one more than the depth of the a_index'th node.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_index</td><td>the node number of the parent </td></tr>
    <tr><td class="paramname">a_child</td><td>the child number (starting from 0) of the parent </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the node number of the child, or -1 if not found</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.01 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome.html#a4e13390ee43bbe0a8a491fddcadae226">org.jgap.gp.IGPChromosome</a>.</p>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00786">786</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00838">org.jgap.gp.impl.ProgramChromosome.getFunctions()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00050">org.jgap.gp.impl.ProgramChromosome.m_depth</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00983">org.jgap.gp.impl.ProgramChromosome.execute_boolean()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01106">org.jgap.gp.impl.ProgramChromosome.execute_double()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01082">org.jgap.gp.impl.ProgramChromosome.execute_float()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01032">org.jgap.gp.impl.ProgramChromosome.execute_int()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01058">org.jgap.gp.impl.ProgramChromosome.execute_long()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01131">org.jgap.gp.impl.ProgramChromosome.execute_object()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01005">org.jgap.gp.impl.ProgramChromosome.execute_void()</a>, <a class="el" href="_j_g_a_p_tree_node_8java_source.html#l00054">org.jgap.gp.impl.JGAPTreeNode.getChildCount()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00298">org.jgap.gp.impl.ProgramChromosome.toString()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00409">org.jgap.gp.impl.ProgramChromosome.toStringDebug()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00350">org.jgap.gp.impl.ProgramChromosome.toStringNorm()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aae34fa625e78f8be2de585462d3bbcf8_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aae34fa625e78f8be2de585462d3bbcf8_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aae34fa625e78f8be2de585462d3bbcf8_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_aae34fa625e78f8be2de585462d3bbcf8_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9814ee68c4893d979b6d6c702f754ed6" title="org.jgap.gp.impl.Program\lChromosome.getFunctions" alt="" coords="221,5,400,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a885c955a0bb1f22ad404c242d51cba45"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.gp.impl.ProgramChromosome.getChild </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>&#160;</td>
          <td class="paramname"><em>a_node</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_child</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00804">804</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00838">org.jgap.gp.impl.ProgramChromosome.getFunctions()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00050">org.jgap.gp.impl.ProgramChromosome.m_depth</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a885c955a0bb1f22ad404c242d51cba45_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a885c955a0bb1f22ad404c242d51cba45_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a885c955a0bb1f22ad404c242d51cba45_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a885c955a0bb1f22ad404c242d51cba45_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9814ee68c4893d979b6d6c702f754ed6" title="org.jgap.gp.impl.Program\lChromosome.getFunctions" alt="" coords="221,5,400,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ac6c8388003452adf7ebd2b0e70c916ac"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.gp.impl.ProgramChromosome.getDepth </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_index</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Gets the depth of the branch starting at the a_index'th node.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_index</td><td>the index of the node at which to check the depth </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the depth of the branch starting at the a_index'th node</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00877">877</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00050">org.jgap.gp.impl.ProgramChromosome.m_depth</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00490">org.jgap.gp.impl.ProgramChromosome.selectNode()</a>, <a class="el" href="_g_p_program_test_8java_source.html#l00046">org.jgap.gp.impl.GPProgramTest.testExecute_0()</a>, <a class="el" href="_program_chromosome_test_8java_source.html#l00673">org.jgap.gp.impl.ProgramChromosomeTest.testGetDepth_0()</a>, <a class="el" href="_program_chromosome_test_8java_source.html#l00691">org.jgap.gp.impl.ProgramChromosomeTest.testGetDepth_1()</a>, <a class="el" href="_program_chromosome_test_8java_source.html#l00711">org.jgap.gp.impl.ProgramChromosomeTest.testGetDepth_2()</a>, and <a class="el" href="_program_chromosome_test_8java_source.html#l00173">org.jgap.gp.impl.ProgramChromosomeTest.testGrowNode_3()</a>.</p>

</div>
</div>
<a class="anchor" id="a9814ee68c4893d979b6d6c702f754ed6"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> [] org.jgap.gp.impl.ProgramChromosome.getFunctions </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome.html#ad7a5cd77b01e717993466858667604ce">org.jgap.gp.IGPChromosome</a>.</p>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00838">838</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00786">org.jgap.gp.impl.ProgramChromosome.getChild()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00298">org.jgap.gp.impl.ProgramChromosome.toString()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00350">org.jgap.gp.impl.ProgramChromosome.toStringNorm()</a>.</p>

</div>
</div>
<a class="anchor" id="a8640ae6ff64b6b0fe5d0972a554104f0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> [] org.jgap.gp.impl.ProgramChromosome.getFunctionSet </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>set of <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> instances allowed</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2.1 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome.html#a340247497b10eadfc097435d5a87b24d">org.jgap.gp.IGPChromosome</a>.</p>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00830">830</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00045">org.jgap.gp.impl.ProgramChromosome.m_functionSet</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00173">org.jgap.gp.impl.ProgramChromosome.clone()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00240">org.jgap.gp.impl.ProgramChromosome.growOrFull()</a>.</p>

</div>
</div>
<a class="anchor" id="a0204f8ead0682a3210428c6291456be7"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">synchronized <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> org.jgap.gp.impl.ProgramChromosome.getGene </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_locus</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Returns the <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> at the given index (locus) within the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. The first gene is at index zero and the last gene is at the index equal to the size of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> - 1.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_locus</td><td>index of the gene value to be returned </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> at the given index</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01326">1326</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p>Referenced by <a class="el" href="_put_stone1_8java_source.html#l00050">examples.gp.tictactoe.PutStone1.execute_void()</a>, <a class="el" href="_branch_typing_cross_test_8java_source.html#l00047">org.jgap.gp.impl.BranchTypingCrossTest.testOperate_0()</a>, <a class="el" href="_branch_typing_cross_test_8java_source.html#l00119">org.jgap.gp.impl.BranchTypingCrossTest.testOperate_1()</a>, <a class="el" href="_branch_typing_cross_test_8java_source.html#l00191">org.jgap.gp.impl.BranchTypingCrossTest.testOperate_2()</a>, and <a class="el" href="_g_p_genotype_test_8java_source.html#l00236">org.jgap.gp.impl.GPGenotypeTest.testReferenceProblem_0()</a>.</p>

</div>
</div>
<a class="anchor" id="a9dcb6045fafdd72b8241f3f1634ec433"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> org.jgap.gp.impl.ProgramChromosome.getNode </td>
          <td>(</td>
          <td class="paramtype">Class&#160;</td>
          <td class="paramname"><em>a_type</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_exactMatch</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Checks whether a node with a given type is contained in the program.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_type</td><td>the type to look for </td></tr>
    <tr><td class="paramname">a_exactMatch</td><td>true: look for exactly the given type: false: also look for sub types </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>true specific node found</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2.1 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00924">924</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00748">org.jgap.gp.impl.ProgramChromosome.redepth()</a>, <a class="el" href="_program_chromosome_test_8java_source.html#l00062">org.jgap.gp.impl.ProgramChromosomeTest.testGrowNode_0()</a>, <a class="el" href="_program_chromosome_test_8java_source.html#l00096">org.jgap.gp.impl.ProgramChromosomeTest.testGrowNode_1()</a>, <a class="el" href="_program_chromosome_test_8java_source.html#l00135">org.jgap.gp.impl.ProgramChromosomeTest.testGrowNode_2()</a>, <a class="el" href="_program_chromosome_test_8java_source.html#l00173">org.jgap.gp.impl.ProgramChromosomeTest.testGrowNode_3()</a>, <a class="el" href="_program_chromosome_test_8java_source.html#l00214">org.jgap.gp.impl.ProgramChromosomeTest.testGrowNode_4()</a>, <a class="el" href="_program_chromosome_test_8java_source.html#l00242">org.jgap.gp.impl.ProgramChromosomeTest.testGrowNode_4_2()</a>, <a class="el" href="_program_chromosome_test_8java_source.html#l00280">org.jgap.gp.impl.ProgramChromosomeTest.testGrowNode_4_3()</a>, and <a class="el" href="_program_chromosome_test_8java_source.html#l00316">org.jgap.gp.impl.ProgramChromosomeTest.testGrowNode_5()</a>.</p>

</div>
</div>
<a class="anchor" id="a2e3b39bebbe5a3f6bb46494341d8d39d"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> org.jgap.gp.impl.ProgramChromosome.getNode </td>
          <td>(</td>
          <td class="paramtype">Class&#160;</td>
          <td class="paramname"><em>a_type</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_exactMatch</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_startIndex</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00928">928</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>, and <a class="el" href="_program_chromosome_8java_source.html#l01178">org.jgap.gp.impl.ProgramChromosome.size()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a2e3b39bebbe5a3f6bb46494341d8d39d_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a2e3b39bebbe5a3f6bb46494341d8d39d_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a2e3b39bebbe5a3f6bb46494341d8d39d_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a2e3b39bebbe5a3f6bb46494341d8d39d_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ade35d12aa45f98f2acea90b52bda3375" title="org.jgap.gp.impl.Program\lChromosome.size" alt="" coords="222,5,389,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a94c73ab9b25ed666e4fabe83e70dd8c4"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.gp.impl.ProgramChromosome.getParentNode </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_child</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Gets the node which is the parent of the given node in this chromosome. If the child is at depth d then the parent is the first function at depth d-1 when iterating backwards through the function list starting from the child.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_child</td><td>the child node </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the parent node, or null if the child is the root node</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00901">901</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00050">org.jgap.gp.impl.ProgramChromosome.m_depth</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p>Referenced by <a class="el" href="_j_g_a_p_tree_node_8java_source.html#l00071">org.jgap.gp.impl.JGAPTreeNode.getParent()</a>.</p>

</div>
</div>
<a class="anchor" id="a6536e99f35877cb53ef5d01ea0c786be"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.gp.impl.ProgramChromosome.getPersistentRepresentation </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>the persistent representation of the chromosome, including all genes</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.3 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome.html#a7fb40e0daa789a54d1a8b3c9488aa60f">org.jgap.gp.IGPChromosome</a>.</p>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01363">1363</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l01348">org.jgap.gp.impl.ProgramChromosome.encode()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a6536e99f35877cb53ef5d01ea0c786be_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a6536e99f35877cb53ef5d01ea0c786be_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a6536e99f35877cb53ef5d01ea0c786be_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a6536e99f35877cb53ef5d01ea0c786be_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1ceca8b2f7a033bf9ab63f70b438c211" title="org.jgap.gp.impl.Program\lChromosome.encode" alt="" coords="323,5,490,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a14b7aa6ec69495ad519f64dbc362488e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.gp.impl.ProgramChromosome.getSize </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_index</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Gets the number of nodes in the branch starting at the a_index'th node.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_index</td><td>the index of the node at which to start counting </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the number of nodes in the branch starting at the a_index'th node</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00856">856</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00050">org.jgap.gp.impl.ProgramChromosome.m_depth</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

</div>
</div>
<a class="anchor" id="a6b798831e4e59f1dbcd58381e090a9d0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.gp.impl.ProgramChromosome.growOrFull </td>
          <td>(</td>
          <td class="paramtype">final int&#160;</td>
          <td class="paramname"><em>a_num</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final int&#160;</td>
          <td class="paramname"><em>a_depth</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final Class&#160;</td>
          <td class="paramname"><em>a_type</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final Class[]&#160;</td>
          <td class="paramname"><em>a_argTypes</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td>
          <td class="paramname"><em>a_functionSet</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_grow</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_tries</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Initialize this chromosome using the grow or the full method.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_num</td><td>the chromosome's index in the individual of this chromosome </td></tr>
    <tr><td class="paramname">a_depth</td><td>the maximum depth of the chromosome to create </td></tr>
    <tr><td class="paramname">a_type</td><td>the type of the chromosome to create </td></tr>
    <tr><td class="paramname">a_argTypes</td><td>the array of argument types for this chromosome </td></tr>
    <tr><td class="paramname">a_functionSet</td><td>the set of nodes valid to pick from </td></tr>
    <tr><td class="paramname">a_grow</td><td>true: use grow method; false: use full method </td></tr>
    <tr><td class="paramname">a_tries</td><td>maximum number of tries for creating a valid program</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00240">240</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00055">org.jgap.gp.impl.ProgramChromosome.argTypes</a>, <a class="el" href="_program_chromosome_8java_source.html#l00830">org.jgap.gp.impl.ProgramChromosome.getFunctionSet()</a>, <a class="el" href="_base_g_p_chromosome_8java_source.html#l00357">org.jgap.gp.BaseGPChromosome.getGPConfiguration()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00591">org.jgap.gp.impl.ProgramChromosome.growOrFullNode()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00045">org.jgap.gp.impl.ProgramChromosome.m_functionSet</a>, <a class="el" href="_program_chromosome_8java_source.html#l00057">org.jgap.gp.impl.ProgramChromosome.m_index</a>, <a class="el" href="_program_chromosome_8java_source.html#l00059">org.jgap.gp.impl.ProgramChromosome.m_maxDepth</a>, <a class="el" href="_program_chromosome_8java_source.html#l00728">org.jgap.gp.impl.ProgramChromosome.redepth()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00834">org.jgap.gp.impl.ProgramChromosome.setFunctionSet()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a6b798831e4e59f1dbcd58381e090a9d0_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a6b798831e4e59f1dbcd58381e090a9d0_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a6b798831e4e59f1dbcd58381e090a9d0_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a6b798831e4e59f1dbcd58381e090a9d0_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a8640ae6ff64b6b0fe5d0972a554104f0" title="org.jgap.gp.impl.Program\lChromosome.getFunctionSet" alt="" coords="837,5,1029,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a57f6a4989aa1e01bda2c1f77cc086c4a" title="org.jgap.gp.BaseGPChromosome.get\lGPConfiguration" alt="" coords="814,201,1053,242"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1ca14815335297629dc3bd4812445405" title="org.jgap.gp.impl.Program\lChromosome.growOrFullNode" alt="" coords="222,401,421,442"/><area shape="rect" id="node51" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a72e5e8ab3c253f4ec7dc10fb17a45180" title="org.jgap.gp.impl.Program\lChromosome.redepth" alt="" coords="238,466,405,507"/><area shape="rect" id="node53" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a81171ec1774c2605dc2080c476768f19" title="org.jgap.gp.impl.Program\lChromosome.setFunctionSet" alt="" coords="225,531,417,573"/><area shape="rect" id="node10" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1e11a46dcd815ed2778c03800abd8e56" title="org.jgap.gp.impl.Program\lChromosome.selectNode" alt="" coords="532,342,703,383"/><area shape="rect" id="node12" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a13d8f2e469a316fd02a863a1d924ad5d" title="org.jgap.gp.BaseGPChromosome.get\lIndividual" alt="" coords="814,266,1053,307"/><area shape="rect" id="node21" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html#a54af1a43cc4c4b0cb84879bb5f6b0eaf" title="org.jgap.gp.impl.GPConfiguration.validateNode" alt="" coords="470,408,765,435"/><area shape="rect" id="node25" href="interfaceorg_1_1jgap_1_1_random_generator.html#a708f273cc71846c2ae973172f20f2c01" title="org.jgap.RandomGenerator.next\lDouble" alt="" coords="514,459,721,501"/><area shape="rect" id="node27" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a92b35be05ab42231ab0842197332fde8" title="org.jgap.gp.BaseGPChromosome.get\lCommandOfClass" alt="" coords="498,525,737,566"/><area shape="rect" id="node31" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html#a6168fe89c9e8f548d8fa50013a9926f1" title="org.jgap.gp.impl.GPConfiguration.get\lDynamizeArityProb" alt="" coords="499,590,735,631"/><area shape="rect" id="node33" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html#a7a2c517e36f51f4e2be5d32fd4b7def2" title="org.jgap.gp.impl.GPConfiguration.is\lNoCommandGeneCloning" alt="" coords="503,655,731,697"/><area shape="rect" id="node35" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#af25451cf53447cdec834e5be9ac77f79" title="org.jgap.gp.impl.Program\lChromosome.clone" alt="" coords="534,123,701,165"/><area shape="rect" id="node45" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html#a764c86ce05fec22c79f4092ecd6a2269" title="org.jgap.gp.CommandGene.get\lChildType" alt="" coords="516,721,719,762"/><area shape="rect" id="node49" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html#a2f2fd7c05e6e948e2fd1a7bd0df37ef1" title="org.jgap.gp.CommandGene.get\lArity" alt="" coords="516,786,719,827"/><area shape="rect" id="node15" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1873ffcdf7eaaf635b85b70f5a4ac59a" title="org.jgap.gp.impl.Program\lChromosome.getArity" alt="" coords="850,331,1017,373"/><area shape="rect" id="node17" href="interfaceorg_1_1jgap_1_1gp_1_1_i_single_node_validator.html#a8c657d357179ee11ac73015301fa2789" title="org.jgap.gp.ISingleNodeValidator.is\lAllowed" alt="" coords="820,397,1047,438"/><area shape="rect" id="node19" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ac6c8388003452adf7ebd2b0e70c916ac" title="org.jgap.gp.impl.Program\lChromosome.getDepth" alt="" coords="850,462,1017,503"/><area shape="rect" id="node23" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html#abc4494312ed10634e36e31a2f620b4be" title="org.jgap.gp.impl.GPConfiguration.get\lNodeValidator" alt="" coords="815,527,1051,569"/><area shape="rect" id="node29" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome.html#ad7a5cd77b01e717993466858667604ce" title="org.jgap.gp.IGPChromosome.get\lFunctions" alt="" coords="827,593,1040,634"/><area shape="rect" id="node40" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a49e32c4ee7a42b4e7fbddae454b325c2" title="org.jgap.gp.impl.Program\lChromosome.ProgramChromosome" alt="" coords="818,70,1049,111"/><area shape="rect" id="node42" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ade35d12aa45f98f2acea90b52bda3375" title="org.jgap.gp.impl.Program\lChromosome.size" alt="" coords="850,135,1017,177"/><area shape="rect" id="node47" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html#a2b466f1bd4416ce97c0d2567be1f65a8" title="org.jgap.gp.CommandGene.get\lReturnType" alt="" coords="832,721,1035,762"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a1ca14815335297629dc3bd4812445405"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> [] org.jgap.gp.impl.ProgramChromosome.growOrFullNode </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_num</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_depth</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Class&#160;</td>
          <td class="paramname"><em>a_returnType</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_subReturnType</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td>
          <td class="paramname"><em>a_functionSet</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>&#160;</td>
          <td class="paramname"><em>a_rootNode</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_recurseLevel</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_grow</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_childNum</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_validateNode</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Create a tree of nodes using the grow or the full method.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_num</td><td>the chromosome's index in the individual of this chromosome </td></tr>
    <tr><td class="paramname">a_depth</td><td>the maximum depth of the tree to create </td></tr>
    <tr><td class="paramname">a_returnType</td><td>the return type the lastly evaluated node must have </td></tr>
    <tr><td class="paramname">a_subReturnType</td><td>the sub return type to look for </td></tr>
    <tr><td class="paramname">a_functionSet</td><td>the set of function valid to pick from </td></tr>
    <tr><td class="paramname">a_rootNode</td><td>null, or parent node of the node to develop </td></tr>
    <tr><td class="paramname">a_recurseLevel</td><td>0 for first call </td></tr>
    <tr><td class="paramname">a_grow</td><td>true: use grow method; false: use full method </td></tr>
    <tr><td class="paramname">a_childNum</td><td>index of the child in the parent node to which it belongs (-1 if node is root node) </td></tr>
    <tr><td class="paramname">a_validateNode</td><td>true: check if node selected is valid (when called recursively a_validateNode is set to true)</td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>possible modified set of functions (e.g. to avoid having a unique command more than once)</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000157">Todo:</a></b></dt><dd>we could optionally use the clone handler </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00591">591</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00173">org.jgap.gp.impl.ProgramChromosome.clone()</a>, <a class="el" href="_command_gene_8java_source.html#l00307">org.jgap.gp.CommandGene.getArity()</a>, <a class="el" href="_command_gene_8java_source.html#l00629">org.jgap.gp.CommandGene.getChildType()</a>, <a class="el" href="_base_g_p_chromosome_8java_source.html#l00295">org.jgap.gp.BaseGPChromosome.getCommandOfClass()</a>, <a class="el" href="_g_p_configuration_8java_source.html#l00385">org.jgap.gp.impl.GPConfiguration.getDynamizeArityProb()</a>, <a class="el" href="_base_g_p_chromosome_8java_source.html#l00357">org.jgap.gp.BaseGPChromosome.getGPConfiguration()</a>, <a class="el" href="_base_g_p_chromosome_8java_source.html#l00059">org.jgap.gp.BaseGPChromosome.getIndividual()</a>, <a class="el" href="_g_p_configuration_8java_source.html#l01206">org.jgap.gp.impl.GPConfiguration.isNoCommandGeneCloning()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00050">org.jgap.gp.impl.ProgramChromosome.m_depth</a>, <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>, <a class="el" href="_program_chromosome_8java_source.html#l00057">org.jgap.gp.impl.ProgramChromosome.m_index</a>, <a class="el" href="_program_chromosome_8java_source.html#l00059">org.jgap.gp.impl.ProgramChromosome.m_maxDepth</a>, <a class="el" href="interfaceorg_1_1jgap_1_1_random_generator.html#a708f273cc71846c2ae973172f20f2c01">org.jgap.RandomGenerator.nextDouble()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00490">org.jgap.gp.impl.ProgramChromosome.selectNode()</a>, and <a class="el" href="_g_p_configuration_8java_source.html#l00841">org.jgap.gp.impl.GPConfiguration.validateNode()</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00240">org.jgap.gp.impl.ProgramChromosome.growOrFull()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a1ca14815335297629dc3bd4812445405_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a1ca14815335297629dc3bd4812445405_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a1ca14815335297629dc3bd4812445405_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a1ca14815335297629dc3bd4812445405_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#af25451cf53447cdec834e5be9ac77f79" title="org.jgap.gp.impl.Program\lChromosome.clone" alt="" coords="318,138,485,179"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a57f6a4989aa1e01bda2c1f77cc086c4a" title="org.jgap.gp.BaseGPChromosome.get\lGPConfiguration" alt="" coords="598,201,837,242"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a13d8f2e469a316fd02a863a1d924ad5d" title="org.jgap.gp.BaseGPChromosome.get\lIndividual" alt="" coords="598,266,837,307"/><area shape="rect" id="node15" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html#a2f2fd7c05e6e948e2fd1a7bd0df37ef1" title="org.jgap.gp.CommandGene.get\lArity" alt="" coords="300,394,503,435"/><area shape="rect" id="node17" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html#a764c86ce05fec22c79f4092ecd6a2269" title="org.jgap.gp.CommandGene.get\lChildType" alt="" coords="300,459,503,501"/><area shape="rect" id="node21" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a92b35be05ab42231ab0842197332fde8" title="org.jgap.gp.BaseGPChromosome.get\lCommandOfClass" alt="" coords="282,525,521,566"/><area shape="rect" id="node25" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html#a6168fe89c9e8f548d8fa50013a9926f1" title="org.jgap.gp.impl.GPConfiguration.get\lDynamizeArityProb" alt="" coords="283,590,519,631"/><area shape="rect" id="node29" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html#a7a2c517e36f51f4e2be5d32fd4b7def2" title="org.jgap.gp.impl.GPConfiguration.is\lNoCommandGeneCloning" alt="" coords="287,655,515,697"/><area shape="rect" id="node31" href="interfaceorg_1_1jgap_1_1_random_generator.html#a708f273cc71846c2ae973172f20f2c01" title="org.jgap.RandomGenerator.next\lDouble" alt="" coords="298,721,505,762"/><area shape="rect" id="node33" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1e11a46dcd815ed2778c03800abd8e56" title="org.jgap.gp.impl.Program\lChromosome.selectNode" alt="" coords="316,329,487,370"/><area shape="rect" id="node43" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html#a54af1a43cc4c4b0cb84879bb5f6b0eaf" title="org.jgap.gp.impl.GPConfiguration.validateNode" alt="" coords="254,787,549,813"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a8640ae6ff64b6b0fe5d0972a554104f0" title="org.jgap.gp.impl.Program\lChromosome.getFunctionSet" alt="" coords="621,135,813,177"/><area shape="rect" id="node11" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a49e32c4ee7a42b4e7fbddae454b325c2" title="org.jgap.gp.impl.Program\lChromosome.ProgramChromosome" alt="" coords="602,5,833,46"/><area shape="rect" id="node13" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ade35d12aa45f98f2acea90b52bda3375" title="org.jgap.gp.impl.Program\lChromosome.size" alt="" coords="634,70,801,111"/><area shape="rect" id="node19" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html#a2b466f1bd4416ce97c0d2567be1f65a8" title="org.jgap.gp.CommandGene.get\lReturnType" alt="" coords="616,527,819,569"/><area shape="rect" id="node23" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome.html#ad7a5cd77b01e717993466858667604ce" title="org.jgap.gp.IGPChromosome.get\lFunctions" alt="" coords="611,593,824,634"/><area shape="rect" id="node37" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1873ffcdf7eaaf635b85b70f5a4ac59a" title="org.jgap.gp.impl.Program\lChromosome.getArity" alt="" coords="634,462,801,503"/><area shape="rect" id="node39" href="interfaceorg_1_1jgap_1_1gp_1_1_i_single_node_validator.html#a8c657d357179ee11ac73015301fa2789" title="org.jgap.gp.ISingleNodeValidator.is\lAllowed" alt="" coords="604,331,831,373"/><area shape="rect" id="node41" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ac6c8388003452adf7ebd2b0e70c916ac" title="org.jgap.gp.impl.Program\lChromosome.getDepth" alt="" coords="634,397,801,438"/><area shape="rect" id="node45" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html#abc4494312ed10634e36e31a2f620b4be" title="org.jgap.gp.impl.GPConfiguration.get\lNodeValidator" alt="" coords="599,779,835,821"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a1d821adfada0d4ba257d1ac1cf15969a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.gp.impl.ProgramChromosome.init </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00158">158</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_base_g_p_chromosome_8java_source.html#l00357">org.jgap.gp.BaseGPChromosome.getGPConfiguration()</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00084">org.jgap.gp.impl.ProgramChromosome.ProgramChromosome()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a1d821adfada0d4ba257d1ac1cf15969a_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a1d821adfada0d4ba257d1ac1cf15969a_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a1d821adfada0d4ba257d1ac1cf15969a_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a1d821adfada0d4ba257d1ac1cf15969a_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a57f6a4989aa1e01bda2c1f77cc086c4a" title="org.jgap.gp.BaseGPChromosome.get\lGPConfiguration" alt="" coords="222,5,461,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ac082a0f6a404ae98a32289ca561b765f"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.gp.impl.ProgramChromosome.init </td>
          <td>(</td>
          <td class="paramtype">final int&#160;</td>
          <td class="paramname"><em>a_size</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00163">163</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00050">org.jgap.gp.impl.ProgramChromosome.m_depth</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

</div>
</div>
<a class="anchor" id="a24c91c7e53c8ad7be9d7badee0d1aad5"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.gp.impl.ProgramChromosome.isCompareApplicationData </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01295">1295</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00082">org.jgap.gp.impl.ProgramChromosome.m_compareAppData</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l01200">org.jgap.gp.impl.ProgramChromosome.compareTo()</a>.</p>

</div>
</div>
<a class="anchor" id="a171447ec4020a6ca9f4e089836147993"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.gp.impl.ProgramChromosome.isPossible </td>
          <td>(</td>
          <td class="paramtype">Class&#160;</td>
          <td class="paramname"><em>a_returnType</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_subReturnType</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td>
          <td class="paramname"><em>a_nodeSet</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_function</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_growing</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Determines whether there exists a function or terminal in the given node set with the given return and sub return type.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_returnType</td><td>the return type to look for </td></tr>
    <tr><td class="paramname">a_subReturnType</td><td>the sub return type to look for </td></tr>
    <tr><td class="paramname">a_nodeSet</td><td>the array of nodes to look through </td></tr>
    <tr><td class="paramname">a_function</td><td>true to look for a function, false to look for a terminal </td></tr>
    <tr><td class="paramname">a_growing</td><td>true: grow mode, false: full mode</td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>true if such a node exists, false otherwise</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00440">440</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l01168">org.jgap.gp.impl.ProgramChromosome.getArity()</a>, and <a class="el" href="_base_g_p_chromosome_8java_source.html#l00059">org.jgap.gp.BaseGPChromosome.getIndividual()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a171447ec4020a6ca9f4e089836147993_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a171447ec4020a6ca9f4e089836147993_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a171447ec4020a6ca9f4e089836147993_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a171447ec4020a6ca9f4e089836147993_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1873ffcdf7eaaf635b85b70f5a4ac59a" title="org.jgap.gp.impl.Program\lChromosome.getArity" alt="" coords="258,5,425,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a13d8f2e469a316fd02a863a1d924ad5d" title="org.jgap.gp.BaseGPChromosome.get\lIndividual" alt="" coords="222,70,461,111"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a72e5e8ab3c253f4ec7dc10fb17a45180"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.gp.impl.ProgramChromosome.redepth </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Recalculate the depth of each node.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome.html#a0f6a34e86c91d5271761093ad69ecc96">org.jgap.gp.IGPChromosome</a>.</p>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00728">728</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00050">org.jgap.gp.impl.ProgramChromosome.m_depth</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00240">org.jgap.gp.impl.ProgramChromosome.growOrFull()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00748">org.jgap.gp.impl.ProgramChromosome.redepth()</a>.</p>

</div>
</div>
<a class="anchor" id="af63010758c5c507c9e58889cc37e985e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.gp.impl.ProgramChromosome.redepth </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_index</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Calculate the depth of the next node and the indices of the children of the current node. The depth of the next node is just one plus the depth of the current node. The index of the first child is always the next node. The index of the second child is found by recursively calling this method on the tree starting with the first child.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_index</td><td>the index of the reference depth </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the index of the next node of the same depth as the current node (i.e. the next sibling node)</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00748">748</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_base_g_p_chromosome_8java_source.html#l00059">org.jgap.gp.BaseGPChromosome.getIndividual()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00924">org.jgap.gp.impl.ProgramChromosome.getNode()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00050">org.jgap.gp.impl.ProgramChromosome.m_depth</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00728">org.jgap.gp.impl.ProgramChromosome.redepth()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_af63010758c5c507c9e58889cc37e985e_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_af63010758c5c507c9e58889cc37e985e_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_af63010758c5c507c9e58889cc37e985e_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_af63010758c5c507c9e58889cc37e985e_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a13d8f2e469a316fd02a863a1d924ad5d" title="org.jgap.gp.BaseGPChromosome.get\lIndividual" alt="" coords="222,5,461,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9dcb6045fafdd72b8241f3f1634ec433" title="org.jgap.gp.impl.Program\lChromosome.getNode" alt="" coords="258,70,425,111"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a72e5e8ab3c253f4ec7dc10fb17a45180" title="org.jgap.gp.impl.Program\lChromosome.redepth" alt="" coords="258,135,425,177"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="abb3f952c7eb74808cf6dc8dda7bfd94a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> [] org.jgap.gp.impl.ProgramChromosome.remove </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td>
          <td class="paramname"><em>a_functionSet</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>&#160;</td>
          <td class="paramname"><em>node</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01330">1330</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l01178">org.jgap.gp.impl.ProgramChromosome.size()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_abb3f952c7eb74808cf6dc8dda7bfd94a_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_abb3f952c7eb74808cf6dc8dda7bfd94a_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_abb3f952c7eb74808cf6dc8dda7bfd94a_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_abb3f952c7eb74808cf6dc8dda7bfd94a_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ade35d12aa45f98f2acea90b52bda3375" title="org.jgap.gp.impl.Program\lChromosome.size" alt="" coords="222,5,389,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a1e11a46dcd815ed2778c03800abd8e56"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> org.jgap.gp.impl.ProgramChromosome.selectNode </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_chromIndex</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">Class&#160;</td>
          <td class="paramname"><em>a_returnType</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_subReturnType</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td>
          <td class="paramname"><em>a_functionSet</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_function</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_growing</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Randomly chooses a valid node from the functions set.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_chromIndex</td><td>index of the chromosome in the individual (0..n-1) </td></tr>
    <tr><td class="paramname">a_returnType</td><td>the return type of node to choose </td></tr>
    <tr><td class="paramname">a_subReturnType</td><td>the sub return type to look for </td></tr>
    <tr><td class="paramname">a_functionSet</td><td>the functions to use </td></tr>
    <tr><td class="paramname">a_function</td><td>true to choose a function, false to choose a terminal </td></tr>
    <tr><td class="paramname">a_growing</td><td>true to ignore the function parameter, false otherwise </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the node chosen</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00490">490</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l01168">org.jgap.gp.impl.ProgramChromosome.getArity()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00877">org.jgap.gp.impl.ProgramChromosome.getDepth()</a>, <a class="el" href="_base_g_p_chromosome_8java_source.html#l00357">org.jgap.gp.BaseGPChromosome.getGPConfiguration()</a>, <a class="el" href="_base_g_p_chromosome_8java_source.html#l00059">org.jgap.gp.BaseGPChromosome.getIndividual()</a>, <a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_single_node_validator.html#a8c657d357179ee11ac73015301fa2789">org.jgap.gp.ISingleNodeValidator.isAllowed()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00057">org.jgap.gp.impl.ProgramChromosome.m_index</a>, and <a class="el" href="_command_gene_8java_source.html#l00059">org.jgap.gp.CommandGene.VoidClass</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00591">org.jgap.gp.impl.ProgramChromosome.growOrFullNode()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a1e11a46dcd815ed2778c03800abd8e56_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a1e11a46dcd815ed2778c03800abd8e56_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a1e11a46dcd815ed2778c03800abd8e56_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a1e11a46dcd815ed2778c03800abd8e56_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1873ffcdf7eaaf635b85b70f5a4ac59a" title="org.jgap.gp.impl.Program\lChromosome.getArity" alt="" coords="261,5,427,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ac6c8388003452adf7ebd2b0e70c916ac" title="org.jgap.gp.impl.Program\lChromosome.getDepth" alt="" coords="261,70,427,111"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a57f6a4989aa1e01bda2c1f77cc086c4a" title="org.jgap.gp.BaseGPChromosome.get\lGPConfiguration" alt="" coords="225,135,463,177"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a13d8f2e469a316fd02a863a1d924ad5d" title="org.jgap.gp.BaseGPChromosome.get\lIndividual" alt="" coords="225,201,463,242"/><area shape="rect" id="node11" href="interfaceorg_1_1jgap_1_1gp_1_1_i_single_node_validator.html#a8c657d357179ee11ac73015301fa2789" title="org.jgap.gp.ISingleNodeValidator.is\lAllowed" alt="" coords="231,266,457,307"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a714709ffa46266e370becc723467c8bb"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.gp.impl.ProgramChromosome.setArgTypes </td>
          <td>(</td>
          <td class="paramtype">Class[]&#160;</td>
          <td class="paramname"><em>a_argTypes</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00169">169</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00055">org.jgap.gp.impl.ProgramChromosome.argTypes</a>.</p>

</div>
</div>
<a class="anchor" id="afdecb0a33331b2ecef9be73815f72ef1"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.gp.impl.ProgramChromosome.setCompareApplicationData </td>
          <td>(</td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_doCompare</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Should we also consider the application data when comparing? Default is "false" as "true" means a <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> is losing its identity when application data is set differently!</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_doCompare</td><td>true: consider application data in method compareTo</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01285">1285</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00082">org.jgap.gp.impl.ProgramChromosome.m_compareAppData</a>.</p>

</div>
</div>
<a class="anchor" id="a9b47c5e0e01b774eed4f0be74a94764e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.gp.impl.ProgramChromosome.setFunctions </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td>
          <td class="paramname"><em>a_functions</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00842">842</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

</div>
</div>
<a class="anchor" id="a81171ec1774c2605dc2080c476768f19"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.gp.impl.ProgramChromosome.setFunctionSet </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>[]&#160;</td>
          <td class="paramname"><em>a_functionSet</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00834">834</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00045">org.jgap.gp.impl.ProgramChromosome.m_functionSet</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00240">org.jgap.gp.impl.ProgramChromosome.growOrFull()</a>.</p>

</div>
</div>
<a class="anchor" id="a705e4148b5d42c772bab87068287b8bf"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.gp.impl.ProgramChromosome.setGene </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>index</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a>&#160;</td>
          <td class="paramname"><em>a_gene</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01157">1157</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

</div>
</div>
<a class="anchor" id="a2ea8008dc2f1662945564f8d0ee67c82"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.gp.impl.ProgramChromosome.setValueFromPersistentRepresentation </td>
          <td>(</td>
          <td class="paramtype">final String&#160;</td>
          <td class="paramname"><em>a_representation</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_unsupported_representation_exception.html">UnsupportedRepresentationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_representation</td><td>String </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_unsupported_representation_exception.html">UnsupportedRepresentationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.3 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01393">1393</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l01437">org.jgap.gp.impl.ProgramChromosome.createGene()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01352">org.jgap.gp.impl.ProgramChromosome.decode()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>, and <a class="el" href="_program_chromosome_8java_source.html#l01459">org.jgap.gp.impl.ProgramChromosome.split()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a2ea8008dc2f1662945564f8d0ee67c82_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a2ea8008dc2f1662945564f8d0ee67c82_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a2ea8008dc2f1662945564f8d0ee67c82_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a2ea8008dc2f1662945564f8d0ee67c82_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a5a09ecbda2b198791e979d3dae3b0f01" title="org.jgap.gp.impl.Program\lChromosome.createGene" alt="" coords="299,5,469,46"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a7a41513bb42bc79e7708c46afd61b4a4" title="org.jgap.gp.impl.Program\lChromosome.decode" alt="" coords="301,70,467,111"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a5db9d5ee745620591027ff0572e2cff3" title="org.jgap.gp.impl.Program\lChromosome.split" alt="" coords="301,135,467,177"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a57f6a4989aa1e01bda2c1f77cc086c4a" title="org.jgap.gp.BaseGPChromosome.get\lGPConfiguration" alt="" coords="518,5,757,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ade35d12aa45f98f2acea90b52bda3375"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.gp.impl.ProgramChromosome.size </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>number of functions and terminals present</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01178">1178</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00173">org.jgap.gp.impl.ProgramChromosome.clone()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01200">org.jgap.gp.impl.ProgramChromosome.compareTo()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00928">org.jgap.gp.impl.ProgramChromosome.getNode()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l01330">org.jgap.gp.impl.ProgramChromosome.remove()</a>.</p>

</div>
</div>
<a class="anchor" id="a5db9d5ee745620591027ff0572e2cff3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">static final List org.jgap.gp.impl.ProgramChromosome.split </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_string</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_unsupported_representation_exception.html">UnsupportedRepresentationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">static</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Splits a_string into individual gene representations.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_string</td><td>the string to split </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the elements of the returned array are the persistent representation strings of the gene's components </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_unsupported_representation_exception.html">UnsupportedRepresentationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.3 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l01459">1459</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l01393">org.jgap.gp.impl.ProgramChromosome.setValueFromPersistentRepresentation()</a>.</p>

</div>
</div>
<a class="anchor" id="addf578c58866aaf237a11d91308d0c04"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.gp.impl.ProgramChromosome.toString </td>
          <td>(</td>
          <td class="paramtype">final int&#160;</td>
          <td class="paramname"><em>a_startNode</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Output program in left-hand notion (e.g.: "+ X Y" for "X + Y").</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_startNode</td><td>node to start with </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>output in left-hand notion</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00298">298</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l01168">org.jgap.gp.impl.ProgramChromosome.getArity()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00786">org.jgap.gp.impl.ProgramChromosome.getChild()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00838">org.jgap.gp.impl.ProgramChromosome.getFunctions()</a>, <a class="el" href="_base_g_p_chromosome_8java_source.html#l00059">org.jgap.gp.BaseGPChromosome.getIndividual()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00350">org.jgap.gp.impl.ProgramChromosome.toStringNorm()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_addf578c58866aaf237a11d91308d0c04_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_addf578c58866aaf237a11d91308d0c04_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_addf578c58866aaf237a11d91308d0c04_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_addf578c58866aaf237a11d91308d0c04_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1873ffcdf7eaaf635b85b70f5a4ac59a" title="org.jgap.gp.impl.Program\lChromosome.getArity" alt="" coords="258,5,425,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aae34fa625e78f8be2de585462d3bbcf8" title="org.jgap.gp.impl.Program\lChromosome.getChild" alt="" coords="258,70,425,111"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9814ee68c4893d979b6d6c702f754ed6" title="org.jgap.gp.impl.Program\lChromosome.getFunctions" alt="" coords="509,99,688,141"/><area shape="rect" id="node10" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a13d8f2e469a316fd02a863a1d924ad5d" title="org.jgap.gp.BaseGPChromosome.get\lIndividual" alt="" coords="222,187,461,229"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a145081f4837b9a7b50af35f8256c0eea"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.gp.impl.ProgramChromosome.toStringDebug </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>debug representation of progrm chromosome, containing class names of all children</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.4 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00409">409</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l01168">org.jgap.gp.impl.ProgramChromosome.getArity()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00786">org.jgap.gp.impl.ProgramChromosome.getChild()</a>, <a class="el" href="_base_g_p_chromosome_8java_source.html#l00059">org.jgap.gp.BaseGPChromosome.getIndividual()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00350">org.jgap.gp.impl.ProgramChromosome.toStringNorm()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a145081f4837b9a7b50af35f8256c0eea_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a145081f4837b9a7b50af35f8256c0eea_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a145081f4837b9a7b50af35f8256c0eea_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a145081f4837b9a7b50af35f8256c0eea_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1873ffcdf7eaaf635b85b70f5a4ac59a" title="org.jgap.gp.impl.Program\lChromosome.getArity" alt="" coords="722,53,889,95"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aae34fa625e78f8be2de585462d3bbcf8" title="org.jgap.gp.impl.Program\lChromosome.getChild" alt="" coords="722,171,889,212"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a13d8f2e469a316fd02a863a1d924ad5d" title="org.jgap.gp.BaseGPChromosome.get\lIndividual" alt="" coords="686,236,925,277"/><area shape="rect" id="node11" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a56c22c94166729d37888a6daaef76bb3" title="org.jgap.gp.impl.Program\lChromosome.toStringNorm" alt="" coords="240,141,421,183"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9814ee68c4893d979b6d6c702f754ed6" title="org.jgap.gp.impl.Program\lChromosome.getFunctions" alt="" coords="973,112,1152,153"/><area shape="rect" id="node16" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#addf578c58866aaf237a11d91308d0c04" title="org.jgap.gp.impl.Program\lChromosome.toString" alt="" coords="470,112,637,153"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a56c22c94166729d37888a6daaef76bb3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.gp.impl.ProgramChromosome.toStringNorm </td>
          <td>(</td>
          <td class="paramtype">final int&#160;</td>
          <td class="paramname"><em>a_startNode</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Output program in "natural" notion (e.g.: "X + Y" for "X + Y").</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_startNode</td><td>the node to start with, e.g. 0 for a complete dump of the program </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>output in normalized notion</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1gp_1_1_i_g_p_chromosome.html#a69a39548f0af1c9746bf6b2352f1a879">org.jgap.gp.IGPChromosome</a>.</p>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00350">350</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>References <a class="el" href="_program_chromosome_8java_source.html#l01168">org.jgap.gp.impl.ProgramChromosome.getArity()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00786">org.jgap.gp.impl.ProgramChromosome.getChild()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00838">org.jgap.gp.impl.ProgramChromosome.getFunctions()</a>, <a class="el" href="_base_g_p_chromosome_8java_source.html#l00059">org.jgap.gp.BaseGPChromosome.getIndividual()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00064">org.jgap.gp.impl.ProgramChromosome.m_genes</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00298">org.jgap.gp.impl.ProgramChromosome.toString()</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00398">org.jgap.gp.impl.ProgramChromosome.getBusinessKey()</a>, <a class="el" href="_program_chromosome_test_8java_source.html#l00345">org.jgap.gp.impl.ProgramChromosomeTest.testToStringNorm_0()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00409">org.jgap.gp.impl.ProgramChromosome.toStringDebug()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a56c22c94166729d37888a6daaef76bb3_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a56c22c94166729d37888a6daaef76bb3_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a56c22c94166729d37888a6daaef76bb3_cgraph" id="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_a56c22c94166729d37888a6daaef76bb3_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1873ffcdf7eaaf635b85b70f5a4ac59a" title="org.jgap.gp.impl.Program\lChromosome.getArity" alt="" coords="487,48,654,89"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#aae34fa625e78f8be2de585462d3bbcf8" title="org.jgap.gp.impl.Program\lChromosome.getChild" alt="" coords="487,230,654,272"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a9814ee68c4893d979b6d6c702f754ed6" title="org.jgap.gp.impl.Program\lChromosome.getFunctions" alt="" coords="739,106,917,148"/><area shape="rect" id="node10" href="classorg_1_1jgap_1_1gp_1_1_base_g_p_chromosome.html#a13d8f2e469a316fd02a863a1d924ad5d" title="org.jgap.gp.BaseGPChromosome.get\lIndividual" alt="" coords="451,165,690,206"/><area shape="rect" id="node12" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#addf578c58866aaf237a11d91308d0c04" title="org.jgap.gp.impl.Program\lChromosome.toString" alt="" coords="235,106,402,148"/></map>
</div>
</p>

</div>
</div>
<h2 class="groupheader">Member Data Documentation</h2>
<a class="anchor" id="a770064f02147733f5077b5b0bfa125fb"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Class [] org.jgap.gp.impl.ProgramChromosome.argTypes</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Array to hold the types of the arguments to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. </p>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00055">55</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00173">org.jgap.gp.impl.ProgramChromosome.clone()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01164">org.jgap.gp.impl.ProgramChromosome.getArgTypes()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00240">org.jgap.gp.impl.ProgramChromosome.growOrFull()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00108">org.jgap.gp.impl.ProgramChromosome.ProgramChromosome()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00169">org.jgap.gp.impl.ProgramChromosome.setArgTypes()</a>.</p>

</div>
</div>
<a class="anchor" id="a1710d7febbb2e909bb1a9d41bba3aa17"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final String org.jgap.gp.impl.ProgramChromosome.CVS_REVISION = &quot;$Revision: 1.52 $&quot;</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>String containing the CVS revision. Read out via reflection! </p>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00030">30</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

</div>
</div>
<a class="anchor" id="a0c09c5995d7929662288b0fc4d0b4858"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">transient Logger org.jgap.gp.impl.ProgramChromosome.LOGGER = Logger.getLogger(ProgramChromosome.class)</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00040">40</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

</div>
</div>
<a class="anchor" id="a918187c443c8cfb61eaa33a32801df64"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.gp.impl.ProgramChromosome.m_applicationData</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Application-specific data that is attached to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. This data may assist the application in evaluating this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in the fitness function. JGAP does not operate on the data, aside from allowing it to be set and retrieved, and considering it with comparations (if user opted in to do so). </p>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00073">73</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l01311">org.jgap.gp.impl.ProgramChromosome.getApplicationData()</a>.</p>

</div>
</div>
<a class="anchor" id="ad4256d733bf0188c9530d7826f1320d3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.gp.impl.ProgramChromosome.m_compareAppData</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Method <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ac9896e142c28c71711133698041fb72f">compareTo()</a>: Should we also consider the application data when comparing? Default is "false" as "true" means a <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>'s losing its identity when application data is set differently!</p>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00082">82</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l01295">org.jgap.gp.impl.ProgramChromosome.isCompareApplicationData()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l01285">org.jgap.gp.impl.ProgramChromosome.setCompareApplicationData()</a>.</p>

</div>
</div>
<a class="anchor" id="a81e14f6750b10a0331646f6b1891f023"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int [] org.jgap.gp.impl.ProgramChromosome.m_depth</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Array to hold the depths of each node. </p>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00050">50</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00173">org.jgap.gp.impl.ProgramChromosome.clone()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00786">org.jgap.gp.impl.ProgramChromosome.getChild()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00877">org.jgap.gp.impl.ProgramChromosome.getDepth()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00901">org.jgap.gp.impl.ProgramChromosome.getParentNode()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00856">org.jgap.gp.impl.ProgramChromosome.getSize()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00591">org.jgap.gp.impl.ProgramChromosome.growOrFullNode()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00163">org.jgap.gp.impl.ProgramChromosome.init()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00728">org.jgap.gp.impl.ProgramChromosome.redepth()</a>.</p>

</div>
</div>
<a class="anchor" id="a9d7f024ebd3113416251c968d06dce6d"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> [] org.jgap.gp.impl.ProgramChromosome.m_functionSet</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>The list of allowed functions/terminals. </p>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00045">45</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00830">org.jgap.gp.impl.ProgramChromosome.getFunctionSet()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00240">org.jgap.gp.impl.ProgramChromosome.growOrFull()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00108">org.jgap.gp.impl.ProgramChromosome.ProgramChromosome()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00834">org.jgap.gp.impl.ProgramChromosome.setFunctionSet()</a>.</p>

</div>
</div>
<a class="anchor" id="ad3903410ad37a12a87ee51b4a3f4858a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html">CommandGene</a> [] org.jgap.gp.impl.ProgramChromosome.m_genes</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>The array of genes contained in this chromosome. </p>

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00064">64</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00216">org.jgap.gp.impl.ProgramChromosome.cleanup()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00173">org.jgap.gp.impl.ProgramChromosome.clone()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01200">org.jgap.gp.impl.ProgramChromosome.compareTo()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01149">org.jgap.gp.impl.ProgramChromosome.execute()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00964">org.jgap.gp.impl.ProgramChromosome.execute_boolean()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01100">org.jgap.gp.impl.ProgramChromosome.execute_double()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01076">org.jgap.gp.impl.ProgramChromosome.execute_float()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01026">org.jgap.gp.impl.ProgramChromosome.execute_int()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01052">org.jgap.gp.impl.ProgramChromosome.execute_long()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01125">org.jgap.gp.impl.ProgramChromosome.execute_object()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01000">org.jgap.gp.impl.ProgramChromosome.execute_void()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00804">org.jgap.gp.impl.ProgramChromosome.getChild()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00877">org.jgap.gp.impl.ProgramChromosome.getDepth()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00838">org.jgap.gp.impl.ProgramChromosome.getFunctions()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01326">org.jgap.gp.impl.ProgramChromosome.getGene()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00928">org.jgap.gp.impl.ProgramChromosome.getNode()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00901">org.jgap.gp.impl.ProgramChromosome.getParentNode()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01363">org.jgap.gp.impl.ProgramChromosome.getPersistentRepresentation()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00856">org.jgap.gp.impl.ProgramChromosome.getSize()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00591">org.jgap.gp.impl.ProgramChromosome.growOrFullNode()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00163">org.jgap.gp.impl.ProgramChromosome.init()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00126">org.jgap.gp.impl.ProgramChromosome.ProgramChromosome()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00842">org.jgap.gp.impl.ProgramChromosome.setFunctions()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01157">org.jgap.gp.impl.ProgramChromosome.setGene()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01393">org.jgap.gp.impl.ProgramChromosome.setValueFromPersistentRepresentation()</a>, <a class="el" href="_program_chromosome_8java_source.html#l01178">org.jgap.gp.impl.ProgramChromosome.size()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00298">org.jgap.gp.impl.ProgramChromosome.toString()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00409">org.jgap.gp.impl.ProgramChromosome.toStringDebug()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00350">org.jgap.gp.impl.ProgramChromosome.toStringNorm()</a>.</p>

</div>
</div>
<a class="anchor" id="a05a7b989fa9e7a7526b3c796bf033fe3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">transient int org.jgap.gp.impl.ProgramChromosome.m_index</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00057">57</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00240">org.jgap.gp.impl.ProgramChromosome.growOrFull()</a>, <a class="el" href="_program_chromosome_8java_source.html#l00591">org.jgap.gp.impl.ProgramChromosome.growOrFullNode()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00490">org.jgap.gp.impl.ProgramChromosome.selectNode()</a>.</p>

</div>
</div>
<a class="anchor" id="add5812f1aa24a114aa44923db580b919"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">transient int org.jgap.gp.impl.ProgramChromosome.m_maxDepth</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00059">59</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

<p>Referenced by <a class="el" href="_program_chromosome_8java_source.html#l00240">org.jgap.gp.impl.ProgramChromosome.growOrFull()</a>, and <a class="el" href="_program_chromosome_8java_source.html#l00591">org.jgap.gp.impl.ProgramChromosome.growOrFullNode()</a>.</p>

</div>
</div>
<a class="anchor" id="a42c376b973ad38ab97754447406a5480"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Map&lt;<a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome_1_1_node_info.html">NodeInfo</a>,Boolean&gt; org.jgap.gp.impl.ProgramChromosome.m_possibleNodes = new HashMap()</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_program_chromosome_8java_source.html#l00423">423</a> of file <a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>LDH_Git/src/org/jgap/gp/impl/<a class="el" href="_program_chromosome_8java_source.html">ProgramChromosome.java</a></li>
</ul>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Sat Nov 23 2013 17:42:45 for Práctica_5_LDH by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
