<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Práctica_5_LDH: org.jgap.BaseGene Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">Práctica_5_LDH
   &#160;<span id="projectnumber">1.2</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li><a href="inherits.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Classes</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Namespaces</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Files</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Functions</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(6)"><span class="SelectionMark">&#160;</span>Pages</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="namespaceorg.html">org</a></li><li class="navelem"><a class="el" href="namespaceorg_1_1jgap.html">jgap</a></li><li class="navelem"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html">BaseGene</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-static-attribs">Static Public Attributes</a> &#124;
<a href="#pro-methods">Protected Member Functions</a> &#124;
<a href="#pri-attribs">Private Attributes</a> &#124;
<a href="#pri-static-attribs">Static Private Attributes</a> &#124;
<a href="classorg_1_1jgap_1_1_base_gene-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">org.jgap.BaseGene Class Reference<span class="mlabels"><span class="mlabel">abstract</span></span></div>  </div>
</div><!--header-->
<div class="contents">
<div class="dynheader">
Inheritance diagram for org.jgap.BaseGene:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_gene__inherit__graph.png" border="0" usemap="#org_8jgap_8_base_gene_inherit__map" alt="Inheritance graph"/></div>
<map name="org_8jgap_8_base_gene_inherit__map" id="org_8jgap_8_base_gene_inherit__map">
<area shape="rect" id="node15" href="classorg_1_1jgap_1_1impl_1_1_boolean_gene.html" title="org.jgap.impl.BooleanGene" alt="" coords="589,72,768,99"/><area shape="rect" id="node17" href="classorg_1_1jgap_1_1impl_1_1_composite_gene.html" title="org.jgap.impl.CompositeGene" alt="" coords="581,123,776,149"/><area shape="rect" id="node19" href="classorg_1_1jgap_1_1impl_1_1_fixed_binary_gene.html" title="org.jgap.impl.FixedBinaryGene" alt="" coords="579,173,779,200"/><area shape="rect" id="node21" href="classorg_1_1jgap_1_1impl_1_1_map_gene.html" title="org.jgap.impl.MapGene" alt="" coords="601,224,757,251"/><area shape="rect" id="node23" href="classorg_1_1jgap_1_1impl_1_1_number_gene.html" title="org.jgap.impl.NumberGene" alt="" coords="590,275,767,301"/><area shape="rect" id="node35" href="classorg_1_1jgap_1_1impl_1_1_set_gene.html" title="org.jgap.impl.SetGene" alt="" coords="603,325,754,352"/><area shape="rect" id="node37" href="classorg_1_1jgap_1_1impl_1_1_string_gene.html" title="org.jgap.impl.StringGene" alt="" coords="596,376,761,403"/><area shape="rect" id="node39" href="classorg_1_1jgap_1_1supergenes_1_1_abstract_supergene.html" title="org.jgap.supergenes.Abstract\lSupergene" alt="" coords="583,442,775,483"/><area shape="rect" id="node2" href="interfaceorg_1_1jgap_1_1_gene.html" title="org.jgap.Gene" alt="" coords="221,199,323,225"/><area shape="rect" id="node8" href="interfaceorg_1_1jgap_1_1_i_unique_key.html" title="org.jgap.IUniqueKey" alt="" coords="6,249,146,276"/><area shape="rect" id="node10" href="interfaceorg_1_1jgap_1_1_i_business_key.html" title="org.jgap.IBusinessKey" alt="" coords="195,249,349,276"/><area shape="rect" id="node25" href="classorg_1_1jgap_1_1impl_1_1_double_gene.html" title="org.jgap.impl.DoubleGene" alt="" coords="870,173,1042,200"/><area shape="rect" id="node27" href="classorg_1_1jgap_1_1impl_1_1_integer_gene.html" title="org.jgap.impl.IntegerGene" alt="" coords="871,224,1041,251"/><area shape="rect" id="node31" href="classorg_1_1jgap_1_1impl_1_1_mutiple_integer_gene.html" title="org.jgap.impl.MutipleIntegerGene" alt="" coords="849,275,1063,301"/><area shape="rect" id="node33" href="classorg_1_1jgap_1_1impl_1_1_number_gene_test_1_1_number_gene_impl.html" title="org.jgap.impl.NumberGene\lTest.NumberGeneImpl" alt="" coords="867,326,1045,367"/><area shape="rect" id="node41" href="classorg_1_1jgap_1_1supergenes_1_1_instantiable_supergene_for_testing.html" title="org.jgap.supergenes.Instantiable\lSupergeneForTesting" alt="" coords="851,391,1061,433"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<div class="dynheader">
Collaboration diagram for org.jgap.BaseGene:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_gene__coll__graph.png" border="0" usemap="#org_8jgap_8_base_gene_coll__map" alt="Collaboration graph"/></div>
<map name="org_8jgap_8_base_gene_coll__map" id="org_8jgap_8_base_gene_coll__map">
<area shape="rect" id="node2" href="interfaceorg_1_1jgap_1_1_gene.html" title="org.jgap.Gene" alt="" coords="406,16,509,43"/><area shape="rect" id="node12" href="classorg_1_1jgap_1_1_configuration.html" title="org.jgap.Configuration" alt="" coords="792,483,941,509"/><area shape="rect" id="node22" href="interfaceorg_1_1jgap_1_1_i_breeder.html" title="org.jgap.IBreeder" alt="" coords="397,224,518,251"/><area shape="rect" id="node35" href="interfaceorg_1_1jgap_1_1_i_chromosome.html" title="org.jgap.IChromosome" alt="" coords="380,141,535,168"/><area shape="rect" id="node27" href="interfaceorg_1_1jgap_1_1_i_j_g_a_p_factory.html" title="org.jgap.IJGAPFactory" alt="" coords="380,355,535,381"/><area shape="rect" id="node32" href="interfaceorg_1_1jgap_1_1_fitness_evaluator.html" title="org.jgap.FitnessEvaluator" alt="" coords="372,456,543,483"/><area shape="rect" id="node41" href="interfaceorg_1_1jgap_1_1_random_generator.html" title="org.jgap.RandomGenerator" alt="" coords="368,557,547,584"/><area shape="rect" id="node44" href="classorg_1_1jgap_1_1_bulk_fitness_function.html" title="org.jgap.BulkFitnessFunction" alt="" coords="362,405,553,432"/><area shape="rect" id="node48" href="classorg_1_1jgap_1_1_configuration_1_1_configuration_configurable.html" title="org.jgap.Configuration.Configuration\lConfigurable" alt="" coords="342,761,573,802"/><area shape="rect" id="node51" href="classorg_1_1jgap_1_1_fitness_function.html" title="org.jgap.FitnessFunction" alt="" coords="375,507,540,533"/><area shape="rect" id="node57" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html" title="org.jgap.IGeneConstraint\lChecker" alt="" coords="598,870,765,911"/><area shape="rect" id="node8" href="interfaceorg_1_1jgap_1_1_i_unique_key.html" title="org.jgap.IUniqueKey" alt="" coords="143,8,283,35"/><area shape="rect" id="node10" href="interfaceorg_1_1jgap_1_1_i_business_key.html" title="org.jgap.IBusinessKey" alt="" coords="790,432,943,459"/><area shape="rect" id="node17" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html" title="org.jgap.util.ICloneable" alt="" coords="136,275,291,301"/><area shape="rect" id="node30" href="interfaceorg_1_1jgap_1_1_i_chromosome_pool.html" title="org.jgap.IChromosomePool" alt="" coords="367,827,547,853"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:acea53de5bb68978322967e8019607f24"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#acea53de5bb68978322967e8019607f24">BaseGene</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration)  throws InvalidConfigurationException </td></tr>
<tr class="separator:acea53de5bb68978322967e8019607f24"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a35e754e8e217bdd26962e1c3c31d0169"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a35e754e8e217bdd26962e1c3c31d0169">getAllele</a> ()</td></tr>
<tr class="separator:a35e754e8e217bdd26962e1c3c31d0169"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af842a106d90ca973956f560ae448464e"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#af842a106d90ca973956f560ae448464e">hashCode</a> ()</td></tr>
<tr class="separator:af842a106d90ca973956f560ae448464e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a569b99cf460d27d2c1acbea2b7775975"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a569b99cf460d27d2c1acbea2b7775975">cleanup</a> ()</td></tr>
<tr class="separator:a569b99cf460d27d2c1acbea2b7775975"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a935dfec06cb77bff0d2a45859d6a35b7"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a935dfec06cb77bff0d2a45859d6a35b7">toString</a> ()</td></tr>
<tr class="separator:a935dfec06cb77bff0d2a45859d6a35b7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a26c00775111b327b178288164ed9c667"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a26c00775111b327b178288164ed9c667">size</a> ()</td></tr>
<tr class="separator:a26c00775111b327b178288164ed9c667"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9fb0e302ae57ee3f2f5d1ee08a75bf43"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a9fb0e302ae57ee3f2f5d1ee08a75bf43">equals</a> (final Object a_other)</td></tr>
<tr class="separator:a9fb0e302ae57ee3f2f5d1ee08a75bf43"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a48f71c4c1282de2ca2cf8f4009c4615a"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a48f71c4c1282de2ca2cf8f4009c4615a">getEnergy</a> ()</td></tr>
<tr class="separator:a48f71c4c1282de2ca2cf8f4009c4615a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afb39fbd8997971a140a9a4380ceecc4a"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#afb39fbd8997971a140a9a4380ceecc4a">setEnergy</a> (final double a_energy)</td></tr>
<tr class="separator:afb39fbd8997971a140a9a4380ceecc4a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afeb9a5f42018c057c8197fb0e7d57354"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#afeb9a5f42018c057c8197fb0e7d57354">setApplicationData</a> (final Object a_newData)</td></tr>
<tr class="separator:afeb9a5f42018c057c8197fb0e7d57354"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0520139dc694dc6034aeb631ae7d70bd"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a0520139dc694dc6034aeb631ae7d70bd">getApplicationData</a> ()</td></tr>
<tr class="separator:a0520139dc694dc6034aeb631ae7d70bd"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:adad4ccd46efd7b2a3dc81bd0ba0c410c"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#adad4ccd46efd7b2a3dc81bd0ba0c410c">setCompareApplicationData</a> (final boolean a_doCompare)</td></tr>
<tr class="separator:adad4ccd46efd7b2a3dc81bd0ba0c410c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:acd6431c06c81bc03b4abbe32e4d9d8ef"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#acd6431c06c81bc03b4abbe32e4d9d8ef">isCompareApplicationData</a> ()</td></tr>
<tr class="separator:acd6431c06c81bc03b4abbe32e4d9d8ef"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a334db8a543f3008719d78ee219826b65"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a334db8a543f3008719d78ee219826b65">setConstraintChecker</a> (final <a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a> a_constraintChecker)</td></tr>
<tr class="separator:a334db8a543f3008719d78ee219826b65"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae9823bef7c550fc1533afb8ab2b34d93"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#ae9823bef7c550fc1533afb8ab2b34d93">getConstraintChecker</a> ()</td></tr>
<tr class="separator:ae9823bef7c550fc1533afb8ab2b34d93"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a40f8f3d249145ad028892cd706ba9dae"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a40f8f3d249145ad028892cd706ba9dae">newGene</a> ()</td></tr>
<tr class="separator:a40f8f3d249145ad028892cd706ba9dae"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a32a316c9cc9e273bd75182cbdf188fb6"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a32a316c9cc9e273bd75182cbdf188fb6">getConfiguration</a> ()</td></tr>
<tr class="separator:a32a316c9cc9e273bd75182cbdf188fb6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac4e7c0fad73465365953b375cfa802a8"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#ac4e7c0fad73465365953b375cfa802a8">getBusinessKey</a> ()</td></tr>
<tr class="separator:ac4e7c0fad73465365953b375cfa802a8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab53d562a62f5c72fc88c7e5315a23ed7"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#ab53d562a62f5c72fc88c7e5315a23ed7">getUniqueID</a> ()</td></tr>
<tr class="separator:ab53d562a62f5c72fc88c7e5315a23ed7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa7d9f1c3c2e727d496d66a85818c17d0"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#aa7d9f1c3c2e727d496d66a85818c17d0">setUniqueIDTemplate</a> (String a_templateID, int a_index)</td></tr>
<tr class="separator:aa7d9f1c3c2e727d496d66a85818c17d0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad3c210e440f2b542391db5a909fda3b7"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#ad3c210e440f2b542391db5a909fda3b7">getUniqueIDTemplate</a> (int a_index)</td></tr>
<tr class="separator:ad3c210e440f2b542391db5a909fda3b7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_methods_interfaceorg_1_1jgap_1_1_gene"><td colspan="2" onclick="javascript:toggleInherit('pub_methods_interfaceorg_1_1jgap_1_1_gene')"><img src="closed.png" alt="-"/>&#160;Public Member Functions inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">org.jgap.Gene</a></td></tr>
<tr class="memitem:af46a0c31e0fecea483497e87ca278127 inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#af46a0c31e0fecea483497e87ca278127">setAllele</a> (Object a_newValue)</td></tr>
<tr class="separator:af46a0c31e0fecea483497e87ca278127 inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a044aa8f0a04ee06bb04bbd35ad4c7f2f inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#a044aa8f0a04ee06bb04bbd35ad4c7f2f">getPersistentRepresentation</a> ()  throws UnsupportedOperationException</td></tr>
<tr class="separator:a044aa8f0a04ee06bb04bbd35ad4c7f2f inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a057e5d6cce5ae768cbf626e0ccf70520 inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#a057e5d6cce5ae768cbf626e0ccf70520">setValueFromPersistentRepresentation</a> (String a_representation)  throws UnsupportedOperationException, UnsupportedRepresentationException</td></tr>
<tr class="separator:a057e5d6cce5ae768cbf626e0ccf70520 inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4ef135da7549ee98139928e21adb841a inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#a4ef135da7549ee98139928e21adb841a">setToRandomValue</a> (<a class="el" href="interfaceorg_1_1jgap_1_1_random_generator.html">RandomGenerator</a> a_numberGenerator)</td></tr>
<tr class="separator:a4ef135da7549ee98139928e21adb841a inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af506b6e1d6ceb3e88e522da4c79adba4 inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#af506b6e1d6ceb3e88e522da4c79adba4">applyMutation</a> (int index, double a_percentage)</td></tr>
<tr class="separator:af506b6e1d6ceb3e88e522da4c79adba4 inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae8405395f244a18424794e8f2e36c608 inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#ae8405395f244a18424794e8f2e36c608">setApplicationData</a> (Object a_newData)</td></tr>
<tr class="separator:ae8405395f244a18424794e8f2e36c608 inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a404ee1527c7dab2f8115f18507d0544b inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#a404ee1527c7dab2f8115f18507d0544b">setCompareApplicationData</a> (boolean a_doCompare)</td></tr>
<tr class="separator:a404ee1527c7dab2f8115f18507d0544b inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af5c429a29c338d2ad5d5514d5beea6e7 inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#af5c429a29c338d2ad5d5514d5beea6e7">setEnergy</a> (double a_energy)</td></tr>
<tr class="separator:af5c429a29c338d2ad5d5514d5beea6e7 inherit pub_methods_interfaceorg_1_1jgap_1_1_gene"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-static-attribs"></a>
Static Public Attributes</h2></td></tr>
<tr class="memitem:acc4b1c46f3508b835bf95b9058eae3a0"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#acc4b1c46f3508b835bf95b9058eae3a0">S_APPLICATION_DATA</a> = &quot;Application data&quot;</td></tr>
<tr class="separator:acc4b1c46f3508b835bf95b9058eae3a0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a86edb3d2c29e5064305fabc2dd59a3b9"><td class="memItemLeft" align="right" valign="top">static final double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a86edb3d2c29e5064305fabc2dd59a3b9">DELTA</a> = 0.0000001</td></tr>
<tr class="separator:a86edb3d2c29e5064305fabc2dd59a3b9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_gene"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_gene')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">org.jgap.Gene</a></td></tr>
<tr class="memitem:a771652182ee68c96d4107eee7692fee2 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_gene"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#a771652182ee68c96d4107eee7692fee2">CVS_REVISION</a> = &quot;$Revision: 1.26 $&quot;</td></tr>
<tr class="separator:a771652182ee68c96d4107eee7692fee2 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_gene"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae49de1c9f46601f1b60d393a9ce3901a inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_gene"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#ae49de1c9f46601f1b60d393a9ce3901a">PERSISTENT_FIELD_DELIMITER</a> = &quot;:&quot;</td></tr>
<tr class="separator:ae49de1c9f46601f1b60d393a9ce3901a inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_gene"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html">org.jgap.IUniqueKey</a></td></tr>
<tr class="memitem:a5b6472b1e40001ee262da59ee449a77b inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html#a5b6472b1e40001ee262da59ee449a77b">CVS_REVISION</a> = &quot;$Revision: 1.1 $&quot;</td></tr>
<tr class="separator:a5b6472b1e40001ee262da59ee449a77b inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_business_key.html">org.jgap.IBusinessKey</a></td></tr>
<tr class="memitem:a6d43b687da0842739b8021f466ca451a inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_business_key.html#a6d43b687da0842739b8021f466ca451a">CVS_REVISION</a> = &quot;$Revision: 1.3 $&quot;</td></tr>
<tr class="separator:a6d43b687da0842739b8021f466ca451a inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_business_key"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pro-methods"></a>
Protected Member Functions</h2></td></tr>
<tr class="memitem:ad20da6b23ba524a95caa28f727c26dcd"><td class="memItemLeft" align="right" valign="top">abstract Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#ad20da6b23ba524a95caa28f727c26dcd">getInternalValue</a> ()</td></tr>
<tr class="separator:ad20da6b23ba524a95caa28f727c26dcd"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a72fa365cdcc9f2eb399fb8346a0a511b"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a72fa365cdcc9f2eb399fb8346a0a511b">compareApplicationData</a> (final Object a_appdata1, final Object a_appdata2)</td></tr>
<tr class="separator:a72fa365cdcc9f2eb399fb8346a0a511b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa423e96ffac5a9589fb3a4fbda791b3c"><td class="memItemLeft" align="right" valign="top">abstract <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#aa423e96ffac5a9589fb3a4fbda791b3c">newGeneInternal</a> ()</td></tr>
<tr class="separator:aa423e96ffac5a9589fb3a4fbda791b3c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5e8fc72c1fccdb665113c0c34e3fa4b5"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a5e8fc72c1fccdb665113c0c34e3fa4b5">encode</a> (String a_string)</td></tr>
<tr class="separator:a5e8fc72c1fccdb665113c0c34e3fa4b5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae9277d0b77c1e8ca244687a8ab8cf42f"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#ae9277d0b77c1e8ca244687a8ab8cf42f">decode</a> (String a_string)</td></tr>
<tr class="separator:ae9277d0b77c1e8ca244687a8ab8cf42f"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-attribs"></a>
Private Attributes</h2></td></tr>
<tr class="memitem:a8553da900cf30fc4b0f5463404b86a8e"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a8553da900cf30fc4b0f5463404b86a8e">m_energy</a></td></tr>
<tr class="separator:a8553da900cf30fc4b0f5463404b86a8e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab2e121e812be7a4dead5ca7e300ac3d7"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#ab2e121e812be7a4dead5ca7e300ac3d7">m_applicationData</a></td></tr>
<tr class="separator:ab2e121e812be7a4dead5ca7e300ac3d7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a27bb3cd52e8cd52fffc4953f172709a2"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a27bb3cd52e8cd52fffc4953f172709a2">m_compareAppData</a></td></tr>
<tr class="separator:a27bb3cd52e8cd52fffc4953f172709a2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4d41d885b45ce031c33c88214f98019e"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a4d41d885b45ce031c33c88214f98019e">m_configuration</a></td></tr>
<tr class="separator:a4d41d885b45ce031c33c88214f98019e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8cd57f3a57f5a892f590df9e06b82e47"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a8cd57f3a57f5a892f590df9e06b82e47">m_uniqueID</a></td></tr>
<tr class="separator:a8cd57f3a57f5a892f590df9e06b82e47"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3f14fef3c7107830f24306cf7bb0abf0"><td class="memItemLeft" align="right" valign="top">Map&lt; Integer, String &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a3f14fef3c7107830f24306cf7bb0abf0">m_uniqueIDTemplates</a></td></tr>
<tr class="separator:a3f14fef3c7107830f24306cf7bb0abf0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a846344a347601679d0935dcdba7f7e6f"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a846344a347601679d0935dcdba7f7e6f">m_geneAlleleChecker</a></td></tr>
<tr class="separator:a846344a347601679d0935dcdba7f7e6f"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-static-attribs"></a>
Static Private Attributes</h2></td></tr>
<tr class="memitem:a7e83775c4b2b9391ec833869808d0154"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a7e83775c4b2b9391ec833869808d0154">CVS_REVISION</a> = &quot;$Revision: 1.28 $&quot;</td></tr>
<tr class="separator:a7e83775c4b2b9391ec833869808d0154"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Abstract base class for all genes. Provides default implementations.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.2 </dd></dl>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00023">23</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="acea53de5bb68978322967e8019607f24"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.BaseGene.BaseGene </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the configuration to use </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00082">82</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_configuration_8java_source.html#l01500">org.jgap.Configuration.getJGAPFactory()</a>, <a class="el" href="_configuration_8java_source.html#l01911">org.jgap.Configuration.isUniqueKeysActive()</a>, <a class="el" href="_base_gene_8java_source.html#l00060">org.jgap.BaseGene.m_configuration</a>, <a class="el" href="_base_gene_8java_source.html#l00066">org.jgap.BaseGene.m_uniqueID</a>, and <a class="el" href="_base_gene_8java_source.html#l00072">org.jgap.BaseGene.m_uniqueIDTemplates</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_gene_acea53de5bb68978322967e8019607f24_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_base_gene_acea53de5bb68978322967e8019607f24_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_base_gene_acea53de5bb68978322967e8019607f24_cgraph" id="classorg_1_1jgap_1_1_base_gene_acea53de5bb68978322967e8019607f24_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_configuration.html#ad71c68a6972d6fd993197773439154a0" title="org.jgap.Configuration.get\lJGAPFactory" alt="" coords="253,5,424,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_configuration.html#a1cbe9f50b9017fe15ac4e27487cc549f" title="org.jgap.Configuration.is\lUniqueKeysActive" alt="" coords="257,70,420,111"/></map>
</div>
</p>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="a569b99cf460d27d2c1acbea2b7775975"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseGene.cleanup </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Executed by the genetic engine when this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> instance is no longer needed and should perform any necessary resource cleanup. If you need a special cleanup, override this method.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#a5e353bbf6a9576b702bf466a5f952b82">org.jgap.Gene</a>.</p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00138">138</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

</div>
</div>
<a class="anchor" id="a72fa365cdcc9f2eb399fb8346a0a511b"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.BaseGene.compareApplicationData </td>
          <td>(</td>
          <td class="paramtype">final Object&#160;</td>
          <td class="paramname"><em>a_appdata1</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">final Object&#160;</td>
          <td class="paramname"><em>a_appdata2</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00306">306</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_base_gene_8java_source.html#l00409">org.jgap.BaseGene.getConfiguration()</a>.</p>

<p>Referenced by <a class="el" href="_number_gene_8java_source.html#l00050">org.jgap.impl.NumberGene.compareTo()</a>, <a class="el" href="_boolean_gene_8java_source.html#l00264">org.jgap.impl.BooleanGene.compareTo()</a>, <a class="el" href="_string_gene_8java_source.html#l00409">org.jgap.impl.StringGene.compareTo()</a>, <a class="el" href="_fixed_binary_gene_8java_source.html#l00423">org.jgap.impl.FixedBinaryGene.compareTo()</a>, <a class="el" href="_composite_gene_8java_source.html#l00450">org.jgap.impl.CompositeGene.compareTo()</a>, and <a class="el" href="_base_gene_8java_source.html#l00193">org.jgap.BaseGene.equals()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_gene_a72fa365cdcc9f2eb399fb8346a0a511b_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_base_gene_a72fa365cdcc9f2eb399fb8346a0a511b_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_base_gene_a72fa365cdcc9f2eb399fb8346a0a511b_cgraph" id="classorg_1_1jgap_1_1_base_gene_a72fa365cdcc9f2eb399fb8346a0a511b_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_gene.html#a32a316c9cc9e273bd75182cbdf188fb6" title="org.jgap.BaseGene.getConfiguration" alt="" coords="240,12,472,39"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ae9277d0b77c1e8ca244687a8ab8cf42f"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.BaseGene.decode </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_string</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00426">426</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>Referenced by <a class="el" href="_abstract_supergene_8java_source.html#l00453">org.jgap.supergenes.AbstractSupergene.createValidator()</a>, <a class="el" href="_string_gene_8java_source.html#l00205">org.jgap.impl.StringGene.setValueFromPersistentRepresentation()</a>, <a class="el" href="_composite_gene_8java_source.html#l00275">org.jgap.impl.CompositeGene.setValueFromPersistentRepresentation()</a>, and <a class="el" href="_abstract_supergene_8java_source.html#l00411">org.jgap.supergenes.AbstractSupergene.setValueFromPersistentRepresentation()</a>.</p>

</div>
</div>
<a class="anchor" id="a5e8fc72c1fccdb665113c0c34e3fa4b5"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.BaseGene.encode </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_string</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00422">422</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>Referenced by <a class="el" href="_string_gene_8java_source.html#l00306">org.jgap.impl.StringGene.getPersistentRepresentation()</a>, <a class="el" href="_composite_gene_8java_source.html#l00339">org.jgap.impl.CompositeGene.getPersistentRepresentation()</a>, and <a class="el" href="_abstract_supergene_8java_source.html#l00363">org.jgap.supergenes.AbstractSupergene.getPersistentRepresentation()</a>.</p>

</div>
</div>
<a class="anchor" id="a9fb0e302ae57ee3f2f5d1ee08a75bf43"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.BaseGene.equals </td>
          <td>(</td>
          <td class="paramtype">final Object&#160;</td>
          <td class="paramname"><em>a_other</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Compares this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> with the given object and returns true if the other object is a <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> of the same type and has the same value (allele) as this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>. Otherwise it returns false.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_other</td><td>the object to compare to this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> for equality </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>true if this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> is equal to the given object, false otherwise</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.1 </dd></dl>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00193">193</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_base_gene_8java_source.html#l00306">org.jgap.BaseGene.compareApplicationData()</a>, <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#a4eb1d079f14465a2324ad41a6ee3b77a">org.jgap.Gene.getApplicationData()</a>, <a class="el" href="_base_gene_8java_source.html#l00278">org.jgap.BaseGene.getApplicationData()</a>, and <a class="el" href="_base_gene_8java_source.html#l00302">org.jgap.BaseGene.isCompareApplicationData()</a>.</p>

<p>Referenced by <a class="el" href="_composite_gene_8java_source.html#l00154">org.jgap.impl.CompositeGene.addGene()</a>, <a class="el" href="_set_gene_test_8java_source.html#l00109">org.jgap.impl.SetGeneTest.testEquals_2()</a>, <a class="el" href="_composite_gene_test_8java_source.html#l00332">org.jgap.impl.CompositeGeneTest.testEquals_5()</a>, <a class="el" href="_composite_gene_test_8java_source.html#l00349">org.jgap.impl.CompositeGeneTest.testEquals_5_2()</a>, <a class="el" href="_integer_gene_test_8java_source.html#l00170">org.jgap.impl.IntegerGeneTest.testEquals_8()</a>, <a class="el" href="_integer_gene_test_8java_source.html#l00191">org.jgap.impl.IntegerGeneTest.testEquals_9()</a>, <a class="el" href="_integer_gene_test_8java_source.html#l00212">org.jgap.impl.IntegerGeneTest.testEquals_9_2()</a>, <a class="el" href="_integer_gene_test_8java_source.html#l00251">org.jgap.impl.IntegerGeneTest.testEquals_9_3()</a>, <a class="el" href="_composite_gene_test_8java_source.html#l00501">org.jgap.impl.CompositeGeneTest.testNewGene_0()</a>, <a class="el" href="_fixed_binary_gene_test_8java_source.html#l01216">org.jgap.impl.FixedBinaryGeneTest.testNewGene_0()</a>, <a class="el" href="_map_gene_test_8java_source.html#l00270">org.jgap.impl.MapGeneTest.testNewGene_2()</a>, and <a class="el" href="_map_gene_test_8java_source.html#l00285">org.jgap.impl.MapGeneTest.testNewGene_3()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_gene_a9fb0e302ae57ee3f2f5d1ee08a75bf43_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_base_gene_a9fb0e302ae57ee3f2f5d1ee08a75bf43_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_base_gene_a9fb0e302ae57ee3f2f5d1ee08a75bf43_cgraph" id="classorg_1_1jgap_1_1_base_gene_a9fb0e302ae57ee3f2f5d1ee08a75bf43_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_gene.html#a72fa365cdcc9f2eb399fb8346a0a511b" title="org.jgap.BaseGene.compare\lApplicationData" alt="" coords="260,5,447,46"/><area shape="rect" id="node7" href="interfaceorg_1_1jgap_1_1_gene.html#a4eb1d079f14465a2324ad41a6ee3b77a" title="org.jgap.Gene.getApplicationData" alt="" coords="245,71,462,97"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_base_gene.html#a0520139dc694dc6034aeb631ae7d70bd" title="org.jgap.BaseGene.getApplicationData" alt="" coords="230,121,477,148"/><area shape="rect" id="node11" href="classorg_1_1jgap_1_1_base_gene.html#acd6431c06c81bc03b4abbe32e4d9d8ef" title="org.jgap.BaseGene.isCompare\lApplicationData" alt="" coords="253,173,453,214"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_gene.html#a32a316c9cc9e273bd75182cbdf188fb6" title="org.jgap.BaseGene.getConfiguration" alt="" coords="525,12,757,39"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a35e754e8e217bdd26962e1c3c31d0169"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.BaseGene.getAllele </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves the allele value represented by this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>.</p>
<dl class="section return"><dt>Returns</dt><dd>the allele value of this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#aa06c80639659ddbcfa1cfe7b7bb109f9">org.jgap.Gene</a>.</p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00103">103</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="classorg_1_1jgap_1_1_base_gene.html#ad20da6b23ba524a95caa28f727c26dcd">org.jgap.BaseGene.getInternalValue()</a>.</p>

<p>Referenced by <a class="el" href="_integer_gene_8java_source.html#l00331">org.jgap.impl.IntegerGene.applyMutation()</a>, <a class="el" href="_mutiple_integer_gene_8java_source.html#l00352">org.jgap.impl.MutipleIntegerGene.applyMutation()</a>, <a class="el" href="_double_gene_8java_source.html#l00236">org.jgap.impl.DoubleGene.doubleValue()</a>, <a class="el" href="_base_gene_8java_source.html#l00413">org.jgap.BaseGene.getBusinessKey()</a>, <a class="el" href="_integer_gene_8java_source.html#l00239">org.jgap.impl.IntegerGene.intValue()</a>, <a class="el" href="_double_gene_8java_source.html#l00288">org.jgap.impl.DoubleGene.mapValueToWithinBounds()</a>, <a class="el" href="_integer_gene_8java_source.html#l00293">org.jgap.impl.IntegerGene.mapValueToWithinBounds()</a>, <a class="el" href="_mutiple_integer_gene_8java_source.html#l00315">org.jgap.impl.MutipleIntegerGene.mapValueToWithinBounds()</a>, <a class="el" href="_number_gene_test_8java_source.html#l00529">org.jgap.impl.NumberGeneTest.NumberGeneImpl.mapValueToWithinBounds()</a>, <a class="el" href="_map_gene_8java_source.html#l00099">org.jgap.impl.MapGene.newGeneInternal()</a>, <a class="el" href="_set_gene_test_8java_source.html#l00087">org.jgap.impl.SetGeneTest.testGetAllele_0()</a>, <a class="el" href="_map_gene_test_8java_source.html#l00134">org.jgap.impl.MapGeneTest.testGetAllele_0()</a>, <a class="el" href="_double_gene_test_8java_source.html#l00943">org.jgap.impl.DoubleGeneTest.testMapValueToWithinBoundsSupportsFullDoubleRange()</a>, <a class="el" href="_double_gene_test_8java_source.html#l00960">org.jgap.impl.DoubleGeneTest.testMapValueToWithinBoundsSupportsFullDoubleRange2()</a>, <a class="el" href="_double_gene_test_8java_source.html#l00978">org.jgap.impl.DoubleGeneTest.testMapValueToWithinBoundsSupportsFullDoubleRange3()</a>, <a class="el" href="_double_gene_test_8java_source.html#l00995">org.jgap.impl.DoubleGeneTest.testMapValueToWithinBoundsSupportsFullDoubleRange4()</a>, <a class="el" href="_double_gene_test_8java_source.html#l01012">org.jgap.impl.DoubleGeneTest.testMapValueToWithinBoundsSupportsFullDoubleRange5()</a>, <a class="el" href="_integer_gene_test_8java_source.html#l00985">org.jgap.impl.IntegerGeneTest.testMapValueToWithinBoundsSupportsFullIntegerRange()</a>, <a class="el" href="_inversion_operator_test_8java_source.html#l00173">org.jgap.impl.InversionOperatorTest.testOperate_2()</a>, <a class="el" href="_composite_gene_test_8java_source.html#l00586">org.jgap.impl.CompositeGeneTest.testPersistentPresentation_1()</a>, <a class="el" href="_map_gene_test_8java_source.html#l00353">org.jgap.impl.MapGeneTest.testPersistentRepresentation_2()</a>, <a class="el" href="_integer_gene_test_8java_source.html#l00392">org.jgap.impl.IntegerGeneTest.testPersistentRepresentation_2()</a>, <a class="el" href="_integer_gene_test_8java_source.html#l00414">org.jgap.impl.IntegerGeneTest.testPersistentRepresentation_3()</a>, <a class="el" href="_composite_gene_test_8java_source.html#l00459">org.jgap.impl.CompositeGeneTest.testSetAllele_4()</a>, <a class="el" href="_composite_gene_test_8java_source.html#l00483">org.jgap.impl.CompositeGeneTest.testSetAllele_5()</a>, <a class="el" href="_string_gene_test_8java_source.html#l00991">org.jgap.impl.StringGeneTest.testSetToRandomValue_0()</a>, <a class="el" href="_string_gene_test_8java_source.html#l01029">org.jgap.impl.StringGeneTest.testSetToRandomValue_3()</a>, <a class="el" href="_string_gene_test_8java_source.html#l01042">org.jgap.impl.StringGeneTest.testSetToRandomValue_4()</a>, <a class="el" href="_set_gene_test_8java_source.html#l00416">org.jgap.impl.SetGeneTest.testSetValueFromPersistentRepresentation_1()</a>, <a class="el" href="_boolean_gene_test_8java_source.html#l00569">org.jgap.impl.BooleanGeneTest.testSetValueFromPersistentRepresentation_1()</a>, <a class="el" href="_set_gene_test_8java_source.html#l00429">org.jgap.impl.SetGeneTest.testSetValueFromPersistentRepresentation_2()</a>, <a class="el" href="_boolean_gene_test_8java_source.html#l00582">org.jgap.impl.BooleanGeneTest.testSetValueFromPersistentRepresentation_2()</a>, <a class="el" href="_set_gene_test_8java_source.html#l00441">org.jgap.impl.SetGeneTest.testSetValueFromPersistentRepresentation_3()</a>, and <a class="el" href="_boolean_gene_test_8java_source.html#l00595">org.jgap.impl.BooleanGeneTest.testSetValueFromPersistentRepresentation_3()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_gene_a35e754e8e217bdd26962e1c3c31d0169_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_base_gene_a35e754e8e217bdd26962e1c3c31d0169_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_base_gene_a35e754e8e217bdd26962e1c3c31d0169_cgraph" id="classorg_1_1jgap_1_1_base_gene_a35e754e8e217bdd26962e1c3c31d0169_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_gene.html#ad20da6b23ba524a95caa28f727c26dcd" title="org.jgap.BaseGene.getInternal\lValue" alt="" coords="241,5,439,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a0520139dc694dc6034aeb631ae7d70bd"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.BaseGene.getApplicationData </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves the application-specific data that is attached to this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>. Attaching application-specific data may be useful for some applications when it comes time to distinguish a <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> from another. JGAP ignores this data functionally.</p>
<dl class="section return"><dt>Returns</dt><dd>the application-specific data previously attached to this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>, or null if there is no data attached</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.4 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#a4eb1d079f14465a2324ad41a6ee3b77a">org.jgap.Gene</a>.</p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00278">278</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_base_gene_8java_source.html#l00049">org.jgap.BaseGene.m_applicationData</a>.</p>

<p>Referenced by <a class="el" href="_number_gene_8java_source.html#l00050">org.jgap.impl.NumberGene.compareTo()</a>, <a class="el" href="_boolean_gene_8java_source.html#l00264">org.jgap.impl.BooleanGene.compareTo()</a>, <a class="el" href="_string_gene_8java_source.html#l00409">org.jgap.impl.StringGene.compareTo()</a>, <a class="el" href="_fixed_binary_gene_8java_source.html#l00423">org.jgap.impl.FixedBinaryGene.compareTo()</a>, <a class="el" href="_composite_gene_8java_source.html#l00450">org.jgap.impl.CompositeGene.compareTo()</a>, <a class="el" href="_base_gene_8java_source.html#l00193">org.jgap.BaseGene.equals()</a>, <a class="el" href="_base_gene_8java_source.html#l00392">org.jgap.BaseGene.newGene()</a>, and <a class="el" href="_base_gene_8java_source.html#l00152">org.jgap.BaseGene.toString()</a>.</p>

</div>
</div>
<a class="anchor" id="ac4e7c0fad73465365953b375cfa802a8"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.BaseGene.getBusinessKey </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>business key</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_business_key.html#ab1b12087d8726e9020441a1fed6be0f8">org.jgap.IBusinessKey</a>.</p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00413">413</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_base_gene_8java_source.html#l00103">org.jgap.BaseGene.getAllele()</a>, and <a class="el" href="_gene_8java_source.html#l00038">org.jgap.Gene.PERSISTENT_FIELD_DELIMITER</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_gene_ac4e7c0fad73465365953b375cfa802a8_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_base_gene_ac4e7c0fad73465365953b375cfa802a8_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_base_gene_ac4e7c0fad73465365953b375cfa802a8_cgraph" id="classorg_1_1jgap_1_1_base_gene_ac4e7c0fad73465365953b375cfa802a8_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_gene.html#a35e754e8e217bdd26962e1c3c31d0169" title="org.jgap.BaseGene.getAllele" alt="" coords="285,12,472,39"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_gene.html#ad20da6b23ba524a95caa28f727c26dcd" title="org.jgap.BaseGene.getInternal\lValue" alt="" coords="521,5,719,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a32a316c9cc9e273bd75182cbdf188fb6"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> org.jgap.BaseGene.getConfiguration </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>the configuration set</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#aa178414303649867b6176de80c6ba892">org.jgap.Gene</a>.</p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00409">409</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_base_gene_8java_source.html#l00060">org.jgap.BaseGene.m_configuration</a>.</p>

<p>Referenced by <a class="el" href="_set_gene_8java_source.html#l00127">org.jgap.impl.SetGene.applyMutation()</a>, <a class="el" href="_map_gene_8java_source.html#l00222">org.jgap.impl.MapGene.applyMutation()</a>, <a class="el" href="_abstract_supergene_8java_source.html#l00222">org.jgap.supergenes.AbstractSupergene.applyMutation()</a>, <a class="el" href="_string_gene_8java_source.html#l00568">org.jgap.impl.StringGene.applyMutation()</a>, <a class="el" href="_fixed_binary_gene_8java_source.html#l00100">org.jgap.impl.FixedBinaryGene.clone()</a>, <a class="el" href="_base_gene_8java_source.html#l00306">org.jgap.BaseGene.compareApplicationData()</a>, <a class="el" href="_set_gene_8java_source.html#l00272">org.jgap.impl.SetGene.compareTo()</a>, <a class="el" href="_composite_gene_8java_source.html#l00319">org.jgap.impl.CompositeGene.createGene()</a>, <a class="el" href="_abstract_supergene_8java_source.html#l00488">org.jgap.supergenes.AbstractSupergene.createGene()</a>, <a class="el" href="_abstract_supergene_8java_source.html#l00453">org.jgap.supergenes.AbstractSupergene.createValidator()</a>, <a class="el" href="_double_gene_8java_source.html#l00288">org.jgap.impl.DoubleGene.mapValueToWithinBounds()</a>, <a class="el" href="_integer_gene_8java_source.html#l00293">org.jgap.impl.IntegerGene.mapValueToWithinBounds()</a>, <a class="el" href="_mutiple_integer_gene_8java_source.html#l00315">org.jgap.impl.MutipleIntegerGene.mapValueToWithinBounds()</a>, <a class="el" href="_set_gene_8java_source.html#l00062">org.jgap.impl.SetGene.newGeneInternal()</a>, <a class="el" href="_fixed_binary_gene_8java_source.html#l00069">org.jgap.impl.FixedBinaryGene.newGeneInternal()</a>, <a class="el" href="_map_gene_8java_source.html#l00099">org.jgap.impl.MapGene.newGeneInternal()</a>, <a class="el" href="_double_gene_8java_source.html#l00108">org.jgap.impl.DoubleGene.newGeneInternal()</a>, <a class="el" href="_integer_gene_8java_source.html#l00113">org.jgap.impl.IntegerGene.newGeneInternal()</a>, <a class="el" href="_mutiple_integer_gene_8java_source.html#l00118">org.jgap.impl.MutipleIntegerGene.newGeneInternal()</a>, <a class="el" href="_boolean_gene_8java_source.html#l00125">org.jgap.impl.BooleanGene.newGeneInternal()</a>, <a class="el" href="_abstract_supergene_8java_source.html#l00192">org.jgap.supergenes.AbstractSupergene.newGeneInternal()</a>, <a class="el" href="_string_gene_8java_source.html#l00383">org.jgap.impl.StringGene.newGeneInternal()</a>, <a class="el" href="_composite_gene_8java_source.html#l00418">org.jgap.impl.CompositeGene.newGeneInternal()</a>, <a class="el" href="_number_gene_test_8java_source.html#l00489">org.jgap.impl.NumberGeneTest.NumberGeneImpl.newGeneInternal()</a>, <a class="el" href="_fixed_binary_gene_8java_source.html#l00198">org.jgap.impl.FixedBinaryGene.substring()</a>, and <a class="el" href="_set_gene_test_8java_source.html#l00032">org.jgap.impl.SetGeneTest.testConstruct_0()</a>.</p>

</div>
</div>
<a class="anchor" id="ae9823bef7c550fc1533afb8ab2b34d93"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a> org.jgap.BaseGene.getConstraintChecker </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd><a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a> the constraint checker to be used whenever method <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#af46a0c31e0fecea483497e87ca278127">setAllele(Object)</a> is called.</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.5 (moved from CompositeGene, where it was since 2.0) </dd></dl>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00370">370</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_base_gene_8java_source.html#l00348">org.jgap.BaseGene.m_geneAlleleChecker</a>.</p>

<p>Referenced by <a class="el" href="_base_gene_8java_source.html#l00392">org.jgap.BaseGene.newGene()</a>, <a class="el" href="_string_gene_8java_source.html#l00383">org.jgap.impl.StringGene.newGeneInternal()</a>, <a class="el" href="_composite_gene_8java_source.html#l00418">org.jgap.impl.CompositeGene.newGeneInternal()</a>, <a class="el" href="_fixed_binary_gene_8java_source.html#l00109">org.jgap.impl.FixedBinaryGene.setAllele()</a>, <a class="el" href="_number_gene_8java_source.html#l00138">org.jgap.impl.NumberGene.setAllele()</a>, <a class="el" href="_string_gene_8java_source.html#l00347">org.jgap.impl.StringGene.setAllele()</a>, <a class="el" href="_composite_gene_8java_source.html#l00389">org.jgap.impl.CompositeGene.setAllele()</a>, <a class="el" href="_double_gene_test_8java_source.html#l00411">org.jgap.impl.DoubleGeneTest.testNewGene_0()</a>, <a class="el" href="_number_gene_test_8java_source.html#l00420">org.jgap.impl.NumberGeneTest.testSetConstraintChecker_0()</a>, <a class="el" href="_fixed_binary_gene_test_8java_source.html#l00529">org.jgap.impl.FixedBinaryGeneTest.testSetConstraintChecker_0()</a>, <a class="el" href="_double_gene_test_8java_source.html#l00828">org.jgap.impl.DoubleGeneTest.testSetConstraintChecker_0()</a>, <a class="el" href="_composite_gene_test_8java_source.html#l00844">org.jgap.impl.CompositeGeneTest.testSetConstraintChecker_0()</a>, and <a class="el" href="_string_gene_test_8java_source.html#l01131">org.jgap.impl.StringGeneTest.testSetConstraintChecker_0()</a>.</p>

</div>
</div>
<a class="anchor" id="a48f71c4c1282de2ca2cf8f4009c4615a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double org.jgap.BaseGene.getEnergy </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>energy of the gene</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.3 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#aa1794038b5018cdb49998639c79cb57a">org.jgap.Gene</a>.</p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00235">235</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_base_gene_8java_source.html#l00039">org.jgap.BaseGene.m_energy</a>.</p>

<p>Referenced by <a class="el" href="_base_gene_8java_source.html#l00392">org.jgap.BaseGene.newGene()</a>, <a class="el" href="_number_gene_test_8java_source.html#l00439">org.jgap.impl.NumberGeneTest.testSetEnergy_0()</a>, <a class="el" href="_set_gene_test_8java_source.html#l00558">org.jgap.impl.SetGeneTest.testSetEnergy_0()</a>, <a class="el" href="_boolean_gene_test_8java_source.html#l00709">org.jgap.impl.BooleanGeneTest.testSetEnergy_0()</a>, <a class="el" href="_double_gene_test_8java_source.html#l00878">org.jgap.impl.DoubleGeneTest.testSetEnergy_0()</a>, <a class="el" href="_integer_gene_test_8java_source.html#l00898">org.jgap.impl.IntegerGeneTest.testSetEnergy_0()</a>, <a class="el" href="_composite_gene_test_8java_source.html#l00915">org.jgap.impl.CompositeGeneTest.testSetEnergy_0()</a>, <a class="el" href="_string_gene_test_8java_source.html#l01166">org.jgap.impl.StringGeneTest.testSetEnergy_0()</a>, <a class="el" href="_fixed_binary_gene_test_8java_source.html#l01299">org.jgap.impl.FixedBinaryGeneTest.testSetEnergy_0()</a>, <a class="el" href="_number_gene_test_8java_source.html#l00451">org.jgap.impl.NumberGeneTest.testSetEnergy_1()</a>, <a class="el" href="_set_gene_test_8java_source.html#l00570">org.jgap.impl.SetGeneTest.testSetEnergy_1()</a>, <a class="el" href="_boolean_gene_test_8java_source.html#l00721">org.jgap.impl.BooleanGeneTest.testSetEnergy_1()</a>, <a class="el" href="_double_gene_test_8java_source.html#l00890">org.jgap.impl.DoubleGeneTest.testSetEnergy_1()</a>, <a class="el" href="_integer_gene_test_8java_source.html#l00910">org.jgap.impl.IntegerGeneTest.testSetEnergy_1()</a>, <a class="el" href="_composite_gene_test_8java_source.html#l00928">org.jgap.impl.CompositeGeneTest.testSetEnergy_1()</a>, <a class="el" href="_string_gene_test_8java_source.html#l01178">org.jgap.impl.StringGeneTest.testSetEnergy_1()</a>, and <a class="el" href="_fixed_binary_gene_test_8java_source.html#l01311">org.jgap.impl.FixedBinaryGeneTest.testSetEnergy_1()</a>.</p>

</div>
</div>
<a class="anchor" id="ad20da6b23ba524a95caa28f727c26dcd"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">abstract Object org.jgap.BaseGene.getInternalValue </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span><span class="mlabel">pure virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Each <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> implementation holds its own m_value object keeping the allele value. In your <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> implementation, just return it with this method (see <a class="el" href="classorg_1_1jgap_1_1impl_1_1_boolean_gene.html">org.jgap.impl.BooleanGene</a> for example) </p>
<dl class="section return"><dt>Returns</dt><dd>the m_value object</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.2 </dd></dl>

<p>Implemented in <a class="el" href="classorg_1_1jgap_1_1impl_1_1_composite_gene.html#a9ee20f6169c0d441cdacb3685d5336f6">org.jgap.impl.CompositeGene</a>, <a class="el" href="classorg_1_1jgap_1_1supergenes_1_1_abstract_supergene.html#af4b0e1f740b6aa78679ab71c51568cea">org.jgap.supergenes.AbstractSupergene</a>, <a class="el" href="classorg_1_1jgap_1_1impl_1_1_string_gene.html#a2510aeb97b5362ab5ce9e00d412873dd">org.jgap.impl.StringGene</a>, <a class="el" href="classorg_1_1jgap_1_1impl_1_1_map_gene.html#a20a2cdecf740152ac3454df33e0d8449">org.jgap.impl.MapGene</a>, <a class="el" href="classorg_1_1jgap_1_1impl_1_1_fixed_binary_gene.html#a9c3d7ad5bd78dbdf8b6c17278fe14f11">org.jgap.impl.FixedBinaryGene</a>, <a class="el" href="classorg_1_1jgap_1_1impl_1_1_boolean_gene.html#ad1d9fb9f74b0905a92472f22050ab5b8">org.jgap.impl.BooleanGene</a>, <a class="el" href="classorg_1_1jgap_1_1impl_1_1_set_gene.html#a5877331914ff77741c6bcde9531008a2">org.jgap.impl.SetGene</a>, and <a class="el" href="classorg_1_1jgap_1_1impl_1_1_number_gene.html#a123a17deb1c4d63e97ade4ec25e2917f">org.jgap.impl.NumberGene</a>.</p>

<p>Referenced by <a class="el" href="_base_gene_8java_source.html#l00103">org.jgap.BaseGene.getAllele()</a>, <a class="el" href="_base_gene_8java_source.html#l00118">org.jgap.BaseGene.hashCode()</a>, and <a class="el" href="_base_gene_8java_source.html#l00152">org.jgap.BaseGene.toString()</a>.</p>

</div>
</div>
<a class="anchor" id="ab53d562a62f5c72fc88c7e5315a23ed7"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.BaseGene.getUniqueID </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>unique ID of the gene, which allows to distinct this instance from others, in the best case worldwide</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.5 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html#a1276412b558e169199771997a190f876">org.jgap.IUniqueKey</a>.</p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00437">437</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_base_gene_8java_source.html#l00066">org.jgap.BaseGene.m_uniqueID</a>.</p>

</div>
</div>
<a class="anchor" id="ad3c210e440f2b542391db5a909fda3b7"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.BaseGene.getUniqueIDTemplate </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_index</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_index</td><td>the index of the template to retrieve the key for </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>String</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.5 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html#af26fbe03487632a16c84b62325489652">org.jgap.IUniqueKey</a>.</p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00465">465</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

</div>
</div>
<a class="anchor" id="af842a106d90ca973956f560ae448464e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.BaseGene.hashCode </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves the hash code value for a <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>. Override if another <a class="el" href="classorg_1_1jgap_1_1_base_gene.html#af842a106d90ca973956f560ae448464e">hashCode()</a> implementation is necessary or more appropriate than this default implementation.</p>
<dl class="section return"><dt>Returns</dt><dd>this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>'s hash code</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00118">118</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="classorg_1_1jgap_1_1_base_gene.html#ad20da6b23ba524a95caa28f727c26dcd">org.jgap.BaseGene.getInternalValue()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_gene_af842a106d90ca973956f560ae448464e_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_base_gene_af842a106d90ca973956f560ae448464e_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_base_gene_af842a106d90ca973956f560ae448464e_cgraph" id="classorg_1_1jgap_1_1_base_gene_af842a106d90ca973956f560ae448464e_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_gene.html#ad20da6b23ba524a95caa28f727c26dcd" title="org.jgap.BaseGene.getInternal\lValue" alt="" coords="251,5,450,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="acd6431c06c81bc03b4abbe32e4d9d8ef"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.BaseGene.isCompareApplicationData </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#a71de32c2c6bf2b71332449f20d604f81">org.jgap.Gene</a>.</p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00302">302</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_base_gene_8java_source.html#l00058">org.jgap.BaseGene.m_compareAppData</a>.</p>

<p>Referenced by <a class="el" href="_number_gene_8java_source.html#l00050">org.jgap.impl.NumberGene.compareTo()</a>, <a class="el" href="_boolean_gene_8java_source.html#l00264">org.jgap.impl.BooleanGene.compareTo()</a>, <a class="el" href="_string_gene_8java_source.html#l00409">org.jgap.impl.StringGene.compareTo()</a>, <a class="el" href="_fixed_binary_gene_8java_source.html#l00423">org.jgap.impl.FixedBinaryGene.compareTo()</a>, <a class="el" href="_composite_gene_8java_source.html#l00450">org.jgap.impl.CompositeGene.compareTo()</a>, and <a class="el" href="_base_gene_8java_source.html#l00193">org.jgap.BaseGene.equals()</a>.</p>

</div>
</div>
<a class="anchor" id="a40f8f3d249145ad028892cd706ba9dae"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> org.jgap.BaseGene.newGene </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Provides implementation-independent means for creating new <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> instances. The new instance that is created and returned should be setup with any implementation-dependent configuration that this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> instance is setup with (aside from the actual value, of course). For example, if this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> were setup with bounds on its value, then the <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> instance returned from this method should also be setup with those same bounds. This is important, as the JGAP core will invoke this method on each <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> in the sample <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in order to create each new <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> in the same respective gene position for a new <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>.</p>
<dl class="section return"><dt>Returns</dt><dd>a new <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> instance of the same type and with the same setup as this concrete <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a></dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rostan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 (since 1.0 in IntegerGene) </dd></dl>
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000041">Todo:</a></b></dt><dd>clone app.data </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#ae4ddccb7bf569c0ed6cdc19def590964">org.jgap.Gene</a>.</p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00392">392</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_base_gene_8java_source.html#l00278">org.jgap.BaseGene.getApplicationData()</a>, <a class="el" href="_base_gene_8java_source.html#l00370">org.jgap.BaseGene.getConstraintChecker()</a>, <a class="el" href="_base_gene_8java_source.html#l00235">org.jgap.BaseGene.getEnergy()</a>, and <a class="el" href="classorg_1_1jgap_1_1_base_gene.html#aa423e96ffac5a9589fb3a4fbda791b3c">org.jgap.BaseGene.newGeneInternal()</a>.</p>

<p>Referenced by <a class="el" href="_supergene_persistent_representation_test_8java_source.html#l00233">org.jgap.supergenes.SupergenePersistentRepresentationTest.testNewGene_0()</a>, <a class="el" href="_map_gene_test_8java_source.html#l00242">org.jgap.impl.MapGeneTest.testNewGene_0()</a>, <a class="el" href="_integer_gene_test_8java_source.html#l00336">org.jgap.impl.IntegerGeneTest.testNewGene_0()</a>, <a class="el" href="_double_gene_test_8java_source.html#l00411">org.jgap.impl.DoubleGeneTest.testNewGene_0()</a>, <a class="el" href="_composite_gene_test_8java_source.html#l00501">org.jgap.impl.CompositeGeneTest.testNewGene_0()</a>, <a class="el" href="_string_gene_test_8java_source.html#l00539">org.jgap.impl.StringGeneTest.testNewGene_0()</a>, <a class="el" href="_fixed_binary_gene_test_8java_source.html#l01216">org.jgap.impl.FixedBinaryGeneTest.testNewGene_0()</a>, <a class="el" href="_supergene_persistent_representation_test_8java_source.html#l00242">org.jgap.supergenes.SupergenePersistentRepresentationTest.testNewGene_1()</a>, <a class="el" href="_map_gene_test_8java_source.html#l00259">org.jgap.impl.MapGeneTest.testNewGene_1()</a>, <a class="el" href="_supergene_persistent_representation_test_8java_source.html#l00254">org.jgap.supergenes.SupergenePersistentRepresentationTest.testNewGene_2()</a>, <a class="el" href="_map_gene_test_8java_source.html#l00270">org.jgap.impl.MapGeneTest.testNewGene_2()</a>, and <a class="el" href="_map_gene_test_8java_source.html#l00285">org.jgap.impl.MapGeneTest.testNewGene_3()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_gene_a40f8f3d249145ad028892cd706ba9dae_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_base_gene_a40f8f3d249145ad028892cd706ba9dae_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_base_gene_a40f8f3d249145ad028892cd706ba9dae_cgraph" id="classorg_1_1jgap_1_1_base_gene_a40f8f3d249145ad028892cd706ba9dae_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_gene.html#a0520139dc694dc6034aeb631ae7d70bd" title="org.jgap.BaseGene.getApplicationData" alt="" coords="246,5,493,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_gene.html#ae9823bef7c550fc1533afb8ab2b34d93" title="org.jgap.BaseGene.getConstraint\lChecker" alt="" coords="262,57,477,98"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_base_gene.html#a48f71c4c1282de2ca2cf8f4009c4615a" title="org.jgap.BaseGene.getEnergy" alt="" coords="271,123,467,149"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_base_gene.html#aa423e96ffac5a9589fb3a4fbda791b3c" title="org.jgap.BaseGene.newGene\lInternal" alt="" coords="273,174,465,215"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="aa423e96ffac5a9589fb3a4fbda791b3c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">abstract <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> org.jgap.BaseGene.newGeneInternal </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span><span class="mlabel">pure virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Implemented in <a class="el" href="classorg_1_1jgap_1_1impl_1_1_number_gene_test_1_1_number_gene_impl.html#a4dcdc2431e0b48f6d53f4cdbdc605996">org.jgap.impl.NumberGeneTest.NumberGeneImpl</a>, <a class="el" href="classorg_1_1jgap_1_1impl_1_1_composite_gene.html#ab71f66feb3b88bb9bdcab5682abd4b03">org.jgap.impl.CompositeGene</a>, <a class="el" href="classorg_1_1jgap_1_1impl_1_1_string_gene.html#ab9957400b5eb890f04032b5ce498cbf4">org.jgap.impl.StringGene</a>, <a class="el" href="classorg_1_1jgap_1_1supergenes_1_1_abstract_supergene.html#a1e30b794dee0e8da4d5a4d213e9fb2c1">org.jgap.supergenes.AbstractSupergene</a>, <a class="el" href="classorg_1_1jgap_1_1impl_1_1_boolean_gene.html#a57df173aaf3d927451500893e5cccaf3">org.jgap.impl.BooleanGene</a>, <a class="el" href="classorg_1_1jgap_1_1impl_1_1_mutiple_integer_gene.html#a0d94e1fe37cbc3e7cda6481117f2a45a">org.jgap.impl.MutipleIntegerGene</a>, <a class="el" href="classorg_1_1jgap_1_1impl_1_1_integer_gene.html#a225ed6a7409f890ce21009051eb7392f">org.jgap.impl.IntegerGene</a>, <a class="el" href="classorg_1_1jgap_1_1impl_1_1_double_gene.html#af714cdfc2b8f7f3e4d397cf149c09b95">org.jgap.impl.DoubleGene</a>, <a class="el" href="classorg_1_1jgap_1_1impl_1_1_map_gene.html#aea5223852eb46cf96ffb04b4b16165b7">org.jgap.impl.MapGene</a>, <a class="el" href="classorg_1_1jgap_1_1impl_1_1_fixed_binary_gene.html#acf83c8bd879cbbcf3e091eaf718594be">org.jgap.impl.FixedBinaryGene</a>, and <a class="el" href="classorg_1_1jgap_1_1impl_1_1_set_gene.html#abe5acebc3f0d099bdb825e6ef7a7c233">org.jgap.impl.SetGene</a>.</p>

<p>Referenced by <a class="el" href="_base_gene_8java_source.html#l00392">org.jgap.BaseGene.newGene()</a>.</p>

</div>
</div>
<a class="anchor" id="afeb9a5f42018c057c8197fb0e7d57354"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseGene.setApplicationData </td>
          <td>(</td>
          <td class="paramtype">final Object&#160;</td>
          <td class="paramname"><em>a_newData</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>This sets the application-specific data that is attached to this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>. Attaching application-specific data may be useful for some applications when it comes time to distinguish a <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> from another. JGAP ignores this data functionally.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_newData</td><td>the new application-specific data to attach to this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.4 </dd></dl>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00262">262</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_base_gene_8java_source.html#l00049">org.jgap.BaseGene.m_applicationData</a>.</p>

</div>
</div>
<a class="anchor" id="adad4ccd46efd7b2a3dc81bd0ba0c410c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseGene.setCompareApplicationData </td>
          <td>(</td>
          <td class="paramtype">final boolean&#160;</td>
          <td class="paramname"><em>a_doCompare</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Should we also consider the application data when comparing? Default is "false" as "true" means a <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> is losing its identity when application data is set differently!</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_doCompare</td><td>true: consider application data in method compareTo</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.4 </dd></dl>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00292">292</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_base_gene_8java_source.html#l00058">org.jgap.BaseGene.m_compareAppData</a>.</p>

</div>
</div>
<a class="anchor" id="a334db8a543f3008719d78ee219826b65"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseGene.setConstraintChecker </td>
          <td>(</td>
          <td class="paramtype">final <a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a>&#160;</td>
          <td class="paramname"><em>a_constraintChecker</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets the constraint checker to be used for this gene whenever method <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#af46a0c31e0fecea483497e87ca278127">setAllele(Object)</a> is called. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_constraintChecker</td><td>the constraint checker to be set</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.5 (moved from CompositeGene, where it was since 2.0) </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#a1ef8e9cd912cc68ebe6cd246cf896633">org.jgap.Gene</a>.</p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00358">358</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_base_gene_8java_source.html#l00348">org.jgap.BaseGene.m_geneAlleleChecker</a>.</p>

</div>
</div>
<a class="anchor" id="afb39fbd8997971a140a9a4380ceecc4a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseGene.setEnergy </td>
          <td>(</td>
          <td class="paramtype">final double&#160;</td>
          <td class="paramname"><em>a_energy</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets the energy of the gene </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_energy</td><td>the energy to set</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.3 </dd></dl>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00246">246</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_base_gene_8java_source.html#l00039">org.jgap.BaseGene.m_energy</a>.</p>

</div>
</div>
<a class="anchor" id="aa7d9f1c3c2e727d496d66a85818c17d0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.BaseGene.setUniqueIDTemplate </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_templateID</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_index</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>A template is a gene that is the logical predecessor of the current gene. A template can occur in mutation or crossing over. In the latter case can be at least two template genes. This is why in this setter method the parameter a_index exists.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_templateID</td><td>the unique ID of the template </td></tr>
    <tr><td class="paramname">a_index</td><td>the index of the template, e.g. in crossing over for the second candidate gene this is 2</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.5 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html#aeac223774f93ba6640bc07cc97a9ef4c">org.jgap.IUniqueKey</a>.</p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00454">454</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

</div>
</div>
<a class="anchor" id="a26c00775111b327b178288164ed9c667"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.BaseGene.size </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>the size of the gene, i.e the number of atomic elements. Always 1 for non-composed <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> types. Override for composed <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> types</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#a7a975e180dd0ac0fd33cdc182f64bea8">org.jgap.Gene</a>.</p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00178">178</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

</div>
</div>
<a class="anchor" id="a935dfec06cb77bff0d2a45859d6a35b7"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.BaseGene.toString </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves a string representation of this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>'s value that may be useful for display purposes.</p>
<dl class="section return"><dt>Returns</dt><dd>a string representation of this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>'s value</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html#ac62239b4b7bf81179fd491a4c4bd1901">org.jgap.Gene</a>.</p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00152">152</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>References <a class="el" href="_base_gene_8java_source.html#l00278">org.jgap.BaseGene.getApplicationData()</a>, <a class="el" href="classorg_1_1jgap_1_1_base_gene.html#ad20da6b23ba524a95caa28f727c26dcd">org.jgap.BaseGene.getInternalValue()</a>, and <a class="el" href="_base_gene_8java_source.html#l00031">org.jgap.BaseGene.S_APPLICATION_DATA</a>.</p>

<p>Referenced by <a class="el" href="_set_gene_8java_source.html#l00222">org.jgap.impl.SetGene.getPersistentRepresentation()</a>, <a class="el" href="_number_gene_test_8java_source.html#l00499">org.jgap.impl.NumberGeneTest.NumberGeneImpl.getPersistentRepresentation()</a>, <a class="el" href="_set_gene_test_8java_source.html#l00056">org.jgap.impl.SetGeneTest.testToString_0()</a>, and <a class="el" href="_set_gene_test_8java_source.html#l00071">org.jgap.impl.SetGeneTest.testToString_1()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_base_gene_a935dfec06cb77bff0d2a45859d6a35b7_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_base_gene_a935dfec06cb77bff0d2a45859d6a35b7_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_base_gene_a935dfec06cb77bff0d2a45859d6a35b7_cgraph" id="classorg_1_1jgap_1_1_base_gene_a935dfec06cb77bff0d2a45859d6a35b7_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_base_gene.html#a0520139dc694dc6034aeb631ae7d70bd" title="org.jgap.BaseGene.getApplicationData" alt="" coords="238,5,485,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_base_gene.html#ad20da6b23ba524a95caa28f727c26dcd" title="org.jgap.BaseGene.getInternal\lValue" alt="" coords="262,57,461,98"/></map>
</div>
</p>

</div>
</div>
<h2 class="groupheader">Member Data Documentation</h2>
<a class="anchor" id="a7e83775c4b2b9391ec833869808d0154"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final String org.jgap.BaseGene.CVS_REVISION = &quot;$Revision: 1.28 $&quot;</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>String containing the CVS revision. Read out via reflection! </p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00026">26</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

</div>
</div>
<a class="anchor" id="a86edb3d2c29e5064305fabc2dd59a3b9"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final double org.jgap.BaseGene.DELTA = 0.0000001</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Delta, useful for comparing doubles and floats. </p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00036">36</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

</div>
</div>
<a class="anchor" id="ab2e121e812be7a4dead5ca7e300ac3d7"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.BaseGene.m_applicationData</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Application-specific data that is attached to the <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>. This data may assist the application in labelling this <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>. JGAP ignores the data, aside from allowing it to be set and retrieved and considering it in clone() and compareTo().</p>
<dl class="section since"><dt>Since</dt><dd>2.4 </dd></dl>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00049">49</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>Referenced by <a class="el" href="_base_gene_8java_source.html#l00278">org.jgap.BaseGene.getApplicationData()</a>, and <a class="el" href="_base_gene_8java_source.html#l00262">org.jgap.BaseGene.setApplicationData()</a>.</p>

</div>
</div>
<a class="anchor" id="a27bb3cd52e8cd52fffc4953f172709a2"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.BaseGene.m_compareAppData</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Method compareTo(): Should we also consider the application data when comparing? Default is "false" as "true" means a <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>'s losing its identity when application data is set differently!</p>
<dl class="section since"><dt>Since</dt><dd>2.4 </dd></dl>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00058">58</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>Referenced by <a class="el" href="_base_gene_8java_source.html#l00302">org.jgap.BaseGene.isCompareApplicationData()</a>, and <a class="el" href="_base_gene_8java_source.html#l00292">org.jgap.BaseGene.setCompareApplicationData()</a>.</p>

</div>
</div>
<a class="anchor" id="a4d41d885b45ce031c33c88214f98019e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> org.jgap.BaseGene.m_configuration</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00060">60</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>Referenced by <a class="el" href="_base_gene_8java_source.html#l00082">org.jgap.BaseGene.BaseGene()</a>, and <a class="el" href="_base_gene_8java_source.html#l00409">org.jgap.BaseGene.getConfiguration()</a>.</p>

</div>
</div>
<a class="anchor" id="a8553da900cf30fc4b0f5463404b86a8e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double org.jgap.BaseGene.m_energy</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Energy of a gene, see RFE 1102206 </p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00039">39</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>Referenced by <a class="el" href="_base_gene_8java_source.html#l00235">org.jgap.BaseGene.getEnergy()</a>, and <a class="el" href="_base_gene_8java_source.html#l00246">org.jgap.BaseGene.setEnergy()</a>.</p>

</div>
</div>
<a class="anchor" id="a846344a347601679d0935dcdba7f7e6f"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a> org.jgap.BaseGene.m_geneAlleleChecker</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Optional helper class for checking if a given allele value to be set for a given gene is valid. If not, the allele value may not be set for the gene or the gene type (e.g. IntegerGene) is not allowed in general!</p>
<dl class="section since"><dt>Since</dt><dd>2.5 (moved from CompositeGene, where it was since 2.0) </dd></dl>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00348">348</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>Referenced by <a class="el" href="_base_gene_8java_source.html#l00370">org.jgap.BaseGene.getConstraintChecker()</a>, and <a class="el" href="_base_gene_8java_source.html#l00358">org.jgap.BaseGene.setConstraintChecker()</a>.</p>

</div>
</div>
<a class="anchor" id="a8cd57f3a57f5a892f590df9e06b82e47"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.BaseGene.m_uniqueID</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Unique ID of the gene that allows to distinct it from other genes. In the best case, this ID is unique worldwide. </p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00066">66</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>Referenced by <a class="el" href="_base_gene_8java_source.html#l00082">org.jgap.BaseGene.BaseGene()</a>, and <a class="el" href="_base_gene_8java_source.html#l00437">org.jgap.BaseGene.getUniqueID()</a>.</p>

</div>
</div>
<a class="anchor" id="a3f14fef3c7107830f24306cf7bb0abf0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Map&lt;Integer,String&gt; org.jgap.BaseGene.m_uniqueIDTemplates</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>In case mutation, crossing over etc. happened, this sequence gives evidence about the parent(s) of the current gene. </p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00072">72</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>Referenced by <a class="el" href="_base_gene_8java_source.html#l00082">org.jgap.BaseGene.BaseGene()</a>.</p>

</div>
</div>
<a class="anchor" id="acc4b1c46f3508b835bf95b9058eae3a0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final String org.jgap.BaseGene.S_APPLICATION_DATA = &quot;Application data&quot;</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Constants for <a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a935dfec06cb77bff0d2a45859d6a35b7">toString()</a> </p>

<p>Definition at line <a class="el" href="_base_gene_8java_source.html#l00031">31</a> of file <a class="el" href="_base_gene_8java_source.html">BaseGene.java</a>.</p>

<p>Referenced by <a class="el" href="_base_gene_test_8java_source.html#l00049">org.jgap.BaseGeneTest.testToString_0()</a>, <a class="el" href="_number_gene_test_8java_source.html#l00053">org.jgap.impl.NumberGeneTest.testToString_0()</a>, <a class="el" href="_base_gene_test_8java_source.html#l00062">org.jgap.BaseGeneTest.testToString_1()</a>, <a class="el" href="_base_gene_test_8java_source.html#l00076">org.jgap.BaseGeneTest.testToString_2()</a>, and <a class="el" href="_base_gene_8java_source.html#l00152">org.jgap.BaseGene.toString()</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>LDH_Git/src/org/jgap/<a class="el" href="_base_gene_8java_source.html">BaseGene.java</a></li>
</ul>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Sat Nov 23 2013 17:42:10 for Práctica_5_LDH by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
