<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Práctica_5_LDH: org.jgap.Genotype Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">Práctica_5_LDH
   &#160;<span id="projectnumber">1.2</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li><a href="inherits.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Classes</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Namespaces</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Files</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Functions</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(6)"><span class="SelectionMark">&#160;</span>Pages</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="namespaceorg.html">org</a></li><li class="navelem"><a class="el" href="namespaceorg_1_1jgap.html">jgap</a></li><li class="navelem"><a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-static-methods">Static Public Member Functions</a> &#124;
<a href="#pro-methods">Protected Member Functions</a> &#124;
<a href="#pri-attribs">Private Attributes</a> &#124;
<a href="#pri-static-attribs">Static Private Attributes</a> &#124;
<a href="classorg_1_1jgap_1_1_genotype-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">org.jgap.Genotype Class Reference</div>  </div>
</div><!--header-->
<div class="contents">
<div class="dynheader">
Inheritance diagram for org.jgap.Genotype:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype__inherit__graph.png" border="0" usemap="#org_8jgap_8_genotype_inherit__map" alt="Inheritance graph"/></div>
<map name="org_8jgap_8_genotype_inherit__map" id="org_8jgap_8_genotype_inherit__map">
<area shape="rect" id="node7" href="classorg_1_1jgap_1_1distr_1_1_distributed_genotype.html" title="org.jgap.distr.Distributed\lGenotype" alt="" coords="21,155,184,197"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<div class="dynheader">
Collaboration diagram for org.jgap.Genotype:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype__coll__graph.png" border="0" usemap="#org_8jgap_8_genotype_coll__map" alt="Collaboration graph"/></div>
<map name="org_8jgap_8_genotype_coll__map" id="org_8jgap_8_genotype_coll__map">
<area shape="rect" id="node6" href="classorg_1_1jgap_1_1_configuration.html" title="org.jgap.Configuration" alt="" coords="811,439,960,465"/><area shape="rect" id="node17" href="interfaceorg_1_1jgap_1_1_i_breeder.html" title="org.jgap.IBreeder" alt="" coords="397,667,518,693"/><area shape="rect" id="node22" href="interfaceorg_1_1jgap_1_1_i_j_g_a_p_factory.html" title="org.jgap.IJGAPFactory" alt="" coords="380,195,535,221"/><area shape="rect" id="node27" href="interfaceorg_1_1jgap_1_1_fitness_evaluator.html" title="org.jgap.FitnessEvaluator" alt="" coords="372,245,543,272"/><area shape="rect" id="node30" href="interfaceorg_1_1jgap_1_1_i_chromosome.html" title="org.jgap.IChromosome" alt="" coords="380,827,535,853"/><area shape="rect" id="node37" href="interfaceorg_1_1jgap_1_1_random_generator.html" title="org.jgap.RandomGenerator" alt="" coords="368,296,547,323"/><area shape="rect" id="node40" href="classorg_1_1jgap_1_1_bulk_fitness_function.html" title="org.jgap.BulkFitnessFunction" alt="" coords="362,616,553,643"/><area shape="rect" id="node44" href="classorg_1_1jgap_1_1_configuration_1_1_configuration_configurable.html" title="org.jgap.Configuration.Configuration\lConfigurable" alt="" coords="342,347,573,389"/><area shape="rect" id="node47" href="classorg_1_1jgap_1_1_fitness_function.html" title="org.jgap.FitnessFunction" alt="" coords="375,413,540,440"/><area shape="rect" id="node53" href="classorg_1_1jgap_1_1_population.html" title="org.jgap.Population" alt="" coords="1153,493,1287,520"/><area shape="rect" id="node11" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html" title="org.jgap.util.ICloneable" alt="" coords="136,635,291,661"/><area shape="rect" id="node25" href="interfaceorg_1_1jgap_1_1_i_chromosome_pool.html" title="org.jgap.IChromosomePool" alt="" coords="367,464,547,491"/><area shape="rect" id="node35" href="interfaceorg_1_1jgap_1_1_i_unique_key.html" title="org.jgap.IUniqueKey" alt="" coords="143,868,283,895"/><area shape="rect" id="node57" href="interfaceorg_1_1jgap_1_1_i_persistent_representation.html" title="org.jgap.IPersistentRepresentation" alt="" coords="774,503,997,529"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a00b11af9a68e9d84bbc52afbf661607f"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a00b11af9a68e9d84bbc52afbf661607f">Genotype</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration, <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>[] a_initialChromosomes)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a00b11af9a68e9d84bbc52afbf661607f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7b08ffd2154cd3ccc8636b795666c996"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a7b08ffd2154cd3ccc8636b795666c996">Genotype</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration, <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> a_population)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a7b08ffd2154cd3ccc8636b795666c996"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afc52d6b649eea1eefd92e376ea3762ff"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#afc52d6b649eea1eefd92e376ea3762ff">Genotype</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration)  throws InvalidConfigurationException </td></tr>
<tr class="separator:afc52d6b649eea1eefd92e376ea3762ff"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad6795433da5538c6f520253ff9d2382e"><td class="memItemLeft" align="right" valign="top">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#ad6795433da5538c6f520253ff9d2382e">getChromosomes</a> ()</td></tr>
<tr class="separator:ad6795433da5538c6f520253ff9d2382e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8241b2d89d32b283abe5c9b42974d64d"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a> ()</td></tr>
<tr class="separator:a8241b2d89d32b283abe5c9b42974d64d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a05380af4c0737841c09d3a5533f9f942"><td class="memItemLeft" align="right" valign="top">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a05380af4c0737841c09d3a5533f9f942">getFittestChromosome</a> ()</td></tr>
<tr class="separator:a05380af4c0737841c09d3a5533f9f942"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6c40835533352287db8d34e5fae6875f"><td class="memItemLeft" align="right" valign="top">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a6c40835533352287db8d34e5fae6875f">getFittestChromosome</a> (int a_startIndex, int a_endIndex)</td></tr>
<tr class="separator:a6c40835533352287db8d34e5fae6875f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a602d6b1413007ef28880cf88ccd2f912"><td class="memItemLeft" align="right" valign="top">synchronized List&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a602d6b1413007ef28880cf88ccd2f912">getFittestChromosomes</a> (int a_numberOfChromosomes)</td></tr>
<tr class="separator:a602d6b1413007ef28880cf88ccd2f912"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a54a6791fbb6e13920e79f4e2dd719eb4"><td class="memItemLeft" align="right" valign="top">synchronized void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a54a6791fbb6e13920e79f4e2dd719eb4">evolve</a> ()</td></tr>
<tr class="separator:a54a6791fbb6e13920e79f4e2dd719eb4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab15d88cf1a6084f9efea2dc935968471"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#ab15d88cf1a6084f9efea2dc935968471">evolve</a> (int a_numberOfEvolutions)</td></tr>
<tr class="separator:ab15d88cf1a6084f9efea2dc935968471"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa740dd753d31620166979199a38d5a50"><td class="memItemLeft" align="right" valign="top">List&lt; String &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#aa740dd753d31620166979199a38d5a50">evolve</a> (IEvolutionMonitor a_monitor)</td></tr>
<tr class="separator:aa740dd753d31620166979199a38d5a50"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a658a541940e9be8ef474375091ff31ae"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a658a541940e9be8ef474375091ff31ae">toString</a> ()</td></tr>
<tr class="separator:a658a541940e9be8ef474375091ff31ae"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac622bbf487c0ab7344d46ca669607cef"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#ac622bbf487c0ab7344d46ca669607cef">fillPopulation</a> (final int a_num)  throws InvalidConfigurationException </td></tr>
<tr class="separator:ac622bbf487c0ab7344d46ca669607cef"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab9bba698920b358bb53131338c9d130a"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#ab9bba698920b358bb53131338c9d130a">equals</a> (Object a_other)</td></tr>
<tr class="separator:ab9bba698920b358bb53131338c9d130a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:adf2dbe6d60d6c74592cb1e6398cece2c"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#adf2dbe6d60d6c74592cb1e6398cece2c">applyGeneticOperators</a> ()</td></tr>
<tr class="separator:adf2dbe6d60d6c74592cb1e6398cece2c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a786848d2da4fd3d87f9f4caa8f9dd58b"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">getConfiguration</a> ()</td></tr>
<tr class="separator:a786848d2da4fd3d87f9f4caa8f9dd58b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abe044a5473e935c2093ee3c6c18775f7"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#abe044a5473e935c2093ee3c6c18775f7">hashCode</a> ()</td></tr>
<tr class="separator:abe044a5473e935c2093ee3c6c18775f7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0d14c5400197e3c43f9d1f0666309a85"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a0d14c5400197e3c43f9d1f0666309a85">run</a> ()</td></tr>
<tr class="separator:a0d14c5400197e3c43f9d1f0666309a85"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a84fd1b399188ea10c0af4b7c1152e0c2"><td class="memItemLeft" align="right" valign="top">List&lt; IEvolveJob &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a84fd1b399188ea10c0af4b7c1152e0c2">getEvolves</a> (IPopulationSplitter a_splitter)  throws Exception </td></tr>
<tr class="separator:a84fd1b399188ea10c0af4b7c1152e0c2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa2b0341edc420ad61e797508b8bf4dd5"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#aa2b0341edc420ad61e797508b8bf4dd5">mergeResults</a> (IPopulationMerger a_merger, EvolveResult[] a_results)  throws Exception </td></tr>
<tr class="separator:aa2b0341edc420ad61e797508b8bf4dd5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abba6c68b3e8388be6884a35f5600e8cb"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#abba6c68b3e8388be6884a35f5600e8cb">setUseMonitor</a> (boolean a_useMonitor)</td></tr>
<tr class="separator:abba6c68b3e8388be6884a35f5600e8cb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af3f49ba90cc8515817366792b7bce320"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#af3f49ba90cc8515817366792b7bce320">setMonitor</a> (IEvolutionMonitor a_monitor)</td></tr>
<tr class="separator:af3f49ba90cc8515817366792b7bce320"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-static-methods"></a>
Static Public Member Functions</h2></td></tr>
<tr class="memitem:a930184e2920b4642f20d0b8bc8412005"><td class="memItemLeft" align="right" valign="top">static <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a930184e2920b4642f20d0b8bc8412005">randomInitialGenotype</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a930184e2920b4642f20d0b8bc8412005"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aaf3d4bf0b470aa4655e85262289443eb"><td class="memItemLeft" align="right" valign="top">static <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#aaf3d4bf0b470aa4655e85262289443eb">getStaticConfiguration</a> ()</td></tr>
<tr class="separator:aaf3d4bf0b470aa4655e85262289443eb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2f44a7709eca83cddd6d794ee59e663c"><td class="memItemLeft" align="right" valign="top">static void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a2f44a7709eca83cddd6d794ee59e663c">setStaticConfiguration</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration)</td></tr>
<tr class="separator:a2f44a7709eca83cddd6d794ee59e663c"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pro-methods"></a>
Protected Member Functions</h2></td></tr>
<tr class="memitem:a6c14c3a36aa8e4291840f0b9c0facb35"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a6c14c3a36aa8e4291840f0b9c0facb35">applyNaturalSelectors</a> (boolean a_processBeforeGeneticOperators)</td></tr>
<tr class="separator:a6c14c3a36aa8e4291840f0b9c0facb35"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac56a4d112334db07ba333602c4a24706"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#ac56a4d112334db07ba333602c4a24706">setPopulation</a> (<a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> a_pop)</td></tr>
<tr class="separator:ac56a4d112334db07ba333602c4a24706"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a554d1eaf21a0a9437cdbc5b0fbeedef4"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a554d1eaf21a0a9437cdbc5b0fbeedef4">applyGeneticOperator</a> (<a class="el" href="interfaceorg_1_1jgap_1_1_genetic_operator.html">GeneticOperator</a> a_operator, <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> a_population, List a_chromosomes)</td></tr>
<tr class="separator:a554d1eaf21a0a9437cdbc5b0fbeedef4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af5e224c5b7c05dbe02b3324d5ec16f2a"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#af5e224c5b7c05dbe02b3324d5ec16f2a">keepPopSizeConstant</a> (<a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> a_pop, int a_maxSize)</td></tr>
<tr class="separator:af5e224c5b7c05dbe02b3324d5ec16f2a"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-attribs"></a>
Private Attributes</h2></td></tr>
<tr class="memitem:aeb3fac2610b8ef8fa8599fcac1b6a199"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#aeb3fac2610b8ef8fa8599fcac1b6a199">m_activeConfiguration</a></td></tr>
<tr class="separator:aeb3fac2610b8ef8fa8599fcac1b6a199"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa2ba4539110aaa8d853c808679f4f0e2"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#aa2ba4539110aaa8d853c808679f4f0e2">m_population</a></td></tr>
<tr class="separator:aa2ba4539110aaa8d853c808679f4f0e2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab1b1b976deab7f2faea463d1b8d52abe"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#ab1b1b976deab7f2faea463d1b8d52abe">m_useMonitor</a></td></tr>
<tr class="separator:ab1b1b976deab7f2faea463d1b8d52abe"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:acac1218542ab91833483ebd42ac8636b"><td class="memItemLeft" align="right" valign="top">IEvolutionMonitor&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#acac1218542ab91833483ebd42ac8636b">m_monitor</a></td></tr>
<tr class="separator:acac1218542ab91833483ebd42ac8636b"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-static-attribs"></a>
Static Private Attributes</h2></td></tr>
<tr class="memitem:ac9f8e9d203b7a4efdf8c2e32a08f84bb"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#ac9f8e9d203b7a4efdf8c2e32a08f84bb">CVS_REVISION</a> = &quot;$Revision: 1.109 $&quot;</td></tr>
<tr class="separator:ac9f8e9d203b7a4efdf8c2e32a08f84bb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8b93a0b2044e40520d4a966ef3ae0bbe"><td class="memItemLeft" align="right" valign="top">static transient <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a8b93a0b2044e40520d4a966ef3ae0bbe">m_staticConfiguration</a></td></tr>
<tr class="separator:a8b93a0b2044e40520d4a966ef3ae0bbe"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Genotypes are fixed-length populations of chromosomes. As an instance of a <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> is evolved, all of its Chromosomes are also evolved. A <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> may be constructed normally via constructor, or the static <a class="el" href="classorg_1_1jgap_1_1_genotype.html#a930184e2920b4642f20d0b8bc8412005">randomInitialGenotype()</a> method can be used to generate a <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> with a randomized <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> population. </p>
<p>Please note that among all created <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instances there may only be one configuration, used by all <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instances.</p>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00033">33</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="a00b11af9a68e9d84bbc52afbf661607f"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Genotype.Genotype </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>[]&#160;</td>
          <td class="paramname"><em>a_initialChromosomes</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Constructs a new <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance with the given array of Chromosomes and the given active <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> instance. Note that the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object must be in a valid state when this method is invoked, or a <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a> will be thrown.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object to use </td></tr>
    <tr><td class="paramname">a_initialChromosomes</td><td>the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> population to be managed by this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td>if the given <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object is in an invalid state</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>
<dl class="deprecated"><dt><b><a class="el" href="deprecated.html#_deprecated000007">Deprecated:</a></b></dt><dd>use <a class="el" href="classorg_1_1jgap_1_1_genotype.html#a7b08ffd2154cd3ccc8636b795666c996">Genotype(Configuration, Population)</a> instead </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00076">76</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>Referenced by <a class="el" href="_genotype_8java_source.html#l00397">org.jgap.Genotype.equals()</a>, and <a class="el" href="_genotype_8java_source.html#l00321">org.jgap.Genotype.randomInitialGenotype()</a>.</p>

</div>
</div>
<a class="anchor" id="a7b08ffd2154cd3ccc8636b795666c996"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Genotype.Genotype </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>&#160;</td>
          <td class="paramname"><em>a_population</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Constructs a new <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance with the given array of Chromosomes and the given active <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> instance. Note that the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object must be in a valid state when this method is invoked, or a InvalidconfigurationException will be thrown.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object to use </td></tr>
    <tr><td class="paramname">a_population</td><td>the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> population to be managed by this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00098">98</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00042">org.jgap.Genotype.m_activeConfiguration</a>, and <a class="el" href="_genotype_8java_source.html#l00050">org.jgap.Genotype.m_population</a>.</p>

</div>
</div>
<a class="anchor" id="afc52d6b649eea1eefd92e376ea3762ff"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Genotype.Genotype </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Don't use this constructor, it's only for internal use.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>not used here! </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00136">136</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="a554d1eaf21a0a9437cdbc5b0fbeedef4"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.applyGeneticOperator </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_genetic_operator.html">GeneticOperator</a>&#160;</td>
          <td class="paramname"><em>a_operator</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>&#160;</td>
          <td class="paramname"><em>a_population</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">List&#160;</td>
          <td class="paramname"><em>a_chromosomes</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Overwritable method that calls a <a class="el" href="interfaceorg_1_1jgap_1_1_genetic_operator.html">GeneticOperator</a> to operate on a given population and asks him to store the result in the list of chromosomes. Override this method if you want to ensure that a_chromosomes is not part of a_population resp. if you want to use a different list.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_operator</td><td>the <a class="el" href="interfaceorg_1_1jgap_1_1_genetic_operator.html">GeneticOperator</a> to call </td></tr>
    <tr><td class="paramname">a_population</td><td>the population to use </td></tr>
    <tr><td class="paramname">a_chromosomes</td><td>the list of <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> objects to return</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.4 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00587">587</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>Referenced by <a class="el" href="_genotype_8java_source.html#l00501">org.jgap.Genotype.applyGeneticOperators()</a>.</p>

</div>
</div>
<a class="anchor" id="adf2dbe6d60d6c74592cb1e6398cece2c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.applyGeneticOperators </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Applies all GeneticOperators registered with the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00501">501</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00587">org.jgap.Genotype.applyGeneticOperator()</a>, <a class="el" href="_genotype_8java_source.html#l00151">org.jgap.Genotype.getChromosomes()</a>, and <a class="el" href="_genotype_8java_source.html#l00167">org.jgap.Genotype.getPopulation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_adf2dbe6d60d6c74592cb1e6398cece2c_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_adf2dbe6d60d6c74592cb1e6398cece2c_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_adf2dbe6d60d6c74592cb1e6398cece2c_cgraph" id="classorg_1_1jgap_1_1_genotype_adf2dbe6d60d6c74592cb1e6398cece2c_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a554d1eaf21a0a9437cdbc5b0fbeedef4" title="org.jgap.Genotype.applyGenetic\lOperator" alt="" coords="275,5,483,46"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_genotype.html#ad6795433da5538c6f520253ff9d2382e" title="org.jgap.Genotype.getChromosomes" alt="" coords="261,71,496,97"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d" title="org.jgap.Genotype.getPopulation" alt="" coords="544,96,755,123"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a6c14c3a36aa8e4291840f0b9c0facb35"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.applyNaturalSelectors </td>
          <td>(</td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_processBeforeGeneticOperators</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Applies all NaturalSelectors registered with the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_processBeforeGeneticOperators</td><td>true apply NaturalSelectors applicable before GeneticOperators, false: apply the ones applicable after GeneticOperators</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000118">Todo:</a></b></dt><dd>optionally use working pool </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00442">442</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00535">org.jgap.Genotype.getConfiguration()</a>, <a class="el" href="_genotype_8java_source.html#l00167">org.jgap.Genotype.getPopulation()</a>, <a class="el" href="_genotype_8java_source.html#l00042">org.jgap.Genotype.m_activeConfiguration</a>, and <a class="el" href="_genotype_8java_source.html#l00570">org.jgap.Genotype.setPopulation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_a6c14c3a36aa8e4291840f0b9c0facb35_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_a6c14c3a36aa8e4291840f0b9c0facb35_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_a6c14c3a36aa8e4291840f0b9c0facb35_cgraph" id="classorg_1_1jgap_1_1_genotype_a6c14c3a36aa8e4291840f0b9c0facb35_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b" title="org.jgap.Genotype.getConfiguration" alt="" coords="259,5,485,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d" title="org.jgap.Genotype.getPopulation" alt="" coords="267,56,477,83"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_genotype.html#ac56a4d112334db07ba333602c4a24706" title="org.jgap.Genotype.setPopulation" alt="" coords="267,107,477,133"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ab9bba698920b358bb53131338c9d130a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Genotype.equals </td>
          <td>(</td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>a_other</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Compares this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> against the specified object. The result is true if the argument is an instance of the <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> class, has exactly the same number of chromosomes as the given <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a>, and, for each <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a>, there is an equal chromosome in the given <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a>. The chromosomes do not need to appear in the same order within the population.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_other</td><td>the object to compare against </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>true if the objects are the same, false otherwise</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00397">397</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00076">org.jgap.Genotype.Genotype()</a>, <a class="el" href="_genotype_8java_source.html#l00151">org.jgap.Genotype.getChromosomes()</a>, and <a class="el" href="_genotype_8java_source.html#l00167">org.jgap.Genotype.getPopulation()</a>.</p>

<p>Referenced by <a class="el" href="_x_m_l_manager_test_8java_source.html#l00186">org.jgap.xml.XMLManagerTest.testGetGenotypeFromDocument_1()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_ab9bba698920b358bb53131338c9d130a_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_ab9bba698920b358bb53131338c9d130a_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_ab9bba698920b358bb53131338c9d130a_cgraph" id="classorg_1_1jgap_1_1_genotype_ab9bba698920b358bb53131338c9d130a_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a00b11af9a68e9d84bbc52afbf661607f" title="org.jgap.Genotype.Genotype" alt="" coords="247,5,435,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_genotype.html#ad6795433da5538c6f520253ff9d2382e" title="org.jgap.Genotype.getChromosomes" alt="" coords="224,56,459,83"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d" title="org.jgap.Genotype.getPopulation" alt="" coords="507,81,717,108"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a54a6791fbb6e13920e79f4e2dd719eb4"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">synchronized void org.jgap.Genotype.evolve </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Evolves the population of Chromosomes within this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a>. This will execute all of the genetic operators added to the present active configuration and then invoke the natural selector to choose which chromosomes will be included in the next generation population. Note that the population size not always remains constant (dependent on the NaturalSelectors used!). To consecutively call this method, use <a class="el" href="classorg_1_1jgap_1_1_genotype.html#ab15d88cf1a6084f9efea2dc935968471">evolve(int)</a>!!!</p>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00231">231</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00535">org.jgap.Genotype.getConfiguration()</a>, <a class="el" href="_genotype_8java_source.html#l00167">org.jgap.Genotype.getPopulation()</a>, and <a class="el" href="_genotype_8java_source.html#l00570">org.jgap.Genotype.setPopulation()</a>.</p>

<p>Referenced by <a class="el" href="_genotype_8java_source.html#l00248">org.jgap.Genotype.evolve()</a>, and <a class="el" href="_genotype_8java_source.html#l00624">org.jgap.Genotype.run()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_a54a6791fbb6e13920e79f4e2dd719eb4_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_a54a6791fbb6e13920e79f4e2dd719eb4_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_a54a6791fbb6e13920e79f4e2dd719eb4_cgraph" id="classorg_1_1jgap_1_1_genotype_a54a6791fbb6e13920e79f4e2dd719eb4_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b" title="org.jgap.Genotype.getConfiguration" alt="" coords="224,5,451,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d" title="org.jgap.Genotype.getPopulation" alt="" coords="232,56,443,83"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_genotype.html#ac56a4d112334db07ba333602c4a24706" title="org.jgap.Genotype.setPopulation" alt="" coords="232,107,443,133"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ab15d88cf1a6084f9efea2dc935968471"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.evolve </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_numberOfEvolutions</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Evolves this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> the specified number of times. This is equivalent to invoking the standard <a class="el" href="classorg_1_1jgap_1_1_genotype.html#a54a6791fbb6e13920e79f4e2dd719eb4">evolve()</a> method the given number of times in a row.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_numberOfEvolutions</td><td>the number of times to evolve this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> before returning</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.1 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00248">248</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00231">org.jgap.Genotype.evolve()</a>, <a class="el" href="_genotype_8java_source.html#l00167">org.jgap.Genotype.getPopulation()</a>, <a class="el" href="_configuration_8java_source.html#l00963">org.jgap.Configuration.getPopulationSize()</a>, <a class="el" href="_configuration_8java_source.html#l01462">org.jgap.Configuration.isKeepPopulationSizeConstant()</a>, <a class="el" href="_genotype_8java_source.html#l00602">org.jgap.Genotype.keepPopSizeConstant()</a>, and <a class="el" href="_genotype_8java_source.html#l00042">org.jgap.Genotype.m_activeConfiguration</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_ab15d88cf1a6084f9efea2dc935968471_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_ab15d88cf1a6084f9efea2dc935968471_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_ab15d88cf1a6084f9efea2dc935968471_cgraph" id="classorg_1_1jgap_1_1_genotype_ab15d88cf1a6084f9efea2dc935968471_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a54a6791fbb6e13920e79f4e2dd719eb4" title="org.jgap.Genotype.evolve" alt="" coords="235,56,405,83"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d" title="org.jgap.Genotype.getPopulation" alt="" coords="472,107,683,133"/><area shape="rect" id="node12" href="classorg_1_1jgap_1_1_configuration.html#a21cd64586b838743538195b6e8a42a11" title="org.jgap.Configuration.get\lPopulationSize" alt="" coords="235,158,405,199"/><area shape="rect" id="node14" href="classorg_1_1jgap_1_1_configuration.html#ad41cca8a908a60f0e4575f02f59b780e" title="org.jgap.Configuration.is\lKeepPopulationSizeConstant" alt="" coords="224,223,416,265"/><area shape="rect" id="node16" href="classorg_1_1jgap_1_1_genotype.html#af5e224c5b7c05dbe02b3324d5ec16f2a" title="org.jgap.Genotype.keepPop\lSizeConstant" alt="" coords="229,289,411,330"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b" title="org.jgap.Genotype.getConfiguration" alt="" coords="464,5,691,32"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_genotype.html#ac56a4d112334db07ba333602c4a24706" title="org.jgap.Genotype.setPopulation" alt="" coords="472,56,683,83"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="aa740dd753d31620166979199a38d5a50"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">List&lt;String&gt; org.jgap.Genotype.evolve </td>
          <td>(</td>
          <td class="paramtype">IEvolutionMonitor&#160;</td>
          <td class="paramname"><em>a_monitor</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Evolves this genotype until the given monitor asks to quit the evolution cycle.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_monitor</td><td>the monitor used to decide when to stop evolution</td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>messages of the registered evolution monitor. May indicate why the evolution was asked to be stopped. May be empty, depending on the implementation of the used monitor</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.4.4 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00271">271</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00231">org.jgap.Genotype.evolve()</a>, <a class="el" href="_genotype_8java_source.html#l00535">org.jgap.Genotype.getConfiguration()</a>, and <a class="el" href="_genotype_8java_source.html#l00167">org.jgap.Genotype.getPopulation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_aa740dd753d31620166979199a38d5a50_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_aa740dd753d31620166979199a38d5a50_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_aa740dd753d31620166979199a38d5a50_cgraph" id="classorg_1_1jgap_1_1_genotype_aa740dd753d31620166979199a38d5a50_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a54a6791fbb6e13920e79f4e2dd719eb4" title="org.jgap.Genotype.evolve" alt="" coords="224,56,395,83"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b" title="org.jgap.Genotype.getConfiguration" alt="" coords="443,5,669,32"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d" title="org.jgap.Genotype.getPopulation" alt="" coords="451,107,661,133"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_genotype.html#ac56a4d112334db07ba333602c4a24706" title="org.jgap.Genotype.setPopulation" alt="" coords="451,56,661,83"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ac622bbf487c0ab7344d46ca669607cef"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.fillPopulation </td>
          <td>(</td>
          <td class="paramtype">final int&#160;</td>
          <td class="paramname"><em>a_num</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Fills up the population with random chromosomes if necessary.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_num</td><td>the number of chromosomes to add </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00354">354</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00535">org.jgap.Genotype.getConfiguration()</a>, and <a class="el" href="_genotype_8java_source.html#l00167">org.jgap.Genotype.getPopulation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_ac622bbf487c0ab7344d46ca669607cef_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_ac622bbf487c0ab7344d46ca669607cef_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_ac622bbf487c0ab7344d46ca669607cef_cgraph" id="classorg_1_1jgap_1_1_genotype_ac622bbf487c0ab7344d46ca669607cef_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b" title="org.jgap.Genotype.getConfiguration" alt="" coords="261,5,488,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d" title="org.jgap.Genotype.getPopulation" alt="" coords="269,56,480,83"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ad6795433da5538c6f520253ff9d2382e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> [] org.jgap.Genotype.getChromosomes </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves the array of Chromosomes that make up the population of this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance.</p>
<dl class="section return"><dt>Returns</dt><dd>the <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> of Chromosomes</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>
<dl class="deprecated"><dt><b><a class="el" href="deprecated.html#_deprecated000008">Deprecated:</a></b></dt><dd>uses <a class="el" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation()</a> instead </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00151">151</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00167">org.jgap.Genotype.getPopulation()</a>.</p>

<p>Referenced by <a class="el" href="_genotype_8java_source.html#l00501">org.jgap.Genotype.applyGeneticOperators()</a>, and <a class="el" href="_genotype_8java_source.html#l00397">org.jgap.Genotype.equals()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_ad6795433da5538c6f520253ff9d2382e_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_ad6795433da5538c6f520253ff9d2382e_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_ad6795433da5538c6f520253ff9d2382e_cgraph" id="classorg_1_1jgap_1_1_genotype_ad6795433da5538c6f520253ff9d2382e_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d" title="org.jgap.Genotype.getPopulation" alt="" coords="288,5,499,32"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a786848d2da4fd3d87f9f4caa8f9dd58b"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> org.jgap.Genotype.getConfiguration </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00535">535</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00042">org.jgap.Genotype.m_activeConfiguration</a>.</p>

<p>Referenced by <a class="el" href="_genotype_8java_source.html#l00442">org.jgap.Genotype.applyNaturalSelectors()</a>, <a class="el" href="_genotype_8java_source.html#l00231">org.jgap.Genotype.evolve()</a>, <a class="el" href="_genotype_8java_source.html#l00354">org.jgap.Genotype.fillPopulation()</a>, <a class="el" href="_genotype_8java_source.html#l00644">org.jgap.Genotype.getEvolves()</a>, and <a class="el" href="_genotype_8java_source.html#l00673">org.jgap.Genotype.mergeResults()</a>.</p>

</div>
</div>
<a class="anchor" id="a84fd1b399188ea10c0af4b7c1152e0c2"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">List&lt;IEvolveJob&gt; org.jgap.Genotype.getEvolves </td>
          <td>(</td>
          <td class="paramtype">IPopulationSplitter&#160;</td>
          <td class="paramname"><em>a_splitter</em></td><td>)</td>
          <td> throws Exception</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Splits a population into pieces that can be evolved independently.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_splitter</td><td>splits the population </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>list of IEvolveJob objects </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Exception</td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00644">644</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00535">org.jgap.Genotype.getConfiguration()</a>, and <a class="el" href="_genotype_8java_source.html#l00167">org.jgap.Genotype.getPopulation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_a84fd1b399188ea10c0af4b7c1152e0c2_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_a84fd1b399188ea10c0af4b7c1152e0c2_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_a84fd1b399188ea10c0af4b7c1152e0c2_cgraph" id="classorg_1_1jgap_1_1_genotype_a84fd1b399188ea10c0af4b7c1152e0c2_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b" title="org.jgap.Genotype.getConfiguration" alt="" coords="251,5,477,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d" title="org.jgap.Genotype.getPopulation" alt="" coords="259,56,469,83"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a05380af4c0737841c09d3a5533f9f942"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> org.jgap.Genotype.getFittestChromosome </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in the <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> with the highest fitness value.</p>
<dl class="section return"><dt>Returns</dt><dd>the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> with the highest fitness value, or null if there are no chromosomes in this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a></dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00182">182</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00167">org.jgap.Genotype.getPopulation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_a05380af4c0737841c09d3a5533f9f942_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_a05380af4c0737841c09d3a5533f9f942_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_a05380af4c0737841c09d3a5533f9f942_cgraph" id="classorg_1_1jgap_1_1_genotype_a05380af4c0737841c09d3a5533f9f942_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d" title="org.jgap.Genotype.getPopulation" alt="" coords="240,12,451,39"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a6c40835533352287db8d34e5fae6875f"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> org.jgap.Genotype.getFittestChromosome </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_startIndex</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_endIndex</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in the <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> with the highest fitness value within the given indices.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_startIndex</td><td>the index to start the determination with </td></tr>
    <tr><td class="paramname">a_endIndex</td><td>the index to end the determination with </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> with the highest fitness value within the given indices, or null if there are no chromosomes in this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a></dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00198">198</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00167">org.jgap.Genotype.getPopulation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_a6c40835533352287db8d34e5fae6875f_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_a6c40835533352287db8d34e5fae6875f_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_a6c40835533352287db8d34e5fae6875f_cgraph" id="classorg_1_1jgap_1_1_genotype_a6c40835533352287db8d34e5fae6875f_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d" title="org.jgap.Genotype.getPopulation" alt="" coords="240,12,451,39"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a602d6b1413007ef28880cf88ccd2f912"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">synchronized List org.jgap.Genotype.getFittestChromosomes </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_numberOfChromosomes</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves the top n Chromsomes in the population (the ones with the best fitness values).</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_numberOfChromosomes</td><td>the number of chromosomes desired </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>the list of Chromosomes with the highest fitness values, or null if there are no chromosomes in this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a></dd></dl>
<dl class="section author"><dt>Author</dt><dd>Charles Kevin Hill </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.4 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00214">214</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00167">org.jgap.Genotype.getPopulation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_a602d6b1413007ef28880cf88ccd2f912_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_a602d6b1413007ef28880cf88ccd2f912_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_a602d6b1413007ef28880cf88ccd2f912_cgraph" id="classorg_1_1jgap_1_1_genotype_a602d6b1413007ef28880cf88ccd2f912_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d" title="org.jgap.Genotype.getPopulation" alt="" coords="240,12,451,39"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a8241b2d89d32b283abe5c9b42974d64d"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> org.jgap.Genotype.getPopulation </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>the current population of chromosomes</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.1 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00167">167</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00050">org.jgap.Genotype.m_population</a>.</p>

<p>Referenced by <a class="el" href="_genotype_8java_source.html#l00501">org.jgap.Genotype.applyGeneticOperators()</a>, <a class="el" href="_genotype_8java_source.html#l00442">org.jgap.Genotype.applyNaturalSelectors()</a>, <a class="el" href="_genotype_test_8java_source.html#l00730">org.jgap.GenotypeTest.doTestEvolve_5()</a>, <a class="el" href="_genotype_8java_source.html#l00397">org.jgap.Genotype.equals()</a>, <a class="el" href="_genotype_8java_source.html#l00231">org.jgap.Genotype.evolve()</a>, <a class="el" href="_genotype_8java_source.html#l00354">org.jgap.Genotype.fillPopulation()</a>, <a class="el" href="_genotype_8java_source.html#l00151">org.jgap.Genotype.getChromosomes()</a>, <a class="el" href="_genotype_8java_source.html#l00644">org.jgap.Genotype.getEvolves()</a>, <a class="el" href="_genotype_8java_source.html#l00182">org.jgap.Genotype.getFittestChromosome()</a>, <a class="el" href="_genotype_8java_source.html#l00214">org.jgap.Genotype.getFittestChromosomes()</a>, <a class="el" href="_genotype_8java_source.html#l00555">org.jgap.Genotype.hashCode()</a>, <a class="el" href="_evaluator_8java_source.html#l00293">org.jgap.audit.Evaluator.storeGenotype()</a>, <a class="el" href="_genotype_test_8java_source.html#l00383">org.jgap.GenotypeTest.testEvolve_1()</a>, and <a class="el" href="_genotype_8java_source.html#l00292">org.jgap.Genotype.toString()</a>.</p>

</div>
</div>
<a class="anchor" id="aaf3d4bf0b470aa4655e85262289443eb"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">static <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> org.jgap.Genotype.getStaticConfiguration </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>the configuration to use with the Genetic Algorithm</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00520">520</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00044">org.jgap.Genotype.m_staticConfiguration</a>.</p>

<p>Referenced by <a class="el" href="_chromosome_for_testing_8java_source.html#l00023">org.jgap.ChromosomeForTesting.ChromosomeForTesting()</a>, <a class="el" href="_chromosome_for_testing2_8java_source.html#l00023">org.jgap.ChromosomeForTesting2.ChromosomeForTesting2()</a>, <a class="el" href="_crossover_operator_8java_source.html#l00085">org.jgap.impl.CrossoverOperator.CrossoverOperator()</a>, <a class="el" href="_greedy_crossover_8java_source.html#l00076">org.jgap.impl.GreedyCrossover.GreedyCrossover()</a>, <a class="el" href="_natural_selector_ext_8java_source.html#l00037">org.jgap.NaturalSelectorExt.NaturalSelectorExt()</a>, <a class="el" href="impl_2_tournament_selector_8java_source.html#l00051">org.jgap.impl.TournamentSelector.TournamentSelector()</a>, and <a class="el" href="_two_way_mutation_operator_8java_source.html#l00061">org.jgap.impl.TwoWayMutationOperator.TwoWayMutationOperator()</a>.</p>

</div>
</div>
<a class="anchor" id="abe044a5473e935c2093ee3c6c18775f7"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.Genotype.hashCode </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00555">555</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00167">org.jgap.Genotype.getPopulation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_abe044a5473e935c2093ee3c6c18775f7_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_abe044a5473e935c2093ee3c6c18775f7_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_abe044a5473e935c2093ee3c6c18775f7_cgraph" id="classorg_1_1jgap_1_1_genotype_abe044a5473e935c2093ee3c6c18775f7_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d" title="org.jgap.Genotype.getPopulation" alt="" coords="245,5,456,32"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="af5e224c5b7c05dbe02b3324d5ec16f2a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.keepPopSizeConstant </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>&#160;</td>
          <td class="paramname"><em>a_pop</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_maxSize</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Cares that the population size does not exceed the given maximum size.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_pop</td><td>the population to keep constant in size </td></tr>
    <tr><td class="paramname">a_maxSize</td><td>the maximum size allowed for the population</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.5 </dd></dl>
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000119">Todo:</a></b></dt><dd>use StandardPostSelector instead? </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00602">602</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>Referenced by <a class="el" href="_genotype_8java_source.html#l00248">org.jgap.Genotype.evolve()</a>.</p>

</div>
</div>
<a class="anchor" id="aa2b0341edc420ad61e797508b8bf4dd5"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.mergeResults </td>
          <td>(</td>
          <td class="paramtype">IPopulationMerger&#160;</td>
          <td class="paramname"><em>a_merger</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">EvolveResult[]&#160;</td>
          <td class="paramname"><em>a_results</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws Exception</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000120">Todo:</a></b></dt><dd>use/enhance IPopulationMerger </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00673">673</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00535">org.jgap.Genotype.getConfiguration()</a>, and <a class="el" href="_genotype_8java_source.html#l00570">org.jgap.Genotype.setPopulation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_aa2b0341edc420ad61e797508b8bf4dd5_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_aa2b0341edc420ad61e797508b8bf4dd5_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_aa2b0341edc420ad61e797508b8bf4dd5_cgraph" id="classorg_1_1jgap_1_1_genotype_aa2b0341edc420ad61e797508b8bf4dd5_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b" title="org.jgap.Genotype.getConfiguration" alt="" coords="267,5,493,32"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_genotype.html#ac56a4d112334db07ba333602c4a24706" title="org.jgap.Genotype.setPopulation" alt="" coords="275,56,485,83"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a930184e2920b4642f20d0b8bc8412005"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">static <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> org.jgap.Genotype.randomInitialGenotype </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Convenience method that returns a newly constructed <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance configured according to the given <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> instance. The population of Chromosomes will be created according to the setup of the sample <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object, but the gene values (alleles) will be set to random legal values.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the current active <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>a newly constructed <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance</dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">IllegalArgumentException</td><td>if the given <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object is null </td></tr>
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td>if the given <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> instance is not in a valid state</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.3 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00321">321</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00076">org.jgap.Genotype.Genotype()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_a930184e2920b4642f20d0b8bc8412005_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_a930184e2920b4642f20d0b8bc8412005_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_a930184e2920b4642f20d0b8bc8412005_cgraph" id="classorg_1_1jgap_1_1_genotype_a930184e2920b4642f20d0b8bc8412005_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a00b11af9a68e9d84bbc52afbf661607f" title="org.jgap.Genotype.Genotype" alt="" coords="230,12,418,39"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a0d14c5400197e3c43f9d1f0666309a85"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.run </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>If used in a Thread: runs the evolution forever. You have to implement a listener to stop computation sometime. See <a class="el" href="namespaceexamples_1_1simple_boolean_threaded.html">examples.simpleBooleanThreaded</a> for a possible implementation of such a listener.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.01 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00624">624</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00231">org.jgap.Genotype.evolve()</a>, <a class="el" href="_genotype_8java_source.html#l00057">org.jgap.Genotype.m_monitor</a>, and <a class="el" href="_genotype_8java_source.html#l00055">org.jgap.Genotype.m_useMonitor</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_a0d14c5400197e3c43f9d1f0666309a85_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_a0d14c5400197e3c43f9d1f0666309a85_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_a0d14c5400197e3c43f9d1f0666309a85_cgraph" id="classorg_1_1jgap_1_1_genotype_a0d14c5400197e3c43f9d1f0666309a85_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a54a6791fbb6e13920e79f4e2dd719eb4" title="org.jgap.Genotype.evolve" alt="" coords="205,56,376,83"/><area shape="rect" id="node5" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b" title="org.jgap.Genotype.getConfiguration" alt="" coords="424,5,651,32"/><area shape="rect" id="node7" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d" title="org.jgap.Genotype.getPopulation" alt="" coords="432,56,643,83"/><area shape="rect" id="node9" href="classorg_1_1jgap_1_1_genotype.html#ac56a4d112334db07ba333602c4a24706" title="org.jgap.Genotype.setPopulation" alt="" coords="432,107,643,133"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="af3f49ba90cc8515817366792b7bce320"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.setMonitor </td>
          <td>(</td>
          <td class="paramtype">IEvolutionMonitor&#160;</td>
          <td class="paramname"><em>a_monitor</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets the evolution monitor to use, activate it via setUseMonitor(true). Only to be used when running <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> as a thread.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_monitor</td><td>the IEvolutionMonitor to use</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.6 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00716">716</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00057">org.jgap.Genotype.m_monitor</a>.</p>

</div>
</div>
<a class="anchor" id="ac56a4d112334db07ba333602c4a24706"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.setPopulation </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>&#160;</td>
          <td class="paramname"><em>a_pop</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00570">570</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00050">org.jgap.Genotype.m_population</a>.</p>

<p>Referenced by <a class="el" href="_genotype_8java_source.html#l00442">org.jgap.Genotype.applyNaturalSelectors()</a>, <a class="el" href="_genotype_8java_source.html#l00231">org.jgap.Genotype.evolve()</a>, and <a class="el" href="_genotype_8java_source.html#l00673">org.jgap.Genotype.mergeResults()</a>.</p>

</div>
</div>
<a class="anchor" id="a2f44a7709eca83cddd6d794ee59e663c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">static void org.jgap.Genotype.setStaticConfiguration </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets the configuration to use with the Genetic Algorithm. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the configuration to use</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00531">531</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00044">org.jgap.Genotype.m_staticConfiguration</a>.</p>

</div>
</div>
<a class="anchor" id="abba6c68b3e8388be6884a35f5600e8cb"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.setUseMonitor </td>
          <td>(</td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_useMonitor</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Use an evolution monitor, only to be used when running <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> as a thread. Otherwise use method <a class="el" href="classorg_1_1jgap_1_1_genotype.html#aa740dd753d31620166979199a38d5a50">evolve(IEvolutionMonitor)</a></p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_useMonitor</td><td>true: use evolution monitor, set it via setMonitor</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.6 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00703">703</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00055">org.jgap.Genotype.m_useMonitor</a>.</p>

</div>
</div>
<a class="anchor" id="a658a541940e9be8ef474375091ff31ae"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.Genotype.toString </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>string representation of this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance, useful for display or debug purposes</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd></dl>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00292">292</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>References <a class="el" href="_genotype_8java_source.html#l00167">org.jgap.Genotype.getPopulation()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_a658a541940e9be8ef474375091ff31ae_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_a658a541940e9be8ef474375091ff31ae_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_a658a541940e9be8ef474375091ff31ae_cgraph" id="classorg_1_1jgap_1_1_genotype_a658a541940e9be8ef474375091ff31ae_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d" title="org.jgap.Genotype.getPopulation" alt="" coords="232,5,443,32"/></map>
</div>
</p>

</div>
</div>
<h2 class="groupheader">Member Data Documentation</h2>
<a class="anchor" id="ac9f8e9d203b7a4efdf8c2e32a08f84bb"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">final String org.jgap.Genotype.CVS_REVISION = &quot;$Revision: 1.109 $&quot;</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>String containing the CVS revision. Read out via reflection! </p>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00036">36</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

</div>
</div>
<a class="anchor" id="aeb3fac2610b8ef8fa8599fcac1b6a199"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> org.jgap.Genotype.m_activeConfiguration</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>The current <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> instance. </p>
<dl class="section since"><dt>Since</dt><dd>1.0 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00042">42</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>Referenced by <a class="el" href="_genotype_8java_source.html#l00442">org.jgap.Genotype.applyNaturalSelectors()</a>, <a class="el" href="_genotype_8java_source.html#l00248">org.jgap.Genotype.evolve()</a>, <a class="el" href="_genotype_8java_source.html#l00098">org.jgap.Genotype.Genotype()</a>, and <a class="el" href="_genotype_8java_source.html#l00535">org.jgap.Genotype.getConfiguration()</a>.</p>

</div>
</div>
<a class="anchor" id="acac1218542ab91833483ebd42ac8636b"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">IEvolutionMonitor org.jgap.Genotype.m_monitor</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00057">57</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>Referenced by <a class="el" href="_genotype_8java_source.html#l00624">org.jgap.Genotype.run()</a>, and <a class="el" href="_genotype_8java_source.html#l00716">org.jgap.Genotype.setMonitor()</a>.</p>

</div>
</div>
<a class="anchor" id="aa2ba4539110aaa8d853c808679f4f0e2"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> org.jgap.Genotype.m_population</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>The array of Chromosomes that make-up the <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a>'s population. </p>
<dl class="section since"><dt>Since</dt><dd>2.0 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00050">50</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>Referenced by <a class="el" href="_genotype_8java_source.html#l00098">org.jgap.Genotype.Genotype()</a>, <a class="el" href="_genotype_8java_source.html#l00167">org.jgap.Genotype.getPopulation()</a>, and <a class="el" href="_genotype_8java_source.html#l00570">org.jgap.Genotype.setPopulation()</a>.</p>

</div>
</div>
<a class="anchor" id="a8b93a0b2044e40520d4a966ef3ae0bbe"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">transient <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> org.jgap.Genotype.m_staticConfiguration</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00044">44</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>Referenced by <a class="el" href="_genotype_8java_source.html#l00520">org.jgap.Genotype.getStaticConfiguration()</a>, and <a class="el" href="_genotype_8java_source.html#l00531">org.jgap.Genotype.setStaticConfiguration()</a>.</p>

</div>
</div>
<a class="anchor" id="ab1b1b976deab7f2faea463d1b8d52abe"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Genotype.m_useMonitor</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Use an enolution monitor </p>
<dl class="section since"><dt>Since</dt><dd>3.6 </dd></dl>

<p>Definition at line <a class="el" href="_genotype_8java_source.html#l00055">55</a> of file <a class="el" href="_genotype_8java_source.html">Genotype.java</a>.</p>

<p>Referenced by <a class="el" href="_genotype_8java_source.html#l00624">org.jgap.Genotype.run()</a>, and <a class="el" href="_genotype_8java_source.html#l00703">org.jgap.Genotype.setUseMonitor()</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>LDH_Git/src/org/jgap/<a class="el" href="_genotype_8java_source.html">Genotype.java</a></li>
</ul>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Sat Nov 23 2013 17:42:12 for Práctica_5_LDH by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
