<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Práctica_5_LDH: org.jgap.GenotypeTest.MyChromosome Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">Práctica_5_LDH
   &#160;<span id="projectnumber">1.2</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li><a href="inherits.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Classes</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Namespaces</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Files</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Functions</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(6)"><span class="SelectionMark">&#160;</span>Pages</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="namespaceorg.html">org</a></li><li class="navelem"><a class="el" href="namespaceorg_1_1jgap.html">jgap</a></li><li class="navelem"><a class="el" href="classorg_1_1jgap_1_1_genotype_test.html">GenotypeTest</a></li><li class="navelem"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html">MyChromosome</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pri-attribs">Private Attributes</a> &#124;
<a href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">org.jgap.GenotypeTest.MyChromosome Class Reference</div>  </div>
</div><!--header-->
<div class="contents">
<div class="dynheader">
Inheritance diagram for org.jgap.GenotypeTest.MyChromosome:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome__inherit__graph.png" border="0" usemap="#org_8jgap_8_genotype_test_8_my_chromosome_inherit__map" alt="Inheritance graph"/></div>
<map name="org_8jgap_8_genotype_test_8_my_chromosome_inherit__map" id="org_8jgap_8_genotype_test_8_my_chromosome_inherit__map">
<area shape="rect" id="node2" href="interfaceorg_1_1jgap_1_1_i_chromosome.html" title="org.jgap.IChromosome" alt="" coords="195,155,348,181"/><area shape="rect" id="node6" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html" title="org.jgap.util.ICloneable" alt="" coords="121,80,275,107"/><area shape="rect" id="node12" href="interfaceorg_1_1jgap_1_1_i_unique_key.html" title="org.jgap.IUniqueKey" alt="" coords="414,80,553,107"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<div class="dynheader">
Collaboration diagram for org.jgap.GenotypeTest.MyChromosome:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome__coll__graph.png" border="0" usemap="#org_8jgap_8_genotype_test_8_my_chromosome_coll__map" alt="Collaboration graph"/></div>
<map name="org_8jgap_8_genotype_test_8_my_chromosome_coll__map" id="org_8jgap_8_genotype_test_8_my_chromosome_coll__map">
<area shape="rect" id="node2" href="interfaceorg_1_1jgap_1_1_i_chromosome.html" title="org.jgap.IChromosome" alt="" coords="380,20,535,47"/><area shape="rect" id="node14" href="classorg_1_1jgap_1_1_configuration.html" title="org.jgap.Configuration" alt="" coords="773,324,923,351"/><area shape="rect" id="node21" href="interfaceorg_1_1jgap_1_1_i_breeder.html" title="org.jgap.IBreeder" alt="" coords="397,172,518,199"/><area shape="rect" id="node6" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html" title="org.jgap.util.ICloneable" alt="" coords="136,239,291,265"/><area shape="rect" id="node38" href="classorg_1_1jgap_1_1_bulk_fitness_function.html" title="org.jgap.BulkFitnessFunction" alt="" coords="362,324,553,351"/><area shape="rect" id="node45" href="classorg_1_1jgap_1_1_fitness_function.html" title="org.jgap.FitnessFunction" alt="" coords="375,375,540,401"/><area shape="rect" id="node26" href="interfaceorg_1_1jgap_1_1_i_j_g_a_p_factory.html" title="org.jgap.IJGAPFactory" alt="" coords="380,476,535,503"/><area shape="rect" id="node31" href="interfaceorg_1_1jgap_1_1_fitness_evaluator.html" title="org.jgap.FitnessEvaluator" alt="" coords="372,527,543,553"/><area shape="rect" id="node35" href="interfaceorg_1_1jgap_1_1_random_generator.html" title="org.jgap.RandomGenerator" alt="" coords="368,577,547,604"/><area shape="rect" id="node42" href="classorg_1_1jgap_1_1_configuration_1_1_configuration_configurable.html" title="org.jgap.Configuration.Configuration\lConfigurable" alt="" coords="342,679,573,721"/><area shape="rect" id="node12" href="interfaceorg_1_1jgap_1_1_i_unique_key.html" title="org.jgap.IUniqueKey" alt="" coords="143,5,283,32"/><area shape="rect" id="node29" href="interfaceorg_1_1jgap_1_1_i_chromosome_pool.html" title="org.jgap.IChromosomePool" alt="" coords="367,425,547,452"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a33590ad6adb4bfbdab418d8e8a78b5e9"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a33590ad6adb4bfbdab418d8e8a78b5e9">MyChromosome</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_conf)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a33590ad6adb4bfbdab418d8e8a78b5e9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae5110cb67ea8a3880b0052890dba7ea8"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#ae5110cb67ea8a3880b0052890dba7ea8">getGene</a> (int a_desiredLocus)</td></tr>
<tr class="separator:ae5110cb67ea8a3880b0052890dba7ea8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae3ea445bb5d440b904a1a5d1edc6387e"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#ae3ea445bb5d440b904a1a5d1edc6387e">getUniqueID</a> ()</td></tr>
<tr class="separator:ae3ea445bb5d440b904a1a5d1edc6387e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4569712147dd31fb4307dfe88e7bb283"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a4569712147dd31fb4307dfe88e7bb283">setUniqueIDTemplate</a> (String a_templateID, int a_index)</td></tr>
<tr class="separator:a4569712147dd31fb4307dfe88e7bb283"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a80c37f479ddf7175e8c2569208dc9364"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a80c37f479ddf7175e8c2569208dc9364">getUniqueIDTemplate</a> (int a_index)</td></tr>
<tr class="separator:a80c37f479ddf7175e8c2569208dc9364"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a18f5051e360985aa85e7dad63266bab5"><td class="memItemLeft" align="right" valign="top"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a18f5051e360985aa85e7dad63266bab5">getGenes</a> ()</td></tr>
<tr class="separator:a18f5051e360985aa85e7dad63266bab5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a42bdfe312e3f05b10ba8b1f2ab1588cf"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a42bdfe312e3f05b10ba8b1f2ab1588cf">size</a> ()</td></tr>
<tr class="separator:a42bdfe312e3f05b10ba8b1f2ab1588cf"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2ce2c7ac07f0581cccc3f94367e43b1a"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a2ce2c7ac07f0581cccc3f94367e43b1a">setFitnessValue</a> (double a_newFitnessValue)</td></tr>
<tr class="separator:a2ce2c7ac07f0581cccc3f94367e43b1a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a60478ceef952e45e61ab80e56192e805"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a60478ceef952e45e61ab80e56192e805">getFitnessValue</a> ()</td></tr>
<tr class="separator:a60478ceef952e45e61ab80e56192e805"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a00e24b5ada37e05d509cd165155fb61c"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a00e24b5ada37e05d509cd165155fb61c">setFitnessValueDirectly</a> (double a_newFitnessValue)</td></tr>
<tr class="separator:a00e24b5ada37e05d509cd165155fb61c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa1642265bd829ee4fdba8dede5d2ae5e"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#aa1642265bd829ee4fdba8dede5d2ae5e">getFitnessValueDirectly</a> ()</td></tr>
<tr class="separator:aa1642265bd829ee4fdba8dede5d2ae5e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af684e72b8fd995f4f39f0009c809c3a4"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#af684e72b8fd995f4f39f0009c809c3a4">compareTo</a> (Object other)</td></tr>
<tr class="separator:af684e72b8fd995f4f39f0009c809c3a4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae183a5fc7e0f40ac707ae3bc69a7fd6f"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#ae183a5fc7e0f40ac707ae3bc69a7fd6f">setGenes</a> (<a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[] a_genes)  throws InvalidConfigurationException </td></tr>
<tr class="separator:ae183a5fc7e0f40ac707ae3bc69a7fd6f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a542e571ec89ab56722d0396d038a3c03"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a542e571ec89ab56722d0396d038a3c03">setIsSelectedForNextGeneration</a> (boolean a_isSelected)</td></tr>
<tr class="separator:a542e571ec89ab56722d0396d038a3c03"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3c465ea8dfe5e6b7d5a1f3dfeb2e38cd"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a3c465ea8dfe5e6b7d5a1f3dfeb2e38cd">isSelectedForNextGeneration</a> ()</td></tr>
<tr class="separator:a3c465ea8dfe5e6b7d5a1f3dfeb2e38cd"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ace23248a8547679cfd6b352a8dbe6e15"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#ace23248a8547679cfd6b352a8dbe6e15">clone</a> ()</td></tr>
<tr class="separator:ace23248a8547679cfd6b352a8dbe6e15"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aee091968753a941ce51377ace0c0d916"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#aee091968753a941ce51377ace0c0d916">setConstraintChecker</a> (<a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a> a_constraintChecker)  throws InvalidConfigurationException </td></tr>
<tr class="separator:aee091968753a941ce51377ace0c0d916"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a20c792fe3836ca8d6e69e6c183217f41"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a20c792fe3836ca8d6e69e6c183217f41">setApplicationData</a> (Object a_newData)</td></tr>
<tr class="separator:a20c792fe3836ca8d6e69e6c183217f41"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a472baec7498af812e73a463d7c6870c8"><td class="memItemLeft" align="right" valign="top">Object&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a472baec7498af812e73a463d7c6870c8">getApplicationData</a> ()</td></tr>
<tr class="separator:a472baec7498af812e73a463d7c6870c8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab38c7c63e7a8468145a13a507939ceb7"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#ab38c7c63e7a8468145a13a507939ceb7">cleanup</a> ()</td></tr>
<tr class="separator:ab38c7c63e7a8468145a13a507939ceb7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac46d8ea344d04de7b648dd6d03ae94b3"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#ac46d8ea344d04de7b648dd6d03ae94b3">getConfiguration</a> ()</td></tr>
<tr class="separator:ac46d8ea344d04de7b648dd6d03ae94b3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a14d03b6c15e6414112ddb39e4a4ec41b"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a14d03b6c15e6414112ddb39e4a4ec41b">increaseAge</a> ()</td></tr>
<tr class="separator:a14d03b6c15e6414112ddb39e4a4ec41b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1ae5b0090b2241c90518cbd386753db8"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a1ae5b0090b2241c90518cbd386753db8">resetAge</a> ()</td></tr>
<tr class="separator:a1ae5b0090b2241c90518cbd386753db8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7a2a95044a2d93b6c1529641dcdbaab9"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a7a2a95044a2d93b6c1529641dcdbaab9">setAge</a> (int a_age)</td></tr>
<tr class="separator:a7a2a95044a2d93b6c1529641dcdbaab9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abe7315264fc2d38625cc4ed0da3baf79"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#abe7315264fc2d38625cc4ed0da3baf79">getAge</a> ()</td></tr>
<tr class="separator:abe7315264fc2d38625cc4ed0da3baf79"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a15c7586b0c9d4c6d88be33c9bbd9dd0e"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a15c7586b0c9d4c6d88be33c9bbd9dd0e">increaseOperatedOn</a> ()</td></tr>
<tr class="separator:a15c7586b0c9d4c6d88be33c9bbd9dd0e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3ea0a85d844419fcd95025681df43a45"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a3ea0a85d844419fcd95025681df43a45">resetOperatedOn</a> ()</td></tr>
<tr class="separator:a3ea0a85d844419fcd95025681df43a45"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa4c90dd4e8970ddbe45a06d590d35e2d"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#aa4c90dd4e8970ddbe45a06d590d35e2d">operatedOn</a> ()</td></tr>
<tr class="separator:aa4c90dd4e8970ddbe45a06d590d35e2d"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-attribs"></a>
Private Attributes</h2></td></tr>
<tr class="memitem:af945beb0c1e47fcb7f1cbbc4d2ecdf59"><td class="memItemLeft" align="right" valign="top">transient <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#af945beb0c1e47fcb7f1cbbc4d2ecdf59">m_conf</a></td></tr>
<tr class="separator:af945beb0c1e47fcb7f1cbbc4d2ecdf59"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="inherited"></a>
Additional Inherited Members</h2></td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">org.jgap.IChromosome</a></td></tr>
<tr class="memitem:a4e7381b89963f2ee6f69839b38ea8dab inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a4e7381b89963f2ee6f69839b38ea8dab">CVS_REVISION</a> = &quot;$Revision: 1.17 $&quot;</td></tr>
<tr class="separator:a4e7381b89963f2ee6f69839b38ea8dab inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a34c8c0933688a10837692d1c0024f663 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a34c8c0933688a10837692d1c0024f663">S_FITNESS_VALUE</a> = &quot;Fitness value&quot;</td></tr>
<tr class="separator:a34c8c0933688a10837692d1c0024f663 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae93666d5923e449f45656ad5fed52f84 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#ae93666d5923e449f45656ad5fed52f84">S_ALLELES</a> = &quot;Alleles&quot;</td></tr>
<tr class="separator:ae93666d5923e449f45656ad5fed52f84 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a03218ba49e2c53108005cd6290252284 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a03218ba49e2c53108005cd6290252284">S_APPLICATION_DATA</a> = &quot;Application data&quot;</td></tr>
<tr class="separator:a03218ba49e2c53108005cd6290252284 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a50371491243a21f494112bf66d2e1a26 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a50371491243a21f494112bf66d2e1a26">S_SIZE</a> = &quot;Size&quot;</td></tr>
<tr class="separator:a50371491243a21f494112bf66d2e1a26 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_chromosome"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html">org.jgap.util.ICloneable</a></td></tr>
<tr class="memitem:a4eaaa85432026f5bc54c3e520b22caa7 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html#a4eaaa85432026f5bc54c3e520b22caa7">CVS_REVISION</a> = &quot;$Revision: 1.3 $&quot;</td></tr>
<tr class="separator:a4eaaa85432026f5bc54c3e520b22caa7 inherit pub_static_attribs_interfaceorg_1_1jgap_1_1util_1_1_i_cloneable"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key"><td colspan="2" onclick="javascript:toggleInherit('pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key')"><img src="closed.png" alt="-"/>&#160;Static Public Attributes inherited from <a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html">org.jgap.IUniqueKey</a></td></tr>
<tr class="memitem:a5b6472b1e40001ee262da59ee449a77b inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key"><td class="memItemLeft" align="right" valign="top">static final String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html#a5b6472b1e40001ee262da59ee449a77b">CVS_REVISION</a> = &quot;$Revision: 1.1 $&quot;</td></tr>
<tr class="separator:a5b6472b1e40001ee262da59ee449a77b inherit pub_static_attribs_interfaceorg_1_1jgap_1_1_i_unique_key"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock">
<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01355">1355</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="a33590ad6adb4bfbdab418d8e8a78b5e9"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.GenotypeTest.MyChromosome.MyChromosome </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_conf</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01359">1359</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

<p>References <a class="el" href="_genotype_test_8java_source.html#l01357">org.jgap.GenotypeTest.MyChromosome.m_conf</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="ab38c7c63e7a8468145a13a507939ceb7"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.GenotypeTest.MyChromosome.cleanup </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Invoked when this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> is no longer needed and should perform any necessary cleanup. Note that this method will attempt to release this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> instance to the active ChromosomePool, if any.</p>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a184e942f9b44185236492ee7261b2b14">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01435">1435</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="ace23248a8547679cfd6b352a8dbe6e15"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.GenotypeTest.MyChromosome.clone </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>clone of the current object instance </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1util_1_1_i_cloneable.html#aa7e7d62077e6428ad7904932b1b4f7d5">org.jgap.util.ICloneable</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01420">1420</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="af684e72b8fd995f4f39f0009c809c3a4"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.GenotypeTest.MyChromosome.compareTo </td>
          <td>(</td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>other</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01405">1405</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="abe7315264fc2d38625cc4ed0da3baf79"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.GenotypeTest.MyChromosome.getAge </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>0: <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> newly created in this generation. This means it does not need being crossed over with another newly created one</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a6186b5e6184d55dbc03adee6bacc8e9a">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01451">1451</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="a472baec7498af812e73a463d7c6870c8"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Object org.jgap.GenotypeTest.MyChromosome.getApplicationData </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves the application-specific data that is attached to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. Attaching application-specific data may be useful for some applications when it comes time to evaluate this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in the fitness function. JGAP ignores this data functionally.</p>
<dl class="section return"><dt>Returns</dt><dd>the application-specific data previously attached to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>, or null if there is no data attached</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a1734f351c78c6d36bec48fdc8a334902">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01431">1431</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="ac46d8ea344d04de7b648dd6d03ae94b3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> org.jgap.GenotypeTest.MyChromosome.getConfiguration </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>the configuration set</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.0 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a4426723f5d26695fd80a0b28b3acb9eb">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01438">1438</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

<p>References <a class="el" href="_genotype_test_8java_source.html#l01357">org.jgap.GenotypeTest.MyChromosome.m_conf</a>.</p>

</div>
</div>
<a class="anchor" id="a60478ceef952e45e61ab80e56192e805"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double org.jgap.GenotypeTest.MyChromosome.getFitnessValue </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves the fitness value of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>, as determined by the active fitness function. If a bulk fitness function is in use and has not yet assigned a fitness value to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>, then -1 is returned.</p>
<p>Attention: should not be called from toString() as the fitness value would be computed if it was initial!</p>
<dl class="section return"><dt>Returns</dt><dd>a positive double value representing the fitness of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>, or -1 if a bulk fitness function is in use and has not yet assigned a fitness value to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a></dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a9b68991c832a2054176d522e49a19ebd">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01394">1394</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

<p>Referenced by <a class="el" href="_genotype_test_8java_source.html#l01401">org.jgap.GenotypeTest.MyChromosome.getFitnessValueDirectly()</a>.</p>

</div>
</div>
<a class="anchor" id="aa1642265bd829ee4fdba8dede5d2ae5e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double org.jgap.GenotypeTest.MyChromosome.getFitnessValueDirectly </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>the lastly computed fitness value, or <a class="el" href="classorg_1_1jgap_1_1_fitness_function.html#a5f1f23e6f8a0f9606aee5d52fac6a06b">FitnessFunction.NO_FITNESS_VALUE</a> in case no value has been computed yet.</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a743303c75d2a001f88af579051862964">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01401">1401</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

<p>References <a class="el" href="_genotype_test_8java_source.html#l01394">org.jgap.GenotypeTest.MyChromosome.getFitnessValue()</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome_aa1642265bd829ee4fdba8dede5d2ae5e_cgraph.png" border="0" usemap="#classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome_aa1642265bd829ee4fdba8dede5d2ae5e_cgraph" alt=""/></div>
<map name="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome_aa1642265bd829ee4fdba8dede5d2ae5e_cgraph" id="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome_aa1642265bd829ee4fdba8dede5d2ae5e_cgraph">
<area shape="rect" id="node3" href="classorg_1_1jgap_1_1_genotype_test_1_1_my_chromosome.html#a60478ceef952e45e61ab80e56192e805" title="org.jgap.GenotypeTest.My\lChromosome.getFitnessValue" alt="" coords="297,5,495,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ae5110cb67ea8a3880b0052890dba7ea8"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> org.jgap.GenotypeTest.MyChromosome.getGene </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_desiredLocus</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Returns the <a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> at the given index (locus) within the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. The first gene is at index zero and the last gene is at the index equal to the size of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> - 1.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_desiredLocus</td><td>index of the gene value to be returned </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> at the given index</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#ac84a3ec514b365e25f023d9e78bfa382">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01364">1364</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

<p>References <a class="el" href="_genotype_test_8java_source.html#l01357">org.jgap.GenotypeTest.MyChromosome.m_conf</a>.</p>

</div>
</div>
<a class="anchor" id="a18f5051e360985aa85e7dad63266bab5"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a> [] org.jgap.GenotypeTest.MyChromosome.getGenes </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves the set of genes that make up this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. This method exists primarily for the benefit of GeneticOperators that require the ability to manipulate Chromosomes at a low level.</p>
<dl class="section return"><dt>Returns</dt><dd>an array of the Genes contained within this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a></dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a995b4203af3bd22b8fd2028749736a03">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01383">1383</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="ae3ea445bb5d440b904a1a5d1edc6387e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.GenotypeTest.MyChromosome.getUniqueID </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>String</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.5 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html#a1276412b558e169199771997a190f876">org.jgap.IUniqueKey</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01372">1372</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="a80c37f479ddf7175e8c2569208dc9364"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.GenotypeTest.MyChromosome.getUniqueIDTemplate </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_index</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_index</td><td>int </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>String</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.5 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html#af26fbe03487632a16c84b62325489652">org.jgap.IUniqueKey</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01379">1379</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="a14d03b6c15e6414112ddb39e4a4ec41b"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.GenotypeTest.MyChromosome.increaseAge </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Increases the number of evolutionary rounds of chromosome in which it has not been changed by one.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a651016ea5a204fc14c7c146e32fd76cf">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01442">1442</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="a15c7586b0c9d4c6d88be33c9bbd9dd0e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.GenotypeTest.MyChromosome.increaseOperatedOn </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Increase information of number of genetic operations performed on chromosome in current evolution round.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a8fda999d5a66640dd577bcff960b368b">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01455">1455</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="a3c465ea8dfe5e6b7d5a1f3dfeb2e38cd"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.GenotypeTest.MyChromosome.isSelectedForNextGeneration </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Retrieves whether this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> has been selected by the natural selector to continue to the next generation.</p>
<dl class="section return"><dt>Returns</dt><dd>true if this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> has been selected, false otherwise</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#af4aa1487bfa13ba0f038b628f44554be">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01416">1416</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="aa4c90dd4e8970ddbe45a06d590d35e2d"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.GenotypeTest.MyChromosome.operatedOn </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>number of genetic operations performed on chromosome in current evolution round</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a47f09dd85e4f9f0e3908c85b63394a7d">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01460">1460</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="a1ae5b0090b2241c90518cbd386753db8"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.GenotypeTest.MyChromosome.resetAge </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Reset age of chromosome because it has been changed.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#aefc4884cc5bbbc511feb8d9ec827a31f">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01445">1445</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="a3ea0a85d844419fcd95025681df43a45"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.GenotypeTest.MyChromosome.resetOperatedOn </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Resets the information of how many genetic operators have been performed on the chromosome in the current round of evolution.</p>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.2 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a4fb2f78d39b20b7f5a9ff8d37402f7bd">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01458">1458</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="a7a2a95044a2d93b6c1529641dcdbaab9"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.GenotypeTest.MyChromosome.setAge </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_age</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_age</td><td>set the age of the chromosome, see BestChromosomesSelector for a use-case</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.3.3 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a9146b114972d86aac3cfb32b249b166f">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01448">1448</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="a20c792fe3836ca8d6e69e6c183217f41"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.GenotypeTest.MyChromosome.setApplicationData </td>
          <td>(</td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>a_newData</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>This sets the application-specific data that is attached to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. Attaching application-specific data may be useful for some applications when it comes time to evaluate this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in the fitness function.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_newData</td><td>the new application-specific data to attach to this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. Should be an instance of <a class="el" href="interfaceorg_1_1jgap_1_1_i_application_data.html">IApplicationData</a></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a99da03ae9bca8f2ca884186aea82c771">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01428">1428</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="aee091968753a941ce51377ace0c0d916"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.GenotypeTest.MyChromosome.setConstraintChecker </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_i_gene_constraint_checker.html">IGeneConstraintChecker</a>&#160;</td>
          <td class="paramname"><em>a_constraintChecker</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets the constraint checker to be used for this gene whenever method setAllele(Object) is called.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_constraintChecker</td><td>the constraint checker to be set </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a05c61ae9fb2caf56455eb66405a164c4">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01424">1424</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="a2ce2c7ac07f0581cccc3f94367e43b1a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.GenotypeTest.MyChromosome.setFitnessValue </td>
          <td>(</td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>a_newFitnessValue</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets the fitness value of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>. This method is for use by bulk fitness functions and should not be invokved from anything else (except test cases).</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_newFitnessValue</td><td>a positive integer representing the fitness of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a56f82d43b102fc5da8e0a071e32dc13d">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01391">1391</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="a00e24b5ada37e05d509cd165155fb61c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.GenotypeTest.MyChromosome.setFitnessValueDirectly </td>
          <td>(</td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>a_newFitnessValue</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets the fitness value of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> directly without any constraint checks, conversions or checks. Only use if you know what you do.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_newFitnessValue</td><td>a positive integer representing the fitness of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#ac2acf33157b233e8c5553fc62861fa23">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01398">1398</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="ae183a5fc7e0f40ac707ae3bc69a7fd6f"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.GenotypeTest.MyChromosome.setGenes </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_gene.html">Gene</a>[]&#160;</td>
          <td class="paramname"><em>a_genes</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets the genes for the chromosome. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_genes</td><td>the genes to set for the chromosome</td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td>in case constraint checker is provided</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a0ff42f1935dab73296301b29ad0a1716">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01409">1409</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="a542e571ec89ab56722d0396d038a3c03"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.GenotypeTest.MyChromosome.setIsSelectedForNextGeneration </td>
          <td>(</td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_isSelected</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets whether this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> has been selected by the natural selector to continue to the next generation or manually (e.g. via an add-method).</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_isSelected</td><td>true if this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> has been selected, false otherwise</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#afeeed3697d5f22154cab618432602ca6">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01413">1413</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="a4569712147dd31fb4307dfe88e7bb283"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.GenotypeTest.MyChromosome.setUniqueIDTemplate </td>
          <td>(</td>
          <td class="paramtype">String&#160;</td>
          <td class="paramname"><em>a_templateID</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_index</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_templateID</td><td>String </td></tr>
    <tr><td class="paramname">a_index</td><td>int</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Since</dt><dd>3.5 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_unique_key.html#aeac223774f93ba6640bc07cc97a9ef4c">org.jgap.IUniqueKey</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01376">1376</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<a class="anchor" id="a42bdfe312e3f05b10ba8b1f2ab1588cf"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.GenotypeTest.MyChromosome.size </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Returns the size of this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> (the number of genes it contains). A <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a>'s size is constant and will not change, until setGenes(...) is used.</p>
<dl class="section return"><dt>Returns</dt><dd>number of genes contained within this <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> instance</dd></dl>
<dl class="section author"><dt>Author</dt><dd>Klaus Meffert </dd>
<dd>
Neil Rotstan </dd></dl>
<dl class="section since"><dt>Since</dt><dd>2.6 </dd></dl>

<p>Implements <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html#a9b460ad665ea1ab4e789061233fc6fa3">org.jgap.IChromosome</a>.</p>

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01387">1387</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Data Documentation</h2>
<a class="anchor" id="af945beb0c1e47fcb7f1cbbc4d2ecdf59"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">transient <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> org.jgap.GenotypeTest.MyChromosome.m_conf</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="_genotype_test_8java_source.html#l01357">1357</a> of file <a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a>.</p>

<p>Referenced by <a class="el" href="_genotype_test_8java_source.html#l01438">org.jgap.GenotypeTest.MyChromosome.getConfiguration()</a>, <a class="el" href="_genotype_test_8java_source.html#l01364">org.jgap.GenotypeTest.MyChromosome.getGene()</a>, and <a class="el" href="_genotype_test_8java_source.html#l01359">org.jgap.GenotypeTest.MyChromosome.MyChromosome()</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>LDH_Git/tests/org/jgap/<a class="el" href="_genotype_test_8java_source.html">GenotypeTest.java</a></li>
</ul>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Sat Nov 23 2013 17:42:17 for Práctica_5_LDH by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
